# Differential abundance analysis

```{r load_data_mag_filtdamr_diffe, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
load("data/data.Rdata")
```

## CC acclimation vs post2
### Enrichment analysis: Ancombc2

```{r zero_phylo1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type== "Control") %>% 
                    filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    #filter(genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filtercapwild, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_CC <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","6_Post-FMT2"))
physeq_sample_CC <- prune_taxa(taxa_sums(physeq_sample_CC)>0, physeq_sample_CC)
```

#### MAG level
```{r ancom_rand_capwild, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_accli_post2_CC = ancombc2(data = physeq_sample_CC, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "time_point", #fixed variable(s)
 #                 rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0.10, 
                        lib_cut = 1, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
#                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_capCC, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  right_join(ancom_result_CC, by=join_by(phylum == phylum)) %>%
  arrange(match(genome, genome_tree$tip.label)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
  unique() %>%
  arrange(phylum) %>%
  dplyr::select(colors) %>%
  pull()

ancom_result_CC <- ancom_rand_output_mag_accli_post2_CC$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_CC %>%
  mutate(significance = ifelse(p_time_point6_Post_FMT2 < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin Accli", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_CC %>%
  filter(p_time_point6_Post_FMT2 < 0.01) %>%
  ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_CC$lfc_time_point6_Post_FMT2)-2,max(ancom_result_CC$lfc_time_point6_Post_FMT2)+2) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_CC$lfc_time_point6_Post_FMT2)+1, 1.5), label = "Enriched\nin Accli", color="#666666") +
  geom_text(aes(max(ancom_result_CC$lfc_time_point6_Post_FMT2)-2, 2), label = "Enriched\nin Post2", color="#666666") +
  theme(legend.position='right',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```

```{r ancom_table_capCC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_CC@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

colnames(ancom_rand_output_mag_accli_post2_CC$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_CC$res))

ancombc_rand_table_mag <- ancom_rand_output_mag_accli_post2_CC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")

tax_color <- tax_color %>%
  mutate(colors = if_else(phylum=="p__Bacillota", '#074759',
          if_else(phylum=="p__Bacillota_A", '#0c80a2',
          if_else(phylum=="p__Bacillota_B", '#10acd9',
          if_else(phylum=="p__Bacillota_C", '#7ddaf5',
          if_else(phylum=="p__Bacteroidota", "#82383880",
          if_else(phylum=="p__Pseudomonadota", '#E0A60880',
          if_else(phylum=="p___Fusobacteriota", '#08EACA80',
          if_else(phylum=="p__Verrucomicrobiota", '#AD8D8D80',
          if_else(phylum=="p__Desulfobacterota", "#39A60880",
          if_else(phylum=="p__Campylobacterota", "#DE2E0880", colors
                  )))))))))))%>%
  dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_CC, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag$genome)) %>% 
ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_reorder(genus, lfc_time_point6_Post_FMT2), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Genus")+
  guides(fill=guide_legend(title="Phylum"))
```

## WC acclimation vs post2
### Enrichment analysis: Ancombc2

```{r zero_phylo1_WC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type== "Hot_control") %>% 
                    filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    #filter(genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filtercapWC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_WC <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","6_Post-FMT2"))
physeq_sample_WC <- prune_taxa(taxa_sums(physeq_sample_WC)>0, physeq_sample_WC)
```


#### MAG level
```{r ancom_rand_capWC, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_accli_post2_WC = ancombc2(data = physeq_sample_WC, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "time_point", #fixed variable(s)
 #                 rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0.10, 
                        lib_cut = 1, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
#                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_capWC_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
colnames(ancom_rand_output_mag_accli_post2_WC$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_WC$res))

ancom_result_WC <- ancom_rand_output_mag_accli_post2_WC$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_WC %>%
  mutate(significance = ifelse(p_time_point6_Post_FMT2 < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin Accli", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_WC %>%
  filter(p_time_point6_Post_FMT2 < 0.01) %>%
  ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_WC$lfc_time_point6_Post_FMT2)-6,max(ancom_result_WC$lfc_time_point6_Post_FMT2)+6) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_WC$lfc_time_point6_Post_FMT2)+1, 1), label = "Enriched\nin Accli", color="#666666") +
  geom_text(aes(max(ancom_result_WC$lfc_time_point6_Post_FMT2)-1, 1), label = "Enriched\nin Post2", color="#666666") +
  theme(legend.position='right',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```  

```{r ancom_table_capWC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_WC@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

colnames(ancom_rand_output_mag_accli_post2_WC$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_WC$res))

ancombc_rand_table_mag_WC <- ancom_rand_output_mag_accli_post2_WC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag_WC$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")

tax_color <- tax_color %>%
  mutate(colors = if_else(phylum=="p__Bacillota", '#074759',
          if_else(phylum=="p__Bacillota_A", '#0c80a2',
          if_else(phylum=="p__Bacillota_B", '#10acd9',
          if_else(phylum=="p__Bacillota_C", '#7ddaf5',
          if_else(phylum=="p__Bacteroidota", "#82383880",
          if_else(phylum=="p__Pseudomonadota", '#E0A60880',
          if_else(phylum=="p___Fusobacteriota", '#08EACA80',
          if_else(phylum=="p__Verrucomicrobiota", '#AD8D8D80',
          if_else(phylum=="p__Desulfobacterota", "#39A60880",
          if_else(phylum=="p__Campylobacterota", "#DE2E0880", colors
                  )))))))))))%>%
  dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_WC, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag_WC %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag_WC$genome)) %>% 
ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_reorder(species, lfc_time_point6_Post_FMT2), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

## CI acclimation vs post2
### Enrichment analysis: Ancombc2

```{r zero_phylo1_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type== "Treatment") %>% 
                    filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    #filter(genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filterCI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_CI <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","6_Post-FMT2"))
physeq_sample_CI <- prune_taxa(taxa_sums(physeq_sample_CI)>0, physeq_sample_CI)
```


#### MAG level
```{r ancom_rand_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_accli_post2_CI = ancombc2(data = physeq_sample_CI, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "time_point", #fixed variable(s)
 #                 rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0.10, 
                        lib_cut = 1, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
#                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_CI_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
colnames(ancom_rand_output_mag_accli_post2_CI$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_CI$res))

ancom_result_CI <- ancom_rand_output_mag_accli_post2_CI$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_CI %>%
  mutate(significance = ifelse(p_time_point6_Post_FMT2 < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin Accli", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_CI %>%
  filter(p_time_point6_Post_FMT2 < 0.01) %>%
  ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_CI$lfc_time_point6_Post_FMT2)-2,max(ancom_result_CI$lfc_time_point6_Post_FMT2)+2) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_CI$lfc_time_point6_Post_FMT2)+2, 2), label = "Enriched\nin Accli", color="#666666") +
  geom_text(aes(max(ancom_result_CI$lfc_time_point6_Post_FMT2)-2, 2), label = "Enriched\nin Post2", color="#666666") +
  theme(legend.position='right',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```  

```{r ancom_table_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_CI@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

ancombc_rand_table_mag_CI <- ancom_rand_output_mag_accli_post2_CI$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag_CI$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")

tax_color <- tax_color %>%
  mutate(colors = if_else(phylum=="p__Bacillota", '#074759',
          if_else(phylum=="p__Bacillota_A", '#0c80a2',
          if_else(phylum=="p__Bacillota_B", '#10acd9',
          if_else(phylum=="p__Bacillota_C", '#7ddaf5',
          if_else(phylum=="p__Bacteroidota", "#82383880",
          if_else(phylum=="p__Pseudomonadota", '#E0A60880',
          if_else(phylum=="p___Fusobacteriota", '#08EACA80',
          if_else(phylum=="p__Verrucomicrobiota", '#AD8D8D80',
          if_else(phylum=="p__Desulfobacterota", "#39A60880",
          if_else(phylum=="p__Campylobacterota", "#DE2E0880", colors
                  )))))))))))%>%
  dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag_CI %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag_CI$genome)) %>% 
ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_reorder(species, lfc_time_point6_Post_FMT2), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

## Community level plots

```{r gift_test_function_plot_1, message=FALSE, warning=FALSE, fig.height=12, fig.width=10, fig.fullwidth=TRUE, eval=TRUE}
accli_post2<-sample_metadata%>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2")
accli_post2$newID<-paste(accli_post2$type, "_", accli_post2$time_point)

GIFTs_functions_community %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(accli_post2, by = join_by(sample == Tube_code)) %>%
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>%
  select(c(1:21, 27,30)) %>%
  pivot_longer(-c(sample,type,time_point),names_to = "trait", values_to = "value") %>%
  mutate(trait = case_when(
    trait %in% GIFT_db$Code_function ~ GIFT_db$Function[match(trait, GIFT_db$Code_function)],
    TRUE ~ trait
  )) %>%
  mutate(trait=factor(trait,levels=unique(GIFT_db$Function))) %>%
  ggplot(aes(x=value, y=time_point, group=time_point, fill=type, color=type)) +
  geom_boxplot() +
  scale_color_manual(name="type",
                     breaks=c("Control","Hot_control", "Treatment"),
                     labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  scale_fill_manual(name="type",
                    breaks=c("Control","Hot_control", "Treatment"),
                    labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
                    values=c("#4477AA50","#d57d2c50","#76b18350")) +
  facet_grid(trait ~ type, space="free", scales="free") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text.y = element_text(angle = 0)) + 
  labs(y="Traits",x="Metabolic capacity index")
```

```{r data_TM6_plot_1, echo=TRUE,results=FALSE}
GIFTs_elements_community_merged<-GIFTs_elements_community %>%
    as.data.frame() %>%
    rownames_to_column(var="sample") %>%
    filter(sample!="AD69") %>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == Tube_code))%>%
    filter(time_point=="1_Acclimation"|time_point == "6_Post-FMT2")%>%
    mutate(functionid = substr(trait, 1, 3)) %>%
    mutate(trait = case_when(
      trait %in% GIFT_db$Code_element ~ GIFT_db$Element[match(trait, GIFT_db$Code_element)],
      TRUE ~ trait
    )) %>%
    mutate(functionid = case_when(
      functionid %in% GIFT_db$Code_function ~ GIFT_db$Function[match(functionid, GIFT_db$Code_function)],
      TRUE ~ functionid
    )) %>%
    mutate(trait=factor(trait,levels=unique(GIFT_db$Element))) %>%
    mutate(functionid=factor(functionid,levels=unique(GIFT_db$Function)))

# Create an interaction variable for time_point and sample
GIFTs_elements_community_merged$interaction_var <- interaction(GIFTs_elements_community_merged$sample, GIFTs_elements_community_merged$time_point)
  
ggplot(GIFTs_elements_community_merged,aes(x=interaction_var,y=trait,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(functionid ~ type, scales="free",space="free") +
        theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=5),
              strip.text.y = element_text(angle = 0)) + 
        labs(y="Traits",x="Time_point",fill="GIFT")+
  scale_x_discrete(labels = function(x) gsub(".*\\.", "", x))
```


## Wilcoxon comparison

### Community elements differences: in CC acclimation vs post2 ####

```{r comunity_elem, comment="", echo=FALSE, message=FALSE, warning=FALSE}
samples_CC <- sample_metadata%>%
  filter(type=="Control") %>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2")%>% 
  mutate(time_point = recode(time_point,
                        "1_Acclimation" = "Acclimation", 
                        "6_Post-FMT2" = "Post2"))

element_gift_CC <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(samples_CC[c(1,10)], by="Tube_code") 
```

```{r commun_wilcox_elem, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_CC <- element_gift_CC %>%
  pivot_longer(-c(Tube_code,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_value < 0.05)%>% #note that p_value is used instead of p_adjust
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_CC <- element_gift_CC  %>% 
  dplyr::select(-c(time_point))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_CC <- subset(element_gift_t_CC, trait %in% significant_elements_CC$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_CC[c(1,10)], by = join_by(Tube_code == Tube_code))

difference_table_CC <- element_gift_filt_CC %>%
  select(-Tube_code) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Post2)%>% 
  mutate(group_color = ifelse(Difference <0, "Post2","Acclimation")) 
```

```{r log_fold_calc, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_CC <- element_gift_filt_CC %>% 
  select(-time_point) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(samples_CC, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(time_point, elements) %>%
  summarise(mean=mean(abundance))

log_fold_CC <- means_gift_CC %>%
  group_by(elements) %>%
  summarise(
    logfc_accli_post2 = log2(mean[time_point == "Acclimation"] / mean[time_point == "Post2"])
  )
```


```{r plot1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
difference_table_CC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CC, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_value), color=group_color)) + #note that p_value was used instead of p_adjust
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
uniqueGIFT <- unique(GIFT_db[c(2,3,4,5,6)])

code_function_CC <- difference_table_CC %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_CC<-unique(code_function_CC$Code_function)

gift_colors_CC <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_CC)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_CC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CC, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_CC, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_accli_post2, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_CC$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```
```{r final_plot_3, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_CC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CC, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_CC, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_accli_post2, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_CC$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
difference_table_CC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CC, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_accli_post2, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE, eval=TRUE}
difference_table_CC %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```
```{r}
difference_table_CC %>% 
  filter(group_color=="Acclimation")
```
```{r sugar_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
sugar_function <- GIFT_db %>% 
  filter(Code_function=="D03") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(sugar_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_CC[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>% 
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r poly_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
poly_function <- GIFT_db %>% 
  filter(Code_function=="D02") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(poly_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_CC[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>% 
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r elements_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_names_CC <- element_gift_filt_CC%>%
  select(-time_point)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_CC[c(1,10)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_CC)[2:16] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_CC, aes(x=time_point, y=.data[[i]], color = time_point, fill=time_point)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}

```

### Community elements differences: in CI acclimation vs post2 ####

```{r comunity_elem_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
samples_CI <- sample_metadata%>%
  filter(type=="Treatment") %>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2")%>% 
  mutate(time_point = recode(time_point,
                             "1_Acclimation" = "Acclimation", 
                             "6_Post-FMT2" = "Post2"))

element_gift_CI <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(samples_CI[c(1,10)], by="Tube_code") 
```

```{r commun_wilcox_elem_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_CI <- element_gift_CI %>%
  pivot_longer(-c(Tube_code,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_CI <- element_gift_CI  %>% 
  dplyr::select(-c(time_point))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_CI <- subset(element_gift_t_CI, trait %in% significant_elements_CI$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_CI[c(1,10)], by = join_by(Tube_code == Tube_code))

difference_table_CI <- element_gift_filt_CI %>%
  select(-Tube_code) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Post2)%>% 
  mutate(group_color = ifelse(Difference <0, "Post2","Acclimation")) 
```

```{r log_fold_calc_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_CI <- element_gift_filt_CI %>% 
  select(-time_point) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(samples_CI, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(time_point, elements) %>%
  summarise(mean=mean(abundance))

log_fold_CI <- means_gift_CI %>%
  group_by(elements) %>%
  summarise(
    logfc_acclimation_post2 = log2(mean[time_point == "Acclimation"] / mean[time_point == "Post2"])
  )
```


```{r plot1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=TRUE}
difference_table_CI %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CI, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_adjust), color=group_color)) + 
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
uniqueGIFT <- unique(GIFT_db[c(2,3,4,5,6)])

code_function_CI <- difference_table_CI %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_CI<-unique(code_function_CI$Code_function)

gift_colors_CI <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_CI)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_CI %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CI, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CI, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_CI, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_acclimation_post2, y = -log(p_adjust), color=legend, size=abs(Difference))) +
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_CI$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```
```{r final_plot_1_1_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_CI %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CI, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CI, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_CI, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_acclimation_post2, y = -log(p_adjust), color=legend, size=abs(Difference))) + 
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_CI$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
difference_table_CI %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CI, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CI, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_acclimation_post2, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE, eval=TRUE}
difference_table_CI %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```
```{r_1}
difference_table_CI %>% 
  filter(group_color=="Acclimation")
```
```{r sugar_plot_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
sugar_function <- GIFT_db %>% 
  filter(Code_function=="D03") %>% 
  select(Code_element) %>% 
  pull()
GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(sugar_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_CI[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>%
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r poly_plot_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
poly_function <- GIFT_db %>% 
  filter(Code_function=="D02") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(poly_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_CI[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>% 
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r elements_plot_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_names_CI <- element_gift_filt_CI%>%
  select(-time_point)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_CI[c(1,10)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_CI)[2:18] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_CI, aes(x=time_point, y=.data[[i]], color = time_point, fill=time_point)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}

```

### Community elements differences: in WC acclimation vs post2 ####
```{r comunity_elem_2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
samples_WC <- sample_metadata%>%
  filter(type=="Hot_control") %>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2")%>% 
  mutate(time_point = recode(time_point,
                             "1_Acclimation" = "Acclimation", 
                             "6_Post-FMT2" = "Post2"))

element_gift_WC <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(samples_WC[c(1,10)], by="Tube_code") 
```

```{r commun_wilcox_elem_2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_WC <- element_gift_WC %>%
  pivot_longer(-c(Tube_code,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_value < 0.05)%>% #note that p_value is used instead of p_adjust
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_WC <- element_gift_WC  %>% 
  dplyr::select(-c(time_point))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_WC <- subset(element_gift_t_WC, trait %in% significant_elements_WC$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_WC[c(1,10)], by = join_by(Tube_code == Tube_code))

difference_table_WC <- element_gift_filt_WC %>%
  select(-Tube_code) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Post2)%>% 
  mutate(group_color = ifelse(Difference <0, "Post2","Acclimation")) 
```

```{r log_fold_calc_2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_WC <- element_gift_filt_WC %>% 
  select(-time_point) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(samples_WC, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(time_point, elements) %>%
  summarise(mean=mean(abundance))

log_fold_WC <- means_gift_WC %>%
  group_by(elements) %>%
  summarise(
    logfc_acclimation_post2 = log2(mean[time_point == "Acclimation"] / mean[time_point == "Post2"])
  )
```


```{r plot1_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
difference_table_WC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_WC, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_value), color=group_color)) + #note that p_value was used instead of p_adjust
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
uniqueGIFT <- unique(GIFT_db[c(2,3,4,5,6)])

code_function_WC <- difference_table_WC %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_WC<-unique(code_function_WC$Code_function)

gift_colors_WC <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_WC)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_WC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_WC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_WC, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_WC, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_acclimation_post2, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_WC$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```

```{r final_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_WC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_WC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_WC, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_WC, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_acclimation_post2, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_WC$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
difference_table_WC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_WC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_WC, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_acclimation_post2, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE, eval=TRUE}
difference_table_WC %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

```{r_2}
difference_table_WC %>% 
  filter(group_color=="Acclimation")
```

```{r sugar_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
sugar_function <- GIFT_db %>% 
  filter(Code_function=="D03") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(sugar_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_WC[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>%
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r poly_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
poly_function <- GIFT_db %>% 
  filter(Code_function=="D02") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(poly_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_WC[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>% 
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r elements_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_names_WC <- element_gift_filt_WC%>%
  select(-time_point)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_WC[c(1,10)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_WC)[2:5] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_WC, aes(x=time_point, y=.data[[i]], color = time_point, fill=time_point)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

### Comparison of both population in wild samples ####

```{r comunity_elem_&, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_wild<- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_wild[c(1,4)], by="Tube_code") 
```

```{r commun_wilcox_elem_6, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_wild <- element_gift_wild %>%
  pivot_longer(-c(Tube_code,Population), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ Population)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_wild <- element_gift_wild  %>% 
  dplyr::select(-c(Population))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_wild <- subset(element_gift_t_wild, trait %in% significant_elements_wild$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_wild[c(1,4)], by = join_by(Tube_code == Tube_code))

difference_table_wild <- element_gift_filt_wild %>%
  select(-Tube_code) %>%
  group_by(Population) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Cold_wet-Hot_dry)%>% 
  mutate(group_color = ifelse(Difference <0, "Hot","Cold")) 
```

```{r log_fold_calc_6, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_wild <- element_gift_filt_wild %>% 
  select(-Population) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(sample_metadata_wild, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(Population, elements) %>%
  summarise(mean=mean(abundance))

log_fold_wild <- means_gift_wild %>%
  group_by(elements) %>%
  summarise(
    logfc_Cold_hot = log2(mean[Population == "Cold_wet"] / mean[Population == "Hot_dry"])
  )
```


```{r plot1_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_wild %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_wild, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_adjust), color=group_color)) + 
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
uniqueGIFT <- unique(GIFT_db[c(2,3,4,5,6)])

code_function_wild <- difference_table_wild %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_wild<-unique(code_function_wild$Code_function)

gift_colors_wild <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_wild)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_wild %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_wild, by=join_by(Elements==trait)) %>%
  left_join(log_fold_wild, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_wild, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_Cold_hot, y = -log(p_adjust), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_wild$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```
```{r final_plot_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_wild %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_wild, by=join_by(Elements==trait)) %>%
  left_join(log_fold_wild, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_wild, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_Cold_hot, y = -log(p_adjust), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_wild$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_wild %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_wild, by=join_by(Elements==trait)) %>%
  left_join(log_fold_wild, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_Cold_hot, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table_wild %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```
```{r}
difference_table_WC %>% 
  filter(group_color=="Acclimation")
```
```{r sugar_plot_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
sugar_function <- GIFT_db %>% 
  filter(Code_function=="D03") %>% 
  select(Code_element) %>% 
  pull()
```

```{r sugar_plot_8, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
poly_function <- GIFT_db %>% 
  filter(Code_function=="D02") %>% 
  select(Code_element) %>% 
  pull()
```

```{r elements_plot_6, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_names_wild <- element_gift_filt_wild%>%
  select(-Population)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_wild[c(1,4)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_wild)[2:18] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_wild, aes(x=Population, y=.data[[i]], color = Population, fill=Population)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}

```

####Butiryc acid biosynthesis

```{r commun_wilcox_elem_plot_9, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE, eval=FALSE}
lipid_function <- GIFT_db %>% 
  filter(Code_function=="D01") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(lipid_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(sample_metadata[c(1,4)], by=join_by(Tube_code==Tube_code)) %>%
  filter(Population!="NA") %>% 
  select(-Tube_code) %>% 
  pivot_longer(!Population,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=Population, y=gift, color=Population, fill=Population)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

### Comparison of both population in acclimation samples ####

```{r comunity_elem_7, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_accli<- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_accli[c(1,4)], by="Tube_code") 
```

```{r commun_wilcox_elem_7, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_accli <- element_gift_accli %>%
  pivot_longer(-c(Tube_code,Population), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ Population)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_accli <- element_gift_accli  %>% 
  dplyr::select(-c(Population))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_accli <- subset(element_gift_t_accli, trait %in% significant_elements_accli$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_accli[c(1,4)], by = join_by(Tube_code == Tube_code))

difference_table_accli <- element_gift_filt_accli %>%
  select(-Tube_code) %>%
  group_by(Population) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Cold_wet-Hot_dry)%>% 
  mutate(group_color = ifelse(Difference <0, "Hot","Cold")) 
```

```{r log_fold_calc_7, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_accli <- element_gift_filt_accli %>% 
  select(-Population) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(sample_metadata_accli, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(Population, elements) %>%
  summarise(mean=mean(abundance))

log_fold_accli <- means_gift_accli %>%
  group_by(elements) %>%
  summarise(
    logfc_Cold_hot = log2(mean[Population == "Cold_wet"] / mean[Population == "Hot_dry"])
  )
```


```{r plot1_7, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_accli %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_accli, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_adjust), color=group_color)) + 
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_7, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
uniqueGIFT <- unique(GIFT_db[c(2,3,4,5,6)])

code_function_accli <- difference_table_accli %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_accli<-unique(code_function_accli$Code_function)

gift_colors_accli <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_accli)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_accli %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_accli, by=join_by(Elements==trait)) %>%
  left_join(log_fold_accli, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_accli, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_Cold_hot, y = -log(p_adjust), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_accli$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```

```{r final_plot_7, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_accli %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_accli, by=join_by(Elements==trait)) %>%
  left_join(log_fold_accli, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_accli, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_Cold_hot, y = -log(p_adjust), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_accli$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_7, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_accli %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_accli, by=join_by(Elements==trait)) %>%
  left_join(log_fold_accli, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_Cold_hot, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_10_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table_accli %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

```{r commun_wilcox_elem_plot_7, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
element_gift_names_accli <- element_gift_filt_accli%>%
  select(-Population)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_accli[c(1,4)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_accli)[2:18] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_accli, aes(x=Population, y=.data[[i]], color = Population, fill=Population)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

### Comparison of CC and CI in post2 samples ####

```{r comunity_elem_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_post2<- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_TM6[c(1,7)], by="Tube_code") 
```

```{r commun_wilcox_elem_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_post2 <- element_gift_post2 %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_post2 <- element_gift_post2  %>% 
  dplyr::select(-c(type))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_post2 <- subset(element_gift_t_post2, trait %in% significant_elements_post2$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM6[c(1,7)], by = join_by(Tube_code == Tube_code))

difference_table_post2 <- element_gift_filt_post2 %>%
  select(-Tube_code) %>%
  group_by(type) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Control-Treatment)%>% 
  mutate(group_color = ifelse(Difference <0, "Treatment","Control")) 
```

```{r log_fold_calc_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_post2 <- element_gift_filt_post2 %>% 
  select(-type) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(sample_metadata_TM6, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(type, elements) %>%
  summarise(mean=mean(abundance))

log_fold_post2 <- means_gift_post2 %>%
  group_by(elements) %>%
  summarise(
    logfc_control_treatment = log2(mean[type == "Control"] / mean[type == "Treatment"])
  )
```


```{r plot1_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_post2 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_adjust), color=group_color)) + 
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
uniqueGIFT <- unique(GIFT_db[c(2,3,4,5,6)])

code_function_post2 <- difference_table_post2 %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_post2<-unique(code_function_post2$Code_function)

gift_colors_post2 <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_post2)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_post2 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_post2, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_control_treatment, y = -log(p_adjust), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_post2$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```

```{r final_plot_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_post2 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_post2, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_control_treatment, y = -log(p_adjust), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_post2$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_post2 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_control_treatment, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_10, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table_post2 %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

```{r commun_wilcox_elem_plot_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
element_gift_names_post2 <- element_gift_filt_post2%>%
  select(-type)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM6[c(1,7)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_post2)[2:3] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_post2, aes(x=type, y=.data[[i]], color = type, fill=type)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```