# Differential abundance analysis

```{r load_data_mag_filtdamr_diffe, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
load("data/data.Rdata")
```

## CC acclimation vs post2
### Enrichment analysis: Ancombc2

```{r zero_phylo1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type== "Control") %>% 
                    filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    #filter(genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filtercapwild, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_CC <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","6_Post-FMT2"))
physeq_sample_CC <- prune_taxa(taxa_sums(physeq_sample_CC)>0, physeq_sample_CC)
```

#### MAG level
```{r ancom_rand_capwild, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_accli_post2_CC = ancombc2(data = physeq_sample_CC, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "time_point", #fixed variable(s)
 #                 rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0.10, 
                        lib_cut = 1, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
#                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_capCC, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  right_join(ancom_result_CC, by=join_by(phylum == phylum)) %>%
  arrange(match(genome, genome_tree$tip.label)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
  unique() %>%
  arrange(phylum) %>%
  dplyr::select(colors) %>%
  pull()

ancom_result_CC <- ancom_rand_output_mag_accli_post2_CC$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_CC %>%
  mutate(significance = ifelse(p_time_point6_Post_FMT2 < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin Accli", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_CC %>%
  filter(p_time_point6_Post_FMT2 < 0.01) %>%
  ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_CC$lfc_time_point6_Post_FMT2)-2,max(ancom_result_CC$lfc_time_point6_Post_FMT2)+2) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_CC$lfc_time_point6_Post_FMT2)+1, 1), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(max(ancom_result_CC$lfc_time_point6_Post_FMT2)-1, 1), label = "Enriched\nin Accli", color="#666666") +
  theme(legend.position='right',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```

```{r ancom_table_capCC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_CC@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

colnames(ancom_rand_output_mag_accli_post2_CC$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_CC$res))

ancombc_rand_table_mag <- ancom_rand_output_mag_accli_post2_CC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_CC, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag$genome)) %>% 
ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_reorder(species, lfc_time_point6_Post_FMT2), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

## WC acclimation vs post2
### Enrichment analysis: Ancombc2

```{r zero_phylo1_WC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type== "Hot_control") %>% 
                    filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    #filter(genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filtercapWC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_WC <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","6_Post-FMT2"))
physeq_sample_WC <- prune_taxa(taxa_sums(physeq_sample_WC)>0, physeq_sample_WC)
```


#### MAG level
```{r ancom_rand_capWC, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_accli_post2_WC = ancombc2(data = physeq_sample_WC, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "time_point", #fixed variable(s)
 #                 rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0.10, 
                        lib_cut = 1, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
#                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_capWC_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
colnames(ancom_rand_output_mag_accli_post2_WC$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_WC$res))

ancom_result_WC <- ancom_rand_output_mag_accli_post2_WC$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_WC %>%
  mutate(significance = ifelse(p_time_point6_Post_FMT2 < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin Accli", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_WC %>%
  filter(p_time_point6_Post_FMT2 < 0.01) %>%
  ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_WC$lfc_time_point6_Post_FMT2)-2,max(ancom_result_WC$lfc_time_point6_Post_FMT2)+2) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_WC$lfc_time_point6_Post_FMT2)+1, 1), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(max(ancom_result_WC$lfc_time_point6_Post_FMT2)-1, 1), label = "Enriched\nin Accli", color="#666666") +
  theme(legend.position='none',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```  

```{r ancom_table_capWC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_WC@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

colnames(ancom_rand_output_mag_accli_post2_WC$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_WC$res))

ancombc_rand_table_mag_WC <- ancom_rand_output_mag_accli_post2_WC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag_WC$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_WC, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag_WC %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag_WC$genome)) %>% 
ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_reorder(species, lfc_time_point6_Post_FMT2), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

## CI acclimation vs post2
### Enrichment analysis: Ancombc2

```{r zero_phylo1_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type== "Treatment") %>% 
                    filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    #filter(genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filterCI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_CI <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","6_Post-FMT2"))
physeq_sample_CI <- prune_taxa(taxa_sums(physeq_sample_CI)>0, physeq_sample_CI)
```


#### MAG level
```{r ancom_rand_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_accli_post2_CI = ancombc2(data = physeq_sample_CI, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "time_point", #fixed variable(s)
 #                 rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0.10, 
                        lib_cut = 1, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
#                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_CI_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
colnames(ancom_rand_output_mag_accli_post2_CI$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_CI$res))

ancom_result_CI <- ancom_rand_output_mag_accli_post2_CI$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_CI %>%
  mutate(significance = ifelse(p_time_point6_Post_FMT2 < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin Accli", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_CI %>%
  filter(p_time_point6_Post_FMT2 < 0.01) %>%
  ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_CI$lfc_time_point6_Post_FMT2)-2,max(ancom_result_CI$lfc_time_point6_Post_FMT2)+2) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_CI$lfc_time_point6_Post_FMT2)+1, 1), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(max(ancom_result_CI$lfc_time_point6_Post_FMT2)-1, 1), label = "Enriched\nin Accli", color="#666666") +
  theme(legend.position='none',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```  

```{r ancom_table_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_CI@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

ancombc_rand_table_mag_CI <- ancom_rand_output_mag_accli_post2_CI$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag_CI$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag_CI %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag_CI$genome)) %>% 
ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_reorder(species, lfc_time_point6_Post_FMT2), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

## Community level plots

```{r gift_test_function_plot_1, message=FALSE, warning=FALSE, fig.height=12, fig.width=10, fig.fullwidth=TRUE, eval=TRUE}
accli_post2<-sample_metadata%>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2")
accli_post2$newID<-paste(accli_post2$type, "_", accli_post2$time_point)

GIFTs_functions_community %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(accli_post2, by = join_by(sample == Tube_code)) %>%
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>%
  select(c(1:21, 27,30)) %>%
  pivot_longer(-c(sample,type,time_point),names_to = "trait", values_to = "value") %>%
  mutate(trait = case_when(
    trait %in% GIFT_db$Code_function ~ GIFT_db$Function[match(trait, GIFT_db$Code_function)],
    TRUE ~ trait
  )) %>%
  mutate(trait=factor(trait,levels=unique(GIFT_db$Function))) %>%
  ggplot(aes(x=value, y=time_point, group=time_point, fill=type, color=type)) +
  geom_boxplot() +
  scale_color_manual(name="type",
                     breaks=c("Control","Hot_control", "Treatment"),
                     labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  scale_fill_manual(name="type",
                    breaks=c("Control","Hot_control", "Treatment"),
                    labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
                    values=c("#4477AA50","#d57d2c50","#76b18350")) +
  facet_grid(trait ~ type, space="free", scales="free") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text.y = element_text(angle = 0)) + 
  labs(y="Traits",x="Metabolic capacity index")
```

```{r data_TM6_plot_1, echo=TRUE,results=FALSE}
GIFTs_elements_community_merged<-GIFTs_elements_community %>%
    as.data.frame() %>%
    rownames_to_column(var="sample") %>%
    filter(sample!="AD69") %>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == Tube_code))%>%
    filter(time_point=="1_Acclimation"|time_point == "6_Post-FMT2")%>%
    mutate(functionid = substr(trait, 1, 3)) %>%
    mutate(trait = case_when(
      trait %in% GIFT_db$Code_element ~ GIFT_db$Element[match(trait, GIFT_db$Code_element)],
      TRUE ~ trait
    )) %>%
    mutate(functionid = case_when(
      functionid %in% GIFT_db$Code_function ~ GIFT_db$Function[match(functionid, GIFT_db$Code_function)],
      TRUE ~ functionid
    )) %>%
    mutate(trait=factor(trait,levels=unique(GIFT_db$Element))) %>%
    mutate(functionid=factor(functionid,levels=unique(GIFT_db$Function)))

# Create an interaction variable for time_point and sample
GIFTs_elements_community_merged$interaction_var <- interaction(GIFTs_elements_community_merged$sample, GIFTs_elements_community_merged$time_point)
  
ggplot(GIFTs_elements_community_merged,aes(x=interaction_var,y=trait,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(functionid ~ type, scales="free",space="free") +
        theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=5),
              strip.text.y = element_text(angle = 0)) + 
        labs(y="Traits",x="Time_point",fill="GIFT")+
  scale_x_discrete(labels = function(x) gsub(".*\\.", "", x))
```


# ANCOM-BC2 functional traits
## CC

```{r phyloseq_ele, comment="", echo=FALSE, message=FALSE, warning=FALSE}
samples_CC <- sample_metadata%>%
  column_to_rownames(var="Tube_code")%>%
  filter(type=="Control") %>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
  sample_data()

count_phy_CC <- GIFTs_elements_community %>% 
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  filter(sample %in% rownames(samples_CC)) %>%
  column_to_rownames("sample") %>% 
  select(which(!colSums(., na.rm=TRUE) %in% 0)) %>% 
  t() %>%
  otu_table(., taxa_are_rows=T)

uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

TAX <- uniqueGIFT_db%>%
  remove_rownames()%>%
  column_to_rownames(var="Code_element")%>%
  as.matrix()%>%
  tax_table()

physeq_function_CC = phyloseq(count_phy_CC, TAX, samples_CC)
```

```{r phyloseq_ele_CC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_element_CC = ancombc2(data = physeq_function_CC, 
                                                  assay_name = "counts", 
                                                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                                                  fix_formula = "time_point", #fixed variable(s)
                                                  p_adj_method = "holm", 
                                                  pseudo_sens = TRUE,
                                                  prv_cut = 0, 
                                                  lib_cut = 0, 
                                                  s0_perc = 0.05,
                                                  group = NULL, 
                                                  struc_zero = FALSE, 
                                                  neg_lb = FALSE,
                                                  alpha = 0.05, 
                                                  n_cl = 2, 
                                                  verbose = TRUE,
                                                  global = FALSE, 
                                                  pairwise = FALSE, 
                                                  dunnet = FALSE, 
                                                  trend = FALSE,
                                                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                                                  em_control = list(tol = 1e-5, max_iter = 100),
                                                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                                                  trend_control = NULL)
```

```{r table_ele1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq_function_CC@tax_table) %>%
  rownames_to_column(., "taxon")
```

```{r ancom_rand_res_elem1, echo=FALSE, comment="", message=FALSE, warning=FALSE}
colnames(ancom_rand_output_element_CC$res) <- gsub("-", "_", colnames(ancom_rand_output_element_CC$res))

ancombc_rand_table_CC <- ancom_rand_output_element_CC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., tax, by="taxon")
```

```{r ancombc_rand_plot_elem1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ancombc_rand_table_CC%>%
  #      mutate(Function=factor(Function,levels=ancombc_rand_table$Function)) %>%
  mutate(Color = ifelse(lfc_time_point6_Post_FMT2 <0, "Acclimation","Post2")) %>%
  ggplot(aes(x=forcats::fct_reorder(Function,lfc_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, fill=Color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Traits") + 
  ylab("log2FoldChange")
```

#### Functional level
```{r phylo_func1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
sample_info_tab_phyCC <- accli_post2%>%
  column_to_rownames(var="Tube_code")%>%
  filter(type=="Control") %>%
  sample_data()

count_phyCC <- GIFTs_functions_community %>% 
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  filter(sample %in% rownames(sample_info_tab_phyCC)) %>%
  column_to_rownames("sample") %>% 
  select(which(!colSums(., na.rm=TRUE) %in% 0)) %>% 
  t() %>%
  otu_table(., taxa_are_rows=T)

unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)

TAX <- unique_funct_db%>%
  remove_rownames()%>%
  column_to_rownames(var="Code_function")%>%
  as.matrix()%>%
  tax_table()

physeq_functional_filtered_CC = phyloseq(count_phyCC, TAX, sample_info_tab_phyCC)
physeq_functional_filtered_clr_CC <- microbiome::transform(physeq_functional_filtered_CC, 'clr')  
```


```{r ancom_rand_func_prewild, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_function_CC = ancombc2(data = physeq_functional_filtered_CC, 
                                               assay_name = "counts", 
                                               tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                                               fix_formula = "time_point", #fixed variable(s)
                                               p_adj_method = "holm", 
                                               pseudo_sens = TRUE,
                                               prv_cut = 0, 
                                               lib_cut = 0, 
                                               s0_perc = 0.05,
                                               group = NULL, 
                                               struc_zero = FALSE, 
                                               neg_lb = FALSE,
                                               alpha = 0.05, 
                                               n_cl = 2, 
                                               verbose = TRUE,
                                               global = FALSE, 
                                               pairwise = FALSE, 
                                               dunnet = FALSE, 
                                               trend = FALSE,
                                               iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                                               em_control = list(tol = 1e-5, max_iter = 100),
                                               mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                                               trend_control = NULL)

```

```{r table_func1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq_functional_filtered_clr_CC@tax_table) %>%
  rownames_to_column(., "taxon")

colnames(ancom_rand_output_function_CC$res) <- gsub("-", "_", colnames(ancom_rand_output_function_CC$res))

ancombc_rand_func_CC <- ancom_rand_output_function_CC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., tax, by="taxon") %>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
```

```{r ancom_rand_res_funct1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancombc_rand_table_func_CC <- ancom_rand_output_function_CC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., tax, by="taxon")
```

```{r ancombc_rand_plot_funct, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=4, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_func_CC%>%
  mutate(Color = ifelse(lfc_time_point6_Post_FMT2 <0, "Acclimation","Post2")) %>%
  ggplot(aes(x=forcats::fct_reorder(Function,lfc_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, fill=Color)) + 
  geom_col() +
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Traits") + 
  ylab("log2FoldChange")
```

## CI

```{r phyloseq_ele_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
samples_CI <- sample_metadata%>%
  column_to_rownames(var="Tube_code")%>%
  filter(type=="Treatment") %>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
  sample_data()

count_phy_CI <- GIFTs_elements_community %>% 
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  filter(sample %in% rownames(samples_CI)) %>%
  column_to_rownames("sample") %>% 
  select(which(!colSums(., na.rm=TRUE) %in% 0)) %>% 
  t() %>%
  otu_table(., taxa_are_rows=T)

uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

TAX <- uniqueGIFT_db%>%
  remove_rownames()%>%
  column_to_rownames(var="Code_element")%>%
  as.matrix()%>%
  tax_table()

physeq_function_CI = phyloseq(count_phy_CI, TAX, samples_CI)
```

```{r phyloseq_ele_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_element_CI = ancombc2(data = physeq_function_CI, 
                                                  assay_name = "counts", 
                                                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                                                  fix_formula = "time_point", #fixed variable(s)
                                                  p_adj_method = "holm", 
                                                  pseudo_sens = TRUE,
                                                  prv_cut = 0, 
                                                  lib_cut = 0, 
                                                  s0_perc = 0.05,
                                                  group = NULL, 
                                                  struc_zero = FALSE, 
                                                  neg_lb = FALSE,
                                                  alpha = 0.05, 
                                                  n_cl = 2, 
                                                  verbose = TRUE,
                                                  global = FALSE, 
                                                  pairwise = FALSE, 
                                                  dunnet = FALSE, 
                                                  trend = FALSE,
                                                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                                                  em_control = list(tol = 1e-5, max_iter = 100),
                                                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                                                  trend_control = NULL)
```

```{r table_ele1CI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq_function_CI@tax_table) %>%
  rownames_to_column(., "taxon")
```

```{r ancom_rand_res_elem1CI, echo=FALSE, comment="", message=FALSE, warning=FALSE}
colnames(ancom_rand_output_element_CI$res) <- gsub("-", "_", colnames(ancom_rand_output_element_CI$res))

ancombc_rand_table_CI <- ancom_rand_output_element_CI$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., tax, by="taxon")
```

```{r ancombc_rand_plot_elem1CI, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ancombc_rand_table_CI%>%
  #      mutate(Function=factor(Function,levels=ancombc_rand_table$Function)) %>%
  mutate(Color = ifelse(lfc_time_point6_Post_FMT2 <0, "Acclimation","Post2")) %>%
  ggplot(aes(x=forcats::fct_reorder(Function,lfc_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, fill=Color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Traits") + 
  ylab("log2FoldChange")
```

#### Functional level
```{r phylo_func1CI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
sample_info_tab_phyCI <- accli_post2%>%
  column_to_rownames(var="Tube_code")%>%
  filter(type=="Treatment") %>%
  sample_data()

count_phyCI<- GIFTs_functions_community %>% 
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  filter(sample %in% rownames(sample_info_tab_phyCI)) %>%
  column_to_rownames("sample") %>% 
  select(which(!colSums(., na.rm=TRUE) %in% 0)) %>% 
  t() %>%
  otu_table(., taxa_are_rows=T)

unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)

TAX <- unique_funct_db%>%
  remove_rownames()%>%
  column_to_rownames(var="Code_function")%>%
  as.matrix()%>%
  tax_table()

physeq_functional_filtered_CI = phyloseq(count_phyCI, TAX, sample_info_tab_phyCI)
physeq_functional_filtered_clr_CI <- microbiome::transform(physeq_functional_filtered_CI, 'clr')  
```


```{r ancom_rand_func_prewildCI, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_function_CI = ancombc2(data = physeq_functional_filtered_CI, 
                                               assay_name = "counts", 
                                               tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                                               fix_formula = "time_point", #fixed variable(s)
                                               p_adj_method = "holm", 
                                               pseudo_sens = TRUE,
                                               prv_cut = 0, 
                                               lib_cut = 0, 
                                               s0_perc = 0.05,
                                               group = NULL, 
                                               struc_zero = FALSE, 
                                               neg_lb = FALSE,
                                               alpha = 0.05, 
                                               n_cl = 2, 
                                               verbose = TRUE,
                                               global = FALSE, 
                                               pairwise = FALSE, 
                                               dunnet = FALSE, 
                                               trend = FALSE,
                                               iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                                               em_control = list(tol = 1e-5, max_iter = 100),
                                               mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                                               trend_control = NULL)

```

```{r table_func1CI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq_functional_filtered_clr_CI@tax_table) %>%
  rownames_to_column(., "taxon")

colnames(ancom_rand_output_function_CI$res) <- gsub("-", "_", colnames(ancom_rand_output_function_CI$res))

ancombc_rand_func_CI <- ancom_rand_output_function_CI$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., tax, by="taxon") %>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
```

```{r ancom_rand_res_funct1CI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancombc_rand_table_func_CI <- ancom_rand_output_function_CI$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., tax, by="taxon")
```

```{r ancombc_rand_plot_functCI, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=4, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_func_CI%>%
  mutate(Color = ifelse(lfc_time_point6_Post_FMT2 <0, "Acclimation","Post2")) %>%
  ggplot(aes(x=forcats::fct_reorder(Function,lfc_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, fill=Color)) + 
  geom_col() +
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Traits") + 
  ylab("log2FoldChange")
```

## WC

```{r phyloseq_ele_1WC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
samples_WC <- sample_metadata%>%
  column_to_rownames(var="Tube_code")%>%
  filter(type=="Hot_control") %>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
  sample_data()

count_phy_WC <- GIFTs_elements_community %>% 
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  filter(sample %in% rownames(samples_WC)) %>%
  column_to_rownames("sample") %>% 
  select(which(!colSums(., na.rm=TRUE) %in% 0)) %>% 
  t() %>%
  otu_table(., taxa_are_rows=T)

uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

TAX <- uniqueGIFT_db%>%
  remove_rownames()%>%
  column_to_rownames(var="Code_element")%>%
  as.matrix()%>%
  tax_table()

physeq_function_WC = phyloseq(count_phy_WC, TAX, samples_WC)
```

```{r phyloseq_ele_WC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_element_WC = ancombc2(data = physeq_function_WC, 
                                                  assay_name = "counts", 
                                                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                                                  fix_formula = "time_point", #fixed variable(s)
                                                  p_adj_method = "holm", 
                                                  pseudo_sens = TRUE,
                                                  prv_cut = 0, 
                                                  lib_cut = 0, 
                                                  s0_perc = 0.05,
                                                  group = NULL, 
                                                  struc_zero = FALSE, 
                                                  neg_lb = FALSE,
                                                  alpha = 0.05, 
                                                  n_cl = 2, 
                                                  verbose = TRUE,
                                                  global = FALSE, 
                                                  pairwise = FALSE, 
                                                  dunnet = FALSE, 
                                                  trend = FALSE,
                                                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                                                  em_control = list(tol = 1e-5, max_iter = 100),
                                                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                                                  trend_control = NULL)
```

```{r table_ele1WC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq_function_WC@tax_table) %>%
  rownames_to_column(., "taxon")
```

```{r ancom_rand_res_elem1WC, echo=FALSE, comment="", message=FALSE, warning=FALSE}
colnames(ancom_rand_output_element_WC$res) <- gsub("-", "_", colnames(ancom_rand_output_element_WC$res))

ancombc_rand_table_WC <- ancom_rand_output_element_WC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., tax, by="taxon")
```

```{r ancombc_rand_plot_elem1WC, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ancombc_rand_table_WC%>%
  #      mutate(Function=factor(Function,levels=ancombc_rand_table$Function)) %>%
  mutate(Color = ifelse(lfc_time_point6_Post_FMT2 <0, "Acclimation","Post2")) %>%
  ggplot(aes(x=forcats::fct_reorder(Function,lfc_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, fill=Color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Traits") + 
  ylab("log2FoldChange")
```

#### Functional level
```{r phylo_func1WC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
sample_info_tab_phyWC <- accli_post2%>%
  column_to_rownames(var="Tube_code")%>%
  filter(type=="Hot_control") %>%
  sample_data()

count_phyWC<- GIFTs_functions_community %>% 
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  filter(sample %in% rownames(sample_info_tab_phyWC)) %>%
  column_to_rownames("sample") %>% 
  select(which(!colSums(., na.rm=TRUE) %in% 0)) %>% 
  t() %>%
  otu_table(., taxa_are_rows=T)

unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)

TAX <- unique_funct_db%>%
  remove_rownames()%>%
  column_to_rownames(var="Code_function")%>%
  as.matrix()%>%
  tax_table()

physeq_functional_filtered_WC = phyloseq(count_phyWC, TAX, sample_info_tab_phyWC)
physeq_functional_filtered_clr_WC <- microbiome::transform(physeq_functional_filtered_WC, 'clr')  
```


```{r ancom_rand_func_prewildWC, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_function_WC = ancombc2(data = physeq_functional_filtered_WC, 
                                               assay_name = "counts", 
                                               tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                                               fix_formula = "time_point", #fixed variable(s)
                                               p_adj_method = "holm", 
                                               pseudo_sens = TRUE,
                                               prv_cut = 0, 
                                               lib_cut = 0, 
                                               s0_perc = 0.05,
                                               group = NULL, 
                                               struc_zero = FALSE, 
                                               neg_lb = FALSE,
                                               alpha = 0.05, 
                                               n_cl = 2, 
                                               verbose = TRUE,
                                               global = FALSE, 
                                               pairwise = FALSE, 
                                               dunnet = FALSE, 
                                               trend = FALSE,
                                               iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                                               em_control = list(tol = 1e-5, max_iter = 100),
                                               mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                                               trend_control = NULL)

```

```{r table_func1WC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq_functional_filtered_clr_WC@tax_table) %>%
  rownames_to_column(., "taxon")

colnames(ancom_rand_output_function_WC$res) <- gsub("-", "_", colnames(ancom_rand_output_function_WC$res))

ancombc_rand_func_WC <- ancom_rand_output_function_WC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., tax, by="taxon") %>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
```

```{r ancom_rand_res_funct1WC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancombc_rand_table_func_WC <- ancom_rand_output_function_WC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., tax, by="taxon")
```

```{r ancombc_rand_plot_functWC, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=4, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_func_WC%>%
  mutate(Color = ifelse(lfc_time_point6_Post_FMT2 <0, "Acclimation","Post2")) %>%
  ggplot(aes(x=forcats::fct_reorder(Function,lfc_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, fill=Color)) + 
  geom_col() +
  scale_fill_manual(values=c("#6b7398","#e5bd5b")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Traits") + 
  ylab("log2FoldChange")
```