# Faecal Microbiota Transplantation (FMT)
```{r knitr_options_FMT, include=FALSE}
knitr::opts_chunk$set(fig.width=10, fig.height=7, fig.fullwidth=TRUE,
fig.path='Figures/', warning=FALSE,
message=FALSE)
set.seed(2802)
```
```{r load_data_FMT}
load("data/data_27022025.Rdata")
load("data/calculations_28022025.Rdata")
```

## Is the GM of the intervention group more similar to warm population after one week from FMT?

### Shapiro test

```{r shapiro_fmt1, comment="", message=FALSE, warning=FALSE,}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT1") %>% 
  filter(metric=="richness") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT1") %>% 
  filter(metric=="neutral") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT1") %>% 
  filter(metric=="phylogenetic") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
```

### Alpha diversity
```{r alpha_div_plot_5, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT1") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(method="t.test",size=3, label.x=.7) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_blank(),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")
```
***p<0.05 when comparing neutral beta diversity between Cold-control and Warm-control

### Beta diversity

***Number of samples used***
```{r beta_div_neutral_treat1_samples_2, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_post1 <- sample_metadata %>%
  filter(time_point == "FMT1") %>% 
  select(Tube_code) %>% 
  pull()
subset_meta_post1 <- sample_metadata %>%
  filter(time_point == "FMT1")
length(samples_to_keep_post1)
```

***Richness***
```{r beta_richness_permanova_post1_2, warning=FALSE, comments="", message=FALSE}
richness_post1 <- as.matrix(beta_q0n$S)
richness_post1 <- as.dist(richness_post1[rownames(richness_post1) %in% samples_to_keep_post1,
               colnames(richness_post1) %in% samples_to_keep_post1])
betadisper(richness_post1, subset_meta_post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post1_1, warning=FALSE, comments="", message=FALSE}
adonis2(richness_post1 ~ type,
        data = subset_meta_post1 %>% arrange(match(Tube_code,labels(richness_post1))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_richness_permanova_post1_3, warning=FALSE, comments="", message=FALSE}
subset_meta_post1_arrange <- column_to_rownames(subset_meta_post1, "Tube_code")
subset_meta_post1_arrange<-subset_meta_post1_arrange[labels(richness_post1),]

pairwise <- pairwise.adonis(richness_post1, subset_meta_post1_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_neutral_permanova_post1_2, warning=FALSE, comments="", message=FALSE}
neutral_post1 <- as.matrix(beta_q1n$S)
neutral_post1 <- as.dist(neutral_post1[rownames(neutral_post1) %in% samples_to_keep_post1,
               colnames(neutral_post1) %in% samples_to_keep_post1])
betadisper(neutral_post1, subset_meta_post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post1_1, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_post1 ~ type,
        data = subset_meta_post1 %>% arrange(match(Tube_code,labels(neutral_post1))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_neutral_permanova_post1_3, warning=FALSE, comments="", message=FALSE}
subset_meta_post1_arrange <- column_to_rownames(subset_meta_post1, "Tube_code")
subset_meta_post1_arrange<-subset_meta_post1_arrange[labels(neutral_post1),]
pairwise <- pairwise.adonis(neutral_post1, subset_meta_post1_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_post1_2, warning=FALSE, comments="", message=FALSE}
phylo_post1 <- as.matrix(beta_q1p$S)
phylo_post1 <- as.dist(phylo_post1[rownames(phylo_post1) %in% samples_to_keep_post1,
               colnames(phylo_post1) %in% samples_to_keep_post1])
betadisper(phylo_post1, subset_meta_post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_post1_1, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_post1 ~ type,
        data = subset_meta_post1 %>% arrange(match(Tube_code,labels(phylo_post1))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_phylopermanova_post1_3, warning=FALSE, comments="", message=FALSE}
subset_meta_post1_arrange <- column_to_rownames(subset_meta_post1, "Tube_code")
subset_meta_post1_arrange<-subset_meta_post1_arrange[labels(phylo_post1),]
pairwise <- pairwise.adonis(phylo_post1, subset_meta_post1_arrange$type, perm=999)
pairwise%>%
  tt()
```

***NMDS***
```{r beta_neutral_nmds_post1, warning=FALSE, comments="", message=FALSE, results="hide"}
richness_post1 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post1, by = join_by(sample == Tube_code)) %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

neutral_post1 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post1, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

phylo_post1 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post1, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")
```

### Differentially abundant MAGs
#### Warm vs Intervention

***Structural zeros***

```{r struct_zero_wild_pop_warm, comment="", echo=FALSE, message=FALSE, warning=FALSE}
warm_samples <- sample_metadata %>% 
  filter(type == "Hot_control") %>% 
  filter(time_point == "FMT1") %>%
  dplyr::select("Tube_code") %>%
  pull()

treatment_samples <- sample_metadata %>% 
  filter(type == "Treatment") %>% 
  filter(time_point == "FMT1")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_wild <- sample_metadata%>% 
  filter(type %in% c("Hot_control", "Treatment") & time_point == "FMT1")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_wild$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_warm = all(c_across(all_of(warm_samples)) == 0)) %>% # set true if all samples in warm have zeros
  mutate(all_zeros_treatment = all(c_across(all_of(treatment_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_warm = mean(c_across(all_of(warm_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
  mutate(average_treatment = mean(c_across(all_of(treatment_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_warm == TRUE | all_zeros_treatment == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_warm & !all_zeros_treatment ~ "treatment",!all_zeros_warm & all_zeros_treatment ~ "warm_control",!all_zeros_warm & !all_zeros_treatment ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "warm_control", average_warm, average_treatment)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r}
struc <- structural_zeros %>% 
  filter(present=="warm_control")%>% 
  count(phylum, name = "Hot_control") %>%
  arrange(desc(Hot_control))
structural_zeros %>% 
  filter(present=="treatment")%>% 
  count(phylum, name = "Treatment") %>%
  arrange(desc(Treatment)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```


```{r FMT1}
physeq_FMT1 <- subset_samples(physeq_all, type %in% c("Hot_control", "Treatment") & time_point == "FMT1")
physeq_FMT1 <- prune_taxa(taxa_sums(physeq_FMT1)>0, physeq_FMT1)
```
```{r ancom_rand_pond_FMT1, comment="", message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_fmt1 = ancombc2(data = physeq_FMT1, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "type", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```


```{r, eval=FALSE}
save(ancom_rand_output_fmt1, file="data/ancom_rand_output_fmt1.RData")
```

```{r 1, eval=TRUE}
load("data/ancom_rand_output_fmt1.RData")
```


```{r structural_zeros_fmt1_1}
ancom_rand_output_fmt1$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05)
```

```{r ancom_rand_FMT1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_FMT1@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_fmt1$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05) %>%
  dplyr::arrange(p_typeTreatment) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_typeTreatment)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag%>%
      mutate(genome=factor(taxon,levels=ancombc_rand_table_mag$taxon)) %>%
ggplot(., aes(x=lfc_typeTreatment, y=forcats::fct_reorder(genome,lfc_typeTreatment), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("MAGs")+
  guides(fill=guide_legend(title="Phylum"))
```

***Phyla of the significant MAGs in warm control***
```{r sign_features_FMT1_warm, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_typeTreatment>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in intervention group***
```{r sign_features_FMT1_cold, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_typeTreatment<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

##### Intervention vs cold control

***Structural zeros***

```{r struct_zero_wild_pop_inter, comment="", echo=FALSE, message=FALSE, warning=FALSE}
cold_samples <- sample_metadata %>% 
  filter(type == "Control") %>% 
  filter(time_point == "FMT1") %>%
  dplyr::select("Tube_code") %>%
  pull()

treatment_samples <- sample_metadata %>% 
  filter(type == "Treatment") %>% 
  filter(time_point == "FMT1")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_wild <- sample_metadata%>% 
  filter(type %in% c("Control", "Treatment") & time_point == "FMT1")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_wild$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_cold = all(c_across(all_of(cold_samples)) == 0)) %>% # set true if all samples in cold have zeros
  mutate(all_zeros_treatment = all(c_across(all_of(treatment_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_cold = mean(c_across(all_of(cold_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
  mutate(average_treatment = mean(c_across(all_of(treatment_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_cold == TRUE | all_zeros_treatment == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_cold & !all_zeros_treatment ~ "treatment",!all_zeros_cold & all_zeros_treatment ~ "cold_control",!all_zeros_cold & !all_zeros_treatment ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "cold_control", average_cold, average_treatment)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r}
struc <- structural_zeros %>% 
  filter(present=="cold_control")%>% 
  count(phylum, name = "Cold_control") %>%
  arrange(desc(Cold_control))
structural_zeros %>% 
  filter(present=="treatment")%>% 
  count(phylum, name = "Treatment") %>%
  arrange(desc(Treatment)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```

```{r}
physeq_FMT1_cold <- subset_samples(physeq_all, type %in% c("Control", "Treatment") & time_point == "FMT1")
physeq_FMT1_cold <- prune_taxa(taxa_sums(physeq_FMT1_cold)>0, physeq_FMT1_cold)
```
```{r ancom_rand_pond_FMT1_cold, comment="", message=FALSE, warning=FALSE}
ancom_rand_output_cold = ancombc2(data = physeq_FMT1_cold, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "type", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r }
save(ancom_rand_output_cold, file="data/ancom_rand_output_cold.RData")
```

```{r 3, eval=TRUE}
load("data/ancom_rand_output_cold.RData")
```


```{r 3_1, eval=TRUE}
ancom_rand_output_cold$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,8,9,10) %>% 
  tt()
```

```{r ancom_rand_FMT1_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_FMT1_cold@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_cold$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05) %>%
  dplyr::arrange(p_typeTreatment) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_typeTreatment)

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT1_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag%>%
      mutate(genome=factor(taxon,levels=ancombc_rand_table_mag$taxon)) %>%
ggplot(., aes(x=lfc_typeTreatment, y=forcats::fct_reorder(genome,lfc_typeTreatment), fill=phylum)) + 
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

***Phyla of the significant MAGs in intervention group***
```{r sign_features_FMT1_int, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_typeTreatment>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in cold control***
```{r sign_features_FMT1_cold_con, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_typeTreatment<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

#### Functional differences
***MCI***
```{r gitfs_elements_tm5, warning=FALSE, comments="", message=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT1") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_functional_tm5_stats_2, warning=FALSE, comments="", message=FALSE}
MCI_tm5_ele <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point == "FMT1")

shapiro.test(MCI_tm5_ele$value)#normality test

res.aov <- aov(value ~ type, data = MCI_tm5_ele)#anova test
summary(res.aov)
```

***Functional differences***

CC and CI
```{r comunity_elem_7_1, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Control", "Treatment") & time_point == "FMT1") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames()
```

CI and WC
```{r comunity_elem_CI_WC, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Hot_control", "Treatment") &
           time_point == "FMT1") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```

CC and WC
```{r comunity_elem_CC_WC, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Hot_control", "Control") &
           time_point == "FMT1") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```

## Is the GM of the intervention group still more similar to warm population after one week from FMT?

### Shapiro test

```{r shapiro_fmt2, comment="", message=FALSE, warning=FALSE,}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT2") %>% 
  filter(metric=="richness") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT2") %>% 
  filter(metric=="neutral") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT2") %>% 
  filter(metric=="phylogenetic") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
```

### Alpha diversity
```{r alpha_div_plot_FMT2, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT2") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
        scale_x_discrete(labels = c("Control" = "Cold-Cold", "Hot_control" = "Hot-Hot", "Treatment" = "Cold-Hot")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(method="t.test",size=3, label.x=.7) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_blank(),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")

```
***p<0.05 when comparing richness and neutral beta diversity between Cold-control and Cold-intervention


### Beta diversity

***Number of samples used***
```{r beta_div_neutral_FMT2, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_FMT2 <- sample_metadata %>%
  filter(time_point == "FMT2") %>% 
  select(Tube_code) %>% 
  pull()
subset_meta_FMT2 <- sample_metadata %>%
  filter(time_point == "FMT2")
length(samples_to_keep_FMT2)
```

***Richness***
```{r beta_richness_permanova_FMT2_2, warning=FALSE, comments="", message=FALSE}
richness_FMT2 <- as.matrix(beta_q0n$S)
richness_FMT2 <- as.dist(richness_FMT2[rownames(richness_FMT2) %in% samples_to_keep_FMT2,
               colnames(richness_FMT2) %in% samples_to_keep_FMT2])
betadisper(richness_FMT2, subset_meta_FMT2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_FMT2_1, warning=FALSE, comments="", message=FALSE}
adonis2(richness_FMT2 ~ type,
        data = subset_meta_FMT2 %>% arrange(match(Tube_code,labels(richness_FMT2))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_richness_permanova_FMT2_3, warning=FALSE, comments="", message=FALSE}
subset_meta_FMT2_arrange <- column_to_rownames(subset_meta_FMT2, "Tube_code")
subset_meta_FMT2_arrange<-subset_meta_FMT2_arrange[labels(richness_FMT2),]

pairwise <- pairwise.adonis(richness_FMT2, subset_meta_FMT2_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_neutral_permanova_FMT2_2, warning=FALSE, comments="", message=FALSE}
neutral_FMT2 <- as.matrix(beta_q1n$S)
neutral_FMT2 <- as.dist(neutral_FMT2[rownames(neutral_FMT2) %in% samples_to_keep_FMT2,
               colnames(neutral_FMT2) %in% samples_to_keep_FMT2])
betadisper(neutral_FMT2, subset_meta_FMT2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_FMT2_1, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_FMT2 ~ type,
        data = subset_meta_FMT2 %>% arrange(match(Tube_code,labels(neutral_FMT2))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_neutral_permanova_FMT2_3, warning=FALSE, comments="", message=FALSE}
subset_meta_FMT2_arrange <- column_to_rownames(subset_meta_FMT2, "Tube_code")
subset_meta_FMT2_arrange<-subset_meta_FMT2_arrange[labels(neutral_FMT2),]
pairwise <- pairwise.adonis(neutral_FMT2, subset_meta_FMT2_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_FMT2_2, warning=FALSE, comments="", message=FALSE}
phylo_FMT2 <- as.matrix(beta_q1p$S)
phylo_FMT2 <- as.dist(phylo_FMT2[rownames(phylo_FMT2) %in% samples_to_keep_FMT2,
               colnames(phylo_FMT2) %in% samples_to_keep_FMT2])
betadisper(phylo_FMT2, subset_meta_FMT2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_FMT2_1, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_FMT2 ~ type,
        data = subset_meta_FMT2 %>% arrange(match(Tube_code,labels(phylo_FMT2))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```
```{r beta_phylo_permanova_FMT2_pair, warning=FALSE, comments="", message=FALSE}
subset_meta_FMT2_arrange <- column_to_rownames(subset_meta_FMT2, "Tube_code")
subset_meta_FMT2_arrange<-subset_meta_FMT2_arrange[labels(phylo_FMT2),]
pairwise <- pairwise.adonis(phylo_FMT2, subset_meta_FMT2_arrange$type, perm=999)
pairwise%>%
  tt()
```

***NMDS***
```{r beta_neutral_nmds_FMT2, warning=FALSE, comments="", message=FALSE, results="hide"}
 richness_FMT2 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_FMT2, by = join_by(sample == Tube_code)) %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

neutral_FMT2 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_FMT2, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

phylo_FMT2 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_FMT2, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")
```


### Differentially abundant MAGs
#### Warm vs Intervention
***Structural zeros***

```{r struct_zero_wild_pop, comment="", echo=FALSE, message=FALSE, warning=FALSE}
warm_samples <- sample_metadata %>% 
  filter(type == "Hot_control") %>% 
  filter(time_point == "FMT2") %>%
  dplyr::select("Tube_code") %>%
  pull()

treatment_samples <- sample_metadata %>% 
  filter(type == "Treatment") %>% 
  filter(time_point == "FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_wild <- sample_metadata%>% 
  filter(type %in% c("Hot_control", "Treatment") & time_point == "FMT2")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_wild$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_warm = all(c_across(all_of(warm_samples)) == 0)) %>% # set true if all samples in warm have zeros
  mutate(all_zeros_treatment = all(c_across(all_of(treatment_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_warm = mean(c_across(all_of(warm_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
  mutate(average_treatment = mean(c_across(all_of(treatment_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_warm == TRUE | all_zeros_treatment == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_warm & !all_zeros_treatment ~ "treatment",!all_zeros_warm & all_zeros_treatment ~ "warm_control",!all_zeros_warm & !all_zeros_treatment ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "warm_control", average_warm, average_treatment)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r}
struc <- structural_zeros %>% 
  filter(present=="warm_control")%>% 
  count(phylum, name = "Hot_control") %>%
  arrange(desc(Hot_control))
structural_zeros %>% 
  filter(present=="treatment")%>% 
  count(phylum, name = "Treatment") %>%
  arrange(desc(Treatment)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```
```{r}
physeq_FMT2 <- subset_samples(physeq_all, type %in% c("Hot_control", "Treatment") & time_point == "FMT2")
physeq_FMT2 <- prune_taxa(taxa_sums(physeq_FMT2)>0, physeq_FMT2)
```
```{r ancom_rand_pond_FMT2, comment="", message=FALSE, warning=FALSE}
ancom_rand_output_fmt2 = ancombc2(data = physeq_FMT2, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "type", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r }
save(ancom_rand_output_fmt2, file="data/ancom_rand_output_fmt2.RData")
```

```{r 5, eval=TRUE}
load("data/ancom_rand_output_fmt2.RData")
```
```{r}
ancom_rand_output_fmt2$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,7,8,9) %>% 
  tt()  
```

```{r ancom_rand_FMT2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_FMT2@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_fmt2$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05) %>%
  dplyr::arrange(p_typeTreatment) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_typeTreatment)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag%>%
      mutate(genome=factor(taxon,levels=ancombc_rand_table_mag$taxon)) %>%
ggplot(., aes(x=lfc_typeTreatment, y=forcats::fct_reorder(genome,lfc_typeTreatment), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

***Phyla of the significant MAGs in warm control***
```{r sign_features_FMT2_warm, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_typeTreatment>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in intervention group***
```{r sign_features_FMT2_cold, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_typeTreatment<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

##### Intervention vs cold control
***Structural zeros***

```{r struct_zero_wild_pop_fmt2_int, comment="", echo=FALSE, message=FALSE, warning=FALSE}
cold_samples <- sample_metadata %>% 
  filter(type == "Control") %>% 
  filter(time_point == "FMT2") %>%
  dplyr::select("Tube_code") %>%
  pull()

treatment_samples <- sample_metadata %>% 
  filter(type == "Treatment") %>% 
  filter(time_point == "FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_wild <- sample_metadata%>% 
  filter(type %in% c("Control", "Treatment") & time_point == "FMT2")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_wild$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_cold = all(c_across(all_of(cold_samples)) == 0)) %>% # set true if all samples in cold have zeros
  mutate(all_zeros_treatment = all(c_across(all_of(treatment_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_cold = mean(c_across(all_of(cold_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
  mutate(average_treatment = mean(c_across(all_of(treatment_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_cold == TRUE | all_zeros_treatment == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_cold & !all_zeros_treatment ~ "treatment",!all_zeros_cold & all_zeros_treatment ~ "cold_control",!all_zeros_cold & !all_zeros_treatment ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "cold_control", average_cold, average_treatment)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r}
struc <- structural_zeros %>% 
  filter(present=="cold_control")%>% 
  count(phylum, name = "Cold_control") %>%
  arrange(desc(Cold_control))
structural_zeros %>% 
  filter(present=="treatment")%>% 
  count(phylum, name = "Treatment") %>%
  arrange(desc(Treatment)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```

```{r}
physeq_FMT2 <- subset_samples(physeq_all, type %in% c("Control", "Treatment") & time_point == "FMT2")
physeq_FMT2 <- prune_taxa(taxa_sums(physeq_FMT2)>0, physeq_FMT2)
```
```{r ancom_rand_pond_FMT2_cold, comment="", message=FALSE, warning=FALSE}
ancom_rand_output_cold_fmt2 = ancombc2(data = physeq_FMT2, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "type", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r 8, eval=FALSE}
save(ancom_rand_output_cold_fmt2, file="data/ancom_rand_output_cold_fmt2.RData")
```

```{r 7, eval=TRUE}
load("data/ancom_rand_output_cold_fmt2.RData")

```
```{r}
ancom_rand_output_cold_fmt2$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,7,8,9) %>% 
  tt()
```

```{r ancom_rand_FMT2_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_FMT2@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_cold_fmt2$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05) %>%
  dplyr::arrange(p_typeTreatment) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_typeTreatment)

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT2_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag%>%
      mutate(genome=factor(taxon,levels=ancombc_rand_table_mag$taxon)) %>%
ggplot(., aes(x=lfc_typeTreatment, y=forcats::fct_reorder(genome,lfc_typeTreatment), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

***Phyla of the significant MAGs in intervention group***
```{r sign_features_FMT2_int, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_typeTreatment>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in cold control***
```{r sign_features_FMT2_cold_con, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_typeTreatment<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

#### Functional differences
***MCI***
```{r gitfs_elements_tm5_FMT2_cold, warning=FALSE, comments="", message=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT2") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_functional_FMT2_cold, warning=FALSE, comments="", message=FALSE}
MCI_FMT2 <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point == "FMT2")

shapiro.test(MCI_FMT2$value)

res.aov <- aov(value ~ type, data = MCI_FMT2)
summary(res.aov)
```

***Functional differences***

CC and CI
```{r comunity_elem_FMT2_cold, comment="", message=FALSE, warning=FALSE}
significant_elements<-element_gift %>%
  filter(type %in% c("Control", "Treatment") & time_point == "FMT2") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

```{r comunity_elem_fm2_1, comment="", message=FALSE, warning=FALSE}
element_gift_sig <- element_gift %>%
  select(Tube_code, all_of(intersect(
    significant_elements$trait,
    colnames(element_gift)
  ))) %>%
  left_join(., sample_metadata[c(1, 7)], by = join_by(Tube_code == Tube_code)) %>%
  filter(type!="Hot_control")

difference_table <- element_gift_sig %>%
  select(-Tube_code) %>%
  group_by(type) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Control-Treatment)%>% 
  mutate(group_color = ifelse(Difference <0, "Control","Treatment")) 
```

```{r commun_wilcox_elem_plot_fmt2, comment="", message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  scale_fill_manual(values=c("#4477AA","#76b183")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

CI and WC
```{r comunity_elem_FMT2_warm, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Hot_control", "Treatment") &
           time_point == "FMT2") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```

CC and WC
```{r comunity_elem_FMT2_control, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Hot_control", "Control") &
           time_point == "FMT2") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```