---
title: "AlberdiLab | Martin-Bideguren et al. 2024"
subtitle: "Study title to be added"
author:
  - Garazi Martin-Bideguren^[University of Copenhagen, garazi.bideguren@sund.ku.dk], Ostaizka Aizpurua^[University of Copenhagen, ostaizka.aizpurua@sund.ku.dk], Carlos Cabido^[Sociedad de Ciencias Aranzadi-Departamento de Herpetología, ccabido@aranzadi.eus] and Antton Alberdi^[University of Copenhagen, antton.alberdi@sund.ku.dk]
date: "Last update: `r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
url: https://alberdilab.github.io/Podarcis_adaptation
description: |
  Data analysis code for the study on the recovery of metagenome‑assembled genomes and derived microbial communities from lizard faecal samples.
link-citations: yes
github-repo: alberdilab/Podarcis_adaptation
---

```{r knitr_opts, echo=FALSE}
knitr::opts_chunk$set(
    class.source = "script-source",
    class.output = "script-output",
    comment = NA)
```

# Introduction

This webbook contains all the code used for data analysis in study of the individual-level metagenomic data of Podarcis muralis and Podarcis liolepis lizards from different environments during an experimental setup.

## Prepare the R environment

### Environment

To reproduce all the analyses locally, clone this repository in your computer using:

```
RStudio > New Project > Version Control > Git
```

And indicating the following git repository:

> https://github.com/alberdilab/Podarcis_adaptation.git

Once the R project has been created, follow the instructions and code chunks shown in this webbook.

### Libraries

The following R packages are required for the data analysis.

```{r load_libraries, warning=FALSE, comments="", message=FALSE}
# Base
library(R.utils)
library(knitr)
library(tidyverse)
library(devtools)
library(tinytable)
library(janitor)

# For tree handling
library(ape)
library(phyloseq)
library(phytools)

# For plotting
library(ggplot2)
library(ggrepel)
library(ggpubr)
library(ggnewscale)
library(gridExtra)
library(ggtreeExtra)
library(ggtree)
library(ggh4x)

# For statistics
library(spaa)
library(vegan)
library(Rtsne)
library(geiger)
library(hilldiv2)
library(distillR)
library(broom.mixed)
library(corrplot)
library(nlme)
library(pairwiseAdonis)
```

<!--chapter:end:index.Rmd-->

# Prepare data

## Load data

Load the original data files outputted by the bioinformatic pipeline.

### Sample metadata

```{r load_sample_metadata, warning=FALSE, comments="", message=FALSE}
sample_metadata <- read_tsv("data/metadata.tsv")
```

### Read counts

```{r load_read_counts, warning=FALSE, comments="", message=FALSE}
read_counts_raw <- read_tsv("data/genome.count.tsv") %>%
    rename(genome=1)

#Transformation of read_counts to combine data from both sequence rounds
merge_and_rename <- function(read_counts_raw) {
  read_counts_raw %>%
    # Gather the columns into long format
    pivot_longer(cols = -genome, names_to = "col") %>%
    # Extract prefix
    mutate(prefix = gsub("^(.*?)_.*", "\\1", col)) %>%
    # Group by prefix and genome, then summarize
    group_by(prefix, genome) %>%
    summarize(value = sum(value)) %>%
    # Spread the data back to wide format
    pivot_wider(names_from = prefix, values_from = value)
}

read_counts <- merge_and_rename(read_counts_raw)
```

### Genome base hits

```{r load_genome_hits, warning=FALSE, comments="", message=FALSE}
genome_hits_raw <- read_tsv("data/genome.covered_bases.tsv") %>%
    rename(genome=1)

#Transformation of genome_hits to combine data from both sequence rounds
merge_and_rename <- function(genome_hits_raw) {
  genome_hits_raw %>%
    # Gather the columns into long format
    pivot_longer(cols = -genome, names_to = "col") %>%
    # Extract prefix
    mutate(prefix = gsub("^(.*?)_.*", "\\1", col)) %>%
    # Group by prefix and genome, then summarize
    group_by(prefix, genome) %>%
    summarize(value = sum(value)) %>%
    # Spread the data back to wide format
    pivot_wider(names_from = prefix, values_from = value)
}

genome_hits <- merge_and_rename(genome_hits_raw)
```

### Genome taxonomy

```{r load_genome_taxonomy, warning=FALSE, comments="", message=FALSE}
genome_taxonomy <- read_tsv("data/gtdbtk.summary.tsv") %>%
  select(mag_id = user_genome, classification) %>%
  separate(
    classification,
    into = c("domain", "phylum", "class", "order", "family", "genus", "species"),
    sep = ";") %>%
    rename(genome=1)
```

### Genome quality

```{r load_genome_quality, warning=FALSE, comments="", message=FALSE}
genome_quality <- read_tsv("data/quality_report.tsv") %>%
  select(
    genome = Name, 
    completeness = Completeness, 
    contamination = Contamination,
    length = Genome_Size, 
    gc = GC_Content
  )
genome_quality<-genome_quality %>% 
  mutate (genome=str_remove_all(genome,".fa"))

#Filter MAGs with over 70% completeness and less than 10% contamination
genome_quality <- genome_quality %>%
  filter(completeness > 70 & contamination < 10)
```

### Genome tree

```{r load_genome_tree, warning=FALSE, comments="", message=FALSE}
genome_tree <- read_tree("data/gtdbtk.backbone.bac120.classify.tree")
genome_tree$tip.label <- str_replace_all(genome_tree$tip.label,"'", "") #remove single quotes in MAG names

#Filter genome_taxonomy to keep MAGs with over 70% completeness and less than 10% contamination
genome_taxonomy <- genome_taxonomy %>%
  semi_join(genome_quality, by = "genome")
genome_tree <- keep.tip(genome_tree, tip=genome_taxonomy$genome) # keep only MAG tips
```

### Genome annotations

```{r load_genome_annotations, warning=FALSE, comments="", message=FALSE}
genome_annotations <- read_tsv("data/annotations.tsv.xz") %>%
    rename(gene=1, genome=2, contig=3)

#Filter only the MAGs with over 70% completeness and less than 10% contamination
genome_annotations <- genome_annotations %>%
  semi_join(genome_quality, by = "genome")
```

## Create working objects

Transform the original data files into working objects for downstream analyses.

### Merge genome taxonomy and quality

```{r generate_genome_metadata, warning=FALSE, comments="", message=FALSE}
genome_metadata <- genome_taxonomy %>%
  inner_join(genome_quality,by=join_by(genome==genome)) #join quality
```

### Calculate genome coverage

```{r calculate_genome_coverage, warning=FALSE, comments="", message=FALSE}
#Filter genome_hits for the MAGs with over 70% completeness and less than 10% contamination
genome_hits <- genome_hits %>%
  semi_join(genome_quality, by = "genome")

genome_coverage <- genome_hits %>%
  mutate(across(where(is.numeric), ~ ./genome_metadata$length))
```

## Filtering

4 samples are removed from the analysis due to their low sequencing depth.

```{r filterging, echo=TRUE, warning=FALSE}
#Counts_raw
columns_to_exclude <- c("AD16","AD23", "AD25", "AD91") # Columns to exclude
read_counts <- read_counts %>%
  select(-columns_to_exclude)

#Metadata
sample_metadata <- sample_metadata %>%
  filter(Tube_code != "AD16") %>%
  filter(Tube_code != "AD23") %>%
  filter(Tube_code != "AD25") %>%
  filter(Tube_code != "AD91")

#Coverage_table
columns_to_exclude <- c("AD16", "AD23", "AD25", "AD91")  # Columns to exclude
genome_coverage <- genome_coverage %>%
  select(-columns_to_exclude)
```

```{r filterging_1, echo=TRUE, warning=FALSE}
##Removal of samples of individual LI1_2nd_6
#Counts_raw
columns_to_exclude <- c("AFV13","AD06", "AD31", "AE03", "AF12") # Columns to exclude
read_counts <- read_counts %>%
  select(-columns_to_exclude)

#Metadata
sample_metadata <- sample_metadata %>%
  filter(Tube_code != "AFV13") %>%
  filter(Tube_code != "AD06") %>%
  filter(Tube_code != "AD31") %>%
  filter(Tube_code != "AE03") %>%
  filter(Tube_code != "AF12") 

#Coverage_table
columns_to_exclude <- c("AFV13","AD06", "AD31", "AE03", "AF12")  # Columns to exclude
genome_coverage <- genome_coverage %>%
  select(-columns_to_exclude)
```


### Filter reads by coverage

```{r filter_coverage, warning=FALSE, comments="", message=FALSE}
#Filter read_counts for the MAGs with over 70% completeness and less than 10% contamination
read_counts <- read_counts %>%
  semi_join(genome_quality, by = "genome")

min_coverage=0.3
read_counts_filt <- genome_coverage %>%
  mutate(across(where(is.numeric), ~ ifelse(. > min_coverage, 1, 0))) %>%
  mutate(across(-1, ~ . * read_counts[[cur_column()]])) 
```

### Transform reads into genome counts

```{r calculate_genome_counts_unfiltered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts <- read_counts %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

```{r calculate_genome_counts_filtered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts_filt <- read_counts_filt %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

### Distill annotations into GIFTs 

```{r distill_annotations, warning=FALSE, comments="", message=FALSE, results='hide'}
genome_gifts <- distill(genome_annotations,GIFT_db,genomecol=2,annotcol=c(9,10,19))
```

## Load data statistics

### Raw reads

```{r raw_reads, warning=FALSE, comments="", message=FALSE}
raw_reads <-
  "data/multiqc_general_stats.txt" %>%
  read_tsv() %>%
  select(
    sample = Sample,
    raw_reads = `total_sequences`
  ) %>%
  mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
  group_by(sample_prefix) %>%
  summarize(raw_reads = sum(raw_reads, na.rm = TRUE)) %>%
  rename(sample = sample_prefix)  %>%
  mutate(sample = str_remove(sample, "^fastp \\|\\s*"))
```

### Quality-filtered reads

```{r filtered_reads, warning=FALSE, comments="", message=FALSE}
fastp_reads <-
  "data/multiqc_general_stats.txt" %>%
  read_tsv() %>%
  select(
    sample = Sample,
    trimmed_reads = `total_sequences`
  ) %>%
  mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
  group_by(sample_prefix) %>%
  summarize(trimmed_reads = sum(trimmed_reads, na.rm = TRUE)) %>%
  rename(sample = sample_prefix) %>% 
  filter(!str_detect(sample, "nonlizard \\|")) %>%
  filter(!str_detect(sample, "lizard \\|")) %>%
  filter(!str_detect(sample, "refseq500 \\|"))  %>%
  mutate(sample = str_remove(sample, "^fastp \\|\\s*"))
```

### Host-mapped reads

```{r host_mapped, warning=FALSE, comments="", message=FALSE}
host_mapped <-
  "data/multiqc_general_stats.txt" %>%
  read_tsv() %>%
  filter(str_detect(Sample, "lizard")) %>%
  select(
    sample = Sample,
    host_mapped = `reads_mapped`,
    mapping_total = `raw_total_sequences`
  ) %>%
  mutate(
    host_unmapped = mapping_total - host_mapped
  ) %>%
  filter(!is.na(host_mapped)) %>%
  separate(
    col = sample,
    into = c("host_name", "sample"),
    sep = " \\| "
  ) %>%
  rename(mapped = host_mapped, unmapped = host_unmapped) %>%
  select(-mapping_total) %>%
  pivot_longer(-host_name:-sample) %>%
  mutate(
    name = str_glue("{name}_{host_name}")
  ) %>%
  select(-host_name) %>%
  pivot_wider() %>%
  mutate(sample = str_extract(sample, "^[^_]+")) %>%
  group_by(sample) %>%
  summarize(
    mapped_lizard = sum(mapped_lizard),
    unmapped_lizard = sum(mapped_lizard)
  ) 
```

### Prokaryotic fraction

```{r singlem, warning=FALSE, comments="", message=FALSE}
singlem <-
  "data/singleM.tsv" %>%
  read_tsv() %>%
  distinct() %>%
  mutate(
    sample = sample %>% str_remove_all("_1$"),
   read_fraction = read_fraction %>% str_remove("%") %>% as.numeric(),
   read_fraction = read_fraction / 100
  ) %>%
  select(
   sample,
   singlem_prokaryotic_bases = bacterial_archaeal_bases,
   singlem_metagenome_size = metagenome_size,
   singlem_read_fraction = read_fraction,
  ) %>%
mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
  group_by(sample_prefix) %>%
  summarize(
    singlem_prokaryotic_bases = sum(singlem_prokaryotic_bases),
    singlem_metagenome_size = sum(singlem_metagenome_size),
    singlem_read_fraction = mean(singlem_read_fraction)
  ) %>%
  rename(sample = sample_prefix)
```

### MAG-mapped reads

```{r mag_mapping, warning=FALSE, comments="", message=FALSE}
mag_mapping <-
  read_tsv("data/contig.count.tsv.xz") %>%
  pivot_longer(-sequence_id) %>%
  summarise(value = sum(value), .by = "name") %>%
  rename(sample = name, mapped_mags = value) %>%
  mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
    group_by(sample_prefix) %>%
    summarize(
      mapped_mags = sum(mapped_mags)
    ) %>%
    rename(sample = sample_prefix)
```

### Wrap data statistics

```{r wrap_statistics, warning=FALSE, comments="", message=FALSE}
data_stats <- raw_reads %>%
  left_join(fastp_reads) %>%
  left_join(host_mapped) %>%
  left_join(singlem) %>%
  left_join(mag_mapping)

data_stats<- data_stats %>%
  filter(!str_detect(sample, "nonlizard \\|")) %>%
  filter(!str_detect(sample, "lizard \\|")) %>%
  filter(!str_detect(sample, "refseq500 \\|"))
```

## Prepare color scheme

[AlberdiLab](www.alberdilab.dk) projects use unified color schemes developed for the [Earth Hologenome Initiative](www.earthhologenome.org), to facilitate figure interpretation.

```{r get_ehi_colors, warning=FALSE, comments="", message=FALSE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(phylum, colors) %>% 
    unique() %>%
    arrange(phylum) %>%
    pull(colors, name=phylum)
```

## Wrap working objects

All working objects are wrapped into a single Rdata object to facilitate downstream usage.

```{r wrap_working_objects, warning=FALSE, comments="", message=FALSE}
save(sample_metadata, 
     genome_metadata, 
     read_counts, 
     genome_counts, 
     genome_counts_filt, 
     genome_tree,
     genome_gifts, 
     phylum_colors,
     data_stats,
     file = "data/data.Rdata")
```

<!--chapter:end:01_data_preparation.Rmd-->

# Data statistics


```{r load_data_stats}
load("data/data.Rdata")
```

## Sequencing reads statistics

```{r reads_stats}
data_stats$raw_reads %>% sum()
data_stats$raw_reads %>% mean()
data_stats$raw_reads %>% sd()
```

## DNA fractions
```{r data_fractions_stats}
#Overall
data_stats %>%
    mutate(mapped_perc=mapped_mags/trimmed_reads) %>%
    summarise(mean=mean(mapped_perc),sd=sd(mapped_perc))
```


```{r data_fractions_plot, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
data_stats %>%
  mutate(
    low_quality = raw_reads - trimmed_reads,
    unmapped_reads = trimmed_reads - mapped_lizard - mapped_mags
  ) %>%
  select(sample, low_quality, mapped_lizard, mapped_mags, unmapped_reads) %>%
  pivot_longer(-sample) %>%
  mutate(name=factor(name,levels=c("low_quality","mapped_lizard","unmapped_reads","mapped_mags"))) %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(!is.na(time_point)) %>%
  ggplot(aes(x = sample, y = value, fill = name)) +
      geom_bar(stat = "identity", position = "fill") +
      scale_fill_manual(name="Sequence type",
                    breaks=c("low_quality","mapped_lizard","unmapped_reads","mapped_mags"),
                    labels=c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                    values=c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c"))+
      facet_grid(~time_point, scales = "free") +
      theme_minimal() +
      theme(axis.text.x = element_text(angle = 90, hjust = 1, size=4)) +
      labs(y="DNA sequence fraction",x="Samples")
```

## Recovered microbial fraction

```{r data_estimations_plot, message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
data_stats %>%
  mutate(
    unmapped_reads = trimmed_reads - mapped_lizard - mapped_mags,
    mag_proportion = mapped_mags / (mapped_mags + unmapped_reads),
    singlem_read_fraction = singlem_read_fraction
  ) %>%
  select(sample, mag_proportion, singlem_read_fraction) %>%
  mutate(
    mag_proportion = if_else(singlem_read_fraction == 0, 0, mag_proportion),
    singlem_read_fraction = if_else(singlem_read_fraction == 0, NA, singlem_read_fraction),
    singlem_read_fraction = if_else(singlem_read_fraction < mag_proportion, NA, singlem_read_fraction),
    singlem_read_fraction = if_else(singlem_read_fraction > 1, 1, singlem_read_fraction)
  ) %>%
  pivot_longer(-sample, names_to = "proportion", values_to = "value") %>%
  mutate(
    proportion = factor(
      proportion,
      levels = c("mag_proportion", "singlem_read_fraction")
    )
  ) %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(!is.na(time_point)) %>%
  ggplot(aes(x = sample, y = value, color = proportion)) +
      geom_line(aes(group = sample), color = "#f8a538") +
      geom_point() +
      scale_color_manual(name="Proportion",
                    breaks=c("mag_proportion","singlem_read_fraction"),
                    labels=c("Recovered","Estimated"),
                    values=c("#52e1e8", "#876b53"))+
      #facet_wrap(tissue~., scales = "free", labeller = labeller(tissue = c(cloaca="Cloaca",feces="Feces"))) +
      theme_minimal() +
      facet_grid(~time_point, scales = "free", space="free")+
      labs(y = "Samples", x = "Prokaryotic fraction") +
      scale_y_continuous(limits = c(0, 1)) +
      theme(
        axis.text.y = element_text(size = 4),
        axis.text.x = element_text( angle = 90, vjust = 0.5, hjust = 1, size = 6),
        legend.position = "right"
      )
```

### Domain-adjusted mapping rate (DAMR)

```{r damr}
data_stats %>%
  mutate(
    unmapped_reads = trimmed_reads - mapped_lizard - mapped_mags,
    mag_proportion = mapped_mags / (mapped_mags + unmapped_reads),
    singlem_read_fraction = singlem_read_fraction
  ) %>%
  mutate(damr=pmin(1, mag_proportion/singlem_read_fraction)) %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(!is.na(time_point)) %>%
  select(sample,damr, time_point, species, type) %>%
  #group_by(type) %>%
  #summarise(mean=mean(damr),sd=sd(damr)) %>%
  tt()
```

<!--chapter:end:02_data_statistics.Rmd-->

# MAG catalogue

```{r load_data_mag}
load("data/data.Rdata")
```

## Genome phylogeny

```{r genome_phylogeny, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
# Generate the phylum color heatmap
phylum_heatmap <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(genome,phylum) %>%
    mutate(phylum = factor(phylum, levels = unique(phylum))) %>%
    column_to_rownames(var = "genome")

# Generate  basal tree
circular_tree <- force.ultrametric(genome_tree, method="extend") %>% # extend to ultrametric for the sake of visualisation
    ggtree(., layout="fan", open.angle=10, size=0.5)

# Add phylum ring
circular_tree <- gheatmap(circular_tree, phylum_heatmap, offset=0.55, width=0.1, colnames=FALSE) +
        scale_fill_manual(values=phylum_colors) +
        geom_tiplab2(size=1, hjust=-0.1) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))

# Flush color scale to enable a new color scheme in the next ring
circular_tree <- circular_tree + new_scale_fill()

# Add completeness ring
circular_tree <- circular_tree +
        new_scale_fill() +
        scale_fill_gradient(low = "#d1f4ba", high = "#f4baba") +
        geom_fruit(
                data=genome_metadata,
                geom=geom_bar,
                mapping = aes(x=completeness, y=genome, fill=contamination),
                offset = 0.55,
                orientation="y",
              stat="identity")

# Add genome-size ring
circular_tree <-  circular_tree +
        new_scale_fill() +
        scale_fill_manual(values = "#cccccc") +
        geom_fruit(
             data=genome_metadata,
             geom=geom_bar,
             mapping = aes(x=length, y=genome),
                 offset = 0.05,
                 orientation="y",
         stat="identity")

# Add text
circular_tree <-  circular_tree +
        annotate('text', x=3.4, y=0, label='              Phylum', family='ArialMT', size=3.5) +
        annotate('text', x=4.2, y=0, label='                         Genome quality', family='ArialMT', size=3.5) +
        annotate('text', x=4.7, y=0, label='                      Genome size', family='ArialMT', size=3.5)

#Plot circular tree
circular_tree %>% open_tree(30) %>% rotate_tree(90)
```

## Genome quality

```{r genome_quality}
genome_metadata$completeness %>% mean()
genome_metadata$completeness %>% sd()
genome_metadata$contamination %>% mean()
genome_metadata$contamination %>% sd()
```

```{r genome_quality_plot, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}

#Generate quality biplot
genome_biplot <- genome_metadata %>%
  select(c(genome,domain,phylum,completeness,contamination,length)) %>%
  arrange(match(genome, rev(genome_tree$tip.label))) %>% #sort MAGs according to phylogenetic tree
  ggplot(aes(x=completeness,y=contamination,size=length,color=phylum)) +
              geom_point(alpha=0.7) +
                    ylim(c(10,0)) +
                    scale_color_manual(values=phylum_colors) +
                    labs(y= "Contamination", x = "Completeness") +
                    theme_classic() +
                    theme(legend.position = "none")

#Generate contamination boxplot
genome_contamination <- genome_metadata %>%
            ggplot(aes(y=contamination)) +
                    ylim(c(10,0)) +
                    geom_boxplot(colour = "#999999", fill="#cccccc") +
                    theme_void() +
                    theme(legend.position = "none",
                        axis.title.x = element_blank(),
                        axis.title.y = element_blank(),
                        axis.text.y=element_blank(),
                        axis.ticks.y=element_blank(),
                        axis.text.x=element_blank(),
                        axis.ticks.x=element_blank(),
                        plot.margin = unit(c(0, 0, 0.40, 0),"inches")) #add bottom-margin (top, right, bottom, left)

#Generate completeness boxplot
genome_completeness <- genome_metadata %>%
        ggplot(aes(x=completeness)) +
                xlim(c(50,100)) +
                geom_boxplot(colour = "#999999", fill="#cccccc") +
                theme_void() +
                theme(legend.position = "none",
                    axis.title.x = element_blank(),
                    axis.title.y = element_blank(),
                    axis.text.y=element_blank(),
                    axis.ticks.y=element_blank(),
                    axis.text.x=element_blank(),
                    axis.ticks.x=element_blank(),
                    plot.margin = unit(c(0, 0, 0, 0.50),"inches")) #add left-margin (top, right, bottom, left)

#Render composite figure
grid.arrange(grobs = list(genome_completeness,genome_biplot,genome_contamination),
        layout_matrix = rbind(c(1,1,1,1,1,1,1,1,1,1,1,4),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3)))

```


## Functional overview

```{r function_heatmap, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}

# Aggregate basal GIFT into elements
function_table <- genome_gifts %>%
    to.elements(., GIFT_db)

# Generate  basal tree
function_tree <- force.ultrametric(genome_tree, method="extend") %>%
                ggtree(., size = 0.3) 

#Add phylum colors next to the tree tips
function_tree <- gheatmap(function_tree, phylum_heatmap, offset=0, width=0.1, colnames=FALSE) +
            scale_fill_manual(values=phylum_colors) +
            labs(fill="Phylum")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

#Add functions heatmap
function_tree <- gheatmap(function_tree, function_table, offset=0.5, width=3.5, colnames=FALSE) +
            vexpand(.08) +
            coord_cartesian(clip = "off") +
            scale_fill_gradient(low = "#f4f4f4", high = "steelblue", na.value="white") +
            labs(fill="GIFT")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

# Add completeness barplots
function_tree <- function_tree +
            geom_fruit(data=genome_metadata,
            geom=geom_bar,
            grid.params=list(axis="x", text.size=2, nbreak = 1),
            axis.params=list(vline=TRUE),
            mapping = aes(x=length, y=genome, fill=completeness),
                 offset = 3.8,
                 orientation="y",
                 stat="identity") +
            scale_fill_gradient(low = "#cf8888", high = "#a2cc87") +
            labs(fill="Genome\ncompleteness")

function_tree
```

## Functional ordination

```{r function_ordination, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
# Generate the tSNE ordination
tSNE_function <- Rtsne(X=function_table, dims = 2, check_duplicates = FALSE)

# Plot the ordination
function_ordination <- tSNE_function$Y %>%
                as.data.frame() %>%
                mutate(genome=rownames(function_table)) %>%
                inner_join(genome_metadata, by="genome") %>%
                rename(tSNE1="V1", tSNE2="V2") %>%
                select(genome,phylum,tSNE1,tSNE2, length) %>%
                ggplot(aes(x = tSNE1, y = tSNE2, color = phylum, size=length))+
                            geom_point(shape=16, alpha=0.7) +
                            scale_color_manual(values=phylum_colors) +
                            theme_minimal() +
                labs(color="Phylum", size="Genome size") +
                guides(color = guide_legend(override.aes = list(size = 5))) # enlarge Phylum dots in legend

function_ordination
```

<!--chapter:end:03_mag_catalogue.Rmd-->

# Community composition

```{r load_data_community}
load("data/data.Rdata")
```

## Taxonomy overview 

### Stacked barplot

```{r taxonomy_barplot, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
# Merge data frames based on sample
merged_data<-genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>% #append sample metadata
  filter(count > 0) #filter 0 counts

# Create an interaction variable for time_point and sample
merged_data$interaction_var <- interaction(merged_data$sample, merged_data$individual)

ggplot(merged_data, aes(x=interaction_var,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~ time_point + type ,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
          axis.title.x = element_blank(),
          panel.background = element_blank(),
          panel.border = element_blank(),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black")) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")+
  scale_x_discrete(labels = function(x) gsub(".*\\.", "", x))
```


### Phylum relative abundances
```{r taxonomy_phylum_summary, warning=FALSE, comments="", message=FALSE}
phylum_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,phylum) %>%
  summarise(relabun=sum(count))

phylum_summary %>%
    group_by(phylum) %>%
    summarise(mean=mean(relabun, na.rm=TRUE),sd=sd(relabun, na.rm=TRUE)) %>%
    arrange(-mean) %>%
    tt()
```

```{r taxonomy_boxplot_phylum, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_arrange <- phylum_summary %>%
    group_by(phylum) %>%
    summarise(mean=mean(relabun)) %>%
    arrange(-mean) %>%
    select(phylum) %>%
    pull()

phylum_summary %>%
    filter(phylum %in% phylum_arrange) %>%
    mutate(phylum=factor(phylum,levels=rev(phylum_arrange))) %>%
    ggplot(aes(x=relabun, y=phylum, group=phylum, color=phylum)) +
        scale_color_manual(values=phylum_colors[rev(phylum_arrange)]) +
        geom_jitter(alpha=0.5) + 
        theme_minimal() + 
        theme(legend.position="none") +
        labs(y="Phylum",x="Relative abundance")
```

## Taxonomy boxplot

### Family
```{r taxonomy_family_summary, warning=FALSE, comments="", message=FALSE}
family_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>% #append sample metadata
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,family) %>%
  summarise(relabun=sum(count))

family_summary %>%
    group_by(family) %>%
    summarise(mean=mean(relabun, na.rm=TRUE),sd=sd(relabun, na.rm=TRUE)) %>%
    arrange(-mean) %>%
    tt()
```


```{r taxonomy_jitterplot_family, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
family_arrange <- family_summary %>%
    group_by(family) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(family) %>%
    pull()

family_summary %>%
    left_join(genome_metadata %>% select(family,phylum) %>% unique(),by=join_by(family==family)) %>%
    left_join(sample_metadata,by=join_by(sample==Tube_code)) %>%
    filter(family %in% family_arrange[1:20]) %>%
    mutate(family=factor(family,levels=rev(family_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=family, group=family, color=phylum)) +
        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~type)+
        theme_minimal() + 
        labs(y="Family", x="Relative abundance", color="Phylum")

```

### Genus
```{r taxonomy_genus_summary, warning=FALSE, comments="", message=FALSE}
genus_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>% #append sample metadata
  left_join(genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,genus) %>%
  summarise(relabun=sum(count)) %>%
  filter(genus != "g__")

genus_summary %>%
    group_by(genus) %>%
    summarise(mean=mean(relabun, na.rm=TRUE),sd=sd(relabun, na.rm=TRUE)) %>%
    arrange(-mean) %>%
    tt()
```

```{r taxonomy_jitterplot_genus, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genus_arrange <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=sum(relabun)) %>%
    filter(genus != "g__")%>%
    arrange(-mean) %>%
    select(genus) %>%
    mutate(genus= sub("^g__", "", genus)) %>%
    pull()

genus_summary %>%
    left_join(genome_metadata %>% select(genus,phylum) %>% unique(),by=join_by(genus==genus)) %>%
    left_join(sample_metadata,by=join_by(sample==Tube_code)) %>%
    mutate(genus= sub("^g__", "", genus)) %>%
    filter(genus %in% genus_arrange[1:20]) %>%
    mutate(genus=factor(genus,levels=rev(genus_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=genus, group=genus, color=phylum)) +
        scale_color_manual(values=phylum_colors[-c(3,4,6,8)]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~type)+
        theme_minimal() + 
        labs(y="Family", x="Relative abundance", color="Phylum")

```

# Diversity analysis

## Alpha diversity

```{r alpha_div, comment="", message=FALSE, warning=FALSE}
# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")

# Aggregate basal GIFT into elements
dist <- genome_gifts %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

functional <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, dist = dist) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(functional = 1) %>%
  rownames_to_column(var = "sample") %>%
  mutate(functional = if_else(is.nan(functional), 1, functional))

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample)) %>%
  full_join(functional, by = join_by(sample == sample))
```

### Wild samples

```{r alpha_div_plot_0, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="0_Wild") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = species, group=species, color=species, fill=species)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="species",
          breaks=c("Podarcis_liolepis","Podarcis_muralis"),
          labels=c("Podarcis liolepis","Podarcis muralis"),
          values=c("#e5bd5b", "#6b7398")) +
      scale_fill_manual(name="species",
          breaks=c("Podarcis_liolepis","Podarcis_muralis"),
          labels=c("Podarcis liolepis","Podarcis muralis"),
          values=c("#e5bd5b50", "#6b739850")) +
      facet_wrap(. ~ metric ) +
      coord_cartesian(xlim = c(1, NA)) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      )
```

### Acclimation samples

```{r alpha_div_plot_1, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="1_Acclimation") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = species, group=species, color=species, fill=species)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="species",
          breaks=c("Podarcis_liolepis","Podarcis_muralis"),
          labels=c("Podarcis liolepis","Podarcis muralis"),
          values=c("#e5bd5b", "#6b7398")) +
      scale_fill_manual(name="species",
          breaks=c("Podarcis_liolepis","Podarcis_muralis"),
          labels=c("Podarcis liolepis","Podarcis muralis"),
          values=c("#e5bd5b50", "#6b739850")) +
      facet_wrap(. ~ metric) +
      coord_cartesian(xlim = c(1, NA)) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      )
```

### Antibiotics samples

```{r alpha_div_plot_2, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="2_Antibiotics") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = species, group=species, color=species, fill=species)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="species",
          breaks=c("Podarcis_liolepis","Podarcis_muralis"),
          labels=c("Podarcis liolepis","Podarcis muralis"),
          values=c("#e5bd5b", "#6b7398")) +
      scale_fill_manual(name="species",
          breaks=c("Podarcis_liolepis","Podarcis_muralis"),
          labels=c("Podarcis liolepis","Podarcis muralis"),
          values=c("#e5bd5b50", "#6b739850")) +
      facet_wrap(. ~ metric) +
      coord_cartesian(xlim = c(1, NA)) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      )
```

### Transplant_1 samples

```{r alpha_div_plot_3, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="3_Transplant1") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
      facet_wrap(. ~ metric) +
      coord_cartesian(xlim = c(1, NA)) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      )
```

### Transplant_2 samples

```{r alpha_div_plot_4, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="4_Transplant2") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
      facet_wrap(. ~ metric) +
      coord_cartesian(xlim = c(1, NA)) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      )
```

### Post-Transplant_1 samples

```{r alpha_div_plot_5, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="5_Post-FMT1") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
      facet_wrap(. ~ metric) +
      coord_cartesian(xlim = c(1, NA)) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      )
```

### Post-Transplant_2 samples

```{r alpha_div_plot_6, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
      facet_wrap(. ~ metric) +
      coord_cartesian(xlim = c(1, NA)) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      )
```

## Beta diversity

```{r beta_div, comment="", message=FALSE, warning=FALSE, eval=FALSE}
beta_q0n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1)

beta_q1p <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1, tree = genome_tree)

beta_q1f <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1, dist = dist)
```

<!--chapter:end:04_community_composition.Rmd-->

## Permanovas

#### Load required data

```{r loaddata_2}
meta <- column_to_rownames(sample_metadata, "Tube_code")
```

### 1. Are the wild populations similar?

#### Wild: P.muralis vs P.liolepis

```{r beta_div_neutral_wild_, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
wild <- meta %>%
  filter(time_point == "0_Wild")

# Create a temporary modified version of genome_counts_filt
temp_genome_counts <- transform(genome_counts_filt, row.names = genome_counts_filt$genome)
temp_genome_counts$genome <- NULL

wild.counts <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(wild))]
identical(sort(colnames(wild.counts)), sort(as.character(rownames(wild))))

wild_nmds <- sample_metadata %>%
  filter(time_point == "0_Wild")
```

#### Number of samples used

```{r beta_div_neutral_wild_samples, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(wild)
```

```{r beta_div_neutral_wild_1, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
beta_div_richness_wild<-hillpair(data=wild.counts, q=0)
beta_div_neutral_wild<-hillpair(data=wild.counts, q=1)
beta_div_phylo_wild<-hillpair(data=wild.counts, q=1, tree=genome_tree)
beta_div_func_wild<-hillpair(data=wild.counts, q=1, dist=dist)
```

#### Richness

```{r beta_richness_permanova_1, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_wild$S, wild$species) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova, echo=FALSE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_richness_wild$S ~ species, data=wild[labels(beta_div_richness_wild$S),], permutations=999) %>%
			as.matrix() %>%
			kable()
```


#### Neutral

```{r beta_neutral_permanova_1, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_wild$S, wild$species) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova, echo=FALSE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_neutral_wild$S ~ species, data=wild[labels(beta_div_neutral_wild$S),], permutations=999) %>%
			as.matrix() %>%
			kable()
```

#### Phylogenetic

```{r beta_phylo_permanova_1, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_wild$S, wild$species) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova, echo=FALSE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_phylo_wild$S ~ species, data=wild[labels(beta_div_phylo_wild$S),], permutations=999) %>%
			as.matrix() %>%
			kable()
```

#### Functional

```{r beta_func_permanova_!, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_wild$S, wild$species) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova, echo=FALSE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_func_wild$S ~ species, data=wild[labels(beta_div_func_wild$S),], permutations=999) %>%
			as.matrix() %>%
			kable()
```


```{r beta_neutral_nmds, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_q0n_nmds_wild <- beta_div_richness_wild$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
				vegan::scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(wild_nmds, by = join_by(sample == Tube_code))

beta_q1n_nmds_wild <- beta_div_neutral_wild$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
				vegan::scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(wild_nmds, by = join_by(sample == Tube_code))

beta_q1p_nmds_wild <- beta_div_phylo_wild$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				vegan::scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(wild_nmds, by = join_by(sample == Tube_code))

beta_q1f_nmds_wild <- beta_div_func_wild$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				vegan::scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(wild_nmds, by = join_by(sample == Tube_code))
```

```{r beta_neutral_nmds_plot, echo=FALSE, results=TRUE}
p0<-beta_q0n_nmds_wild %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population)) +
				scale_color_manual(name="Population",
                    breaks=c("Cold_wet","Hot_dry"),
                    labels=c("Cold","Hot"),
                    values=c('#008080',"#d57d2c")) +
        scale_shape_manual(name="Type",
                           breaks=c("Control", "Hot_control", "Treatment"),
                           labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                         values=c("circle","square","triangle")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_q1n_nmds_wild %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=type)) +
				scale_color_manual(name="Population",
                    breaks=c("Cold_wet","Hot_dry"),
                    labels=c("Cold","Hot"),
                    values=c('#008080',"#d57d2c")) +
        scale_shape_manual(name="Type",
                           breaks=c("Control", "Hot_control", "Treatment"),
                           labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                         values=c("circle","square","triangle")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p2<-beta_q1p_nmds_wild %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population)) +
				scale_color_manual(name="Population",
                    breaks=c("Cold_wet","Hot_dry"),
                    labels=c("Cold","Hot"),
                    values=c('#008080',"#d57d2c")) +
        scale_shape_manual(name="Type",
                           breaks=c("Control", "Hot_control", "Treatment"),
                           labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                         values=c("circle","square","triangle")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (),x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none") 

p3<-beta_q1f_nmds_wild %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, label=sample)) +
				scale_color_manual(name="Population",
                    breaks=c("Cold_wet","Hot_dry"),
                    labels=c("Cold","Hot"),
                    values=c('#008080',"#d57d2c")) +
        scale_shape_manual(name="Type",
                           breaks=c("Control", "Hot_control", "Treatment"),
                           labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                         values=c("circle","square","triangle")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic() +
				theme(legend.position="none") +
				guides(color=guide_legend(title="Species"))
```

```{r beta_neutral_nmds_plot_final, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```


### 2. Do the antibiotics work?

#### Acclimation vs antibiotics

```{r beta_div_neutral_treat, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}
treat <- meta  %>% #antibiotics samples giving error when plotting
  filter(time_point == "1_Acclimation" | time_point == "2_Antibiotics")

temp_genome_counts <- transform(genome_counts_filt, row.names = genome_counts_filt$genome)
temp_genome_counts$genome <- NULL

treat.counts <- temp_genome_counts[,which(colnames(temp_genome_counts) %in% rownames(treat))]
identical(sort(colnames(treat.counts)),sort(as.character(rownames(treat))))

treat_nmds <- sample_metadata %>%
  filter(time_point == "1_Acclimation" | time_point == "2_Antibiotics")
```

#### Number of samples used

```{r beta_div_neutral_treat_samples, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(treat)
```

```{r beta_div_neutral_treat_1, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}
beta_div_richness_treat<-hillpair(data=treat.counts, q=0)
beta_div_neutral_treat<-hillpair(data=treat.counts, q=1)
beta_div_phylo_treat<-hillpair(data=treat.counts, q=1, tree=genome_tree)
beta_div_func_treat<-hillpair(data=treat.counts, q=1, dist=dist)
```

##### Richness

```{r beta_div_richness_treat_5, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_treat$S, treat$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_richness_treat_4, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_treat$S ~ time_point+species+individual, data=treat[labels(beta_div_neutral_treat$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

##### Neutral

```{r beta_div_neutral_treat_5, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_treat$S, treat$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_neutral_treat_4, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_treat$S ~ time_point+species+individual, data=treat[labels(beta_div_neutral_treat$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```


##### Phylogenetic

```{r beta_div_phylo_treat_5, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_treat$S, treat$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_phylo_treat_4, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_treat$S ~ time_point+species+individual, data=treat[labels(beta_div_phylo_treat$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

##### Functional

```{r beta_div_func_treat_5, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_treat$S, treat$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_func_treat_4, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_treat$S ~ time_point+species+individual, data=treat[labels(beta_div_func_treat$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```


```{r beta_neutral_nmds_treat, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_treat <- beta_div_richness_treat$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat_nmds, by = c("sample" = "Tube_code"))

beta_neutral_nmds_treat <- beta_div_neutral_treat$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat_nmds, by = c("sample" = "Tube_code"))

beta_phylo_nmds_treat <- beta_div_phylo_treat$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat_nmds, by = join_by(sample == Tube_code))

beta_func_nmds_treat <- beta_div_func_treat$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat_nmds, by = join_by(sample == Tube_code))
```

```{r beta_neutral_nmds_treat_plot, echo=FALSE, results=FALSE}
p0<-beta_richness_nmds_treat %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_neutral_nmds_treat %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")
  
p2<-beta_phylo_nmds_treat %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p3<-beta_func_nmds_treat %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic()+
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_treat_final, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```


### 3. Do the FMT work?

#### Comparison between FMT2 vs Post-FMT2

```{r beta_div_neutral_transplant3, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}

#Create newID to identify duplicated samples
transplants_metadata<-sample_metadata%>%
  mutate(Tube_code=str_remove_all(Tube_code, "_a"))
transplants_metadata$newID <- paste(transplants_metadata$Tube_code, "_", transplants_metadata$individual)

transplant3<-transplants_metadata%>%
  filter(time_point == "4_Transplant2" | time_point == "6_Post-FMT2")%>%
  column_to_rownames("newID")

transplant3_nmds <- transplants_metadata %>%
  filter(time_point == "4_Transplant2" | time_point == "6_Post-FMT2")

full_counts<-temp_genome_counts %>%
    t()%>%
    as.data.frame()%>%
    rownames_to_column("Tube_code")%>%
    full_join(transplants_metadata,by = join_by(Tube_code == Tube_code))

transplant3_counts<-full_counts %>%
  filter(time_point == "4_Transplant2" | time_point == "6_Post-FMT2") %>%
  subset(select=-c(315:324)) %>%
  column_to_rownames("newID")%>%
  subset(select=-c(1))%>%
  t() %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)

identical(sort(colnames(transplant3_counts)),sort(as.character(rownames(transplant3))))
```

#### Number of samples used

```{r beta_div_neutral_treat_samples_1, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(transplant3)
```


```{r beta_div_neutral_transplant3_1, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}
beta_div_richness_transplant3<-hillpair(data=transplant3_counts, q=0)
beta_div_neutral_transplant3<-hillpair(data=transplant3_counts, q=1)
beta_div_phylo_transplant3<-hillpair(data=transplant3_counts, q=1, tree=genome_tree)
beta_div_func_transplant3<-hillpair(data=transplant3_counts, q=1, dist=dist)

#Arrange of metadata dataframe
transplant3_arrange<-transplant3[labels(beta_div_neutral_transplant3$S),]
```

##### Richness

```{r beta_richness_permanova_transplant3_4, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_transplant3$S ~ species+time_point+type+individual, data=transplant3[labels(beta_div_richness_transplant3$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_richness_permanova_transplant3_5, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_richness_transplant3$S,transplant3_arrange$type, perm=999)
pairwise
```

##### Neutral

```{r beta_neutral_permanova_transplant3_4, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_transplant3$S ~ species+time_point+type+individual, data=transplant3[labels(beta_div_neutral_transplant3$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_neutral_permanova_transplant3_5, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_neutral_transplant3$S,transplant3_arrange$type, perm=999)
pairwise
```

##### Phylogenetic

```{r beta_phylo_permanova_transplant3_4, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_transplant3$S ~ species+time_point+type+individual, data=transplant3[labels(beta_div_phylo_transplant3$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_phylo_permanova_transplant3_5, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_phylo_transplant3$S,transplant3_arrange$type, perm=999)
pairwise
```

##### Functional

```{r beta_func_permanova_transplant3_4, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_transplant3$S ~ species+time_point+type+individual, data=transplant3[labels(beta_div_func_transplant3$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_func_permanova_transplant3_5, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_func_transplant3$S,transplant3_arrange$type, perm=999)
pairwise
```

```{r beta_neutral_nmds_transplant3, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_transplant3 <- beta_div_richness_transplant3$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(transplant3_nmds, by = join_by(sample == newID))

beta_neutral_nmds_transplant3 <- beta_div_neutral_transplant3$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(transplant3_nmds, by = join_by(sample == newID))

beta_phylo_nmds_transplant3 <- beta_div_phylo_transplant3$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(transplant3_nmds, by = join_by(sample == newID))

beta_func_nmds_transplant3 <- beta_div_func_transplant3$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(transplant3_nmds, by = join_by(sample == newID))
```

```{r beta_neutral_nmds_transplant3_plot, eho=FALSE, results=FALSE}
p0<-beta_richness_nmds_transplant3 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_neutral_nmds_transplant3 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")
  
p2<-beta_phylo_nmds_transplant3 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p3<-beta_func_nmds_transplant3 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic()+
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_transplant3_final, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```


### 4. Are there differences between the control and the treatment group?

#### After 1 week --> Post-FMT1

```{r beta_div_phylo_post1, echo=TRUE, results=FALSE}
post1 <- meta %>%
  filter(time_point == "5_Post-FMT1")

post1.counts <- temp_genome_counts[,which(colnames(temp_genome_counts) %in% rownames(post1))]
identical(sort(colnames(post1.counts)),sort(as.character(rownames(post1))))

post1_nmds <- sample_metadata %>%
  filter(time_point == "5_Post-FMT1")
```

#### Number of samples used

```{r beta_div_neutral_treat1_samples_2, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(post1)
```

```{r beta_div_phylo_post1_1, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
beta_div_richness_post1<-hillpair(data=post1.counts, q=0)
beta_div_neutral_post1<-hillpair(data=post1.counts, q=1)
beta_div_phylo_post1<-hillpair(data=post1.counts, q=1, tree=genome_tree)
beta_div_func_post1<-hillpair(data=post1.counts, q=1, dist=dist)

#Arrange of metadata dataframe
post1_arrange<-post1[labels(beta_div_neutral_post1$S),]
```

##### Richness

```{r beta_richness_permanova_post1_2, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_post1$S, post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post1_1, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_post1$S ~ species+type, data=post1[labels(beta_div_richness_post1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_richness_permanova_post1_3, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_richness_post1$S,post1_arrange$type, perm=999)
pairwise
```

##### Neutral

```{r beta_neutral_permanova_post1_2, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_post1$S, post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post1_1, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_post1$S ~ species+type, data=post1[labels(beta_div_neutral_post1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_neutral_permanova_post1_3, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_neutral_post1$S,post1_arrange$type, perm=999)
pairwise
```

##### Phylogenetic

```{r beta_phylo_permanova_post1_2, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_post1$S, post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_post1_1, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_post1$S ~ species+type, data=post1[labels(beta_div_phylo_post1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_phylo_permanova_post1_3, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_phylo_post1$S,post1_arrange$type, perm=999)
pairwise
```

##### Functional

```{r beta_func_permanova_post1_2, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_post1$S, post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova_post1_1, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_post1$S ~ species+type, data=post1[labels(beta_div_func_post1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_func_permanova_post1_3, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_func_post1$S,post1_arrange$type, perm=999)
pairwise
```


```{r beta_neutral_nmds_post1, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post1 <- beta_div_richness_post1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post1_nmds, by = join_by(sample == Tube_code))

beta_neutral_nmds_post1 <- beta_div_neutral_post1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post1_nmds, by = join_by(sample == Tube_code))

beta_phylogenetic_nmds_post1 <- beta_div_phylo_post1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post1_nmds, by = join_by(sample == Tube_code))

beta_functional_nmds_post1 <- beta_div_func_post1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post1_nmds, by = join_by(sample == Tube_code))
```

```{r beta_neutral_nmds_post1_plot, eho=FALSE, results=FALSE}
p0<-beta_richness_nmds_post1 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_neutral_nmds_post1 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")
  
p2<-beta_phylogenetic_nmds_post1 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p3<-beta_functional_nmds_post1 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic()+
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post1_final, fig.height=7, eho=FALSE, results=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```


#### After 2 weeks -->Post-FMT2

```{r beta_div_phylo_post2, echo=TRUE, results=FALSE}
post2 <- meta %>%
  filter(time_point == "6_Post-FMT2")

post2.counts <- temp_genome_counts[,which(colnames(temp_genome_counts) %in% rownames(post2))]
identical(sort(colnames(post2.counts)),sort(as.character(rownames(post2))))

post2_nmds <- sample_metadata %>%
  filter(time_point == "6_Post-FMT2")
```

#### Number of samples used

```{r beta_div_neutral_treat_samples_2, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(post2)
```

```{r beta_div_phylo_post2_1, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
beta_div_richness_post2<-hillpair(data=post2.counts, q=0)
beta_div_neutral_post2<-hillpair(data=post2.counts, q=1)
beta_div_phylo_post2<-hillpair(data=post2.counts, q=1, tree=genome_tree)
beta_div_func_post2<-hillpair(data=post2.counts, q=1, dist=dist)

#Arrange of metadata dataframe
post2_arrange<-post2[labels(beta_div_neutral_post2$S),]
```

##### Richness

```{r beta_richness_permanova_post2_2, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_post2$S, post2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post2_1, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_post2$S ~ type, data=post2[labels(beta_div_richness_post2$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_richness_permanova_post2_3, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_richness_post2$S,post2_arrange$type, perm=999)
pairwise
```

##### Neutral

```{r beta_neutral_permanova_post2_2, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_post2$S, post2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post2_1, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_post2$S ~ type, data=post2[labels(beta_div_neutral_post2$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_neutral_permanova_post2_3, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_neutral_post2$S,post2_arrange$type, perm=999)
pairwise
```

##### Phylogenetic

```{r beta_phylo_permanova_post2_3, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_post2$S, post2$type) %>% permutest(., pairwise = TRUE)
```


```{r beta_phylo_permanova_post2_1, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_post2$S ~ type, data=post2[labels(beta_div_phylo_post2$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_phylo_permanova_post2_2, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_phylo_post2$S,post2_arrange$type, perm=999)
pairwise
```

##### Functional

```{r beta_func_permanova_post2_2, echo=FALSE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_post2$S, post2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova_post2_1, echo=FALSE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_post2$S ~ type, data=post2[labels(beta_div_func_post2$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_func_permanova_post2_3, echo=FALSE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_func_post2$S,post2_arrange$type, perm=999)
pairwise
```


```{r beta_neutral_nmds_post2, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post2 <- beta_div_richness_post2$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post2_nmds, by = join_by(sample == Tube_code))

beta_neutral_nmds_post2 <- beta_div_neutral_post2$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post2_nmds, by = join_by(sample == Tube_code))

beta_phylogenetic_nmds_post2 <- beta_div_phylo_post2$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post2_nmds, by = join_by(sample == Tube_code))

beta_functional_nmds_post2 <- beta_div_func_post2$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post2_nmds, by = join_by(sample == Tube_code))
```


```{r beta_neutral_nmds_post2_plot, eho=FALSE, results=FALSE}
p0<-beta_richness_nmds_post2 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_neutral_nmds_post2 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")
  
p2<-beta_phylogenetic_nmds_post2 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p3<-beta_functional_nmds_post2 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic()+
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post2_final, fig.height=7, eho=FALSE, results=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

<!--chapter:end:05_community_composition_comparisons.Rmd-->

# Functional differences

```{r load_data_functional}
load("data/data.Rdata")
```

## Data preparation

```{r gift_analyses, warning=FALSE}
# Aggregate bundle-level GIFTs into the compound level
GIFTs_elements <- to.elements(genome_gifts, GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts$genome, ]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>%
  select_if(~ !is.numeric(.) || sum(.) != 0)

elements <- GIFTs_elements_filtered %>%
  as.data.frame()

# Aggregate element-level GIFTs into the function level
GIFTs_functions <- to.functions(GIFTs_elements_filtered, GIFT_db)
functions <- GIFTs_functions %>%
  as.data.frame()

# Aggregate function-level GIFTs into overall Biosynthesis, Degradation and Structural GIFTs
GIFTs_domains <- to.domains(GIFTs_functions, GIFT_db)
domains <- GIFTs_domains %>%
  as.data.frame()

# Get community-weighed average GIFTs per sample
GIFTs_elements_community <- to.community(GIFTs_elements_filtered, genome_counts_filt %>% column_to_rownames(., "genome") %>% tss(), GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions, genome_counts_filt %>% column_to_rownames(., "genome") %>% tss(), GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains, genome_counts_filt %>% column_to_rownames(., "genome") %>% tss(), GIFT_db)

uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)
```

## Genomes GIFT profiles

```{r gift_function_heatmap, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
GIFTs_elements %>%
  as_tibble(., rownames = "MAG") %>%
  reshape2::melt() %>%
  rename(Code_element = variable, GIFT = value) %>%
  inner_join(GIFT_db,by="Code_element") %>%
  ggplot(., aes(x=Code_element, y=MAG, fill=GIFT, group=Code_function))+
    geom_tile()+
    scale_y_discrete(guide = guide_axis(check.overlap = TRUE))+
    scale_x_discrete(guide = guide_axis(check.overlap = TRUE))+
    scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
    facet_grid(. ~ Code_function, scales = "free", space = "free")+
    theme_grey(base_size=8)+
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),strip.text.x = element_text(angle = 90))

```

## Function level

```{r gift_function_heatmap_1, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
GIFTs_functions_community %>%
    as.data.frame() %>%
    rownames_to_column(var="sample") %>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
    ggplot(aes(x=trait,y=time_point,fill=gift)) +
        geom_tile(colour="white", size=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(type ~ ., scales="free",space="free")
```

## Element level

```{r gift_element_heatmap, comment="", message=FALSE, warning=FALSE, fig.height=16, fig.width=10, fig.fullwidth=TRUE}
GIFTs_elements_community_merged<-GIFTs_elements_community %>%
    as.data.frame() %>%
    rownames_to_column(var="sample") %>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == Tube_code))%>%
    mutate(functionid = substr(trait, 1, 3)) %>%
    mutate(trait = case_when(
      trait %in% GIFT_db$Code_element ~ GIFT_db$Element[match(trait, GIFT_db$Code_element)],
      TRUE ~ trait
    )) %>%
    mutate(functionid = case_when(
      functionid %in% GIFT_db$Code_function ~ GIFT_db$Function[match(functionid, GIFT_db$Code_function)],
      TRUE ~ functionid
    )) %>%
    mutate(trait=factor(trait,levels=unique(GIFT_db$Element))) %>%
    mutate(functionid=factor(functionid,levels=unique(GIFT_db$Function)))

# Create an interaction variable for time_point and sample
GIFTs_elements_community_merged$interaction_var <- interaction(GIFTs_elements_community_merged$sample, GIFTs_elements_community_merged$time_point)
  
ggplot(GIFTs_elements_community_merged,aes(x=interaction_var,y=trait,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(functionid ~ type, scales="free",space="free") +
        theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
              strip.text.y = element_text(angle = 0)) + 
        labs(y="Traits",x="Time_point",fill="GIFT")+
  scale_x_discrete(labels = function(x) gsub(".*\\.", "", x))
```

## Comparison of samples from the 0 Time_point (0_Wild)

### GIFTs Functional community

```{r gitfs_functional_wild, echo=TRUE,results=TRUE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="0_Wild") %>%
  group_by(species) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

#### GIFT test visualisation

```{r gift_test_function_plot_wild, message=FALSE, warning=FALSE, fig.height=12, fig.width=10, fig.fullwidth=TRUE, eval=TRUE}
GIFTs_functions_community %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="0_Wild") %>%
  select(c(1:21, 27)) %>%
  pivot_longer(-c(sample,type),names_to = "trait", values_to = "value") %>%
  mutate(trait = case_when(
      trait %in% GIFT_db$Code_function ~ GIFT_db$Function[match(trait, GIFT_db$Code_function)],
      TRUE ~ trait
    )) %>%
  mutate(trait=factor(trait,levels=unique(GIFT_db$Function))) %>%
  ggplot(aes(x=value, y=type, group=type, fill=type, color=type)) +
    geom_boxplot() +
    scale_color_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA","#d57d2c","#76b183")) +
      scale_fill_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
    facet_grid(trait ~ ., space="free", scales="free") +
              theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
              strip.text.y = element_text(angle = 0)) + 
        labs(y="Traits",x="Metabolic capacity index")
```

### GIFTs Domain community

```{r gitfs_domain_wild, echo=TRUE,results=TRUE}
GIFTs_domains_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="0_Wild") %>%
  group_by(species) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

### GIFTs Elements community

```{r gitfs_elements_wild, echo=TRUE,results=TRUE}
GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="0_Wild") %>%
  group_by(species) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_filtering_4, echo=TRUE,results=FALSE}
sample_metadata_wild <- sample_metadata%>% 
  filter(time_point == "0_Wild")

element_gift_wild <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(., sample_metadata_wild[c(1,3)], by="Tube_code")
```

```{r gitfs_filtering_5, echo=TRUE,results=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_wild, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_wild[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
filtered_data <- element_gift_wild[, !numeric_cols | colnames(element_gift_wild) %in% nonzero_numeric_cols]
```

```{r gitfs_filtering_6, echo=TRUE,results=FALSE}
significant_elements_wild <- filtered_data %>%
  pivot_longer(-c(Tube_code,species), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ species, exact=FALSE)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift_wild  %>% 
  dplyr::select(-c(species))  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements_wild$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_wild[c(1,3)], by = join_by(Tube_code == Tube_code))

element_gift_filt %>%
  dplyr::select(-Tube_code)%>%
  group_by(species)  %>%
  summarise(across(everything(), mean))%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))

element_gift_names <- element_gift_filt%>%
  dplyr::select(-species)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  dplyr::select(-Elements)%>%
  dplyr::select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_wild[c(1,3)], by = join_by(Tube_code == Tube_code))
```

```{r gitfs_plot_1, echo=TRUE,results=FALSE, warning=FALSE}
colNames <- names(element_gift_names)[2:30] #always check names(element_gift_names) first to know were your traits finish
for(i in colNames){
  plt <- ggplot(element_gift_names, aes(x=species, y=.data[[i]], color = species)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

## Comparison of samples from the 6th Time_point (6_Post-FMT2)

### GIFTs Functional community

```{r gitfs_functional_tm6, echo=TRUE,results=TRUE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

#### GIFT test visualisation

```{r gift_test_function_plot, message=FALSE, warning=FALSE, fig.height=12, fig.width=10, fig.fullwidth=TRUE, eval=TRUE}
GIFTs_functions_community %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2") %>%
  select(c(1:21, 27)) %>%
  pivot_longer(-c(sample,type),names_to = "trait", values_to = "value") %>%
  mutate(trait = case_when(
      trait %in% GIFT_db$Code_function ~ GIFT_db$Function[match(trait, GIFT_db$Code_function)],
      TRUE ~ trait
    )) %>%
  mutate(trait=factor(trait,levels=unique(GIFT_db$Function))) %>%
  ggplot(aes(x=value, y=type, group=type, fill=type, color=type)) +
    geom_boxplot() +
    scale_color_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA","#d57d2c","#76b183")) +
      scale_fill_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
    facet_grid(trait ~ ., space="free", scales="free") +
              theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
              strip.text.y = element_text(angle = 0)) + 
        labs(y="Traits",x="Metabolic capacity index")
```

### GIFTs Domain community

```{r gitfs_domain_tm6, echo=TRUE,results=TRUE}
GIFTs_domains_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

### GIFTs Elements community

```{r gitfs_elements_tm6, echo=TRUE,results=TRUE}
GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r data_TM6, echo=TRUE,results=FALSE}
sample_metadata_TM6 <- sample_metadata%>% 
  filter(time_point == "6_Post-FMT2")%>% 
  filter(type != "Hot_control")

element_gift_TM6 <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_TM6 %>% select(1, 7), by = "Tube_code")
```

```{r gitfs_filtering_2, echo=TRUE,results=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_TM6, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_TM6[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
filtered_data <- element_gift_TM6[, !numeric_cols | colnames(element_gift_TM6) %in% nonzero_numeric_cols]
```

```{r gitfs_filtering_3, echo=TRUE,results=FALSE}
significant_elements_TM6 <- filtered_data %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type, exact=FALSE)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_value < 0.05)  %>% #take into account that p_value is used and not p_adjust
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift_TM6  %>% 
  dplyr::select(-c(type))  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements_TM6$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM6[c(1,7)], by = join_by(Tube_code == Tube_code))

element_gift_filt %>%
  dplyr::select(-Tube_code)%>%
  group_by(type)  %>%
  summarise(across(everything(), mean))%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))

element_gift_names <- element_gift_filt%>%
  dplyr::select(-type)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  dplyr::select(-Elements)%>%
  dplyr::select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM6[c(1,7)], by = join_by(Tube_code == Tube_code))
```

```{r gitfs_plot, echo=TRUE,results=FALSE, warning=FALSE}
colNames <- names(element_gift_names)[2:20] #always check names(element_gift_names) first to now were your traits finish
for(i in colNames){
  plt <- ggplot(element_gift_names, aes(x=type, y=.data[[i]], color = type)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

## Domain level

### Comparison of samples from the 0 Time_point (0_Wild)

```{r gitfs_merge_1, echo=TRUE,results=FALSE}
#Merge the functional domains with the metadata
merge_gift_wild<- GIFTs_domains_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(., sample_metadata_wild, by="Tube_code")
```


```{r biosynthesis_plot_1, echo=TRUE, results=TRUE}
#Biosynthesis
p1 <-merge_gift_wild %>%
  ggplot(aes(x=species,y=Biosynthesis,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Degradation
p2 <-merge_gift_wild %>%
  ggplot(aes(x=species,y=Degradation,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.45, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Structure
p3 <-merge_gift_wild %>%
  ggplot(aes(x=species,y=Structure,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Overall
p4 <-merge_gift_wild %>%
  ggplot(aes(x=species,y=Overall,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")
```

```{r final_plot_1, echo=FALSE, results=TRUE,fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(p1, p2,p3, p4, ncol = 2))
```


### Comparison of samples from the 6th Time_point (6_Post-FMT2)

```{r gitfs_merge, echo=TRUE,results=FALSE}
#Merge the functional domains with the metadata
merge_gift_TM6 <- GIFTs_domains_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(., sample_metadata_TM6, by="Tube_code")
```

```{r biosynthesis_plot, echo=TRUE, results=TRUE}
#Biosynthesis
p1 <-merge_gift_TM6 %>%
  ggplot(aes(x=type,y=Biosynthesis,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Type")

#Degradation
p2 <-merge_gift_TM6 %>%
  ggplot(aes(x=type,y=Degradation,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "type")

#Structure
p3 <-merge_gift_TM6 %>%
  ggplot(aes(x=type,y=Structure,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "type")

#Overall
p4 <-merge_gift_TM6 %>%
  ggplot(aes(x=type,y=Overall,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "type")
```

```{r final_plot, echo=FALSE, results=TRUE,fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(p1, p2,p3, p4, ncol = 2))
```



<!--chapter:end:06_functional_differences.Rmd-->

