---
title: "AlberdiLab | Martin-Bideguren et al. 2024"
subtitle: "Study title to be added"
author:
  - Garazi Martin-Bideguren^[University of Copenhagen, garazi.bideguren@sund.ku.dk], Ostaizka Aizpurua^[University of Copenhagen, ostaizka.aizpurua@sund.ku.dk], Carlos Cabido^[Sociedad de Ciencias Aranzadi-Departamento de Herpetología, ccabido@aranzadi.eus] and Antton Alberdi^[University of Copenhagen, antton.alberdi@sund.ku.dk]
date: "Last update: `r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
url: https://alberdilab.github.io/Podarcis_adaptation
description: |
  Data analysis code for the study on the recovery of metagenome‑assembled genomes and derived microbial communities from lizard faecal samples.
link-citations: yes
github-repo: alberdilab/Podarcis_adaptation
---

```{r knitr_opts, echo=FALSE}
knitr::opts_chunk$set(
    class.source = "script-source",
    class.output = "script-output",
    comment = NA)
```

# Introduction

This webbook contains all the code used for data analysis in study of the individual-level metagenomic data of Podarcis muralis and Podarcis liolepis lizards from different environments during an experimental setup.

## Prepare the R environment

### Environment

To reproduce all the analyses locally, clone this repository in your computer using:

```
RStudio > New Project > Version Control > Git
```

And indicating the following git repository:

> https://github.com/alberdilab/Podarcis_adaptation.git

Once the R project has been created, follow the instructions and code chunks shown in this webbook.

### Libraries

The following R packages are required for the data analysis.

```{r load_libraries, warning=FALSE, comments="", message=FALSE}
# Base
library(R.utils)
library(knitr)
library(tidyverse)
library(devtools)
library(tinytable)
library(janitor)
library(readxl)

# For tree handling
library(ape)
library(phyloseq)
library(phytools)

# For plotting
library(ggplot2)
library(ggrepel)
library(ggpubr)
library(ggnewscale)
library(gridExtra)
library(ggtreeExtra)
library(ggtree)
library(ggh4x)
library(rstatix)
library(ggpmisc)

# For statistics
library(spaa)
library(vegan)
library(Rtsne)
library(geiger)
library(hilldiv2)
library(distillR)
library(broom.mixed)
library(corrplot)
library(nlme)
library(pairwiseAdonis)
library(ANCOMBC)
```

<!--chapter:end:index.Rmd-->

# Prepare data

## Load data

Load the original data files outputted by the bioinformatic pipeline.

### Sample metadata

```{r load_sample_metadata, warning=FALSE, comments="", message=FALSE}
sample_metadata <- read_tsv("data/metadata.tsv")
```

### Read counts

```{r load_read_counts, warning=FALSE, comments="", message=FALSE}
read_counts_raw <- read_tsv("data/genome.count.tsv") %>%
    rename(genome=1)

#Transformation of read_counts to combine data from both sequence rounds
merge_and_rename <- function(read_counts_raw) {
  read_counts_raw %>%
    # Gather the columns into long format
    pivot_longer(cols = -genome, names_to = "col") %>%
    # Extract prefix
    mutate(prefix = gsub("^(.*?)_.*", "\\1", col)) %>%
    # Group by prefix and genome, then summarize
    group_by(prefix, genome) %>%
    summarize(value = sum(value)) %>%
    # Spread the data back to wide format
    pivot_wider(names_from = prefix, values_from = value)
}

read_counts <- merge_and_rename(read_counts_raw)
```

### Genome base hits

```{r load_genome_hits, warning=FALSE, comments="", message=FALSE}
genome_hits_raw <- read_tsv("data/genome.covered_bases.tsv") %>%
    rename(genome=1)

#Transformation of genome_hits to combine data from both sequence rounds
merge_and_rename <- function(genome_hits_raw) {
  genome_hits_raw %>%
    # Gather the columns into long format
    pivot_longer(cols = -genome, names_to = "col") %>%
    # Extract prefix
    mutate(prefix = gsub("^(.*?)_.*", "\\1", col)) %>%
    # Group by prefix and genome, then summarize
    group_by(prefix, genome) %>%
    summarize(value = sum(value)) %>%
    # Spread the data back to wide format
    pivot_wider(names_from = prefix, values_from = value)
}

genome_hits <- merge_and_rename(genome_hits_raw)
```

### Genome taxonomy

```{r load_genome_taxonomy, warning=FALSE, comments="", message=FALSE}
genome_taxonomy <- read_tsv("data/gtdbtk.summary.tsv") %>%
  select(mag_id = user_genome, classification) %>%
  separate(
    classification,
    into = c("domain", "phylum", "class", "order", "family", "genus", "species"),
    sep = ";") %>%
    rename(genome=1)
```

### Genome quality

```{r load_genome_quality, warning=FALSE, comments="", message=FALSE}
genome_quality <- read_tsv("data/quality_report.tsv") %>%
  select(
    genome = Name, 
    completeness = Completeness, 
    contamination = Contamination,
    length = Genome_Size, 
    gc = GC_Content
  )
genome_quality<-genome_quality %>% 
  mutate (genome=str_remove_all(genome,".fa"))

#Filter MAGs with over 70% completeness and less than 10% contamination
genome_quality <- genome_quality %>%
  filter(completeness > 70 & contamination < 10)
```

### Genome tree

```{r load_genome_tree, warning=FALSE, comments="", message=FALSE}
genome_tree <- read_tree("data/gtdbtk.backbone.bac120.classify.tree")
genome_tree$tip.label <- str_replace_all(genome_tree$tip.label,"'", "") #remove single quotes in MAG names

#Filter genome_taxonomy to keep MAGs with over 70% completeness and less than 10% contamination
genome_taxonomy <- genome_taxonomy %>%
  semi_join(genome_quality, by = "genome")
genome_tree <- keep.tip(genome_tree, tip=genome_taxonomy$genome) # keep only MAG tips
```

### Genome annotations

```{r load_genome_annotations, warning=FALSE, comments="", message=FALSE}
genome_annotations <- read_tsv("data/annotations.tsv.xz") %>%
    rename(gene=1, genome=2, contig=3)

#Filter only the MAGs with over 70% completeness and less than 10% contamination
genome_annotations <- genome_annotations %>%
  semi_join(genome_quality, by = "genome")
```

## Create working objects

Transform the original data files into working objects for downstream analyses.

### Merge genome taxonomy and quality

```{r generate_genome_metadata, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_metadata <- genome_taxonomy %>%
  inner_join(genome_quality,by=join_by(genome==genome)) #join quality
```

### Calculate genome coverage

```{r calculate_genome_coverage, warning=FALSE, comments="", message=FALSE}
#Filter genome_hits for the MAGs with over 70% completeness and less than 10% contamination
genome_hits <- genome_hits %>%
  semi_join(genome_quality, by = "genome")

genome_coverage <- genome_hits %>%
  mutate(across(where(is.numeric), ~ ./genome_metadata$length))
```

## Filtering

4 samples are removed from the analysis due to their low sequencing depth.

```{r filterging, echo=TRUE, warning=FALSE}
#Counts_raw
columns_to_exclude <- c("AD16","AD23", "AD25", "AD91") # Columns to exclude
read_counts <- read_counts %>%
  select(-columns_to_exclude)

#Metadata
sample_metadata <- sample_metadata %>%
  filter(Tube_code != "AD16") %>%
  filter(Tube_code != "AD23") %>%
  filter(Tube_code != "AD25") %>%
  filter(Tube_code != "AD91")

#Coverage_table
columns_to_exclude <- c("AD16", "AD23", "AD25", "AD91")  # Columns to exclude
genome_coverage <- genome_coverage %>%
  select(-columns_to_exclude)
```

```{r filterging_1, echo=TRUE, warning=FALSE}
##Removal of samples of individual LI1_2nd_6
#Counts_raw
columns_to_exclude <- c("AFV13","AD06", "AD31", "AE03", "AF12") # Columns to exclude
read_counts <- read_counts %>%
  select(-columns_to_exclude)

#Metadata
sample_metadata <- sample_metadata %>%
  filter(Tube_code != "AFV13") %>%
  filter(Tube_code != "AD06") %>%
  filter(Tube_code != "AD31") %>%
  filter(Tube_code != "AE03") %>%
  filter(Tube_code != "AF12") 

#Coverage_table
columns_to_exclude <- c("AFV13","AD06", "AD31", "AE03", "AF12")  # Columns to exclude
genome_coverage <- genome_coverage %>%
  select(-columns_to_exclude)
```


### Filter reads by coverage

```{r filter_coverage, warning=FALSE, comments="", message=FALSE}
#Filter read_counts for the MAGs with over 70% completeness and less than 10% contamination
read_counts <- read_counts %>%
  semi_join(genome_quality, by = "genome")

min_coverage=0.3
read_counts_filt <- genome_coverage %>%
  mutate(across(where(is.numeric), ~ ifelse(. > min_coverage, 1, 0))) %>%
  mutate(across(-1, ~ . * read_counts[[cur_column()]])) 
```

### Transform reads into genome counts

```{r calculate_genome_counts_unfiltered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts <- read_counts %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

```{r calculate_genome_counts_filtered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts_filt <- read_counts_filt %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

### Distill annotations into GIFTs 

```{r distill_annotations, comment="", message=FALSE, warning=FALSE, eval=FALSE}
genome_gifts <- distill(genome_annotations,GIFT_db3,genomecol=2,annotcol=c(9,10,19))
```

## Load data statistics

### Raw reads

```{r raw_reads, warning=FALSE, comments="", message=FALSE}
raw_reads <-
  "data/multiqc_general_stats_2.txt" %>%
  read_tsv() %>%
  select(
    sample = Sample,
    raw_reads = `total_sequences`
  ) %>%
  mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
  group_by(sample_prefix) %>%
  summarize(raw_reads = sum(raw_reads, na.rm = TRUE)) %>%
  rename(sample = sample_prefix)  %>%
  mutate(sample = str_remove(sample, "^fastp \\|\\s*"))
```

### Quality-filtered reads

```{r filtered_reads, warning=FALSE, comments="", message=FALSE}
fastp_reads <-
  "data/multiqc_general_stats.txt" %>%
  read_tsv() %>%
  select(
    sample = Sample,
    trimmed_reads = `total_sequences`
  ) %>%
  mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
  group_by(sample_prefix) %>%
  summarize(trimmed_reads = sum(trimmed_reads, na.rm = TRUE)) %>%
  rename(sample = sample_prefix) %>% 
  filter(!str_detect(sample, "nonlizard \\|")) %>%
  filter(!str_detect(sample, "lizard \\|")) %>%
  filter(!str_detect(sample, "refseq500 \\|"))  %>%
  mutate(sample = str_remove(sample, "^fastp \\|\\s*"))
```

### Host-mapped reads

```{r host_mapped, warning=FALSE, comments="", message=FALSE}
host_mapped <-
  "data/multiqc_general_stats.txt" %>%
  read_tsv() %>%
  filter(str_detect(Sample, "lizard")) %>%
  select(
    sample = Sample,
    host_mapped = `reads_mapped`,
    mapping_total = `raw_total_sequences`
  ) %>%
  mutate(
    host_unmapped = mapping_total - host_mapped
  ) %>%
  filter(!is.na(host_mapped)) %>%
  separate(
    col = sample,
    into = c("host_name", "sample"),
    sep = " \\| "
  ) %>%
  rename(mapped = host_mapped, unmapped = host_unmapped) %>%
  select(-mapping_total) %>%
  pivot_longer(-host_name:-sample) %>%
  mutate(
    name = str_glue("{name}_{host_name}")
  ) %>%
  select(-host_name) %>%
  pivot_wider() %>%
  mutate(sample = str_extract(sample, "^[^_]+")) %>%
  group_by(sample) %>%
  summarize(
    mapped_lizard = sum(mapped_lizard),
    unmapped_lizard = sum(unmapped_lizard)
  ) 
```

### Prokaryotic fraction

```{r singlem, warning=FALSE, comments="", message=FALSE}
singlem <-
  "data/singleM.tsv" %>%
  read_tsv() %>%
  distinct() %>%
  mutate(
    sample = sample %>% str_remove_all("_1$"),
   read_fraction = read_fraction %>% str_remove("%") %>% as.numeric(),
   read_fraction = read_fraction / 100
  ) %>%
  select(
   sample,
   singlem_prokaryotic_bases = bacterial_archaeal_bases,
   singlem_metagenome_size = metagenome_size,
   singlem_read_fraction = read_fraction,
  ) %>%
mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
  group_by(sample_prefix) %>%
  summarize(
    singlem_prokaryotic_bases = sum(singlem_prokaryotic_bases),
    singlem_metagenome_size = sum(singlem_metagenome_size),
    singlem_read_fraction = mean(singlem_read_fraction)
  ) %>%
  rename(sample = sample_prefix)
```

### MAG-mapped reads

```{r mag_mapping, warning=FALSE, comments="", message=FALSE}
mag_mapping <-
  read_tsv("data/contig.count.tsv.xz") %>%
  pivot_longer(-sequence_id) %>%
  summarise(value = sum(value), .by = "name") %>%
  rename(sample = name, mapped_mags = value) %>%
  mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
    group_by(sample_prefix) %>%
    summarize(
      mapped_mags = sum(mapped_mags)
    ) %>%
    rename(sample = sample_prefix)
```

### Wrap data statistics

```{r wrap_statistics, warning=FALSE, comments="", message=FALSE}
data_stats <- raw_reads %>%
  left_join(fastp_reads) %>%
  left_join(host_mapped) %>%
  left_join(singlem) %>%
  left_join(mag_mapping)

data_stats<- data_stats %>%
  filter(!str_detect(sample, "nonlizard \\|")) %>%
  filter(!str_detect(sample, "lizard \\|")) %>%
  filter(!str_detect(sample, "refseq500 \\|"))
```

## Prepare color scheme

[AlberdiLab](www.alberdilab.dk) projects use unified color schemes developed for the [Earth Hologenome Initiative](www.earthhologenome.org), to facilitate figure interpretation.

```{r get_ehi_colors, warning=FALSE, comments="", message=FALSE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(phylum, colors) %>% 
    unique() %>%
    arrange(phylum) %>%
    pull(colors, name=phylum)
```

## Wrap working objects

All working objects are wrapped into a single Rdata object to facilitate downstream usage.

```{r wrap_working_objects, warning=FALSE, comments="", message=FALSE}
save(sample_metadata, 
     genome_metadata, 
     read_counts, 
     genome_counts, 
     genome_counts_filt, 
     genome_tree,
     genome_gifts, 
     phylum_colors,
     data_stats,
     file = "data/data.Rdata")
```

<!--chapter:end:01_data_preparation.Rmd-->

# Data statistics

```{r load_data_stats}
load("data/data.Rdata")
```

## Sequencing reads statistics

```{r reads_stats}
data_stats$raw_reads %>% sum()
data_stats$raw_reads %>% mean()
data_stats$raw_reads %>% sd()
```

## DNA fractions
```{r data_fractions_stats}
#Overall
data_stats %>%
    mutate(mapped_perc=mapped_mags/trimmed_reads) %>%
    summarise(mean=mean(mapped_perc),sd=sd(mapped_perc))
```


```{r data_fractions_plot, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
data_stats %>%
  mutate(
    low_quality = raw_reads - trimmed_reads,
    unmapped_reads = trimmed_reads - mapped_lizard - mapped_mags
  ) %>%
  select(sample, low_quality, mapped_lizard, mapped_mags, unmapped_reads) %>%
  pivot_longer(-sample) %>%
  mutate(name=factor(name,levels=c("low_quality","mapped_lizard","unmapped_reads","mapped_mags"))) %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(!is.na(time_point)) %>%
  ggplot(aes(x = sample, y = value, fill = name)) +
      geom_bar(stat = "identity", position = "fill") +
      scale_fill_manual(name="Sequence type",
                    breaks=c("low_quality","mapped_lizard","unmapped_reads","mapped_mags"),
                    labels=c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                    values=c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c"))+
      facet_grid(~time_point, scales = "free", labeller = labeller(time_point=c("0_Wild"="Wild", "1_Acclimation"="Acclimation", "2_Antibiotics"="Antibiotics",
                                                                                "3_Transplant1"="Transplant1", "4_Transplant2"="Transplant2", "5_Post-FMT1"="Post1",
                                                                                "6_Post-FMT2"="Post2"))) +
      theme_minimal() +
      theme(axis.text.x = element_text(angle = 90, hjust = 1, size=0)) +
      labs(y="DNA sequence fraction",x="Samples")
```

## Recovered microbial fraction

```{r data_estimations_plot, message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
data_stats %>%
  mutate(
    unmapped_reads = trimmed_reads - mapped_lizard - mapped_mags,
    mag_proportion = mapped_mags / (mapped_mags + unmapped_reads),
    singlem_read_fraction = singlem_read_fraction
  ) %>%
  select(sample, mag_proportion, singlem_read_fraction) %>%
  mutate(
    mag_proportion = if_else(singlem_read_fraction == 0, 0, mag_proportion),
    singlem_read_fraction = if_else(singlem_read_fraction == 0, NA, singlem_read_fraction),
    singlem_read_fraction = if_else(singlem_read_fraction < mag_proportion, NA, singlem_read_fraction),
    singlem_read_fraction = if_else(singlem_read_fraction > 1, 1, singlem_read_fraction)
  ) %>%
  pivot_longer(-sample, names_to = "proportion", values_to = "value") %>%
  mutate(
    proportion = factor(
      proportion,
      levels = c("mag_proportion", "singlem_read_fraction")
    )
  ) %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(!is.na(time_point)) %>%
  ggplot(aes(x = sample, y = value, color = proportion)) +
      geom_line(aes(group = sample), color = "#f8a538") +
      geom_point() +
      scale_color_manual(name="Proportion",
                    breaks=c("mag_proportion","singlem_read_fraction"),
                    labels=c("Recovered","Estimated"),
                    values=c("#52e1e8", "#876b53"))+
      theme_minimal() +
      facet_grid(~time_point, scales = "free", space="free", labeller = labeller(time_point=c("0_Wild"="Wild", "1_Acclimation"="Acclimation", "2_Antibiotics"="Antibiotics",
                                                                                "3_Transplant1"="Transplant1", "4_Transplant2"="Transplant2", "5_Post-FMT1"="Post1",
                                                                                "6_Post-FMT2"="Post2")))+
      labs(y = "Samples", x = "Prokaryotic fraction") +
      scale_y_continuous(limits = c(0, 1)) +
      theme(
        axis.text.y = element_text(size = 4),
        axis.text.x = element_text( angle = 90, vjust = 0.5, hjust = 1, size = 0),
        legend.position = "right"
      )
```

### Domain-adjusted mapping rate (DAMR)

```{r damr}
data_stats %>%
  mutate(
    unmapped_reads = trimmed_reads - mapped_lizard - mapped_mags,
    mag_proportion = mapped_mags / (mapped_mags + unmapped_reads),
    singlem_read_fraction = singlem_read_fraction
  ) %>%
  mutate(damr=pmin(1, mag_proportion/singlem_read_fraction)) %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(!is.na(time_point)) %>%
  select(sample,damr, time_point, species, type) %>%
  #group_by(type) %>%
  #summarise(mean=mean(damr),sd=sd(damr)) %>%
  tt()
```

<!--chapter:end:02_data_statistics.Rmd-->

# MAG catalogue

```{r load_data_mag}
load("data/data.Rdata")
```

## Genome phylogeny

```{r genome_phylogeny, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
# Generate the phylum color heatmap
phylum_heatmap <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(genome,phylum) %>%
    mutate(phylum = factor(phylum, levels = unique(phylum))) %>%
    column_to_rownames(var = "genome")

# Generate  basal tree
circular_tree <- force.ultrametric(genome_tree, method="extend") %>% # extend to ultrametric for the sake of visualisation
    ggtree(., layout="fan", open.angle=10, size=0.5)

# Add phylum ring
circular_tree <- gheatmap(circular_tree, phylum_heatmap, offset=0.55, width=0.1, colnames=FALSE) +
        scale_fill_manual(values=phylum_colors) +
        geom_tiplab2(size=1, hjust=-0.1) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))

# Flush color scale to enable a new color scheme in the next ring
circular_tree <- circular_tree + new_scale_fill()

# Add completeness ring
circular_tree <- circular_tree +
        new_scale_fill() +
        scale_fill_gradient(low = "#d1f4ba", high = "#f4baba") +
        geom_fruit(
                data=genome_metadata,
                geom=geom_bar,
                mapping = aes(x=completeness, y=genome, fill=contamination),
                offset = 0.55,
                orientation="y",
              stat="identity")

# Add genome-size ring
circular_tree <-  circular_tree +
        new_scale_fill() +
        scale_fill_manual(values = "#cccccc") +
        geom_fruit(
             data=genome_metadata,
             geom=geom_bar,
             mapping = aes(x=length, y=genome),
                 offset = 0.05,
                 orientation="y",
         stat="identity")

# Add text
circular_tree <-  circular_tree +
        annotate('text', x=3.4, y=0, label='              Phylum', family='ArialMT', size=3.5) +
        annotate('text', x=4.2, y=0, label='                         Genome quality', family='ArialMT', size=3.5) +
        annotate('text', x=4.7, y=0, label='                      Genome size', family='ArialMT', size=3.5)

#Plot circular tree
circular_tree %>% open_tree(30) %>% rotate_tree(90)
```

## Genome quality

```{r genome_quality}
genome_metadata$completeness %>% mean()
genome_metadata$completeness %>% sd()
genome_metadata$contamination %>% mean()
genome_metadata$contamination %>% sd()
```

```{r genome_quality_plot, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}

#Generate quality biplot
genome_biplot <- genome_metadata %>%
  select(c(genome,domain,phylum,completeness,contamination,length)) %>%
  arrange(match(genome, rev(genome_tree$tip.label))) %>% #sort MAGs according to phylogenetic tree
  ggplot(aes(x=completeness,y=contamination,size=length,color=phylum)) +
              geom_point(alpha=0.7) +
                    ylim(c(10,0)) +
                    scale_color_manual(values=phylum_colors) +
                    labs(y= "Contamination", x = "Completeness") +
                    theme_classic() +
                    theme(legend.position = "none")

#Generate contamination boxplot
genome_contamination <- genome_metadata %>%
            ggplot(aes(y=contamination)) +
                    ylim(c(10,0)) +
                    geom_boxplot(colour = "#999999", fill="#cccccc") +
                    theme_void() +
                    theme(legend.position = "none",
                        axis.title.x = element_blank(),
                        axis.title.y = element_blank(),
                        axis.text.y=element_blank(),
                        axis.ticks.y=element_blank(),
                        axis.text.x=element_blank(),
                        axis.ticks.x=element_blank(),
                        plot.margin = unit(c(0, 0, 0.40, 0),"inches")) #add bottom-margin (top, right, bottom, left)

#Generate completeness boxplot
genome_completeness <- genome_metadata %>%
        ggplot(aes(x=completeness)) +
                xlim(c(50,100)) +
                geom_boxplot(colour = "#999999", fill="#cccccc") +
                theme_void() +
                theme(legend.position = "none",
                    axis.title.x = element_blank(),
                    axis.title.y = element_blank(),
                    axis.text.y=element_blank(),
                    axis.ticks.y=element_blank(),
                    axis.text.x=element_blank(),
                    axis.ticks.x=element_blank(),
                    plot.margin = unit(c(0, 0, 0, 0.50),"inches")) #add left-margin (top, right, bottom, left)

#Render composite figure
grid.arrange(grobs = list(genome_completeness,genome_biplot,genome_contamination),
        layout_matrix = rbind(c(1,1,1,1,1,1,1,1,1,1,1,4),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3)))

```


## Functional overview

```{r function_heatmap, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}

# Aggregate basal GIFT into elements
function_table <- genome_gifts %>%
    to.elements(., GIFT_db3)

# Generate  basal tree
function_tree <- force.ultrametric(genome_tree, method="extend") %>%
                ggtree(., size = 0.3) 

#Add phylum colors next to the tree tips
function_tree <- gheatmap(function_tree, phylum_heatmap, offset=0, width=0.1, colnames=FALSE) +
            scale_fill_manual(values=phylum_colors) +
            labs(fill="Phylum")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

#Add functions heatmap
function_tree <- gheatmap(function_tree, function_table, offset=0.5, width=3.5, colnames=FALSE) +
            vexpand(.08) +
            coord_cartesian(clip = "off") +
            scale_fill_gradient(low = "#f4f4f4", high = "steelblue", na.value="white") +
            labs(fill="GIFT")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

# Add completeness barplots
function_tree <- function_tree +
            geom_fruit(data=genome_metadata,
            geom=geom_bar,
            grid.params=list(axis="x", text.size=2, nbreak = 1),
            axis.params=list(vline=TRUE),
            mapping = aes(x=length, y=genome, fill=completeness),
                 offset = 3.8,
                 orientation="y",
                 stat="identity") +
            scale_fill_gradient(low = "#cf8888", high = "#a2cc87") +
            labs(fill="Genome\ncompleteness")

function_tree
```

## Functional ordination

```{r function_ordination, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
# Generate the tSNE ordination
tSNE_function <- Rtsne(X=function_table, dims = 2, check_duplicates = FALSE)

# Plot the ordination
function_ordination <- tSNE_function$Y %>%
                as.data.frame() %>%
                mutate(genome=rownames(function_table)) %>%
                inner_join(genome_metadata, by="genome") %>%
                rename(tSNE1="V1", tSNE2="V2") %>%
                select(genome,phylum,tSNE1,tSNE2, length) %>%
                ggplot(aes(x = tSNE1, y = tSNE2, color = phylum, size=length))+
                            geom_point(shape=16, alpha=0.7) +
                            scale_color_manual(values=phylum_colors) +
                            theme_minimal() +
                labs(color="Phylum", size="Genome size") +
                guides(color = guide_legend(override.aes = list(size = 5))) # enlarge Phylum dots in legend

function_ordination
```

<!--chapter:end:03_mag_catalogue.Rmd-->

# Community composition

```{r load_data_community}
load("data/data.Rdata")
```

## Taxonomy overview 

### Stacked barplot

```{r taxonomy_barplot, fig.height=8, fig.width=18, fig.fullwidth=TRUE, warning=FALSE}
# Merge data frames based on sample
transplants_metadata<-sample_metadata%>%
  mutate(Tube_code=str_remove_all(Tube_code, "_a"))
transplants_metadata$newID <- paste(transplants_metadata$Tube_code, "_", transplants_metadata$individual)

merged_data<-genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., transplants_metadata, by = join_by(sample == Tube_code)) %>% #append sample metadata
  filter(count > 0) #filter 0 counts

ggplot(merged_data, aes(x=newID,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~ time_point + type ,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    labs(fill="Phylum",y = "Relative abundance",x="Sample")+
    theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size=0))
```

#### Wild samples

```{r taxonomy_barplot_wild, fig.height=8, fig.width=18, fig.fullwidth=TRUE, warning=FALSE}
merged_data  %>%
  filter(time_point=="0_Wild")  %>%
  ggplot(aes(x=newID,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~ Population,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    labs(fill="Phylum",y = "Relative abundance",x="Sample")+
    theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size=0),
    strip.text.x = element_text(size = 12)
    )
```


### Phylum relative abundances
```{r taxonomy_phylum_summary, warning=FALSE, comments="", message=FALSE}
phylum_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,phylum) %>%
  summarise(relabun=sum(count))

phylum_summary %>%
    group_by(phylum) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T))  %>%
    mutate(total=str_c(round(total_mean,2),"±",round(total_sd,2))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(phylum,total) %>% 
    tt()

```

#### Wild samples

```{r taxonomy_phylum_summary_wild, warning=FALSE, comments="", message=FALSE}
#Cold and wet population
phylum_summary %>%
    left_join(sample_metadata, by = join_by(sample == Tube_code))  %>%
    group_by(phylum, Population) %>%
    filter(Population=="Cold_wet") %>%
    filter(time_point=="0_Wild") %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T))  %>%
    mutate(total=str_c(round(total_mean,2),"±",round(total_sd,2))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(phylum,total, Population) %>% 
    tt()

#Hot and dry population
phylum_summary %>%
    left_join(sample_metadata, by = join_by(sample == Tube_code))  %>%
    group_by(phylum, Population) %>%
    filter(Population=="Hot_dry")%>%
    filter(time_point=="0_Wild") %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T))  %>%
    mutate(total=str_c(round(total_mean,2),"±",round(total_sd,2))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(phylum,total, Population) %>% 
    tt()
```

```{r taxonomy_boxplot_phylum, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_arrange <- phylum_summary %>%
    group_by(phylum) %>%
    summarise(mean=mean(relabun)) %>%
    arrange(-mean) %>%
    select(phylum) %>%
    pull()

phylum_summary %>%
    filter(phylum %in% phylum_arrange) %>%
    mutate(phylum=factor(phylum,levels=rev(phylum_arrange))) %>%
    ggplot(aes(x=relabun, y=phylum, group=phylum, color=phylum)) +
        scale_color_manual(values=phylum_colors[rev(phylum_arrange)]) +
        geom_jitter(alpha=0.5) + 
        theme_minimal() + 
        theme(legend.position="none") +
        labs(y="Phylum",x="Relative abundance")
```

## Taxonomy boxplot

### Family
```{r taxonomy_family_summary, warning=FALSE, comments="", message=FALSE}
family_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>% #append sample metadata
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,family) %>%
  summarise(relabun=sum(count))

family_summary %>%
    group_by(family) %>%
    summarise(mean=mean(relabun, na.rm=TRUE),sd=sd(relabun, na.rm=TRUE)) %>%
    arrange(-mean) %>%
    tt()
```


```{r taxonomy_jitterplot_family, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
family_arrange <- family_summary %>%
    group_by(family) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(family) %>%
    pull()

family_summary %>%
    left_join(genome_metadata %>% select(family,phylum) %>% unique(),by=join_by(family==family)) %>%
    left_join(sample_metadata,by=join_by(sample==Tube_code)) %>%
    filter(family %in% family_arrange[1:20]) %>%
    mutate(family=factor(family,levels=rev(family_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=family, group=family, color=phylum)) +
        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~type)+
        theme_minimal() + 
        labs(y="Family", x="Relative abundance", color="Phylum")

```

### Genus
```{r taxonomy_genus_summary, warning=FALSE, comments="", message=FALSE}
genus_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>% #append sample metadata
  left_join(genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,genus) %>%
  summarise(relabun=sum(count)) %>%
  filter(genus != "g__")

genus_summary %>%
    group_by(genus) %>%
    summarise(mean=mean(relabun, na.rm=TRUE),sd=sd(relabun, na.rm=TRUE)) %>%
    arrange(-mean) %>%
    tt()
```

```{r taxonomy_jitterplot_genus, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genus_arrange <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=sum(relabun)) %>%
    filter(genus != "g__")%>%
    arrange(-mean) %>%
    select(genus) %>%
    mutate(genus= sub("^g__", "", genus)) %>%
    pull()

genus_summary %>%
    left_join(genome_metadata %>% select(genus,phylum) %>% unique(),by=join_by(genus==genus)) %>%
    left_join(sample_metadata,by=join_by(sample==Tube_code)) %>%
    mutate(genus= sub("^g__", "", genus)) %>%
    filter(genus %in% genus_arrange[1:20]) %>%
    mutate(genus=factor(genus,levels=rev(genus_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=genus, group=genus, color=phylum)) +
        scale_color_manual(values=phylum_colors[-c(3,4,6,8)]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~type)+
        theme_minimal() + 
        labs(y="Family", x="Relative abundance", color="Phylum")

```

# Diversity analysis

## Alpha diversity

```{r alpha_div, comment="", message=FALSE, warning=FALSE}
# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")

# Aggregate basal GIFT into elements
dist <- genome_gifts %>%
  to.elements(., GIFT_db3) %>%
  traits2dist(., method = "gower")

functional <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, dist = dist) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(functional = 1) %>%
  rownames_to_column(var = "sample") %>%
  mutate(functional = if_else(is.nan(functional), 1, functional))

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample)) %>%
  full_join(functional, by = join_by(sample == sample))
```

### Wild samples

```{r alpha_div_plot_0, comment="", message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="0_Wild") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = Population, group=Population, color=Population, fill=Population)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#008080', "#d57d2c")) +
      scale_fill_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#00808050', "#d57d2c50")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=3, label.x=.58) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")
```

### Acclimation samples

```{r alpha_div_plot_1, comment="", message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="1_Acclimation") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = Population, group=Population, color=Population, fill=Population)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#008080', "#d57d2c")) +
      scale_fill_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#00808050', "#d57d2c50")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=3, label.x=.58) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")
```

### Antibiotics samples

```{r alpha_div_plot_2, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="2_Antibiotics") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = Population, group=Population, color=Population, fill=Population)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#008080', "#d57d2c")) +
      scale_fill_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#00808050', "#d57d2c50")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=3, label.x=.58) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")
```

### Transplant_1 samples

```{r alpha_div_plot_3, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="3_Transplant1") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=3, label.x=.7) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")
```

### Transplant_2 samples

```{r alpha_div_plot_4, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="4_Transplant2") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=3, label.x=.7) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")
```

### Post-Transplant_1 samples

```{r alpha_div_plot_5, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="5_Post-FMT1") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=3, label.x=.7) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")
```

### Post-Transplant_2 samples

```{r alpha_div_plot_6, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=3, label.x=.7) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_text(angle = 45, hjust = 1),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")
```

## Beta diversity

```{r beta_div, comment="", message=FALSE, warning=FALSE, eval=FALSE}
beta_q0n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1)

beta_q1p <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1, tree = genome_tree)

beta_q1f <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1, dist = dist)
```

<!--chapter:end:04_community_composition.Rmd-->

## Permanovas

#### Load required data

```{r loaddata_2}
meta <- column_to_rownames(sample_metadata, "Tube_code")
```

### 1. Are the wild populations similar?

#### Wild: P.muralis vs P.liolepis

```{r beta_div_neutral_wild_, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
wild <- meta %>%
  filter(time_point == "0_Wild")

# Create a temporary modified version of genome_counts_filt
temp_genome_counts <- transform(genome_counts_filt, row.names = genome_counts_filt$genome)
temp_genome_counts$genome <- NULL

wild.counts <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(wild))]
identical(sort(colnames(wild.counts)), sort(as.character(rownames(wild))))

wild_nmds <- sample_metadata %>%
  filter(time_point == "0_Wild")
```

##### Number of samples used

```{r beta_div_neutral_wild_samples, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(wild)
```

```{r beta_div_neutral_wild_1, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_wild<-hillpair(data=wild.counts, q=0)
beta_div_neutral_wild<-hillpair(data=wild.counts, q=1)
beta_div_phylo_wild<-hillpair(data=wild.counts, q=1, tree=genome_tree)
beta_div_func_wild<-hillpair(data=wild.counts, q=1, dist=dist)
```

#### Richness

```{r beta_richness_permanova_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_wild$S, wild$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova, echo=TRUE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_richness_wild$S ~ Population, data=wild[labels(beta_div_richness_wild$S),], permutations=999) %>%
			as.matrix() %>%
			kable()
```


#### Neutral

```{r beta_neutral_permanova_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_wild$S, wild$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova, echo=TRUE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_neutral_wild$S ~ Population, data=wild[labels(beta_div_neutral_wild$S),], permutations=999) %>%
			as.matrix() %>%
			kable()
```

#### Phylogenetic

```{r beta_phylo_permanova_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_wild$S, wild$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova, echo=TRUE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_phylo_wild$S ~ Population, data=wild[labels(beta_div_phylo_wild$S),], permutations=999) %>%
			as.matrix() %>%
			kable()
```

#### Functional

```{r beta_func_permanova_!, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_wild$S, wild$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova, echo=TRUE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_func_wild$S ~ Population, data=wild[labels(beta_div_func_wild$S),], permutations=999) %>%
			as.matrix() %>%
			kable()
```

```{r beta_neutral_nmds, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_q0n_nmds_wild <- beta_div_richness_wild$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
				vegan::scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(wild_nmds, by = join_by(sample == Tube_code))

beta_q1n_nmds_wild <- beta_div_neutral_wild$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
				vegan::scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(wild_nmds, by = join_by(sample == Tube_code))

beta_q1p_nmds_wild <- beta_div_phylo_wild$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				vegan::scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(wild_nmds, by = join_by(sample == Tube_code))

beta_q1f_nmds_wild <- beta_div_func_wild$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				vegan::scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(wild_nmds, by = join_by(sample == Tube_code))
```

```{r beta_neutral_nmds_plot, echo=FALSE, results=TRUE}
p0<-beta_q0n_nmds_wild %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population)) +
				scale_color_manual(name="Population",
                    breaks=c("Cold_wet","Hot_dry"),
                    labels=c("Cold","Hot"),
                    values=c('#008080',"#d57d2c")) +
        scale_shape_manual(name="Type",
                           breaks=c("Control", "Hot_control", "Treatment"),
                           labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                         values=c("circle","square","triangle")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_q1n_nmds_wild %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=type)) +
				scale_color_manual(name="Population",
                    breaks=c("Cold_wet","Hot_dry"),
                    labels=c("Cold","Hot"),
                    values=c('#008080',"#d57d2c")) +
        scale_shape_manual(name="Type",
                           breaks=c("Control", "Hot_control", "Treatment"),
                           labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                         values=c("circle","square","triangle")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p2<-beta_q1p_nmds_wild %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population)) +
				scale_color_manual(name="Population",
                    breaks=c("Cold_wet","Hot_dry"),
                    labels=c("Cold","Hot"),
                    values=c('#008080',"#d57d2c")) +
        scale_shape_manual(name="Type",
                           breaks=c("Control", "Hot_control", "Treatment"),
                           labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                         values=c("circle","square","triangle")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (),x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none") 

p3<-beta_q1f_nmds_wild %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, label=sample)) +
				scale_color_manual(name="Population",
                    breaks=c("Cold_wet","Hot_dry"),
                    labels=c("Cold","Hot"),
                    values=c('#008080',"#d57d2c")) +
        scale_shape_manual(name="Type",
                           breaks=c("Control", "Hot_control", "Treatment"),
                           labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                         values=c("circle","square","triangle")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic() +
				theme(legend.position="none") +
				guides(color=guide_legend(title="Species"))
```

```{r beta_neutral_nmds_plot_final, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

### 2. Effect of acclimation

```{r beta_div_neutral_accli, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
accli <- meta %>%
  filter(time_point == "1_Acclimation")

# Create a temporary modified version of genome_counts_filt
temp_genome_counts <- transform(genome_counts_filt, row.names = genome_counts_filt$genome)
temp_genome_counts$genome <- NULL

accli.counts <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(accli))]
identical(sort(colnames(accli.counts)), sort(as.character(rownames(accli))))

accli_nmds <- sample_metadata %>%
  filter(time_point == "1_Acclimation")
```

##### Number of samples used

```{r beta_div_neutral_accli_samples, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(accli)
```

```{r beta_div_neutral_accli_1, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_accli<-hillpair(data=accli.counts, q=0)
beta_div_neutral_accli<-hillpair(data=accli.counts, q=1)
beta_div_phylo_accli<-hillpair(data=accli.counts, q=1, tree=genome_tree)
beta_div_func_accli<-hillpair(data=accli.counts, q=1, dist=dist)
```

#### Richness

```{r beta_richness_permanova_1_accli, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_accli$S, accli$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_accli, echo=TRUE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_richness_accli$S ~ Population, data=accli[labels(beta_div_richness_accli$S),], permutations=999) %>%
  as.matrix() %>%
  kable()
```


#### Neutral

```{r beta_neutral_permanova_1_accli, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_accli$S, accli$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_accli, echo=TRUE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_neutral_accli$S ~ Population, data=accli[labels(beta_div_neutral_accli$S),], permutations=999) %>%
  as.matrix() %>%
  kable()
```

#### Phylogenetic

```{r beta_phylo_permanova_1_accli, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_accli$S, accli$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_accli, echo=TRUE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_phylo_accli$S ~ Population, data=accli[labels(beta_div_phylo_accli$S),], permutations=999) %>%
  as.matrix() %>%
  kable()
```

#### Functional

```{r beta_func_permanova_accli_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_accli$S, accli$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova_accli, echo=TRUE, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(formula=beta_div_func_accli$S ~ Population, data=accli[labels(beta_div_func_accli$S),], permutations=999) %>%
  as.matrix() %>%
  kable()
```


```{r beta_neutral_nmds_accli, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_q0n_nmds_accli <- beta_div_richness_accli$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(accli_nmds, by = join_by(sample == Tube_code))

beta_q1n_nmds_accli <- beta_div_neutral_accli$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(accli_nmds, by = join_by(sample == Tube_code))

beta_q1p_nmds_accli <- beta_div_phylo_accli$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(accli_nmds, by = join_by(sample == Tube_code))

beta_q1f_nmds_accli <- beta_div_func_accli$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(accli_nmds, by = join_by(sample == Tube_code))
```

```{r beta_neutral_nmds_plot_accli, echo=FALSE, results=TRUE}
p0<-beta_q0n_nmds_accli %>%
  group_by(Population) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=Population)) +
  scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
  scale_shape_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("circle","square","triangle")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
  theme_classic() +
  theme(legend.position="none")

p1<-beta_q1n_nmds_accli %>%
  group_by(Population) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=type)) +
  scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
  scale_shape_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("circle","square","triangle")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
  theme_classic() +
  theme(legend.position="none")

p2<-beta_q1p_nmds_accli %>%
  group_by(Population) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=Population)) +
  scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
  scale_shape_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("circle","square","triangle")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y= element_blank (),x="NMDS1 \n Phylogenetic beta diversity") +
  theme_classic() +
  theme(legend.position="none") 

p3<-beta_q1f_nmds_accli %>%
  group_by(Population) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, label=sample)) +
  scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
  scale_shape_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("circle","square","triangle")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
  theme_classic() +
  theme(legend.position="none") +
  guides(color=guide_legend(title="Species"))
```

```{r beta_neutral_nmds_plot_final_accli, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

### 3. Comparison between Wild and Acclimation

```{r beta_div_neutral_accli1, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}
accli1 <- meta  %>%
  filter(time_point == "0_Wild" | time_point == "1_Acclimation")

temp_genome_counts <- transform(genome_counts_filt, row.names = genome_counts_filt$genome)
temp_genome_counts$genome <- NULL

accli1.counts <- temp_genome_counts[,which(colnames(temp_genome_counts) %in% rownames(accli1))]
identical(sort(colnames(accli1.counts)),sort(as.character(rownames(accli1))))

accli1_nmds <- sample_metadata %>%
  filter(time_point == "0_Wild" | time_point == "1_Acclimation")
```

##### Number of samples used

```{r beta_div_neutral_accli1_samples, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(accli1)
```

```{r beta_div_neutral_accli1_1, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_accli1<-hillpair(data=accli1.counts, q=0)
beta_div_neutral_accli1<-hillpair(data=accli1.counts, q=1)
beta_div_phylo_accli1<-hillpair(data=accli1.counts, q=1, tree=genome_tree)
beta_div_func_accli1<-hillpair(data=accli1.counts, q=1, dist=dist)
```

##### Richness

```{r beta_div_richness_accli1_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_accli1$S, accli1$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_richness_accli1_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_accli1$S ~ time_point*Population, data=accli1[labels(beta_div_neutral_accli1$S),], permutations=999, strata=accli1$individual) %>%
		as.matrix() %>%
		kable()
```

##### Neutral

```{r beta_div_neutral_accli1_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_accli1$S, accli1$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_neutral_accli1_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_accli1$S ~ time_point*Population, data=accli1[labels(beta_div_neutral_accli1$S),], permutations=999, strata=accli1$individual) %>%
		as.matrix() %>%
		kable()
```

##### Phylogenetic

```{r beta_div_phylo_accli1_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_accli1$S, accli1$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_phylo_accli1_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_accli1$S ~ time_point*Population, data=accli1[labels(beta_div_phylo_accli1$S),], permutations=999, strata=accli1$individual) %>%
		as.matrix() %>%
		kable()
```

##### Functional

```{r beta_div_func_accli1_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_accli1$S, accli1$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_func_accli1_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_accli1$S ~ time_point*Population, data=accli1[labels(beta_div_func_accli1$S),], permutations=999, strata=accli1$individual) %>%
		as.matrix() %>%
		kable()
```


```{r beta_neutral_nmds_accli1, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_accli1 <- beta_div_richness_accli1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(accli1_nmds, by = c("sample" = "Tube_code"))

beta_neutral_nmds_accli1 <- beta_div_neutral_accli1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(accli1_nmds, by = c("sample" = "Tube_code"))

beta_phylo_nmds_accli1 <- beta_div_phylo_accli1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(accli1_nmds, by = join_by(sample == Tube_code))

beta_func_nmds_accli1 <- beta_div_func_accli1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(accli1_nmds, by = join_by(sample == Tube_code))
```

```{r beta_neutral_nmds_accli1_plot, echo=FALSE, results=FALSE}
p0<-beta_richness_nmds_accli1 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=time_point)) +
        scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_neutral_nmds_accli1 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=time_point)) +
        scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")
  
p2<-beta_phylo_nmds_accli1 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=time_point)) +
				scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p3<-beta_func_nmds_accli1 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=time_point)) +
				scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic()+
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_accli1_final, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

### 4. Do the antibiotics work?

#### Antibiotics

```{r beta_div_neutral_treat1, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}
treat1 <- meta  %>%
  filter(time_point == "2_Antibiotics")

temp_genome_counts <- transform(genome_counts_filt, row.names = genome_counts_filt$genome)
temp_genome_counts$genome <- NULL

treat1.counts <- temp_genome_counts[,which(colnames(temp_genome_counts) %in% rownames(treat1))]
identical(sort(colnames(treat1.counts)),sort(as.character(rownames(treat1))))

treat1_nmds <- sample_metadata %>%
  filter(time_point == "2_Antibiotics")
```

##### Number of samples used

```{r beta_div_neutral_treat1_samples, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(treat1)
```

```{r beta_div_neutral_treat1_1, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_treat1<-hillpair(data=treat1.counts, q=0)
beta_div_neutral_treat1<-hillpair(data=treat1.counts, q=1)
beta_div_phylo_treat1<-hillpair(data=treat1.counts, q=1, tree=genome_tree)
beta_div_func_treat1<-hillpair(data=treat1.counts, q=1, dist=dist)
```

##### Richness

```{r beta_div_richness_treat1_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_treat1$S, treat1$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_richness_treat1_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_treat1$S ~ Population, data=treat1[labels(beta_div_neutral_treat1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

##### Neutral

```{r beta_div_neutral_treat1_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_treat1$S, treat1$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_neutral_treat1_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_treat1$S ~ Population, data=treat1[labels(beta_div_neutral_treat1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

##### Phylogenetic

```{r beta_div_phylo_treat1_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_treat1$S, treat1$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_phylo_treat1_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_treat1$S ~ Population, data=treat1[labels(beta_div_phylo_treat1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

##### Functional

```{r beta_div_func_treat1_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_treat1$S, treat1$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_func_treat1_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_treat1$S ~ Population, data=treat1[labels(beta_div_func_treat1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_neutral_nmds_treat1, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_treat1 <- beta_div_richness_treat1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat1_nmds, by = c("sample" = "Tube_code"))

beta_neutral_nmds_treat1 <- beta_div_neutral_treat1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat1_nmds, by = c("sample" = "Tube_code"))

beta_phylo_nmds_treat1 <- beta_div_phylo_treat1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat1_nmds, by = join_by(sample == Tube_code))

beta_func_nmds_treat1 <- beta_div_func_treat1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat1_nmds, by = join_by(sample == Tube_code))
```

```{r beta_neutral_nmds_treat1_plot, echo=FALSE, results=FALSE}
p0<-beta_richness_nmds_treat1 %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=type)) +
        scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_neutral_nmds_treat1 %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=type)) +
        scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")
  
p2<-beta_phylo_nmds_treat1 %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=type)) +
				scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p3<-beta_func_nmds_treat1 %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=type)) +
				scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic()+
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_treat1_final, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

#### Acclimation vs antibiotics

```{r beta_div_neutral_treat, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}
treat <- meta  %>%
  filter(time_point == "1_Acclimation" | time_point == "2_Antibiotics")

temp_genome_counts <- transform(genome_counts_filt, row.names = genome_counts_filt$genome)
temp_genome_counts$genome <- NULL

treat.counts <- temp_genome_counts[,which(colnames(temp_genome_counts) %in% rownames(treat))]
identical(sort(colnames(treat.counts)),sort(as.character(rownames(treat))))

treat_nmds <- sample_metadata %>%
  filter(time_point == "1_Acclimation" | time_point == "2_Antibiotics")
```

##### Number of samples used

```{r beta_div_neutral_treat_samples, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(treat)
```

```{r beta_div_neutral_treat_1, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_treat<-hillpair(data=treat.counts, q=0)
beta_div_neutral_treat<-hillpair(data=treat.counts, q=1)
beta_div_phylo_treat<-hillpair(data=treat.counts, q=1, tree=genome_tree)
beta_div_func_treat<-hillpair(data=treat.counts, q=1, dist=dist)
```

##### Richness

```{r beta_div_richness_treat_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_treat$S, treat$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_richness_treat_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_treat$S ~ time_point*Population, data=treat[labels(beta_div_neutral_treat$S),], permutations=999,strata=treat$individual) %>%
		as.matrix() %>%
		kable()
```

##### Neutral

```{r beta_div_neutral_treat_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_treat$S, treat$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_neutral_treat_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_treat$S ~ time_point*Population, data=treat[labels(beta_div_neutral_treat$S),], permutations=999,strata=treat$individual) %>%
		as.matrix() %>%
		kable()
```

##### Phylogenetic

```{r beta_div_phylo_treat_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_treat$S, treat$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_phylo_treat_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_treat$S ~ time_point*Population, data=treat[labels(beta_div_phylo_treat$S),], permutations=999,strata=treat$individual) %>%
		as.matrix() %>%
		kable()
```

##### Functional

```{r beta_div_func_treat_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_treat$S, treat$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_func_treat_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_treat$S ~ time_point*Population, data=treat[labels(beta_div_func_treat$S),], permutations=999,strata=treat$individual) %>%
		as.matrix() %>%
		kable()
```

```{r beta_neutral_nmds_treat, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_treat <- beta_div_richness_treat$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat_nmds, by = c("sample" = "Tube_code"))

beta_neutral_nmds_treat <- beta_div_neutral_treat$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat_nmds, by = c("sample" = "Tube_code"))

beta_phylo_nmds_treat <- beta_div_phylo_treat$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat_nmds, by = join_by(sample == Tube_code))

beta_func_nmds_treat <- beta_div_func_treat$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(treat_nmds, by = join_by(sample == Tube_code))
```

```{r beta_neutral_nmds_treat_plot, echo=FALSE, results=FALSE}
p0<-beta_richness_nmds_treat %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_neutral_nmds_treat %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")
  
p2<-beta_phylo_nmds_treat %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p3<-beta_func_nmds_treat %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic()+
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_treat_final, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```


### 5. Does the FMT work?

#### Comparison between FMT2 vs Post-FMT2

```{r beta_div_neutral_transplant3, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}
#Create newID to identify duplicated samples
transplants_metadata<-sample_metadata%>%
  mutate(Tube_code=str_remove_all(Tube_code, "_a"))
transplants_metadata$newID <- paste(transplants_metadata$Tube_code, "_", transplants_metadata$individual)

transplant3<-transplants_metadata%>%
  filter(time_point == "4_Transplant2" | time_point == "6_Post-FMT2")%>%
  column_to_rownames("newID")

transplant3_nmds <- transplants_metadata %>%
  filter(time_point == "4_Transplant2" | time_point == "6_Post-FMT2")

full_counts<-temp_genome_counts %>%
    t()%>%
    as.data.frame()%>%
    rownames_to_column("Tube_code")%>%
    full_join(transplants_metadata,by = join_by(Tube_code == Tube_code))

transplant3_counts<-full_counts %>%
  filter(time_point == "4_Transplant2" | time_point == "6_Post-FMT2") %>%
  subset(select=-c(315:324)) %>%
  column_to_rownames("newID")%>%
  subset(select=-c(1))%>%
  t() %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)

identical(sort(colnames(transplant3_counts)),sort(as.character(rownames(transplant3))))
```

##### Number of samples used

```{r beta_div_neutral_treat_samples_1, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(transplant3)
```


```{r beta_div_neutral_transplant3_1, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_transplant3<-hillpair(data=transplant3_counts, q=0)
beta_div_neutral_transplant3<-hillpair(data=transplant3_counts, q=1)
beta_div_phylo_transplant3<-hillpair(data=transplant3_counts, q=1, tree=genome_tree)
beta_div_func_transplant3<-hillpair(data=transplant3_counts, q=1, dist=dist)
```

```{r beta_div_neutral_transplant3_2, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}
#Arrange of metadata dataframe
transplant3_arrange<-transplant3[labels(beta_div_neutral_transplant3$S),]
```

##### Richness

```{r beta_richness_permanova_transplant3_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_transplant3$S ~ Population+time_point+type, data=transplant3[labels(beta_div_richness_transplant3$S),], permutations=999, strata=transplant3$individual) %>%
		as.matrix() %>%
		kable()
```

```{r beta_richness_permanova_transplant3_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_richness_transplant3$S,transplant3_arrange$type, perm=999)
pairwise%>%
  tt()
```

##### Neutral

```{r beta_neutral_permanova_transplant3_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_transplant3$S ~ Population+time_point+type, data=transplant3[labels(beta_div_neutral_transplant3$S),], permutations=999, strata=transplant3$individual) %>%
		as.matrix() %>%
		kable()
```

```{r beta_neutral_permanova_transplant3_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_neutral_transplant3$S,transplant3_arrange$type, perm=999)
pairwise%>%
  tt()
```

##### Phylogenetic

```{r beta_phylo_permanova_transplant3_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_transplant3$S ~ Population+time_point+type, data=transplant3[labels(beta_div_phylo_transplant3$S),], permutations=999, strata=transplant3$individual) %>%
		as.matrix() %>%
		kable()
```

```{r beta_phylo_permanova_transplant3_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_phylo_transplant3$S,transplant3_arrange$type, perm=999)
pairwise%>%
  tt()
```

##### Functional

```{r beta_func_permanova_transplant3_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_transplant3$S ~ Population+time_point+type, data=transplant3[labels(beta_div_func_transplant3$S),], permutations=999, strata=transplant3$individual) %>%
		as.matrix() %>%
		kable()
```

```{r beta_func_permanova_transplant3_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_func_transplant3$S,transplant3_arrange$type, perm=999)
pairwise%>%
  tt()
```

```{r beta_neutral_nmds_transplant3, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_transplant3 <- beta_div_richness_transplant3$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(transplant3_nmds, by = join_by(sample == newID))

beta_neutral_nmds_transplant3 <- beta_div_neutral_transplant3$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(transplant3_nmds, by = join_by(sample == newID))

beta_phylo_nmds_transplant3 <- beta_div_phylo_transplant3$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(transplant3_nmds, by = join_by(sample == newID))

beta_func_nmds_transplant3 <- beta_div_func_transplant3$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(transplant3_nmds, by = join_by(sample == newID))
```

```{r beta_neutral_nmds_transplant3_plot, eho=FALSE, results=FALSE}
p0<-beta_richness_nmds_transplant3 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_neutral_nmds_transplant3 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")
  
p2<-beta_phylo_nmds_transplant3 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p3<-beta_func_nmds_transplant3 %>%
			group_by(individual) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic()+
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_transplant3_final, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```



#### Comparison between the different experimental time points (Acclimation vs Transplant samples)

```{r diss_data1_3, echo=FALSE, results=FALSE}
transplants_metadata<-transplants_metadata%>%
  filter(!(Tube_code == "AD45" | Tube_code == "AD48"))

#Create newID to identify duplicated samples
transplant6<-transplants_metadata%>%
  filter(time_point == "3_Transplant1" | time_point == "4_Transplant2" | time_point == "6_Post-FMT2" |time_point == "1_Acclimation")%>%
  column_to_rownames("newID")

#Transform time_point names to combine the transplant samples
transplant6 <- transplant6 %>%
  mutate(time_point = recode(time_point,
                             "3_Transplant1" = "Transplant",
                             "4_Transplant2" = "Transplant"))

#Transform time_point names to combine the transplant samples
transplant6_pairwise <- transplants_metadata %>%
  filter(time_point == "3_Transplant1" | time_point == "4_Transplant2" | time_point == "6_Post-FMT2"|time_point == "1_Acclimation")

transplant6_pairwise <- transplant6_pairwise %>%
  mutate(time_point = recode(time_point,
                             "3_Transplant1" = "Transplant",
                             "4_Transplant2" = "Transplant"))

full_counts_new<-temp_genome_counts %>%
  t()%>%
  as.data.frame()%>%
  rownames_to_column("Tube_code")%>%
  full_join(transplants_metadata,by = join_by(Tube_code == Tube_code))

transplant6_counts<-full_counts %>%
  filter(time_point == "3_Transplant1" | time_point == "4_Transplant2" | time_point == "6_Post-FMT2"|time_point == "1_Acclimation") %>%
  subset(select=-c(315:324)) %>%
  column_to_rownames("newID")%>%
  subset(select=-c(1))%>%
  t() %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)

identical(sort(colnames(transplant6_counts)),sort(as.character(rownames(transplant6))))
```

```{r diss_data1_4, echo=FALSE, results=FALSE}
beta_div_neutral_transplant6<-hillpair(data=transplant6_counts, q=1)

transplant6_beta_div_neutral<-as.matrix(beta_div_neutral_transplant6$S) #convert the pairwise comparisons into a matrix

transplant6_beta_div_neutral_df <- as.data.frame(as.table(as.matrix(transplant6_beta_div_neutral))) #convert the matrix to a data frame with pairwise values

transplant6_beta_div_neutral_df <- transplant6_beta_div_neutral_df %>% #remove the duplicated pairwise comparisons
  filter(Var1 != Var2) %>%
  mutate(pair = pmap_chr(list(Var1, Var2), ~paste(sort(c(..1, ..2)), collapse = "_"))) %>%
  distinct(pair, .keep_all = TRUE) %>%
  select(Sample_1 = Var1, Sample_2 = Var2, Distance = Freq) %>%
  arrange(Sample_1, Sample_2)


transplant6_beta_div_neutral_df <- transplant6_beta_div_neutral_df %>% #keep only the pairwise comparisons between the same individual
  mutate(Sample_1_part = sub("^[^_]*_", "", Sample_1),
         Sample_2_part = sub("^[^_]*_", "", Sample_2)) %>%
  filter(Sample_1_part == Sample_2_part) 
#%>%
#select(Sample_1, Sample_2, Distance)


transplant6_beta_div_neutral_met<-transplant6_beta_div_neutral_df %>% #merge with the metadata associated to the pairwise comparisons for Sample_1
  inner_join(transplant6_pairwise, by = join_by(Sample_1 == "newID"))

# Extract the part before the first `_` for Sample_1 and Sample_2
transplant6_beta_div_neutral_met$Sample_1_part <- sub("_.*", "", transplant6_beta_div_neutral_met$Sample_1)

# Create a named vector for matching Tube_code with Time_point
time_point_map_new <- setNames(transplant6_beta_div_neutral_met$time_point, transplant6_beta_div_neutral_met$Tube_code)

# Function to replace the part before the first _ with the corresponding Time_point
replace_with_time_point_new <- function(sample_part, tube_code, time_point_map_new) {
  if (tube_code %in% names(time_point_map_new)) {
    return(time_point_map_new[tube_code])
  } else {
    return(sample_part)
  }
}

# Apply the function to Sample_1_part
transplant6_beta_div_neutral_met$Sample_1_part <- mapply(replace_with_time_point_new,
                                                         transplant6_beta_div_neutral_met$Sample_1_part,
                                                         transplant6_beta_div_neutral_met$Tube_code,
                                                         list(time_point_map_new))

transplant6_beta_div_neutral_met<-transplant6_beta_div_neutral_met %>%
  subset(select=-c(6:16))

transplant6_beta_div_neutral_met$Sample_2_part <- sub("_.*", "", transplant6_beta_div_neutral_met$Sample_2)


transplant6_beta_div_neutral_met<-transplant6_beta_div_neutral_met %>% #merge with the metadata associated to the pairwise comparisons for Sample_2
  inner_join(transplant6_pairwise, by = join_by(Sample_2 == "newID"))

# Create a named vector again for matching Tube_code with Time_point
time_point_map_new <- setNames(transplant6_beta_div_neutral_met$time_point, transplant6_beta_div_neutral_met$Tube_code)


replace_with_time_point_new <- function(sample_part, tube_code, time_point_map_new) {
  if (tube_code %in% names(time_point_map_new)) {
    return(time_point_map_new[tube_code])
  } else {
    return(sample_part)
  }
}

# Apply the function to Sample_2_part
transplant6_beta_div_neutral_met$Sample_2_part <- mapply(replace_with_time_point_new,
                                                         transplant6_beta_div_neutral_met$Sample_2_part,
                                                         transplant6_beta_div_neutral_met$Tube_code,
                                                         list(time_point_map_new))


# Create a new variable to plot the distances between the time_points
transplant6_beta_div_neutral_met$comparison <- paste(transplant6_beta_div_neutral_met$Sample_1_part, "-",transplant6_beta_div_neutral_met$Sample_2_part)


##plot the differences between acclimation and post-transplant and transplant and post-transplant

# Reorder the 'comparison' factor
transplant6_beta_div_neutral_met$comparison <- factor(
  transplant6_beta_div_neutral_met$comparison,
  levels = c("Transplant - 1_Acclimation",  # Acclimation-Transplant first
             "6_Post-FMT2 - Transplant",    # Transplant-Treatment second
             "6_Post-FMT2 - 1_Acclimation")    # Treatment-Acclimation last
)

transplant6_beta_div_neutral_met_filtered <- transplant6_beta_div_neutral_met %>%
  filter(!(Sample_1_part == "Transplant" & Sample_2_part == "Transplant"))
```

```{r diss_plot2, echo=FALSE, results=FALSE}
Acclimation_Transplant<-transplant6_beta_div_neutral_met_filtered %>% 
  filter(comparison == "Transplant - 1_Acclimation")
stat.test_accli <- Acclimation_Transplant %>% 
  t_test(Distance ~ type)%>% 
  add_xy_position(x = "type")%>% 
  filter(p.adj < 0.05)


bxp1 <- Acclimation_Transplant %>%
  ggplot(aes(x = type, y = Distance, color = type, fill = type)) +
  geom_boxplot() +
  scale_color_manual(name = "Type",
                     breaks = c("Control", "Hot_control", "Treatment"),
                     labels = c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values = c("#4477AA", "#d57d2c", "#76b183")) +
  scale_fill_manual(name = "Type",
                    breaks = c("Control", "Hot_control", "Treatment"),
                    labels = c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                    values = c("#4477AA50", "#d57d2c50", "#76b18350")) +
  scale_x_discrete(labels = c("Control" = "Cold-Cold", "Hot_control" = "Hot-Hot", "Treatment" = "Cold-Hot")) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ylab("Beta diversity dissimilarity distance")

# Add significance bars with p-values
p1<-bxp1 + 
  geom_signif(comparisons = list(c("Control", "Hot_control"), c("Control", "Treatment"), c("Hot_control", "Treatment")),
              map_signif_level = TRUE)


Transplant_Treatment<-transplant6_beta_div_neutral_met_filtered %>% 
  filter(comparison == "6_Post-FMT2 - Transplant")
bxp2 <- Transplant_Treatment %>%
  ggplot(aes(x = type, y = Distance, color = type, fill = type)) +
  geom_boxplot() +
  scale_color_manual(name = "Type",
                     breaks = c("Control", "Hot_control", "Treatment"),
                     labels = c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values = c("#4477AA", "#d57d2c", "#76b183")) +
  scale_fill_manual(name = "Type",
                    breaks = c("Control", "Hot_control", "Treatment"),
                    labels = c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                    values = c("#4477AA50", "#d57d2c50", "#76b18350")) +
  scale_x_discrete(labels = c("Control" = "Cold-Cold", "Hot_control" = "Hot-Hot", "Treatment" = "Cold-Hot")) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ylab("Beta diversity dissimilarity distance")
# Add significance bars with p-values
p2<-bxp2 + 
  geom_signif(comparisons = list(c("Control", "Hot_control"), c("Control", "Treatment"), c("Hot_control", "Treatment")),
              map_signif_level = TRUE)

Treatment_Acclimation<-transplant6_beta_div_neutral_met_filtered %>% 
  filter(comparison == "6_Post-FMT2 - 1_Acclimation")
bxp3 <- Treatment_Acclimation %>%
  ggplot(aes(x = type, y = Distance, color = type, fill = type)) +
  geom_boxplot() +
  scale_color_manual(name = "Type",
                     breaks = c("Control", "Hot_control", "Treatment"),
                     labels = c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values = c("#4477AA", "#d57d2c", "#76b183")) +
  scale_fill_manual(name = "Type",
                    breaks = c("Control", "Hot_control", "Treatment"),
                    labels = c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                    values = c("#4477AA50", "#d57d2c50", "#76b18350")) +
  scale_x_discrete(labels = c("Control" = "Cold-Cold", "Hot_control" = "Hot-Hot", "Treatment" = "Cold-Hot")) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ylab("Beta diversity dissimilarity distance")

# Add significance bars with p-values
p3<-bxp3 + 
  geom_signif(comparisons = list(c("Control", "Hot_control"), c("Control", "Treatment"), c("Hot_control", "Treatment")),
              map_signif_level = TRUE)
```

```{r diss_plot3, , fig.height=7, eho=FALSE, results=TRUE}
ggarrange( p1, p2, p3, ncol=3, nrow=1, common.legend = TRUE, legend="right")
```

#### Comparison of acclimation samples to transplant samples

```{r transplant7_data, eho=FALSE, results=TRUE}
transplant7<-transplants_metadata%>%
  filter(time_point == "4_Transplant2" | time_point == "1_Acclimation"| time_point == "3_Transplant1")%>%
  column_to_rownames("newID")

transplant7_nmds <- transplants_metadata %>%
  filter(time_point == "4_Transplant2" | time_point == "1_Acclimation"| time_point == "3_Transplant1")

transplant7_counts<-full_counts %>%
  filter(time_point == "4_Transplant2" | time_point == "1_Acclimation"| time_point == "3_Transplant1") %>%
  subset(select=-c(315:324)) %>%
  column_to_rownames("newID")%>%
  subset(select=-c(1))%>%
  t() %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)

transplant7_counts <- transplant7_counts[, !names(transplant7_counts) %in% c("AD45 _ LI1_2nd_2", "AD48 _ LI1_2nd_6")]

identical(sort(colnames(transplant7_counts)),sort(as.character(rownames(transplant7))))
```

##### Number of samples used

```{r beta_div_neutral_transplant7_samples_1, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(transplant7)
```


```{r beta_div_neutral_transplant7_1, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_transplant7<-hillpair(data=transplant7_counts, q=0)
beta_div_neutral_transplant7<-hillpair(data=transplant7_counts, q=1)
beta_div_phylo_transplant7<-hillpair(data=transplant7_counts, q=1, tree=genome_tree)
beta_div_func_transplant7<-hillpair(data=transplant7_counts, q=1, dist=dist)
```

```{r beta_div_neutral_transplant7_2, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}
#Arrange of metadata dataframe
transplant7_arrange<-transplant7[labels(beta_div_neutral_transplant7$S),]
transplant7_arrange <- transplant7_arrange %>%
  mutate(time_point = recode(time_point,
                             "3_Transplant1" = "Transplant",
                             "4_Transplant2" = "Transplant"))

transplant7_arrange$type_time <- interaction(transplant7_arrange$type, transplant7_arrange$time_point)
```

##### Richness

```{r beta_richness_permanova_transplant7_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_transplant7$S ~ Population*time_point+type, data=transplant7[labels(beta_div_richness_transplant7$S),], permutations=999, strata=transplant7$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_richness_permanova_transplant7_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_richness_transplant7$S,transplant7_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Neutral

```{r beta_neutral_permanova_transplant7_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_transplant7$S ~ Population+time_point*type, data=transplant7[labels(beta_div_neutral_transplant7$S),], permutations=999, strata=transplant7$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_neutral_permanova_transplant7_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_neutral_transplant7$S,transplant7_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Phylogenetic

```{r beta_phylo_permanova_transplant7_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_transplant7$S ~ Population+time_point+type, data=transplant7[labels(beta_div_phylo_transplant7$S),], permutations=999, strata=transplant7$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_phylo_permanova_transplant7_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_phylo_transplant7$S,transplant7_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Functional

```{r beta_func_permanova_transplant7_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_transplant7$S ~ Population+time_point+type, data=transplant7[labels(beta_div_func_transplant7$S),], permutations=999, strata=transplant7$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_func_permanova_transplant7_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_func_transplant7$S,transplant7_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

```{r beta_neutral_nmds_transplant7, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_transplant7 <- beta_div_richness_transplant7$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(transplant7_nmds, by = join_by(sample == newID))

beta_neutral_nmds_transplant7 <- beta_div_neutral_transplant7$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(transplant7_nmds, by = join_by(sample == newID))

beta_phylo_nmds_transplant7 <- beta_div_phylo_transplant7$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(transplant7_nmds, by = join_by(sample == newID))

beta_func_nmds_transplant7 <- beta_div_func_transplant7$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(transplant7_nmds, by = join_by(sample == newID))
```

```{r beta_neutral_nmds_transplant7_plot, eho=FALSE, results=FALSE}
p0<-beta_richness_nmds_transplant7 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  scale_shape_manual(name="time_point",
                     breaks=c("1_Acclimation", "3_Transplant1", "4_Transplant2"),
                     labels=c("Acclimation", "Transplant", "Transplant"),
                     values=c("circle","square","square")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
  theme_classic() +
  theme(legend.position="none")

p1<-beta_neutral_nmds_transplant7 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  scale_shape_manual(name="time_point",
                     breaks=c("1_Acclimation", "3_Transplant1", "4_Transplant2"),
                     labels=c("Acclimation", "Transplant", "Transplant"),
                     values=c("circle","square","square")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
  theme_classic() +
  theme(legend.position="none")

p2<-beta_phylo_nmds_transplant7 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  scale_shape_manual(name="time_point",
                     breaks=c("1_Acclimation", "3_Transplant1", "4_Transplant2"),
                     labels=c("Acclimation", "Transplant", "Transplant"),
                     values=c("circle","square","square")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
  theme_classic() +
  theme(legend.position="none")

p3<-beta_func_nmds_transplant7 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  scale_shape_manual(name="time_point",
                     breaks=c("1_Acclimation", "3_Transplant1", "4_Transplant2"),
                     labels=c("Acclimation", "Transplant", "Transplant"),
                     values=c("circle","square","square")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
  theme_classic()+
  theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_transplant7_final, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

#### Comparison between Acclimation vs Post-FMT1
```{r beta_div_neutral_post3_data, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}
post3 <- meta  %>%
  filter(time_point == "1_Acclimation" | time_point == "5_Post-FMT1")

temp_genome_counts <- transform(genome_counts_filt, row.names = genome_counts_filt$genome)
temp_genome_counts$genome <- NULL

post3.counts <- temp_genome_counts[,which(colnames(temp_genome_counts) %in% rownames(post3))]
identical(sort(colnames(post3.counts)),sort(as.character(rownames(post3))))

post3_nmds <- sample_metadata %>%
  filter(time_point == "1_Acclimation" | time_point == "5_Post-FMT1")
```

##### Number of samples used

```{r beta_div_neutral_post3_samples, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(post3)
```

```{r beta_div_neutral_post3_1, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_post3<-hillpair(data=post3.counts, q=0)
beta_div_neutral_post3<-hillpair(data=post3.counts, q=1)
beta_div_phylo_post3<-hillpair(data=post3.counts, q=1, tree=genome_tree)
beta_div_func_post3<-hillpair(data=post3.counts, q=1, dist=dist)
```

```{r data_post3_2, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
#Arrange of metadata dataframe
post3_arrange<-post3[labels(beta_div_neutral_post3$S),]
post3_arrange$type_time <- interaction(post3_arrange$type, post3_arrange$time_point)
```

##### Richness

```{r beta_div_richness_post3_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_post3$S, post3_arrange$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_richness_post3_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_post3$S ~ time_point*type, data=post3[labels(beta_div_neutral_post3$S),], permutations=999,strata=post3$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_div_pairwise_post3_8, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(beta_div_richness_post3$S, post3_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Neutral

```{r beta_div_neutral_post3_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_post3$S, post3$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_neutral_post3_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_post3$S ~ time_point*type, data=post3[labels(beta_div_neutral_post3$S),], permutations=999,strata=post3$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_div_pairwise_post3_9, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(beta_div_neutral_post3$S, post3_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Phylogenetic

```{r beta_div_phylo_post3_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_post3$S, post3$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_phylo_post3_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_post3$S ~ time_point*type, data=post3[labels(beta_div_phylo_post3$S),], permutations=999,strata=post3$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_div_pairwise_post3_6, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(beta_div_phylo_post3$S, post3_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Functional

```{r beta_div_func_post3_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_post3$S, post3$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_func_post3_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_post3$S ~ time_point*type, data=post3[labels(beta_div_func_post3$S),], permutations=999,strata=post3$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_div_pairwise_post3_7, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(beta_div_func_post3$S, post3_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

```{r beta_neutral_nmds_post3, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post3 <- beta_div_richness_post3$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(post3_nmds, by = c("sample" = "Tube_code"))

beta_neutral_nmds_post3 <- beta_div_neutral_post3$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(post3_nmds, by = c("sample" = "Tube_code"))

beta_phylo_nmds_post3 <- beta_div_phylo_post3$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(post3_nmds, by = join_by(sample == Tube_code))

beta_func_nmds_post3 <- beta_div_func_post3$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(post3_nmds, by = join_by(sample == Tube_code))
```

```{r beta_neutral_nmds_post3_plot, echo=FALSE, results=FALSE}
p0<-beta_richness_nmds_post3 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
  theme_classic() +
  theme(legend.position="none")

p1<-beta_neutral_nmds_post3 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
  theme_classic() +
  theme(legend.position="none")

p2<-beta_phylo_nmds_post3 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
  theme_classic() +
  theme(legend.position="none")

p3<-beta_func_nmds_post3 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
  theme_classic()+
  theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post3_final, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

#### Comparison between Acclimation vs Post-FMT2

```{r beta_div_neutral_post3, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE}
post4 <- meta  %>%
  filter(time_point == "1_Acclimation" | time_point == "6_Post-FMT2")

temp_genome_counts <- transform(genome_counts_filt, row.names = genome_counts_filt$genome)
temp_genome_counts$genome <- NULL

post4.counts <- temp_genome_counts[,which(colnames(temp_genome_counts) %in% rownames(post4))]
identical(sort(colnames(post4.counts)),sort(as.character(rownames(post4))))

post4_nmds <- sample_metadata %>%
  filter(time_point == "1_Acclimation" | time_point == "6_Post-FMT2")
```

##### Number of samples used

```{r beta_div_neutral_post4_samples, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(post4)
```

```{r beta_div_neutral_post4_1, echo=TRUE, results=FALSE,message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_post4<-hillpair(data=post4.counts, q=0)
beta_div_neutral_post4<-hillpair(data=post4.counts, q=1)
beta_div_phylo_post4<-hillpair(data=post4.counts, q=1, tree=genome_tree)
beta_div_func_post4<-hillpair(data=post4.counts, q=1, dist=dist)
```

```{r data_post4_2, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
#Arrange of metadata dataframe
post4_arrange<-post4[labels(beta_div_neutral_post4$S),]
post4_arrange$type_time <- interaction(post4_arrange$type, post4_arrange$time_point)
```

##### Richness

```{r beta_div_richness_post4_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_post4$S, post4_arrange$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_richness_post4_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_post4$S ~ time_point*Population, data=post4[labels(beta_div_neutral_post4$S),], permutations=999,strata=post4$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_div_pairwise_post4_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(beta_div_richness_post4$S, post4_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Neutral

```{r beta_div_neutral_post4_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_post4$S, post4_arrange$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_neutral_post4_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_post4$S ~ time_point*Population, data=post4[labels(beta_div_neutral_post4$S),], permutations=999,strata=post4$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_div_pairwise_post4_6, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(beta_div_neutral_post4$S, post4_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Phylogenetic

```{r beta_div_phylo_post4_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_post4$S, post4$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_phylo_post4_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_post4$S ~ time_point*Population, data=post4[labels(beta_div_phylo_post4$S),], permutations=999,strata=post4$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_div_pairwise_post4_7, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(beta_div_phylo_post4$S, post4_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Functional

```{r beta_div_func_post4_5, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_post4$S, post4$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_func_post4_4, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_post4$S ~ time_point*Population, data=post4[labels(beta_div_func_post4$S),], permutations=999,strata=post4$individual) %>%
  as.matrix() %>%
  kable()
```

```{r beta_div_pairwise_post4_8, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(beta_div_func_post4$S, post4_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

```{r beta_neutral_nmds_post4, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post4 <- beta_div_richness_post4$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(post4_nmds, by = c("sample" = "Tube_code"))

beta_neutral_nmds_post4 <- beta_div_neutral_post4$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(post4_nmds, by = c("sample" = "Tube_code"))

beta_phylo_nmds_post4 <- beta_div_phylo_post4$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(post4_nmds, by = join_by(sample == Tube_code))

beta_func_nmds_post4 <- beta_div_func_post4$S %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
  scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(post4_nmds, by = join_by(sample == Tube_code))
```

```{r beta_neutral_nmds_post4_plot, echo=FALSE, results=FALSE}
p0<-beta_richness_nmds_post4 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
  theme_classic() +
  theme(legend.position="none")

p1<-beta_neutral_nmds_post4 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
  theme_classic() +
  theme(legend.position="none")

p2<-beta_phylo_nmds_post4 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
  theme_classic() +
  theme(legend.position="none")

p3<-beta_func_nmds_post4 %>%
  group_by(type) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
  scale_color_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
  theme_classic()+
  theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post4_final, echo=FALSE, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

### 6. Are there differences between the control and the treatment group?

#### After 1 week --> Post-FMT1

```{r beta_div_phylo_post1, echo=TRUE, results=FALSE}
post1 <- meta %>%
  filter(time_point == "5_Post-FMT1")

post1.counts <- temp_genome_counts[,which(colnames(temp_genome_counts) %in% rownames(post1))]
identical(sort(colnames(post1.counts)),sort(as.character(rownames(post1))))

post1_nmds <- sample_metadata %>%
  filter(time_point == "5_Post-FMT1")
```

##### Number of samples used

```{r beta_div_neutral_treat1_samples_2, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(post1)
```

```{r beta_div_phylo_post1_1, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_post1<-hillpair(data=post1.counts, q=0)
beta_div_neutral_post1<-hillpair(data=post1.counts, q=1)
beta_div_phylo_post1<-hillpair(data=post1.counts, q=1, tree=genome_tree)
beta_div_func_post1<-hillpair(data=post1.counts, q=1, dist=dist)
```

```{r beta_div_phylo_post1_2, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
#Arrange of metadata dataframe
post1_arrange<-post1[labels(beta_div_neutral_post1$S),]
```

##### Richness

```{r beta_richness_permanova_post1_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_post1$S, post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post1_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_post1$S ~ Population+type, data=post1[labels(beta_div_richness_post1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_richness_permanova_post1_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_richness_post1$S,post1_arrange$type, perm=999)
pairwise
```

##### Neutral

```{r beta_neutral_permanova_post1_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_post1$S, post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post1_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_post1$S ~ Population+type, data=post1[labels(beta_div_neutral_post1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_neutral_permanova_post1_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_neutral_post1$S,post1_arrange$type, perm=999)
pairwise
```

##### Phylogenetic

```{r beta_phylo_permanova_post1_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_post1$S, post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_post1_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_post1$S ~ Population+type, data=post1[labels(beta_div_phylo_post1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_phylo_permanova_post1_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_phylo_post1$S,post1_arrange$type, perm=999)
pairwise
```

##### Functional

```{r beta_func_permanova_post1_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_post1$S, post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova_post1_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_post1$S ~ Population+type, data=post1[labels(beta_div_func_post1$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_func_permanova_post1_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_func_post1$S,post1_arrange$type, perm=999)
pairwise
```


```{r beta_neutral_nmds_post1, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post1 <- beta_div_richness_post1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post1_nmds, by = join_by(sample == Tube_code))

beta_neutral_nmds_post1 <- beta_div_neutral_post1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post1_nmds, by = join_by(sample == Tube_code))

beta_phylogenetic_nmds_post1 <- beta_div_phylo_post1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post1_nmds, by = join_by(sample == Tube_code))

beta_functional_nmds_post1 <- beta_div_func_post1$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post1_nmds, by = join_by(sample == Tube_code))
```

```{r beta_neutral_nmds_post1_plot, eho=FALSE, results=FALSE}
p0<-beta_richness_nmds_post1 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_neutral_nmds_post1 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")
  
p2<-beta_phylogenetic_nmds_post1 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p3<-beta_functional_nmds_post1 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic()+
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post1_final, fig.height=7, eho=FALSE, results=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```


#### After 2 weeks -->Post-FMT2

```{r beta_div_phylo_post2, echo=TRUE, results=FALSE}
post2 <- meta %>%
  filter(time_point == "6_Post-FMT2")

post2.counts <- temp_genome_counts[,which(colnames(temp_genome_counts) %in% rownames(post2))]
identical(sort(colnames(post2.counts)),sort(as.character(rownames(post2))))

post2_nmds <- sample_metadata %>%
  filter(time_point == "6_Post-FMT2")
```


##### Number of samples used

```{r beta_div_neutral_treat_samples_2, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(post2)
```

```{r beta_div_phylo_post2_1, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_post2<-hillpair(data=post2.counts, q=0)
beta_div_neutral_post2<-hillpair(data=post2.counts, q=1)
beta_div_phylo_post2<-hillpair(data=post2.counts, q=1, tree=genome_tree)
beta_div_func_post2<-hillpair(data=post2.counts, q=1, dist=dist)
```

```{r beta_div_phylo_post2_2, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
#Arrange of metadata dataframe
post2_arrange<-post2[labels(beta_div_neutral_post2$S),]
```

##### Richness

```{r beta_richness_permanova_post2_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_post2$S, post2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post2_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_post2$S ~ type, data=post2[labels(beta_div_richness_post2$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_richness_permanova_post2_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_richness_post2$S,post2_arrange$type, perm=999)
pairwise%>%
  tt()
```

##### Neutral

```{r beta_neutral_permanova_post2_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_post2$S, post2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post2_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_post2$S ~ type, data=post2[labels(beta_div_neutral_post2$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_neutral_permanova_post2_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_neutral_post2$S,post2_arrange$type, perm=999)
pairwise%>%
  tt()
```

##### Phylogenetic

```{r beta_phylo_permanova_post2_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_post2$S, post2$type) %>% permutest(., pairwise = TRUE)
```


```{r beta_phylo_permanova_post2_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_post2$S ~ type, data=post2[labels(beta_div_phylo_post2$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_phylo_permanova_post2_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_phylo_post2$S,post2_arrange$type, perm=999)
pairwise%>%
  tt()
```

##### Functional

```{r beta_func_permanova_post2_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_post2$S, post2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova_post2_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_post2$S ~ type, data=post2[labels(beta_div_func_post2$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_func_permanova_post2_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_func_post2$S,post2_arrange$type, perm=999)
pairwise%>%
  tt()
```


```{r beta_neutral_nmds_post2, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post2 <- beta_div_richness_post2$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post2_nmds, by = join_by(sample == Tube_code))

beta_neutral_nmds_post2 <- beta_div_neutral_post2$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post2_nmds, by = join_by(sample == Tube_code))

beta_phylogenetic_nmds_post2 <- beta_div_phylo_post2$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post2_nmds, by = join_by(sample == Tube_code))

beta_functional_nmds_post2 <- beta_div_func_post2$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post2_nmds, by = join_by(sample == Tube_code))
```


```{r beta_neutral_nmds_post2_plot, eho=FALSE, results=FALSE}
p0<-beta_richness_nmds_post2 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_neutral_nmds_post2 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")
  
p2<-beta_phylogenetic_nmds_post2 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p3<-beta_functional_nmds_post2 %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic()+
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post2_final, fig.height=7, eho=FALSE, results=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

#### Post1 vs Post2

```{r beta_div_phylo_post5, echo=TRUE, results=FALSE}
post5 <- meta %>%
  filter(time_point == "6_Post-FMT2" | time_point == "5_Post-FMT1")

post5.counts <- temp_genome_counts[,which(colnames(temp_genome_counts) %in% rownames(post5))]
identical(sort(colnames(post5.counts)),sort(as.character(rownames(post5))))

post5_nmds <- sample_metadata %>%
  filter(time_point == "6_Post-FMT2"| time_point == "5_Post-FMT1")
```


##### Number of samples used

```{r post5_samples_, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(post5)
```

```{r beta_div_phylo_post5_1, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_post5<-hillpair(data=post5.counts, q=0)
beta_div_neutral_post5<-hillpair(data=post5.counts, q=1)
beta_div_phylo_post5<-hillpair(data=post5.counts, q=1, tree=genome_tree)
beta_div_func_post5<-hillpair(data=post5.counts, q=1, dist=dist)
```

```{r beta_div_phylo_post5_2, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
#Arrange of metadata dataframe
post5_arrange<-post5[labels(beta_div_neutral_post5$S),]
post5_arrange$type_time <- interaction(post5_arrange$type, post5_arrange$time_point)
```

##### Richness

```{r beta_richness_permanova_post5_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_richness_post5$S, post5$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post5_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_richness_post5$S ~ type, data=post5[labels(beta_div_richness_post5$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_richness_permanova_post5_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_richness_post5$S,post5_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Neutral

```{r beta_neutral_permanova_post5_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_neutral_post5$S, post5$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post5_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_neutral_post5$S ~ type, data=post5[labels(beta_div_neutral_post5$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_neutral_permanova_post5_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_neutral_post5$S,post5_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Phylogenetic

```{r beta_phylo_permanova_post5_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_phylo_post5$S, post5$type) %>% permutest(., pairwise = TRUE)
```


```{r beta_phylo_permanova_post5_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_phylo_post5$S ~ type, data=post5[labels(beta_div_phylo_post5$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_phylo_permanova_post5_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_phylo_post5$S,post5_arrange$type_time, perm=999)
pairwise%>%
  tt()
```

##### Functional

```{r beta_func_permanova_post5_2, echo=TRUE, warning=FALSE, comments="", message=FALSE}
betadisper(beta_div_func_post5$S, post5$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova_post5_1, echo=TRUE, warning=FALSE, comments="", message=FALSE}
adonis2(formula=beta_div_func_post5$S ~ type, data=post5[labels(beta_div_func_post5$S),], permutations=999) %>%
		as.matrix() %>%
		kable()
```

```{r beta_func_permanova_post5_3, echo=TRUE, warning=FALSE, comments="", message=FALSE}
pairwise<-pairwise.adonis(beta_div_func_post5$S,post5_arrange$type_time, perm=999)
pairwise%>%
  tt()
```


```{r beta_neutral_nmds_post5, echo=TRUE, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post5 <- beta_div_richness_post5$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post5_nmds, by = join_by(sample == Tube_code))

beta_neutral_nmds_post5 <- beta_div_neutral_post5$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post5_nmds, by = join_by(sample == Tube_code))

beta_phylogenetic_nmds_post5 <- beta_div_phylo_post5$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post5_nmds, by = join_by(sample == Tube_code))

beta_functional_nmds_post5 <- beta_div_func_post5$S %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(post5_nmds, by = join_by(sample == Tube_code))
```


```{r beta_neutral_nmds_post5_plot, eho=FALSE, results=FALSE}
p0<-beta_richness_nmds_post5 %>%
			group_by(type, time_point) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p1<-beta_neutral_nmds_post5 %>%
			group_by(type, time_point) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")
  
p2<-beta_phylogenetic_nmds_post5 %>%
			group_by(type, time_point) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")

p3<-beta_functional_nmds_post5 %>%
			group_by(type, time_point) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type, shape=time_point)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Functional beta diversity") +
				theme_classic()+
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post5_final, fig.height=7, eho=FALSE, results=TRUE}
ggarrange(p0, p1, p2, p3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

<!--chapter:end:05_community_composition_comparisons.Rmd-->

# Functional differences

```{r load_data_functional}
load("data/data.Rdata")
```

## Data preparation

```{r gift_analyses, warning=FALSE}
# Aggregate bundle-level GIFTs into the compound level
GIFTs_elements <- to.elements(genome_gifts, GIFT_db3)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts$genome, ]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>%
  select_if(~ !is.numeric(.) || sum(.) != 0)

elements <- GIFTs_elements_filtered %>%
  as.data.frame()

# Aggregate element-level GIFTs into the function level
GIFTs_functions <- to.functions(GIFTs_elements_filtered, GIFT_db3)
functions <- GIFTs_functions %>%
  as.data.frame()

# Aggregate function-level GIFTs into overall Biosynthesis, Degradation and Structural GIFTs
GIFTs_domains <- to.domains(GIFTs_functions, GIFT_db3)
domains <- GIFTs_domains %>%
  as.data.frame()

# Get community-weighed average GIFTs per sample
GIFTs_elements_community <- to.community(GIFTs_elements_filtered, genome_counts_filt %>% column_to_rownames(., "genome") %>% tss(), GIFT_db3)
GIFTs_functions_community <- to.community(GIFTs_functions, genome_counts_filt %>% column_to_rownames(., "genome") %>% tss(), GIFT_db3)
GIFTs_domains_community <- to.community(GIFTs_domains, genome_counts_filt %>% column_to_rownames(., "genome") %>% tss(), GIFT_db3)

uniqueGIFT_db<- unique(GIFT_db3[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)
```

## Genomes GIFT profiles

```{r gift_function_heatmap, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
GIFTs_elements %>%
  as_tibble(., rownames = "MAG") %>%
  reshape2::melt() %>%
  rename(Code_element = variable, GIFT = value) %>%
  inner_join(GIFT_db3,by="Code_element") %>%
  ggplot(., aes(x=Code_element, y=MAG, fill=GIFT, group=Code_function))+
    geom_tile()+
    scale_y_discrete(guide = guide_axis(check.overlap = TRUE))+
    scale_x_discrete(guide = guide_axis(check.overlap = TRUE))+
    scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
    facet_grid(. ~ Code_function, scales = "free", space = "free")+
    theme_grey(base_size=8)+
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),strip.text.x = element_text(angle = 90))

```

## Function level

```{r gift_function_heatmap_1, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
GIFTs_functions_community %>%
    as.data.frame() %>%
    rownames_to_column(var="sample") %>%
    filter(sample!="AD69") %>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
    ggplot(aes(x=trait,y=time_point,fill=gift)) +
        geom_tile(colour="white", size=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(type ~ ., scales="free",space="free")
```

## Element level

```{r gift_element_heatmap, comment="", message=FALSE, warning=FALSE, fig.height=16, fig.width=10, fig.fullwidth=TRUE}
GIFTs_elements_community_merged<-GIFTs_elements_community %>%
    as.data.frame() %>%
    rownames_to_column(var="sample") %>%
    filter(sample!="AD69") %>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == Tube_code))%>%
    mutate(functionid = substr(trait, 1, 3)) %>%
    mutate(trait = case_when(
      trait %in% GIFT_db3$Code_element ~ GIFT_db3$Element[match(trait, GIFT_db3$Code_element)],
      TRUE ~ trait
    )) %>%
    mutate(functionid = case_when(
      functionid %in% GIFT_db3$Code_function ~ GIFT_db3$Function[match(functionid, GIFT_db3$Code_function)],
      TRUE ~ functionid
    )) %>%
    mutate(trait=factor(trait,levels=unique(GIFT_db3$Element))) %>%
    mutate(functionid=factor(functionid,levels=unique(GIFT_db3$Function)))

# Create an interaction variable for time_point and sample
GIFTs_elements_community_merged$interaction_var <- interaction(GIFTs_elements_community_merged$sample, GIFTs_elements_community_merged$time_point)
  
ggplot(GIFTs_elements_community_merged,aes(x=interaction_var,y=trait,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(functionid ~ type, scales="free",space="free") +
        theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=5),
              strip.text.y = element_text(angle = 0)) + 
        labs(y="Traits",x="Time_point",fill="GIFT")+
  scale_x_discrete(labels = function(x) gsub(".*\\.", "", x))
```

## Comparison of samples from the 0 Time_point (0_Wild)

### GIFTs Functional community

```{r gitfs_functional_wild, echo=TRUE,results=TRUE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="0_Wild") %>%
  group_by(species) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

#### GIFT test visualisation

```{r gift_test_function_plot_wild, message=FALSE, warning=FALSE, fig.height=12, fig.width=10, fig.fullwidth=TRUE, eval=TRUE}
GIFTs_functions_community %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="0_Wild") %>%
  select(c(1:30, 33)) %>%
  pivot_longer(-c(sample,Population),names_to = "trait", values_to = "value") %>%
  mutate(trait = case_when(
      trait %in% GIFT_db3$Code_function ~ GIFT_db3$Function[match(trait, GIFT_db3$Code_function)],
      TRUE ~ trait
    )) %>%
  mutate(trait=factor(trait,levels=unique(GIFT_db3$Function))) %>%
  ggplot(aes(x=value, y=Population, group=Population, fill=Population, color=Population)) +
    geom_boxplot() +
    scale_color_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#008080', "#d57d2c")) +
      scale_fill_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#00808050', "#d57d2c50")) +
    facet_grid(trait ~ ., space="free", scales="free") +
              theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
              strip.text.y = element_text(angle = 0)) + 
        labs(y="Traits",x="Metabolic capacity index")
```

### GIFTs Domain community

```{r gitfs_domain_wild, echo=TRUE,results=TRUE}
GIFTs_domains_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="0_Wild") %>%
  group_by(species) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

### GIFTs Elements community

```{r gitfs_elements_wild, echo=TRUE,results=TRUE}
GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="0_Wild") %>%
  group_by(species) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_filtering_4, echo=TRUE,results=FALSE}
sample_metadata_wild <- sample_metadata%>% 
  filter(time_point == "0_Wild")

element_gift_wild <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(., sample_metadata_wild[c(1,3)], by="Tube_code")
```

```{r gitfs_filtering_5, echo=TRUE,results=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_wild, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_wild[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
filtered_data <- element_gift_wild[, !numeric_cols | colnames(element_gift_wild) %in% nonzero_numeric_cols]
```

```{r gitfs_filtering_6, echo=TRUE,results=FALSE}
significant_elements_wild <- filtered_data %>%
  pivot_longer(-c(Tube_code,species), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ species, exact=FALSE)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift_wild  %>% 
  dplyr::select(-c(species))  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements_wild$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_wild[c(1,3)], by = join_by(Tube_code == Tube_code))

element_gift_filt %>%
  dplyr::select(-Tube_code)%>%
  group_by(species)  %>%
  summarise(across(everything(), mean))%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))

element_gift_names <- element_gift_filt%>%
  dplyr::select(-species)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  dplyr::select(-Elements)%>%
  dplyr::select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_wild[c(1,4)], by = join_by(Tube_code == Tube_code))
```

```{r gitfs_plot_1, echo=TRUE,results=FALSE, warning=FALSE}
colNames <- names(element_gift_names)[2:72] #always check names(element_gift_names) first to know where your traits finish
for(i in colNames){
  plt <- ggplot(element_gift_names, aes(x=Population, y=.data[[i]], color = Population)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

## Comparison of samples from the 1st Time_point (1_Acclimation)

### GIFTs Functional community

```{r gitfs_functional_accli, echo=TRUE,results=TRUE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="1_Acclimation") %>%
  group_by(species) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

#### GIFT test visualisation

```{r gift_test_function_plot_accli, message=FALSE, warning=FALSE, fig.height=12, fig.width=10, fig.fullwidth=TRUE, eval=TRUE}
GIFTs_functions_community %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="1_Acclimation") %>%
  select(c(1:30, 33)) %>%
  pivot_longer(-c(sample,Population),names_to = "trait", values_to = "value") %>%
  mutate(trait = case_when(
      trait %in% GIFT_db3$Code_function ~ GIFT_db3$Function[match(trait, GIFT_db3$Code_function)],
      TRUE ~ trait
    )) %>%
  mutate(trait=factor(trait,levels=unique(GIFT_db3$Function))) %>%
  ggplot(aes(x=value, y=Population, group=Population, fill=Population, color=Population)) +
    geom_boxplot() +
    scale_color_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#008080', "#d57d2c")) +
      scale_fill_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#00808050', "#d57d2c50")) +
    facet_grid(trait ~ ., space="free", scales="free") +
              theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
              strip.text.y = element_text(angle = 0)) + 
        labs(y="Traits",x="Metabolic capacity index")
```

### GIFTs Domain community

```{r gitfs_domain_accli, echo=TRUE,results=TRUE}
GIFTs_domains_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="1_Acclimation") %>%
  group_by(species) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

### GIFTs Elements community

```{r gitfs_elements_accli, echo=TRUE,results=TRUE}
GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="1_Acclimation") %>%
  group_by(species) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_filtering_4_1, echo=TRUE,results=FALSE}
sample_metadata_accli <- sample_metadata%>% 
  filter(time_point == "1_Acclimation")

element_gift_accli <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(., sample_metadata_accli[c(1,3)], by="Tube_code")
```

```{r gitfs_filtering_5_1, echo=TRUE,results=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_accli, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_accli[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
filtered_data <- element_gift_accli[, !numeric_cols | colnames(element_gift_accli) %in% nonzero_numeric_cols]
```

```{r gitfs_filtering_6_1, echo=TRUE,results=FALSE}
significant_elements_accli <- filtered_data %>%
  pivot_longer(-c(Tube_code,species), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ species, exact=FALSE)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift_accli  %>% 
  dplyr::select(-c(species))  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements_accli$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_accli[c(1,3)], by = join_by(Tube_code == Tube_code))

element_gift_filt %>%
  dplyr::select(-Tube_code)%>%
  group_by(species)  %>%
  summarise(across(everything(), mean))%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))

element_gift_names <- element_gift_filt%>%
  dplyr::select(-species)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  dplyr::select(-Elements)%>%
  dplyr::select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_accli[c(1,4)], by = join_by(Tube_code == Tube_code))
```

```{r gitfs_plot_1_1, echo=TRUE,results=FALSE, warning=FALSE}
colNames <- names(element_gift_names)[2:51] #always check names(element_gift_names) first to know where your traits finish
for(i in colNames){
  plt <- ggplot(element_gift_names, aes(x=Population, y=.data[[i]], color = Population)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```



## Comparison of samples from the 5th Time_point (5_Post-FMT1)

### GIFTs Functional community

```{r gitfs_functional_tm5, echo=TRUE,results=TRUE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="5_Post-FMT1") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

#### GIFT test visualisation

```{r gift_test_function_plot_tm5, message=FALSE, warning=FALSE, fig.height=12, fig.width=10, fig.fullwidth=TRUE, eval=TRUE}
GIFTs_functions_community %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="5_Post-FMT1") %>%
  select(c(1:30, 36)) %>%
  pivot_longer(-c(sample,type),names_to = "trait", values_to = "value") %>%
  mutate(trait = case_when(
      trait %in% GIFT_db3$Code_function ~ GIFT_db3$Function[match(trait, GIFT_db3$Code_function)],
      TRUE ~ trait
    )) %>%
  mutate(trait=factor(trait,levels=unique(GIFT_db3$Function))) %>%
  ggplot(aes(x=value, y=type, group=type, fill=type, color=type)) +
    geom_boxplot() +
    scale_color_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA","#d57d2c","#76b183")) +
      scale_fill_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
    facet_grid(trait ~ ., space="free", scales="free") +
              theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
              strip.text.y = element_text(angle = 0)) + 
        labs(y="Traits",x="Metabolic capacity index")
```

### GIFTs Domain community

```{r gitfs_domain_tm5, echo=TRUE,results=TRUE}
GIFTs_domains_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="5_Post-FMT1") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

### GIFTs Elements community

```{r gitfs_elements_tm5, echo=TRUE,results=TRUE}
GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="5_Post-FMT1") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r data_tm5, echo=TRUE,results=FALSE}
sample_metadata_tm5 <- sample_metadata%>% 
  filter(time_point == "5_Post-FMT1")%>% 
  filter(type != "Hot_control")

element_gift_tm5 <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_tm5 %>% select(1, 7), by = "Tube_code")
```

```{r gitfs_filtering_2, echo=TRUE,results=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_tm5, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_tm5[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
filtered_data <- element_gift_tm5[, !numeric_cols | colnames(element_gift_tm5) %in% nonzero_numeric_cols]
```

```{r gitfs_filtering_3, echo=TRUE,results=FALSE}
significant_elements_tm5 <- filtered_data %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type, exact=FALSE)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_value < 0.05)  %>% #take into account that p_value is used and not p_adjust
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift_tm5  %>% 
  dplyr::select(-c(type))  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements_tm5$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_tm5[c(1,7)], by = join_by(Tube_code == Tube_code))

element_gift_filt %>%
  dplyr::select(-Tube_code)%>%
  group_by(type)  %>%
  summarise(across(everything(), mean))%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))

element_gift_names <- element_gift_filt%>%
  dplyr::select(-type)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  dplyr::select(-Elements)%>%
  dplyr::select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_tm5[c(1,7)], by = join_by(Tube_code == Tube_code))
```

```{r gitfs_plot, echo=TRUE,results=FALSE, warning=FALSE}
colNames <- names(element_gift_names)[2:14] #always check names(element_gift_names) first to now where your traits finish
for(i in colNames){
  plt <- ggplot(element_gift_names, aes(x=type, y=.data[[i]], color = type)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

## Comparison of samples from the 6th Time_point (6_Post-FMT2)

### GIFTs Functional community

```{r gitfs_functional_tm6, echo=TRUE,results=TRUE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_functional_tm6_stats, echo=TRUE,results=TRUE}
MCI_tm6 <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2")

shapiro.test(MCI_tm6$value)#normality test

res.aov<-aov(value ~ type, data=MCI_tm6)#anova test
summary(res.aov)
```

#### GIFT test visualisation

```{r gift_test_function_plot, message=FALSE, warning=FALSE, fig.height=12, fig.width=10, fig.fullwidth=TRUE, eval=TRUE}
GIFTs_functions_community %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2") %>%
  select(c(1:30, 36)) %>%
  pivot_longer(-c(sample,type),names_to = "trait", values_to = "value") %>%
  mutate(trait = case_when(
      trait %in% GIFT_db3$Code_function ~ GIFT_db3$Function[match(trait, GIFT_db3$Code_function)],
      TRUE ~ trait
    )) %>%
  mutate(trait=factor(trait,levels=unique(GIFT_db3$Function))) %>%
  ggplot(aes(x=value, y=type, group=type, fill=type, color=type)) +
    geom_boxplot() +
    scale_color_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA","#d57d2c","#76b183")) +
      scale_fill_manual(name="type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
    facet_grid(trait ~ ., space="free", scales="free") +
              theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
              strip.text.y = element_text(angle = 0)) + 
        labs(y="Traits",x="Metabolic capacity index")
```

### GIFTs Domain community

```{r gitfs_domain_tm6, echo=TRUE,results=TRUE}
GIFTs_domains_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_domain_tm6_stats, echo=TRUE,results=TRUE}
MCI_domains_tm6 <- GIFTs_domains_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2")

shapiro.test(MCI_domains_tm6$value)#normality test

res.aov<-aov(value ~ type, data=MCI_domains_tm6)#anova test
summary(res.aov)
```

### GIFTs Elements community

```{r gitfs_elements_tm6, echo=TRUE,results=TRUE}
GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_ele_tm6_stats, echo=TRUE,results=TRUE}
MCI_ele_tm6 <- GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="6_Post-FMT2")

shapiro.test(MCI_ele_tm6$value) #normality test

res.aov<-aov(value ~ type, data=MCI_ele_tm6) #anova test
summary(res.aov)
```
#### CI vs CC

```{r data_TM6, echo=TRUE,results=FALSE}
sample_metadata_TM6 <- sample_metadata%>% 
  filter(time_point == "6_Post-FMT2")%>% 
  filter(type != "Hot_control")

element_gift_TM6 <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_TM6 %>% select(1, 7), by = "Tube_code")
```

```{r gitfs_filtering_2_1, echo=TRUE,results=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_TM6, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_TM6[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
filtered_data <- element_gift_TM6[, !numeric_cols | colnames(element_gift_TM6) %in% nonzero_numeric_cols]
```

```{r gitfs_filtering_3_1, echo=TRUE,results=FALSE}
significant_elements_TM6 <- filtered_data %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type, exact=FALSE)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_value < 0.05)  %>% #take into account that p_value is used and not p_adjust
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift_TM6  %>% 
  dplyr::select(-c(type))  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements_TM6$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM6[c(1,7)], by = join_by(Tube_code == Tube_code))

element_gift_filt %>%
  dplyr::select(-Tube_code)%>%
  group_by(type)  %>%
  summarise(across(everything(), mean))%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))

element_gift_names <- element_gift_filt%>%
  dplyr::select(-type)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  dplyr::select(-Elements)%>%
  dplyr::select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM6[c(1,7)], by = join_by(Tube_code == Tube_code))
```

```{r gitfs_plot_1_1_1, echo=TRUE,results=FALSE, warning=FALSE}
colNames <- names(element_gift_names)[2:20] #always check names(element_gift_names) first to now where your traits finish
for(i in colNames){
  plt <- ggplot(element_gift_names, aes(x=type, y=.data[[i]], color = type)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

#### CI vs WC

```{r data_TM8, echo=TRUE,results=FALSE}
sample_metadata_TM8 <- sample_metadata%>% 
  filter(time_point == "6_Post-FMT2")%>% 
  filter(type != "Control")

element_gift_TM8 <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_TM8 %>% select(1, 7), by = "Tube_code")
```

```{r gitfs_filtering_2_1_1, echo=TRUE,results=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_TM8, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_TM8[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
filtered_data <- element_gift_TM8[, !numeric_cols | colnames(element_gift_TM8) %in% nonzero_numeric_cols]
```

```{r gitfs_filtering_3_1_1, echo=TRUE,results=FALSE}
significant_elements_TM8 <- filtered_data %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type, exact=FALSE)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_value < 0.05)  %>% #take into account that p_value is used and not p_adjust
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift_TM8  %>% 
  dplyr::select(-c(type))  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements_TM8$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM8[c(1,7)], by = join_by(Tube_code == Tube_code))

element_gift_filt %>%
  dplyr::select(-Tube_code)%>%
  group_by(type)  %>%
  summarise(across(everything(), mean))%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))

element_gift_names <- element_gift_filt%>%
  dplyr::select(-type)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  dplyr::select(-Elements)%>%
  dplyr::select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM8[c(1,7)], by = join_by(Tube_code == Tube_code))
```

```{r gitfs_plot_1_1_1_1, echo=TRUE,results=FALSE, warning=FALSE}
colNames <- names(element_gift_names)[2:13] #always check names(element_gift_names) first to now where your traits finish
for(i in colNames){
  plt <- ggplot(element_gift_names, aes(x=type, y=.data[[i]], color = type)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

## Domain level

### Comparison of samples from the 0 Time_point (0_Wild)

```{r gitfs_merge_1_2, echo=TRUE,results=FALSE}
#Merge the functional domains with the metadata
merge_gift_wild<- GIFTs_domains_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(., sample_metadata_wild, by="Tube_code")
```


```{r biosynthesis_plot_1, echo=TRUE, results=TRUE}
#Biosynthesis
p1 <-merge_gift_wild %>%
  ggplot(aes(x=species,y=Biosynthesis,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Degradation
p2 <-merge_gift_wild %>%
  ggplot(aes(x=species,y=Degradation,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.45, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Structure
p3 <-merge_gift_wild %>%
  ggplot(aes(x=species,y=Structure,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Transport
p4 <-merge_gift_wild %>%
  ggplot(aes(x=species,y=Transport,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")
```

```{r final_plot_1_2, echo=FALSE, results=TRUE,fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(p1, p2,p3,p4, ncol = 2))
```

### Comparison of samples from the 1st Time_point (1_Acclimation)

```{r gitfs_merge_1_1, echo=TRUE,results=FALSE}
#Merge the functional domains with the metadata
merge_gift_accli<- GIFTs_domains_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(., sample_metadata_accli, by="Tube_code")
```


```{r biosynthesis_plot_1_1, echo=TRUE, results=TRUE}
#Biosynthesis
p1 <-merge_gift_accli %>%
  ggplot(aes(x=species,y=Biosynthesis,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Degradation
p2 <-merge_gift_accli %>%
  ggplot(aes(x=species,y=Degradation,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.45, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Structure
p3 <-merge_gift_accli %>%
  ggplot(aes(x=species,y=Structure,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Transport
p4 <-merge_gift_accli %>%
  ggplot(aes(x=species,y=Transport,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")
```

```{r final_plot_1_1, echo=FALSE, results=TRUE,fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(p1, p2,p3,p4, ncol = 2))
```

### Comparison of samples from the 5th Time_point (5_Post-FMT1)

```{r gitfs_merge_1, echo=TRUE,results=FALSE}
#Merge the functional domains with the metadata
merge_gift_tm5<- GIFTs_domains_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(., sample_metadata_tm5, by="Tube_code")
```


```{r biosynthesis_plot_1_2, echo=TRUE, results=TRUE}
#Biosynthesis
p1 <-merge_gift_tm5 %>%
  ggplot(aes(x=type,y=Biosynthesis,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Degradation
p2 <-merge_gift_tm5 %>%
  ggplot(aes(x=type,y=Degradation,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.45, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Structure
p3 <-merge_gift_tm5 %>%
  ggplot(aes(x=type,y=Structure,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")

#Transport
p4 <-merge_gift_tm5 %>%
  ggplot(aes(x=species,y=Transport,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")
```

```{r final_plot_1, echo=FALSE, results=TRUE,fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(p1, p2,p3,p4, ncol = 2))
```

### Comparison of samples from the 6th Time_point (6_Post-FMT2)

#### CI vs CC

```{r gitfs_merge, echo=TRUE,results=FALSE}
#Merge the functional domains with the metadata
merge_gift_TM6 <- GIFTs_domains_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(., sample_metadata_TM6, by="Tube_code")
```

```{r biosynthesis_plot, echo=TRUE, results=TRUE}
#Biosynthesis
p1 <-merge_gift_TM6 %>%
  ggplot(aes(x=type,y=Biosynthesis,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Type")

#Degradation
p2 <-merge_gift_TM6 %>%
  ggplot(aes(x=type,y=Degradation,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "type")

#Structure
p3 <-merge_gift_TM6 %>%
  ggplot(aes(x=type,y=Structure,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "type")

#Transport
p4 <-merge_gift_TM6 %>%
  ggplot(aes(x=type,y=Transport,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")
```

```{r final_plot, echo=FALSE, results=TRUE,fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(p1, p2,p3,p4, ncol = 2))
```

#### CI vs WC

```{r gitfs_merge_1_1_1_1, echo=TRUE,results=FALSE}
#Merge the functional domains with the metadata
merge_gift_TM8 <- GIFTs_domains_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(., sample_metadata_TM8, by="Tube_code")
```

```{r biosynthesis_plot_1_1_1, echo=TRUE, results=TRUE}
#Biosynthesis
p1 <-merge_gift_TM8 %>%
  ggplot(aes(x=type,y=Biosynthesis,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Type")

#Degradation
p2 <-merge_gift_TM8 %>%
  ggplot(aes(x=type,y=Degradation,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "type")

#Structure
p3 <-merge_gift_TM8 %>%
  ggplot(aes(x=type,y=Structure,color=type,fill=type))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "type")

#Transport
p4 <-merge_gift_TM8 %>%
  ggplot(aes(x=species,y=Transport,color=species,fill=species))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10),
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Species")
```

```{r final_plot_1_1_1, echo=FALSE, results=TRUE,fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(p1, p2,p3,p4, ncol = 2))
```

<!--chapter:end:06_functional_differences.Rmd-->

# Differential abundance analysis

```{r load_data_mag_filtdamr_diffe, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
load("data/data.Rdata")
```

## CC acclimation vs post2
### Enrichment analysis: Ancombc2

#### Structural zeros

```{r struct_zero_cc_acclipost2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
cc_accli_samples <- sample_metadata %>% 
  filter(type== "Control") %>% 
  filter(time_point == "1_Acclimation") %>%
  dplyr::select("Tube_code") %>%
  pull()

cc_post2_samples <- sample_metadata %>% 
  filter(type== "Control") %>% 
  filter(time_point == "6_Post-FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

structural_zeros <- genome_counts_filt %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_accli = all(c_across(all_of(cc_accli_samples)) == 0)) %>% # set true if all samples in accli have zeros
   mutate(all_zeros_post2 = all(c_across(all_of(cc_post2_samples)) == 0)) %>% # set true if all samples in post2 have zeros
   mutate(average_accli = mean(c_across(all_of(cc_accli_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
   mutate(average_post2 = mean(c_across(all_of(cc_post2_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
   filter(all_zeros_accli == TRUE || all_zeros_post2==TRUE)  %>% # filter only genomes with structural zeros
   mutate(present = case_when(
      all_zeros_accli & !all_zeros_post2 ~ "post2",
      !all_zeros_accli & all_zeros_post2 ~ "accli",
      !all_zeros_accli & !all_zeros_post2 ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "accli", average_accli, average_post2)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
structural_zeros
struct_mag <- structural_zeros %>% 
  filter(present %in% c("accli","post2")) %>% 
  select(genome) %>% 
  pull()
struct_mag_meta <- structural_zeros %>% 
  filter(present %in% c("accli","post2"))
```


```{r zero_phylo1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type== "Control") %>% 
                    filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    filter(!genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filtercapwild, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_CC <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","6_Post-FMT2"))
physeq_sample_CC <- prune_taxa(taxa_sums(physeq_sample_CC)>0, physeq_sample_CC)
```

#### MAG level
```{r ancom_rand_capwild, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_accli_post2_CC = ancombc2(data = physeq_sample_CC, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "time_point", #fixed variable(s)
                  rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                        lib_cut = 0, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_capCC, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
colnames(ancom_rand_output_mag_accli_post2_CC$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_CC$res))

ancom_result_CC <- ancom_rand_output_mag_accli_post2_CC$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  right_join(ancom_result_CC, by=join_by(phylum == phylum)) %>%
  arrange(match(genome, genome_tree$tip.label)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
  unique() %>%
  arrange(phylum) %>%
  dplyr::select(colors) %>%
  pull()

ancom_result_CC %>%
  mutate(significance = ifelse(p_time_point6_Post_FMT2 < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin Accli", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_CC %>%
  filter(p_time_point6_Post_FMT2 < 0.01) %>%
  ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_CC$lfc_time_point6_Post_FMT2)-3,max(ancom_result_CC$lfc_time_point6_Post_FMT2)+3) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_CC$lfc_time_point6_Post_FMT2)+1, 1.5), label = "Enriched\nin Accli", color="#666666") +
  geom_text(aes(max(ancom_result_CC$lfc_time_point6_Post_FMT2)-2, 1.5), label = "Enriched\nin Post2", color="#666666") +
  theme(legend.position='right',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```

```{r ancom_table_capCC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_CC@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

colnames(ancom_rand_output_mag_accli_post2_CC$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_CC$res))

ancombc_rand_table_mag <- ancom_rand_output_mag_accli_post2_CC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")

tax_color <- tax_color %>%
  mutate(colors = if_else(phylum=="p__Bacillota", '#074759',
          if_else(phylum=="p__Bacillota_A", '#0c80a2',
          if_else(phylum=="p__Bacillota_B", '#10acd9',
          if_else(phylum=="p__Bacillota_C", '#7ddaf5',
          if_else(phylum=="p__Bacteroidota", "#82383880",
          if_else(phylum=="p__Pseudomonadota", '#E0A60880',
          if_else(phylum=="p___Fusobacteriota", '#08EACA80',
          if_else(phylum=="p__Verrucomicrobiota", '#AD8D8D80',
          if_else(phylum=="p__Desulfobacterota", "#39A60880",
          if_else(phylum=="p__Campylobacterota", "#DE2E0880", colors
                  )))))))))))%>%
  dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_CC, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag$genome)) %>% 
ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_reorder(genus, lfc_time_point6_Post_FMT2), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Genus")+
  guides(fill=guide_legend(title="Phylum"))
```

## CC acclimation vs post1
### Enrichment analysis: Ancombc2
#### Structural zeros

```{r struct_zero_cc_acclipost1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
cc_accli_samples <- sample_metadata %>% 
  filter(type== "Control") %>% 
  filter(time_point == "1_Acclimation") %>%
  dplyr::select("Tube_code") %>%
  pull()

cc_post1_samples <- sample_metadata %>% 
  filter(type== "Control") %>% 
  filter(time_point == "5_Post-FMT1")%>%
  dplyr::select("Tube_code") %>%
  pull()

structural_zeros <- genome_counts_filt %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_accli = all(c_across(all_of(cc_accli_samples)) == 0)) %>% # set true if all samples in accli have zeros
   mutate(all_zeros_post1 = all(c_across(all_of(cc_post1_samples)) == 0)) %>% # set true if all samples in post1 have zeros
   mutate(average_accli = mean(c_across(all_of(cc_accli_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
   mutate(average_post1 = mean(c_across(all_of(cc_post1_samples)), na.rm = TRUE)) %>% # get average genome counts across post1
   filter(all_zeros_accli == TRUE || all_zeros_post1==TRUE)  %>% # filter only genomes with structural zeros
   mutate(present = case_when(
      all_zeros_accli & !all_zeros_post1 ~ "post1",
      !all_zeros_accli & all_zeros_post1 ~ "accli",
      !all_zeros_accli & !all_zeros_post1 ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "accli", average_accli, average_post1)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
structural_zeros
struct_mag <- structural_zeros %>% 
  filter(present %in% c("accli","post1")) %>% 
  select(genome) %>% 
  pull()
struct_mag_meta <- structural_zeros %>% 
  filter(present %in% c("accli","post1"))
```

```{r zero_phylo1_CC_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type== "Control") %>% 
                    filter(time_point == "1_Acclimation"|time_point == "5_Post-FMT1") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    filter(!genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filterCC_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_CC <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","5_Post-FMT1"))
physeq_sample_CC <- prune_taxa(taxa_sums(physeq_sample_CC)>0, physeq_sample_CC)
```


#### MAG level
```{r ancom_rand_CC_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_accli_post1_CC = ancombc2(data = physeq_sample_CC, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "time_point", #fixed variable(s)
                  rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                        lib_cut = 0, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_CC_1_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
colnames(ancom_rand_output_mag_accli_post1_CC$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post1_CC$res))

ancom_result_CC_post1 <- ancom_rand_output_mag_accli_post1_CC$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_CC_post1 %>%
  mutate(significance = ifelse(p_time_point5_Post_FMT1 < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_time_point5_Post_FMT1), y=lfc_time_point5_Post_FMT1, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin Post1", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin Accli", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_CC_post1 %>%
  filter(p_time_point5_Post_FMT1 < 0.01) %>%
  ggplot(., aes(x=lfc_time_point5_Post_FMT1, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_CC_post1$lfc_time_point5_Post_FMT1)-2,max(ancom_result_CC_post1$lfc_time_point5_Post_FMT1)+2) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_CC_post1$lfc_time_point5_Post_FMT1)+2, 2), label = "Enriched\nin Accli", color="#666666") +
  geom_text(aes(max(ancom_result_CC_post1$lfc_time_point5_Post_FMT1)-2, 2), label = "Enriched\nin Post1", color="#666666") +
  theme(legend.position='right',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```  

```{r ancom_table_CC_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_CC@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

ancombc_rand_table_mag_CC_post1 <- ancom_rand_output_mag_accli_post1_CC$res %>%
  dplyr::select(taxon, lfc_time_point5_Post_FMT1, p_time_point5_Post_FMT1) %>%
  filter(p_time_point5_Post_FMT1 < 0.05) %>%
  dplyr::arrange(p_time_point5_Post_FMT1) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_point5_Post_FMT1)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag_CC_post1$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")

tax_color <- tax_color %>%
  mutate(colors = if_else(phylum=="p__Bacillota", '#074759',
          if_else(phylum=="p__Bacillota_A", '#0c80a2',
          if_else(phylum=="p__Bacillota_B", '#10acd9',
          if_else(phylum=="p__Bacillota_C", '#7ddaf5',
          if_else(phylum=="p__Bacteroidota", "#82383880",
          if_else(phylum=="p__Pseudomonadota", '#E0A60880',
          if_else(phylum=="p___Fusobacteriota", '#08EACA80',
          if_else(phylum=="p__Verrucomicrobiota", '#AD8D8D80',
          if_else(phylum=="p__Desulfobacterota", "#39A60880",
          if_else(phylum=="p__Campylobacterota", "#DE2E0880", colors
                  )))))))))))%>%
  dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_CC_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag_CC_post1 %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag_CC_post1$genome)) %>% 
ggplot(., aes(x=lfc_time_point5_Post_FMT1, y=forcats::fct_reorder(species, lfc_time_point5_Post_FMT1), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

## WC acclimation vs post2
### Enrichment analysis: Ancombc2
#### Structural zeros

```{r struct_zero_wc_acclipost2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
wc_accli_samples <- sample_metadata %>% 
  filter(type== "Hot_control") %>% 
  filter(time_point == "1_Acclimation") %>%
  dplyr::select("Tube_code") %>%
  pull()

wc_post2_samples <- sample_metadata %>% 
  filter(type== "Hot_control") %>% 
  filter(time_point == "6_Post-FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

structural_zeros <- genome_counts_filt %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_accli = all(c_across(all_of(wc_accli_samples)) == 0)) %>% # set true if all samples in accli have zeros
   mutate(all_zeros_post2 = all(c_across(all_of(wc_post2_samples)) == 0)) %>% # set true if all samples in post2 have zeros
   mutate(average_accli = mean(c_across(all_of(wc_accli_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
   mutate(average_post2 = mean(c_across(all_of(wc_post2_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
   filter(all_zeros_accli == TRUE || all_zeros_post2==TRUE)  %>% # filter only genomes with structural zeros
   mutate(present = case_when(
      all_zeros_accli & !all_zeros_post2 ~ "post2",
      !all_zeros_accli & all_zeros_post2 ~ "accli",
      !all_zeros_accli & !all_zeros_post2 ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "accli", average_accli, average_post2)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
structural_zeros
struct_mag <- structural_zeros %>% 
  filter(present %in% c("accli","post2")) %>% 
  select(genome) %>% 
  pull()
struct_mag_meta <- structural_zeros %>% 
  filter(present %in% c("accli","post2"))
```

```{r zero_phylo1_WC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type== "Hot_control") %>% 
                    filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    filter(!genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filtercapWC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_WC <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","6_Post-FMT2"))
physeq_sample_WC <- prune_taxa(taxa_sums(physeq_sample_WC)>0, physeq_sample_WC)
```


#### MAG level
```{r ancom_rand_capWC, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_accli_post2_WC = ancombc2(data = physeq_sample_WC, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "time_point", #fixed variable(s)
                  rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                        lib_cut = 0, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_capWC_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
colnames(ancom_rand_output_mag_accli_post2_WC$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_WC$res))

ancom_result_WC <- ancom_rand_output_mag_accli_post2_WC$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_WC %>%
  mutate(significance = ifelse(p_time_point6_Post_FMT2 < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin Accli", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_WC %>%
  filter(p_time_point6_Post_FMT2 < 0.01) %>%
  ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_WC$lfc_time_point6_Post_FMT2)-6,max(ancom_result_WC$lfc_time_point6_Post_FMT2)+6) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_WC$lfc_time_point6_Post_FMT2)+1, 1), label = "Enriched\nin Accli", color="#666666") +
  geom_text(aes(max(ancom_result_WC$lfc_time_point6_Post_FMT2)-1, 1), label = "Enriched\nin Post2", color="#666666") +
  theme(legend.position='right',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```  

```{r ancom_table_capWC, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_WC@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

colnames(ancom_rand_output_mag_accli_post2_WC$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_WC$res))

ancombc_rand_table_mag_WC <- ancom_rand_output_mag_accli_post2_WC$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag_WC$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")

tax_color <- tax_color %>%
  mutate(colors = if_else(phylum=="p__Bacillota", '#074759',
          if_else(phylum=="p__Bacillota_A", '#0c80a2',
          if_else(phylum=="p__Bacillota_B", '#10acd9',
          if_else(phylum=="p__Bacillota_C", '#7ddaf5',
          if_else(phylum=="p__Bacteroidota", "#82383880",
          if_else(phylum=="p__Pseudomonadota", '#E0A60880',
          if_else(phylum=="p___Fusobacteriota", '#08EACA80',
          if_else(phylum=="p__Verrucomicrobiota", '#AD8D8D80',
          if_else(phylum=="p__Desulfobacterota", "#39A60880",
          if_else(phylum=="p__Campylobacterota", "#DE2E0880", colors
                  )))))))))))%>%
  dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_WC, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag_WC %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag_WC$genome)) %>% 
ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_reorder(species, lfc_time_point6_Post_FMT2), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

## CI acclimation vs post2
### Enrichment analysis: Ancombc2
#### Structural zeros

```{r struct_zero_ci_acclipost2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ci_accli_samples <- sample_metadata %>% 
  filter(type== "Treatment") %>% 
  filter(time_point == "1_Acclimation") %>%
  dplyr::select("Tube_code") %>%
  pull()

ci_post2_samples <- sample_metadata %>% 
  filter(type== "Treatment") %>% 
  filter(time_point == "6_Post-FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

structural_zeros <- genome_counts_filt %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_accli = all(c_across(all_of(ci_accli_samples)) == 0)) %>% # set true if all samples in accli have zeros
   mutate(all_zeros_post2 = all(c_across(all_of(ci_post2_samples)) == 0)) %>% # set true if all samples in post2 have zeros
   mutate(average_accli = mean(c_across(all_of(ci_accli_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
   mutate(average_post2 = mean(c_across(all_of(ci_post2_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
   filter(all_zeros_accli == TRUE || all_zeros_post2==TRUE)  %>% # filter only genomes with structural zeros
   mutate(present = case_when(
      all_zeros_accli & !all_zeros_post2 ~ "post2",
      !all_zeros_accli & all_zeros_post2 ~ "accli",
      !all_zeros_accli & !all_zeros_post2 ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "accli", average_accli, average_post2)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
structural_zeros
struct_mag <- structural_zeros %>% 
  filter(present %in% c("accli","post2")) %>% 
  select(genome) %>% 
  pull()
struct_mag_meta <- structural_zeros %>% 
  filter(present %in% c("accli","post2"))
```

```{r zero_phylo1_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type== "Treatment") %>% 
                    filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    filter(!genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filterCI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_CI <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","6_Post-FMT2"))
physeq_sample_CI <- prune_taxa(taxa_sums(physeq_sample_CI)>0, physeq_sample_CI)
```


#### MAG level
```{r ancom_rand_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_accli_post2_CI = ancombc2(data = physeq_sample_CI, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "time_point", #fixed variable(s)
                  rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                        lib_cut = 0, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_CI_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
colnames(ancom_rand_output_mag_accli_post2_CI$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post2_CI$res))

ancom_result_CI <- ancom_rand_output_mag_accli_post2_CI$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_CI %>%
  mutate(significance = ifelse(p_time_point6_Post_FMT2 < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_time_point6_Post_FMT2), y=lfc_time_point6_Post_FMT2, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin Post2", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin Accli", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_CI %>%
  filter(p_time_point6_Post_FMT2 < 0.01) %>%
  ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_CI$lfc_time_point6_Post_FMT2)-2,max(ancom_result_CI$lfc_time_point6_Post_FMT2)+2) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_CI$lfc_time_point6_Post_FMT2)+2, 2), label = "Enriched\nin Accli", color="#666666") +
  geom_text(aes(max(ancom_result_CI$lfc_time_point6_Post_FMT2)-2, 2), label = "Enriched\nin Post2", color="#666666") +
  theme(legend.position='right',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```  

```{r ancom_table_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_CI@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

ancombc_rand_table_mag_CI <- ancom_rand_output_mag_accli_post2_CI$res %>%
  dplyr::select(taxon, lfc_time_point6_Post_FMT2, p_time_point6_Post_FMT2) %>%
  filter(p_time_point6_Post_FMT2 < 0.05) %>%
  dplyr::arrange(p_time_point6_Post_FMT2) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_point6_Post_FMT2)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag_CI$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")

tax_color <- tax_color %>%
  mutate(colors = if_else(phylum=="p__Bacillota", '#074759',
          if_else(phylum=="p__Bacillota_A", '#0c80a2',
          if_else(phylum=="p__Bacillota_B", '#10acd9',
          if_else(phylum=="p__Bacillota_C", '#7ddaf5',
          if_else(phylum=="p__Bacteroidota", "#82383880",
          if_else(phylum=="p__Pseudomonadota", '#E0A60880',
          if_else(phylum=="p___Fusobacteriota", '#08EACA80',
          if_else(phylum=="p__Verrucomicrobiota", '#AD8D8D80',
          if_else(phylum=="p__Desulfobacterota", "#39A60880",
          if_else(phylum=="p__Campylobacterota", "#DE2E0880", colors
                  )))))))))))%>%
  dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_CI, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag_CI %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag_CI$genome)) %>% 
ggplot(., aes(x=lfc_time_point6_Post_FMT2, y=forcats::fct_reorder(species, lfc_time_point6_Post_FMT2), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

## CI acclimation vs post1
### Enrichment analysis: Ancombc2
#### Structural zeros

```{r struct_zero_ci_acclipost1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ci_accli_samples <- sample_metadata %>% 
  filter(type== "Treatment") %>% 
  filter(time_point == "1_Acclimation") %>%
  dplyr::select("Tube_code") %>%
  pull()

ci_post1_samples <- sample_metadata %>% 
  filter(type== "Treatment") %>% 
  filter(time_point == "5_Post-FMT1")%>%
  dplyr::select("Tube_code") %>%
  pull()

structural_zeros <- genome_counts_filt %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_accli = all(c_across(all_of(ci_accli_samples)) == 0)) %>% # set true if all samples in accli have zeros
   mutate(all_zeros_post1 = all(c_across(all_of(ci_post1_samples)) == 0)) %>% # set true if all samples in post1 have zeros
   mutate(average_accli = mean(c_across(all_of(ci_accli_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
   mutate(average_post1 = mean(c_across(all_of(ci_post1_samples)), na.rm = TRUE)) %>% # get average genome counts across post1
   filter(all_zeros_accli == TRUE || all_zeros_post1==TRUE)  %>% # filter only genomes with structural zeros
   mutate(present = case_when(
      all_zeros_accli & !all_zeros_post1 ~ "post1",
      !all_zeros_accli & all_zeros_post1 ~ "accli",
      !all_zeros_accli & !all_zeros_post1 ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "accli", average_accli, average_post1)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
structural_zeros
struct_mag <- structural_zeros %>% 
  filter(present %in% c("accli","post1")) %>% 
  select(genome) %>% 
  pull()
struct_mag_meta <- structural_zeros %>% 
  filter(present %in% c("accli","post1"))
```

```{r zero_phylo1_CI_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type== "Treatment") %>% 
                    filter(time_point == "1_Acclimation"|time_point == "5_Post-FMT1") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    filter(!genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filterCI_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_CI <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","5_Post-FMT1"))
physeq_sample_CI <- prune_taxa(taxa_sums(physeq_sample_CI)>0, physeq_sample_CI)
```


#### MAG level
```{r ancom_rand_CI_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_accli_post1_CI = ancombc2(data = physeq_sample_CI, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "time_point", #fixed variable(s)
                  rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                        lib_cut = 0, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_CI_1_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
colnames(ancom_rand_output_mag_accli_post1_CI$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_accli_post1_CI$res))

ancom_result_CI_post1 <- ancom_rand_output_mag_accli_post1_CI$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_CI_post1 %>%
  mutate(significance = ifelse(p_time_point5_Post_FMT1 < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_time_point5_Post_FMT1), y=lfc_time_point5_Post_FMT1, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin Post1", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin Accli", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_CI_post1 %>%
  filter(p_time_point5_Post_FMT1 < 0.01) %>%
  ggplot(., aes(x=lfc_time_point5_Post_FMT1, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_CI_post1$lfc_time_point5_Post_FMT1)-2,max(ancom_result_CI_post1$lfc_time_point5_Post_FMT1)+2) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_CI_post1$lfc_time_point5_Post_FMT1)+2, 2), label = "Enriched\nin Accli", color="#666666") +
  geom_text(aes(max(ancom_result_CI_post1$lfc_time_point5_Post_FMT1)-2, 2), label = "Enriched\nin Post1", color="#666666") +
  theme(legend.position='right',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```  

```{r ancom_table_CI_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_CI@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

ancombc_rand_table_mag_CI_post1 <- ancom_rand_output_mag_accli_post1_CI$res %>%
  dplyr::select(taxon, lfc_time_point5_Post_FMT1, p_time_point5_Post_FMT1) %>%
  filter(p_time_point5_Post_FMT1 < 0.05) %>%
  dplyr::arrange(p_time_point5_Post_FMT1) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_point5_Post_FMT1)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag_CI_post1$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")

tax_color <- tax_color %>%
  mutate(colors = if_else(phylum=="p__Bacillota", '#074759',
          if_else(phylum=="p__Bacillota_A", '#0c80a2',
          if_else(phylum=="p__Bacillota_B", '#10acd9',
          if_else(phylum=="p__Bacillota_C", '#7ddaf5',
          if_else(phylum=="p__Bacteroidota", "#82383880",
          if_else(phylum=="p__Pseudomonadota", '#E0A60880',
          if_else(phylum=="p___Fusobacteriota", '#08EACA80',
          if_else(phylum=="p__Verrucomicrobiota", '#AD8D8D80',
          if_else(phylum=="p__Desulfobacterota", "#39A60880",
          if_else(phylum=="p__Campylobacterota", "#DE2E0880", colors
                  )))))))))))%>%
  dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_CI_post1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag_CI_post1 %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag_CI_post1$genome)) %>% 
ggplot(., aes(x=lfc_time_point5_Post_FMT1, y=forcats::fct_reorder(species, lfc_time_point5_Post_FMT1), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```


## post2 CC and CI
### Enrichment analysis: Ancombc2

#### Structural zeros

```{r struct_zero_ci_cc_post2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ci_post2 <- sample_metadata %>% 
  filter(type== "Treatment") %>% 
  filter(time_point == "6_Post-FMT2") %>%
  dplyr::select("Tube_code") %>%
  pull()

cc_post2 <- sample_metadata %>% 
  filter(type== "Control") %>% 
  filter(time_point == "6_Post-FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

structural_zeros <- genome_counts_filt %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_ci_post2 = all(c_across(all_of(ci_post2)) == 0)) %>% # set true if all samples in accli have zeros
   mutate(all_zeros_cc_post2 = all(c_across(all_of(cc_post2)) == 0)) %>% # set true if all samples in post2 have zeros
   mutate(average_ci_post2 = mean(c_across(all_of(ci_post2)), na.rm = TRUE)) %>% # get average genome counts across accli
   mutate(average_cc_post2 = mean(c_across(all_of(cc_post2)), na.rm = TRUE)) %>% # get average genome counts across post2
   filter(all_zeros_ci_post2 == TRUE || all_zeros_cc_post2==TRUE)  %>% # filter only genomes with structural zeros
   mutate(present = case_when(
      all_zeros_ci_post2 & !all_zeros_cc_post2 ~ "cc_post2",
      !all_zeros_ci_post2 & all_zeros_cc_post2 ~ "ci_post2",
      !all_zeros_ci_post2 & !all_zeros_cc_post2 ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "ci_post2", average_ci_post2, average_cc_post2)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
structural_zeros
struct_mag <- structural_zeros %>% 
  filter(present %in% c("ci_post2","cc_post2")) %>% 
  select(genome) %>% 
  pull()
struct_mag_meta <- structural_zeros %>% 
  filter(present %in% c("ci_post2","cc_post2"))
```

```{r zero_phylo1_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type!= "Hot_control") %>% 
                    filter(time_point == "6_Post-FMT2") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    filter(!genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filtercold, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_cold <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","6_Post-FMT2"))
physeq_sample_cold <- prune_taxa(taxa_sums(physeq_sample_cold)>0, physeq_sample_cold)
```


#### MAG level
```{r ancom_rand_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_post2_cold = ancombc2(data = physeq_sample_cold, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "type", #fixed variable(s)
 #                 rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                        lib_cut = 0, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
#                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_cold_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
colnames(ancom_rand_output_mag_post2_cold$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_post2_cold$res))

ancom_result_cold <- ancom_rand_output_mag_post2_cold$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_cold %>%
  mutate(significance = ifelse(p_typeTreatment < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_typeTreatment), y=lfc_typeTreatment, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin CI", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin CC", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_cold %>%
  filter(p_typeTreatment < 0.01) %>%
  ggplot(., aes(x=lfc_typeTreatment, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_cold$lfc_typeTreatment)-2,max(ancom_result_cold$lfc_typeTreatment)+2) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_cold$lfc_typeTreatment)+2, 2), label = "Enriched\nin CC", color="#666666") +
  geom_text(aes(max(ancom_result_cold$lfc_typeTreatment)-2, 2), label = "Enriched\nin CI", color="#666666") +
  theme(legend.position='right',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```  

```{r ancom_table_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_cold@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

ancombc_rand_table_mag_cold <- ancom_rand_output_mag_post2_cold$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05) %>%
  dplyr::arrange(p_typeTreatment) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_typeTreatment)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag_cold$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")

tax_color <- tax_color %>%
  mutate(colors = if_else(phylum=="p__Bacillota", '#074759',
          if_else(phylum=="p__Bacillota_A", '#0c80a2',
          if_else(phylum=="p__Bacillota_B", '#10acd9',
          if_else(phylum=="p__Bacillota_C", '#7ddaf5',
          if_else(phylum=="p__Bacteroidota", "#82383880",
          if_else(phylum=="p__Pseudomonadota", '#E0A60880',
          if_else(phylum=="p___Fusobacteriota", '#08EACA80',
          if_else(phylum=="p__Verrucomicrobiota", '#AD8D8D80',
          if_else(phylum=="p__Desulfobacterota", "#39A60880",
          if_else(phylum=="p__Campylobacterota", "#DE2E0880", colors
                  )))))))))))%>%
  dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag_cold %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag_cold$genome)) %>% 
ggplot(., aes(x=lfc_typeTreatment, y=forcats::fct_reorder(species, lfc_typeTreatment), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

## post2 WC and CI
### Enrichment analysis: Ancombc2
####Structural zeros
```{r struct_zero_ci_wc_post2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ci_post2 <- sample_metadata %>% 
  filter(type== "Treatment") %>% 
  filter(time_point == "6_Post-FMT2") %>%
  dplyr::select("Tube_code") %>%
  pull()

wc_post2 <- sample_metadata %>% 
  filter(type== "Hot_control") %>% 
  filter(time_point == "6_Post-FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

structural_zeros <- genome_counts_filt %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_ci_post2 = all(c_across(all_of(ci_post2)) == 0)) %>% # set true if all samples in accli have zeros
   mutate(all_zeros_wc_post2 = all(c_across(all_of(wc_post2)) == 0)) %>% # set true if all samples in post2 have zeros
   mutate(average_ci_post2 = mean(c_across(all_of(ci_post2)), na.rm = TRUE)) %>% # get average genome counts across accli
   mutate(average_wc_post2 = mean(c_across(all_of(wc_post2)), na.rm = TRUE)) %>% # get average genome counts across post2
   filter(all_zeros_ci_post2 == TRUE || all_zeros_wc_post2==TRUE)  %>% # filter only genomes with structural zeros
   mutate(present = case_when(
      all_zeros_ci_post2 & !all_zeros_wc_post2 ~ "wc_post2",
      !all_zeros_ci_post2 & all_zeros_wc_post2 ~ "ci_post2",
      !all_zeros_ci_post2 & !all_zeros_wc_post2 ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "ci_post2", average_ci_post2, average_wc_post2)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
structural_zeros
struct_mag <- structural_zeros %>% 
  filter(present %in% c("ci_post2","wc_post2")) %>% 
  select(genome) %>% 
  pull()
struct_mag_meta <- structural_zeros %>% 
  filter(present %in% c("ci_post2","wc_post2"))
```

```{r zero_phylo1_treat, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#phyloseq object considering structual zeros
phylo_samples <- sample_metadata %>% 
                    filter(type!= "Control") %>% 
                    filter(time_point == "6_Post-FMT2") %>% 
                    column_to_rownames("Tube_code") %>% 
                    sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
                    filter(!genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues (note this approach can be improved!)
                    otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
                    filter(genome %in% rownames(phylo_genome)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% #add an additional taxonomic level to ensure genome-level analysis (as no all genomes have species-level taxonomic assignments. Otherwise, ANCOMBC2 aggregates analyses per species)
                    as.matrix() %>% 
                    tax_table() #convert to phyloseq tax_table object

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_filtered_clr <- microbiome::transform(physeq_genome_filtered, 'clr')
```

```{r filtertreat, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_sample_treat <- subset_samples(physeq_genome_filtered, time_point %in% c("1_Acclimation","6_Post-FMT2"))
physeq_sample_treat <- prune_taxa(taxa_sums(physeq_sample_treat)>0, physeq_sample_treat)
```


#### MAG level
```{r ancom_rand_treat, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_mag_post2_treat = ancombc2(data = physeq_sample_treat, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "type", #fixed variable(s)
 #                 rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                        lib_cut = 0, 
                        s0_perc = 0.05,
                        group = NULL, 
                        struc_zero = FALSE, 
                        neg_lb = FALSE,
                        alpha = 0.05, 
                        n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
#                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_treat_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=TRUE}
colnames(ancom_rand_output_mag_post2_treat$res) <- gsub("-", "_", colnames(ancom_rand_output_mag_post2_treat$res))

ancom_result_treat <- ancom_rand_output_mag_post2_treat$res %>%
  rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))

ancom_result_treat %>%
  mutate(significance = ifelse(p_typeTreatment < 0.01, "1", "0")) %>%
  ggplot(., aes(x=-log(p_typeTreatment), y=lfc_typeTreatment, color=significance)) +
  geom_point() +
  scale_color_manual(values = c("#cccccc","#00FFFF")) +
  geom_text(aes(2.5, 10), label = "Enriched\nin CI", color="#666666") +
  geom_text(aes(2.5, -10), label = "Enriched\nin WC", color="#666666") +
  labs(color="Significance", y="Difference between treatments", x="p-value") +
  theme_classic()

ancom_result_treat %>%
  filter(p_typeTreatment < 0.01) %>%
  ggplot(., aes(x=lfc_typeTreatment, y=forcats::fct_rev(genome), color=phylum)) +
  geom_jitter(size=3) +
  geom_vline(xintercept=0) + 
  xlim(-max(ancom_result_treat$lfc_typeTreatment)-2,max(ancom_result_treat$lfc_typeTreatment)+2) +
  scale_color_manual(values=phylum_colors) +
  geom_text(aes(-max(ancom_result_treat$lfc_typeTreatment)+2, 2), label = "Enriched\nin WC", color="#666666") +
  geom_text(aes(max(ancom_result_treat$lfc_typeTreatment)-2, 2), label = "Enriched\nin CI", color="#666666") +
  theme(legend.position='right',
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title.x=element_blank())+
  labs(y="MAGs",x="Genome counts") + 
  guides(col=guide_legend("Phylum"))
```  

```{r ancom_table_treat, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_sample_treat@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  mutate(across(c(family,genus, species), na_if, ""))%>%
    mutate(family = coalesce(family, paste('Unclassified', order)),
           genus = coalesce(genus, 
                              if_else(grepl('^Unclassified', family),
                                      family, paste('Unclassified', family))),
           species = coalesce(species, 
                              if_else(grepl('^Unclassified', genus),
                                      genus, paste('Unclassified', genus))))

ancombc_rand_table_mag_treat <- ancom_rand_output_mag_post2_treat$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05) %>%
  dplyr::arrange(p_typeTreatment) %>%
  merge(., taxonomy, by="taxon") %>%
#  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_typeTreatment)
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
#  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag_treat$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")

tax_color <- tax_color %>%
  mutate(colors = if_else(phylum=="p__Bacillota", '#074759',
          if_else(phylum=="p__Bacillota_A", '#0c80a2',
          if_else(phylum=="p__Bacillota_B", '#10acd9',
          if_else(phylum=="p__Bacillota_C", '#7ddaf5',
          if_else(phylum=="p__Bacteroidota", "#82383880",
          if_else(phylum=="p__Pseudomonadota", '#E0A60880',
          if_else(phylum=="p___Fusobacteriota", '#08EACA80',
          if_else(phylum=="p__Verrucomicrobiota", '#AD8D8D80',
          if_else(phylum=="p__Desulfobacterota", "#39A60880",
          if_else(phylum=="p__Campylobacterota", "#DE2E0880", colors
                  )))))))))))%>%
  dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_treat, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE, eval=TRUE}
ancombc_rand_table_mag_treat %>%
  mutate(genome=factor(genome,levels=ancombc_rand_table_mag_treat$genome)) %>% 
ggplot(., aes(x=lfc_typeTreatment, y=forcats::fct_reorder(species, lfc_typeTreatment), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

## Community level plots

```{r gift_test_function_plot_1, message=FALSE, warning=FALSE, fig.height=12, fig.width=10, fig.fullwidth=TRUE, eval=TRUE}
accli_post2<-sample_metadata%>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2")
accli_post2$newID<-paste(accli_post2$type, "_", accli_post2$time_point)

GIFTs_functions_community %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(accli_post2, by = join_by(sample == Tube_code)) %>%
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2") %>%
  select(c(1:29, 36,39)) %>%
  pivot_longer(-c(sample,type,time_point),names_to = "trait", values_to = "value") %>%
  mutate(trait = case_when(
    trait %in% GIFT_db3$Code_function ~ GIFT_db3$Function[match(trait, GIFT_db3$Code_function)],
    TRUE ~ trait
  )) %>%
  mutate(trait=factor(trait,levels=unique(GIFT_db3$Function))) %>%
  ggplot(aes(x=value, y=time_point, group=time_point, fill=type, color=type)) +
  geom_boxplot() +
  scale_color_manual(name="type",
                     breaks=c("Control","Hot_control", "Treatment"),
                     labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
                     values=c("#4477AA","#d57d2c","#76b183")) +
  scale_fill_manual(name="type",
                    breaks=c("Control","Hot_control", "Treatment"),
                    labels=c("Cold-Cold","Hot-Hot", "Cold-Hot"),
                    values=c("#4477AA50","#d57d2c50","#76b18350")) +
  facet_grid(trait ~ type, space="free", scales="free") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text.y = element_text(angle = 0)) + 
  labs(y="Traits",x="Metabolic capacity index")
```

```{r data_TM6_plot_1, echo=TRUE,results=FALSE}
GIFTs_elements_community_merged<-GIFTs_elements_community %>%
    as.data.frame() %>%
    rownames_to_column(var="sample") %>%
    filter(sample!="AD69") %>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == Tube_code))%>%
    filter(time_point=="1_Acclimation"|time_point == "6_Post-FMT2")%>%
    mutate(functionid = substr(trait, 1, 3)) %>%
    mutate(trait = case_when(
      trait %in% GIFT_db3$Code_element ~ GIFT_db3$Element[match(trait, GIFT_db3$Code_element)],
      TRUE ~ trait
    )) %>%
    mutate(functionid = case_when(
      functionid %in% GIFT_db3$Code_function ~ GIFT_db3$Function[match(functionid, GIFT_db3$Code_function)],
      TRUE ~ functionid
    )) %>%
    mutate(trait=factor(trait,levels=unique(GIFT_db3$Element))) %>%
    mutate(functionid=factor(functionid,levels=unique(GIFT_db3$Function)))

# Create an interaction variable for time_point and sample
GIFTs_elements_community_merged$interaction_var <- interaction(GIFTs_elements_community_merged$sample, GIFTs_elements_community_merged$time_point)
  
ggplot(GIFTs_elements_community_merged,aes(x=interaction_var,y=trait,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(functionid ~ type, scales="free",space="free") +
        theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=5),
              strip.text.y = element_text(angle = 0)) + 
        labs(y="Traits",x="Time_point",fill="GIFT")+
  scale_x_discrete(labels = function(x) gsub(".*\\.", "", x))
```


## Wilcoxon comparison

### Community elements differences: in CC acclimation vs post2 ####

```{r comunity_elem, comment="", echo=FALSE, message=FALSE, warning=FALSE}
samples_CC <- sample_metadata%>%
  filter(type=="Control") %>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2")%>% 
  mutate(time_point = recode(time_point,
                        "1_Acclimation" = "Acclimation", 
                        "6_Post-FMT2" = "Post2"))

element_gift_CC <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(samples_CC[c(1,10)], by="Tube_code") 
```

```{r comunity_elem_filter, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_CC, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_CC[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
element_gift_CC_filter <- element_gift_CC[, !numeric_cols | colnames(element_gift_CC) %in% nonzero_numeric_cols]
```

```{r commun_wilcox_elem, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db3[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_CC <- element_gift_CC_filter %>%
  pivot_longer(-c(Tube_code,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_value < 0.05)%>% #note that p_value is used instead of p_adjust
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_CC <- element_gift_CC_filter  %>% 
  dplyr::select(-c(time_point))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_CC <- subset(element_gift_t_CC, trait %in% significant_elements_CC$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_CC[c(1,10)], by = join_by(Tube_code == Tube_code))

difference_table_CC <- element_gift_filt_CC %>%
  select(-Tube_code) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Post2)%>% 
  mutate(group_color = ifelse(Difference <0, "Post2","Acclimation")) 
```

```{r log_fold_calc, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_CC <- element_gift_filt_CC %>% 
  select(-time_point) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(samples_CC, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(time_point, elements) %>%
  summarise(mean=mean(abundance))

log_fold_CC <- means_gift_CC %>%
  group_by(elements) %>%
  summarise(
    logfc_accli_post2 = log2(mean[time_point == "Acclimation"] / mean[time_point == "Post2"])
  )
```


```{r plot1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
difference_table_CC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CC, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_value), color=group_color)) + #note that p_value was used instead of p_adjust
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
uniqueGIFT <- unique(GIFT_db3[c(2,3,4,5,6)])

code_function_CC <- difference_table_CC %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_CC<-unique(code_function_CC$Code_function)

gift_colors_CC <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_CC)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_CC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CC, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_CC, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_accli_post2, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_CC$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```
```{r final_plot_3, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_CC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CC, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_CC, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_accli_post2, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_CC$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
difference_table_CC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CC, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_accli_post2, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE, eval=TRUE}
difference_table_CC %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```
```{r}
difference_table_CC %>% 
  filter(group_color=="Acclimation")
```
```{r sugar_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
sugar_function <- GIFT_db3 %>% 
  filter(Code_function=="D03") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(sugar_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_CC[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>% 
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r poly_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
poly_function <- GIFT_db3 %>% 
  filter(Code_function=="D02") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(poly_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_CC[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>% 
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r elements_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_names_CC <- element_gift_filt_CC%>%
  select(-time_point)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_CC[c(1,10)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_CC)[2:16] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_CC, aes(x=time_point, y=.data[[i]], color = time_point, fill=time_point)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}

```

### Community elements differences: in CI acclimation vs post2 ####

```{r comunity_elem_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
samples_CI <- sample_metadata%>%
  filter(type=="Treatment") %>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2")%>% 
  mutate(time_point = recode(time_point,
                             "1_Acclimation" = "Acclimation", 
                             "6_Post-FMT2" = "Post2"))

element_gift_CI <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(samples_CI[c(1,10)], by="Tube_code") 
```

```{r comunity_elem_1_filter, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_CI, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_CI[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
element_gift_CI_filter <- element_gift_CI[, !numeric_cols | colnames(element_gift_CI) %in% nonzero_numeric_cols]
```

```{r commun_wilcox_elem_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db3[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_CI <- element_gift_CI_filter %>%
  pivot_longer(-c(Tube_code,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_CI <- element_gift_CI_filter  %>% 
  dplyr::select(-c(time_point))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_CI <- subset(element_gift_t_CI, trait %in% significant_elements_CI$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_CI[c(1,10)], by = join_by(Tube_code == Tube_code))

difference_table_CI <- element_gift_filt_CI %>%
  select(-Tube_code) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Post2)%>% 
  mutate(group_color = ifelse(Difference <0, "Post2","Acclimation")) 
```

```{r log_fold_calc_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_CI <- element_gift_filt_CI %>% 
  select(-time_point) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(samples_CI, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(time_point, elements) %>%
  summarise(mean=mean(abundance))

log_fold_CI <- means_gift_CI %>%
  group_by(elements) %>%
  summarise(
    logfc_acclimation_post2 = log2(mean[time_point == "Acclimation"] / mean[time_point == "Post2"])
  )
```


```{r plot1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=TRUE}
difference_table_CI %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CI, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_adjust), color=group_color)) + 
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
uniqueGIFT <- unique(GIFT_db3[c(2,3,4,5,6)])

code_function_CI <- difference_table_CI %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_CI<-unique(code_function_CI$Code_function)

gift_colors_CI <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_CI)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_CI %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CI, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CI, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_CI, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_acclimation_post2, y = -log(p_adjust), color=legend, size=abs(Difference))) +
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_CI$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```
```{r final_plot_1_1_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_CI %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CI, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CI, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_CI, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_acclimation_post2, y = -log(p_adjust), color=legend, size=abs(Difference))) + 
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_CI$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
difference_table_CI %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CI, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CI, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_acclimation_post2, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE, eval=TRUE}
difference_table_CI %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```
```{r_1}
difference_table_CI %>% 
  filter(group_color=="Acclimation")
```
```{r sugar_plot_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
sugar_function <- GIFT_db3 %>% 
  filter(Code_function=="D03") %>% 
  select(Code_element) %>% 
  pull()
GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(sugar_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_CI[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>%
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r poly_plot_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
poly_function <- GIFT_db3 %>% 
  filter(Code_function=="D02") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(poly_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_CI[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>% 
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r elements_plot_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_names_CI <- element_gift_filt_CI%>%
  select(-time_point)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_CI[c(1,10)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_CI)[2:17] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_CI, aes(x=time_point, y=.data[[i]], color = time_point, fill=time_point)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}

```

### Community elements differences: in WC acclimation vs post2 ####
```{r comunity_elem_2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
samples_WC <- sample_metadata%>%
  filter(type=="Hot_control") %>% 
  filter(time_point == "1_Acclimation"|time_point == "6_Post-FMT2")%>% 
  mutate(time_point = recode(time_point,
                             "1_Acclimation" = "Acclimation", 
                             "6_Post-FMT2" = "Post2"))

element_gift_WC <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(samples_WC[c(1,10)], by="Tube_code") 
```

```{r comunity_elem_2_filter, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_WC, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_WC[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
element_gift_WC_filter <- element_gift_WC[, !numeric_cols | colnames(element_gift_WC) %in% nonzero_numeric_cols]
```

```{r commun_wilcox_elem_2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db3[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_WC <- element_gift_WC_filter %>%
  pivot_longer(-c(Tube_code,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_value < 0.05)%>% #note that p_value is used instead of p_adjust
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_WC <- element_gift_WC_filter  %>% 
  dplyr::select(-c(time_point))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_WC <- subset(element_gift_t_WC, trait %in% significant_elements_WC$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_WC[c(1,10)], by = join_by(Tube_code == Tube_code))

difference_table_WC <- element_gift_filt_WC %>%
  select(-Tube_code) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Post2)%>% 
  mutate(group_color = ifelse(Difference <0, "Post2","Acclimation")) 
```

```{r log_fold_calc_2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_WC <- element_gift_filt_WC %>% 
  select(-time_point) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(samples_WC, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(time_point, elements) %>%
  summarise(mean=mean(abundance))

log_fold_WC <- means_gift_WC %>%
  group_by(elements) %>%
  summarise(
    logfc_acclimation_post2 = log2(mean[time_point == "Acclimation"] / mean[time_point == "Post2"])
  )
```


```{r plot1_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
difference_table_WC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_WC, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_value), color=group_color)) + #note that p_value was used instead of p_adjust
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
uniqueGIFT <- unique(GIFT_db3[c(2,3,4,5,6)])

code_function_WC <- difference_table_WC %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_WC<-unique(code_function_WC$Code_function)

gift_colors_WC <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_WC)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_WC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_WC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_WC, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_WC, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_acclimation_post2, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_WC$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```

```{r final_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_WC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_WC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_WC, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_WC, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_acclimation_post2, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_WC$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}
difference_table_WC %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_WC, by=join_by(Elements==trait)) %>%
  left_join(log_fold_WC, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_acclimation_post2, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE, eval=TRUE}
difference_table_WC %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

```{r_2}
difference_table_WC %>% 
  filter(group_color=="Acclimation")
```

```{r sugar_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
sugar_function <- GIFT_db3 %>% 
  filter(Code_function=="D03") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(sugar_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_WC[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>%
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r poly_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
poly_function <- GIFT_db3 %>% 
  filter(Code_function=="D02") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(poly_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(samples_WC[c(1,10)], by=join_by(Tube_code==Tube_code)) %>%
  filter(time_point!="NA") %>% 
  select(-Tube_code) %>% 
  pivot_longer(!time_point,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=time_point, y=gift, color=time_point, fill=time_point)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

```{r elements_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_names_WC <- element_gift_filt_WC%>%
  select(-time_point)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., samples_WC[c(1,10)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_WC)[2:10] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_WC, aes(x=time_point, y=.data[[i]], color = time_point, fill=time_point)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

### Comparison of both population in wild samples ####

```{r comunity_elem_&, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_wild<- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_wild[c(1,4)], by="Tube_code") 
```

```{r commun_wilcox_elem_6_filter, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_wild, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_wild[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
element_gift_wild_filter <- element_gift_wild[, !numeric_cols | colnames(element_gift_wild) %in% nonzero_numeric_cols]
```

```{r commun_wilcox_elem_6, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db3[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_wild <- element_gift_wild_filter %>%
  pivot_longer(-c(Tube_code,Population), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ Population)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_wild <- element_gift_wild_filter  %>% 
  dplyr::select(-c(Population))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_wild <- subset(element_gift_t_wild, trait %in% significant_elements_wild$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_wild[c(1,4)], by = join_by(Tube_code == Tube_code))

difference_table_wild <- element_gift_filt_wild %>%
  select(-Tube_code) %>%
  group_by(Population) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Cold_wet-Hot_dry)%>% 
  mutate(group_color = ifelse(Difference <0, "Hot","Cold")) 
```

```{r log_fold_calc_6, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_wild <- element_gift_filt_wild %>% 
  select(-Population) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(sample_metadata_wild, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(Population, elements) %>%
  summarise(mean=mean(abundance))

log_fold_wild <- means_gift_wild %>%
  group_by(elements) %>%
  summarise(
    logfc_Cold_hot = log2(mean[Population == "Cold_wet"] / mean[Population == "Hot_dry"])
  )
```


```{r plot1_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_wild %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_wild, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_adjust), color=group_color)) + 
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
uniqueGIFT <- unique(GIFT_db3[c(2,3,4,5,6)])

code_function_wild <- difference_table_wild %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_wild<-unique(code_function_wild$Code_function)

gift_colors_wild <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_wild)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_wild %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_wild, by=join_by(Elements==trait)) %>%
  left_join(log_fold_wild, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_wild, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_Cold_hot, y = -log(p_adjust), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_wild$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```
```{r final_plot_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_wild %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_wild, by=join_by(Elements==trait)) %>%
  left_join(log_fold_wild, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_wild, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_Cold_hot, y = -log(p_adjust), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_wild$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_wild %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_wild, by=join_by(Elements==trait)) %>%
  left_join(log_fold_wild, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_Cold_hot, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table_wild %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```
```{r}
difference_table_WC %>% 
  filter(group_color=="Acclimation")
```
```{r sugar_plot_6, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
sugar_function <- GIFT_db3 %>% 
  filter(Code_function=="D03") %>% 
  select(Code_element) %>% 
  pull()
```

```{r sugar_plot_8, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
poly_function <- GIFT_db3 %>% 
  filter(Code_function=="D02") %>% 
  select(Code_element) %>% 
  pull()
```

```{r elements_plot_6, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_names_wild <- element_gift_filt_wild%>%
  select(-Population)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_wild[c(1,4)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_wild)[2:73] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_wild, aes(x=Population, y=.data[[i]], color = Population, fill=Population)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}

```

####Butiryc acid biosynthesis

```{r commun_wilcox_elem_plot_9, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE, eval=FALSE}
lipid_function <- GIFT_db3 %>% 
  filter(Code_function=="D01") %>% 
  select(Code_element) %>% 
  pull()

GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(any_of(lipid_function)) %>% 
  rownames_to_column(., "Tube_code") %>% 
  left_join(sample_metadata[c(1,4)], by=join_by(Tube_code==Tube_code)) %>%
  filter(Population!="NA") %>% 
  select(-Tube_code) %>% 
  pivot_longer(!Population,names_to="trait",values_to="gift") %>% 
  left_join(uniqueGIFT_db, by=join_by(trait==Code_element)) %>% 
  ggplot(aes(x=Population, y=gift, color=Population, fill=Population)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.3,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_fill_manual(values=c("#e5bd5b50", "#6b739850"))+
  facet_grid(.~Element,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
  )
```

### Comparison of both population in acclimation samples ####

```{r comunity_elem_7, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_accli<- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_accli[c(1,4)], by="Tube_code") 
```

```{r comunity_elem_7_accli_filter, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_accli, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_accli[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
element_gift_accli_filter <- element_gift_accli[, !numeric_cols | colnames(element_gift_accli) %in% nonzero_numeric_cols]
```

```{r commun_wilcox_elem_7, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db3[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_accli <- element_gift_accli_filter %>%
  pivot_longer(-c(Tube_code,Population), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ Population)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_accli <- element_gift_accli_filter  %>% 
  dplyr::select(-c(Population))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_accli <- subset(element_gift_t_accli, trait %in% significant_elements_accli$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_accli[c(1,4)], by = join_by(Tube_code == Tube_code))

difference_table_accli <- element_gift_filt_accli %>%
  select(-Tube_code) %>%
  group_by(Population) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Cold_wet-Hot_dry)%>% 
  mutate(group_color = ifelse(Difference <0, "Hot","Cold")) 
```

```{r log_fold_calc_7, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_accli <- element_gift_filt_accli %>% 
  select(-Population) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(sample_metadata_accli, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(Population, elements) %>%
  summarise(mean=mean(abundance))

log_fold_accli <- means_gift_accli %>%
  group_by(elements) %>%
  summarise(
    logfc_Cold_hot = log2(mean[Population == "Cold_wet"] / mean[Population == "Hot_dry"])
  )
```


```{r plot1_7, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_accli %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_accli, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_adjust), color=group_color)) + 
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_7, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
uniqueGIFT <- unique(GIFT_db3[c(2,3,4,5,6)])

code_function_accli <- difference_table_accli %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_accli<-unique(code_function_accli$Code_function)

gift_colors_accli <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_accli)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_accli %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_accli, by=join_by(Elements==trait)) %>%
  left_join(log_fold_accli, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_accli, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_Cold_hot, y = -log(p_adjust), color=legend, size=abs(Difference)))+
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_accli$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```

```{r final_plot_7, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_accli %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_accli, by=join_by(Elements==trait)) %>%
  left_join(log_fold_accli, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_accli, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_Cold_hot, y = -log(p_adjust), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_accli$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_7, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_accli %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_accli, by=join_by(Elements==trait)) %>%
  left_join(log_fold_accli, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_Cold_hot, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_10_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table_accli %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

```{r commun_wilcox_elem_plot_7, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
element_gift_names_accli <- element_gift_filt_accli%>%
  select(-Population)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_accli[c(1,4)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_accli)[2:53] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_accli, aes(x=Population, y=.data[[i]], color = Population, fill=Population)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

### Comparison of CC and CI in post2 samples ####

```{r comunity_elem_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_post2<- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_TM6[c(1,7)], by="Tube_code") 
```

```{r comunity_elem_7_1_filter, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_post2, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_post2[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
element_gift_post2_filter <- element_gift_post2[, !numeric_cols | colnames(element_gift_post2) %in% nonzero_numeric_cols]
```

```{r commun_wilcox_elem_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db3[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_post2 <- element_gift_post2_filter %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_value < 0.05)%>% #note p_value was used
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_post2 <- element_gift_post2_filter  %>% 
  dplyr::select(-c(type))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_post2 <- subset(element_gift_t_post2, trait %in% significant_elements_post2$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM6[c(1,7)], by = join_by(Tube_code == Tube_code))

difference_table_post2 <- element_gift_filt_post2 %>%
  select(-Tube_code) %>%
  group_by(type) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Control-Treatment)%>% 
  mutate(group_color = ifelse(Difference <0, "Treatment","Control")) 
```

```{r log_fold_calc_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_post2 <- element_gift_filt_post2 %>% 
  select(-type) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(sample_metadata_TM6, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(type, elements) %>%
  summarise(mean=mean(abundance))

log_fold_post2 <- means_gift_post2 %>%
  group_by(elements) %>%
  summarise(
    logfc_control_treatment = log2(mean[type == "Control"] / mean[type == "Treatment"])
  )
```


```{r plot1_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_post2 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_value), color=group_color)) + 
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
uniqueGIFT <- unique(GIFT_db3[c(2,3,4,5,6)])

code_function_post2 <- difference_table_post2 %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_post2<-unique(code_function_post2$Code_function)

gift_colors_post2 <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_post2)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_post2 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_post2, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_control_treatment, y = -log(p_adjust), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_post2$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```

```{r final_plot_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_post2 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_post2, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_control_treatment, y = -log(p_adjust), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_post2$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_post2 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_control_treatment, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_10, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table_post2 %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

```{r commun_wilcox_elem_plot_7_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
element_gift_names_post2 <- element_gift_filt_post2%>%
  select(-type)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM6[c(1,7)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_post2)[2:3] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_post2, aes(x=type, y=.data[[i]], color = type, fill=type)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

### Comparison of CI and WC in post2 samples ####

```{r comunity_elem_7_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_post2_1<- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_TM8[c(1,7)], by="Tube_code") 
```

```{r comunity_elem_7_1_filter_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_post2_1, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_post2_1[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
element_gift_post2_1_filter <- element_gift_post2_1[, !numeric_cols | colnames(element_gift_post2_1) %in% nonzero_numeric_cols]
```

```{r commun_wilcox_elem_7_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db3[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_post2_1 <- element_gift_post2_1_filter %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_value < 0.05)%>% #note than p_value was used instead of p_adjust
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_post2_1 <- element_gift_post2_1_filter  %>% 
  dplyr::select(-c(type))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_post2_1 <- subset(element_gift_t_post2_1, trait %in% significant_elements_post2_1$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM8[c(1,7)], by = join_by(Tube_code == Tube_code))

difference_table_post2_1 <- element_gift_filt_post2_1 %>%
  select(-Tube_code) %>%
  group_by(type) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Hot_control-Treatment)%>% 
  mutate(group_color = ifelse(Difference <0, "Treatment","Hot_control")) 
```

```{r log_fold_calc_7_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_post2_1 <- element_gift_filt_post2_1 %>% 
  select(-type) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(sample_metadata_TM8, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(type, elements) %>%
  summarise(mean=mean(abundance))

log_fold_post2_1 <- means_gift_post2_1 %>%
  group_by(elements) %>%
  summarise(
    logfc_control_treatment = log2(mean[type == "Hot_control"] / mean[type == "Treatment"])
  )
```


```{r plot1_7_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_post2_1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2_1, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_value), color=group_color)) + 
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_7_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
uniqueGIFT <- unique(GIFT_db3[c(2,3,4,5,6)])

code_function_post2_1 <- difference_table_post2_1 %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_post2_1<-unique(code_function_post2_1$Code_function)

gift_colors_post2_1 <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_post2_1)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_post2_1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2_1, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2_1, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_post2_1, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_control_treatment, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_post2_1$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```

```{r final_plot_7_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_post2_1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2_1, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2_1, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_post2_1, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_control_treatment, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_post2_1$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_7_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_post2_1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2_1, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2_1, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_control_treatment, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_10_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table_post2_1 %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

```{r commun_wilcox_elem_plot_7_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
element_gift_names_post2_1 <- element_gift_filt_post2_1%>%
  select(-type)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM8[c(1,7)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_post2_1)[2:13] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_post2_1, aes(x=type, y=.data[[i]], color = type, fill=type)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

### Comparison of CI in accli and post1 samples ####

```{r comunity_elem_8_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
sample_metadata_CI_accli_post1 <- sample_metadata%>% 
  filter(time_point == "5_Post-FMT1" | time_point == "1_Acclimation") %>% 
  filter(type == "Treatment")%>% 
  mutate(time_point = recode(time_point,
                        "1_Acclimation" = "Acclimation", 
                        "5_Post-FMT1" = "Post1"))

element_gift_CI_accli_post1<- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_CI_accli_post1[c(1,10)], by="Tube_code") 
```

```{r comunity_elem_8_1_filter_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_CI_accli_post1, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_CI_accli_post1[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
element_gift_CI_accli_post1_filter <- element_gift_CI_accli_post1[, !numeric_cols | colnames(element_gift_CI_accli_post1) %in% nonzero_numeric_cols]
```

```{r commun_wilcox_elem_7_1_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db3[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_CI_accli_post1 <- element_gift_CI_accli_post1_filter %>%
  pivot_longer(-c(Tube_code,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>% #note than p_value was used instead of p_adjust
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_CI_accli_post1 <- element_gift_CI_accli_post1_filter  %>% 
  dplyr::select(-c(time_point))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_CI_accli_post1  <- subset(element_gift_t_CI_accli_post1, trait %in% significant_elements_CI_accli_post1$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_CI_accli_post1[c(1,10)], by = join_by(Tube_code == Tube_code))

difference_table_CI_accli_post1 <- element_gift_filt_CI_accli_post1 %>%
  select(-Tube_code) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Post1)%>% 
  mutate(group_color = ifelse(Difference <0, "Post1","Acclimation")) 
```

```{r log_fold_calc_7_1_1_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_post2_1 <- element_gift_filt_post2_1 %>% 
  select(-type) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(sample_metadata_TM8, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(type, elements) %>%
  summarise(mean=mean(abundance))

log_fold_post2_1 <- means_gift_post2_1 %>%
  group_by(elements) %>%
  summarise(
    logfc_control_treatment = log2(mean[type == "Hot_control"] / mean[type == "Treatment"])
  )
```


```{r plot1_7_1_1_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_post2_1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2_1, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_value), color=group_color)) + 
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_7_1_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
uniqueGIFT <- unique(GIFT_db3[c(2,3,4,5,6)])

code_function_post2_1 <- difference_table_post2_1 %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_post2_1<-unique(code_function_post2_1$Code_function)

gift_colors_post2_1 <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_post2_1)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_post2_1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2_1, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2_1, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_post2_1, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_control_treatment, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_post2_1$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```

```{r final_plot_7_1_3, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_post2_1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2_1, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2_1, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_post2_1, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_control_treatment, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_post2_1$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_7_1_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_post2_1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2_1, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2_1, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_control_treatment, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_10_1_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table_post2_1 %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

```{r commun_wilcox_elem_plot_7_1_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
element_gift_names_post2_1 <- element_gift_filt_post2_1%>%
  select(-type)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM8[c(1,7)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_post2_1)[2:13] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_post2_1, aes(x=type, y=.data[[i]], color = type, fill=type)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```

### Comparison of CC in accli and post1 samples ####

```{r comunity_elem_9_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
sample_metadata_CC_accli_post1 <- sample_metadata%>% 
  filter(time_point == "5_Post-FMT1" | time_point == "1_Acclimation") %>% 
  filter(type == "Control")%>% 
  mutate(time_point = recode(time_point,
                        "1_Acclimation" = "Acclimation", 
                        "5_Post-FMT1" = "Post1"))

element_gift_CC_accli_post1<- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "Tube_code") %>% 
  inner_join(sample_metadata_CC_accli_post1[c(1,10)], by="Tube_code") 
```

```{r comunity_elem_9_1_filter_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Find numeric columns
numeric_cols <- sapply(element_gift_CC_accli_post1, is.numeric)

# Calculate column sums for numeric columns only
col_sums_numeric <- colSums(element_gift_CC_accli_post1[, numeric_cols])

# Identify numeric columns with sums not equal to zero
nonzero_numeric_cols <- names(col_sums_numeric)[col_sums_numeric != 0]

# Remove numeric columns with sums not equal to zero
element_gift_CC_accli_post1_filter <- element_gift_CC_accli_post1[, !numeric_cols | colnames(element_gift_CC_accli_post1) %in% nonzero_numeric_cols]
```

```{r commun_wilcox_elem_9_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db3[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_CC_accli_post1 <- element_gift_CC_accli_post1_filter %>%
  pivot_longer(-c(Tube_code,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>% #note than p_value was used instead of p_adjust
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)
# %>%
#   left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_CC_accli_post1 <- element_gift_CC_accli_post1_filter  %>% 
  dplyr::select(-c(time_point))  %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_CC_accli_post1  <- subset(element_gift_t_CC_accli_post1, trait %in% significant_elements_CC_accli_post1$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_CC_accli_post1[c(1,10)], by = join_by(Tube_code == Tube_code))

difference_table_CC_accli_post1 <- element_gift_filt_CC_accli_post1 %>%
  select(-Tube_code) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Post1)%>% 
  mutate(group_color = ifelse(Difference <0, "Post1","Acclimation")) 
```

```{r log_fold_calc_7_1_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
means_gift_CC_accli_post1 <- element_gift_filt_CC_accli_post1 %>% 
  select(-time_point) %>% 
  pivot_longer(!Tube_code, names_to = "elements", values_to = "abundance") %>% 
  left_join(sample_metadata_CC_accli_post1, by=join_by(Tube_code==Tube_code)) %>% 
  group_by(time_point, elements) %>%
  summarise(mean=mean(abundance))

log_fold_CC_accli_post1 <- means_gift_CC_accli_post1 %>%
  group_by(elements) %>%
  summarise(
    logfc_accli_post1 = log2(mean[time_point == "Acclimation"] / mean[time_point == "Post1"])
  )
```


```{r plot1_7_1_1_1, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_CC_accli_post1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CC_accli_post1, by=join_by(Elements==trait)) %>% 
  ggplot(., aes(x = Difference, y = -log(p_adjust), color=group_color)) + 
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2)
```

```{r plot3_7_1_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
uniqueGIFT <- unique(GIFT_db3[c(2,3,4,5,6)])

code_function_post2_1 <- difference_table_post2_1 %>%
  left_join(uniqueGIFT[c(1:3)], by=join_by(Elements==Code_element))

unique_codes_post2_1<-unique(code_function_post2_1$Code_function)

gift_colors_post2_1 <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes_post2_1)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

code_function_post2_1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2_1, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2_1, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_post2_1, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_control_treatment, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_point()+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_post2_1$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log p-value")

# +
#   geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)

```

```{r final_plot_7_1_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
code_function_post2_1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_post2_1, by=join_by(Elements==trait)) %>%
  left_join(log_fold_post2_1, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors_post2_1, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_control_treatment, y = -log(p_value), color=legend, size=abs(Difference))) + #note that p_value was used instead of p_adjust
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors_post2_1$Color)+
  #xlim(c(-10,4)) +
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```

```{r plot4_7_1_3, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
difference_table_CC_accli_post1 %>%
  #  mutate(Difference_abs = abs(Difference)) %>% 
  left_join(significant_elements_CC_accli_post1, by=join_by(Elements==trait)) %>%
  left_join(log_fold_CC_accli_post1, by=join_by(Elements==elements)) %>% 
  ggplot(., aes(x = Difference, y = logfc_accli_post1, color=group_color)) +
  geom_point(size=2, show.legend = FALSE) + 
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  #xlim(c(-10,4)) +
  theme_classic()+
  geom_text_repel(aes(label = Elements), size=2, max.overlaps = Inf)
```

```{r commun_wilcox_elem_plot_10_1_3, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table_CC_accli_post1 %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  #  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

```{r commun_wilcox_elem_plot_7_1_3, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
element_gift_names_post2_1 <- element_gift_filt_post2_1%>%
  select(-type)%>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
  t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Tube_code")%>% 
  left_join(., sample_metadata_TM8[c(1,7)], by = join_by(Tube_code == Tube_code))


colNames <- names(element_gift_names_post2_1)[2:13] #check the column that has the last code
for(i in colNames){
  plt <- ggplot(element_gift_names_post2_1, aes(x=type, y=.data[[i]], color = type, fill=type)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
    geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
    theme_minimal() +
    theme(
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank())
  print(plt)
}
```


<!--chapter:end:07_differential_analysis.Rmd-->

# Physiological measurements

## Respirometry

```{r load_data_resp}
load("data/data.Rdata")
respirometry_resp2 <- read_excel("data/respirometry_resp2.xlsx")
```

```{r dataframe_modify, warning=FALSE}
respirometry_resp2$individual<-as.factor(respirometry_resp2$individual)
respirometry_resp2$cage<-as.factor(respirometry_resp2$cage)
respirometry_resp2$time_point<-as.factor(respirometry_resp2$time_point)
respirometry_resp2$type<-as.factor(respirometry_resp2$type)
respirometry_resp2$population<-as.factor(respirometry_resp2$population)
respirometry_resp2$QC_normalized<-as.numeric(respirometry_resp2$QC_normalized)
respirometry_resp2$r2<-as.numeric(respirometry_resp2$r2)


#Filter NA values
respirometry_resp2<-respirometry_resp2 %>%
  filter(QC_normalized!="NA")
```

### Acclimation oxygen consumption

```{r plot}
respirometry_resp2 %>%
  filter(time_point=="0")%>%
  filter(individual!="AH1_2nd_8") %>%
  filter(individual!="LI1_2nd_4") %>%
  filter(individual!="LI1_2nd_3") %>% #lizard doesn't stop moving during the whole measurement, increasing the value and making it and outlier
  filter(individual!="LI1_2nd_10") %>%
  ggplot(aes(x = population, y = QC_normalized, color=population, fill=population, alpha=0.2)) +
    geom_boxplot()+
    geom_jitter() +
      scale_color_manual(name="Population",
          breaks=c("Cold","Warm"),
          labels=c("Cold","Warm"),
          values=c('#008080', "#d57d2c")) +
      scale_fill_manual(name="Population",
          breaks=c("Cold","Warm"),
          labels=c("Cold","Warm"),
          values=c('#00808050', "#d57d2c50"))+
    #geom_text(aes(label = individual), hjust = 0, nudge_x = 0.05) +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=1))+
    stat_compare_means(size=3,vjust = 3, hjust=1 )+
    theme(legend.position="none",
          panel.background = element_blank(),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank())+
    theme_minimal()+
    labs(x = "Population", y= "Normalized O2 consumption (ml/s/g)")
```

### Time-point oxygen consumption

```{r plot_1}
respirometry_resp2 %>%
  filter(weight!=5.19)%>%
  filter(weight!=5.91)%>%
  filter(weight!=5.03)%>%
  filter(r2!=0.808)%>%
  filter(QC_normalized!=0.000036) %>% #r2 low
  filter(r2!=0.728) %>%  #r2 low
  ggplot(aes(x = time_point, y = QC_normalized, color=time_point, fill=time_point, alpha=0.2)) +
    geom_boxplot()+
    geom_jitter() +
      scale_color_manual(name="Time_point",
          breaks=c("0","1","2"),
          labels=c("Acclimation","1 week post FMT", "2 weeks post FMT"),
          values=c('#BFA366', "#dec14b", "#6E5244")) +
      scale_fill_manual(name="Time_point",
          breaks=c("0","1","2"),
          labels=c("Acclimation","1 week post FMT", "2 weeks post FMT"),
          values=c('#BFA36650', "#dec14b50","#6E524450"))+
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
    facet_wrap(~ factor(type))+
    stat_compare_means(size=3)+
    theme(legend.position="none")+
    labs(x = "Time_point", y= "Normalized O2 consumption (ml/s/g)")
```

```{r plot_1_1}
respirometry_resp2 %>%
  filter(weight!=5.19)%>%
  filter(weight!=5.91)%>%
  filter(weight!=5.03)%>%
  filter(r2!=0.808)%>%
  filter(QC_normalized!=0.000036) %>% #r2 low
  filter(r2!=0.728) %>%
  ggplot(aes(x = time_point, y = QC_normalized, color=time_point, fill=time_point, alpha=0.2)) +
    geom_boxplot()+
    geom_jitter() +
    scale_color_manual(name="Time_point",
          breaks=c("0","1","2"),
          labels=c("Acclimation","1 week post FMT", "2 weeks post FMT"),
          values=c('#BFA366', "#dec14b", "#6E5244")) +
      scale_fill_manual(name="Time_point",
          breaks=c("0","1","2"),
          labels=c("Acclimation","1 week post FMT", "2 weeks post FMT"),
          values=c('#BFA36650', "#dec14b50","#6E524450"))+
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
    facet_wrap(~ factor(type))+
    stat_compare_means(size=3)+
    theme(legend.position="none")+
    labs(x = "Time_point", y= "Normalized O2 consumption (ml/s/g)")+
    geom_signif(comparisons = list(c("0", "1"), c("0", "2"), c("1", "2")),
              map_signif_level = TRUE)
```

### Data modelling 

#### Raw data modelling (LMM)

```{r model, eval=TRUE}
#subset all the wrong measurements
respirometry_resp2_subset<-respirometry_resp2%>%
  filter(weight!=5.19)%>%
  filter(weight!=5.91)%>%
  filter(weight!=5.03)%>%
  filter(r2!=0.808)%>%
  filter(QC_normalized!=0.000036) %>% #r2 low
  filter(r2!=0.728)#r2 low

#Create linear model formula
model <- lme(fixed = QC_normalized ~  time_point*type, data = respirometry_resp2_subset,
               random = ~ 1 | individual)

#Print the model summary
summary(model)
MuMIn::r.squaredGLMM(model)
```


#### Raw data modelling (LM) without random effect

```{r model_1}
#Create linear model formula
formula1 <- formula(QC_normalized ~ time_point*type) #QC_normalized=o2 consumption normalized (ml/s)

#Fit the linear model
model1 <- lm(formula1, data = respirometry_resp2)

#Print the model summary
summary(model1)
```

### Data modeling without WC

#### Raw data modelling (LMM)

```{r model_2}
#Filter out WC from the dataframe and create a subset
respirometry_resp2_cold<-respirometry_resp2%>%
  filter(type!="WC")%>%
  filter(QC_normalized!=0.000036) %>% #r2 low
  filter(r2!=0.728) #r2 low


model4 <- lme(fixed = QC_normalized ~  time_point*type, data = respirometry_resp2_cold,
               random = ~ 1 | individual)

#Print the model summary
summary(model4)
MuMIn::r.squaredGLMM(model4)
```


#### Raw data modelling (LM) without random effect

```{r model_3}
#Create linear model formula
formula5 <- formula(QC_normalized ~ time_point*type)

#Fit the linear model
model5 <- lm(formula5, data = respirometry_resp2_cold)

#Print the model summary
summary(model5)
```


## PHA test

```{r load_data_pha, warning=FALSE, comment=FALSE}
pha_rep2 <- read_tsv("data/Fito_2023.tsv")
```

```{r dataframe_modify_2}
pha_rep2$Individual<-as.factor(pha_rep2$Individual)
pha_rep2$Cage<-as.factor(pha_rep2$Cage)
pha_rep2$time_point<-as.factor(pha_rep2$time_point)
pha_rep2$Type<-as.factor(pha_rep2$Type)
```

### PHA measurements

```{r plot_pha}
pha_rep2 %>%
  ggplot(aes(x = Weight, y = Mean_value)) +
    geom_point() +
    geom_smooth(method = lm, formula = y ~ x) +
        stat_poly_eq()+
  facet_nested(~ time_point+Type)+
    labs(x = "Weight (g)", y="Measurement (mm)")
```

#### Average PHA measurements

```{r pha_measurement}
pha_rep2 %>%
  rowwise() %>%
  mutate(
    mean = round(mean(c(Value1, Value2, Value3, Value4, Value5)), 2),
    sd = round(sd(c(Value1, Value2, Value3, Value4, Value5)), 2)
  ) %>%
  unite("Average", mean, sd, sep = " ± ", remove = TRUE) %>%
  select(Individual, time_point, Average) %>%
    tt()
```

### Time-point measurements

```{r plot_pha_5}
pha_rep2 %>%
  ggplot(aes(x = time_point, y = normalized, color=time_point, fill=time_point, alpha=0.2 )) +
  geom_boxplot()+
  geom_jitter() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  scale_color_manual(name="Time_point",
          breaks=c("0","1"),
          labels=c("Pre","24h"),
          values=c('#BFA366', "#dec14b")) +
      scale_fill_manual(name="Time_point",
          breaks=c("0","1"),
          labels=c("Pre","24h"),
          values=c('#BFA36650', "#dec14b50"))+
  facet_wrap(~ factor(Type))+
  stat_compare_means(size=3)+
  theme(legend.position="none")+
  labs(x = "Time_point", y= "Normalized measurement (mm/g)")
```

### Time-point measurements, linking individuals

```{r plot_pha_1, warning=FALSE}
pha_rep2 %>%
  ggplot(aes(x = interaction(time_point, Individual), y = normalized, color=time_point, fill=time_point, alpha=0.2 )) +
  geom_boxplot()+
  geom_point(position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8), alpha = 0.5) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  geom_line(aes(group=Individual)) +
  scale_color_manual(name="Time_point",
          breaks=c("0","1"),
          labels=c("Pre","24h"),
          values=c('#BFA366', "#dec14b")) +
      scale_fill_manual(name="Time_point",
          breaks=c("0","1"),
          labels=c("Pre","24h"),
          values=c('#BFA36650', "#dec14b50"))+
  facet_wrap(~ factor(Type),scales = "free")+
  theme(legend.position="none")+
  labs(x = "Time_point", y= "Normalized measurement (mm/g)")
```


### Data modelling

#### Raw data modelling (LMM)

```{r model_pha}
#Create linear model formula
model6 <- lme(fixed = normalized ~  time_point*Type, data = pha_rep2,
               random = ~ 1 | Individual)

#Print the model summary
summary(model6)
MuMIn::r.squaredGLMM(model6)
```

#### Raw data modelling (LM) without random effect

```{r model_pha_1}
#Create linear model formula
formula7 <- formula(normalized ~ time_point*Type)

#Fit the linear model
model7 <- lm(formula7, data = pha_rep2)

#Print the model summary
summary(model7)
```

### Data modeling without WC

#### Raw data modelling (LMM)

```{r model_pha_2}
#Filter out WC measurements
pha_rep2_cold<-pha_rep2 %>%
  filter(Type!="WC")

#Create linear model formula
model8 <- lme(fixed = normalized ~  time_point*Type, data = pha_rep2_cold,
               random = ~ 1 | Individual)

#Print the model summary
summary(model8)
MuMIn::r.squaredGLMM(model8)
```

#### Raw data modelling (LM) without random effect

```{r model_pha2_1}
#Create linear model formula
formula9 <- formula(normalized ~ time_point*Type)

#Fit the linear model
model9 <- lm(formula9, data = pha_rep2_cold)

#Print the model summary
summary(model9)
```

<!--chapter:end:08_physiology.Rmd-->

