---
title: "AlberdiLab | Martin-Bideguren et al. 2025"
subtitle: "Study title to be added"
author:
  - Garazi Martin-Bideguren^[University of Copenhagen, garazi.bideguren@sund.ku.dk],Carlos Cabido^[Sociedad de Ciencias Aranzadi-Departamento de Herpetología, ccabido@aranzadi.eus], Kevin Kohl^[University of Pittsburgh, kkohl78@gmail.com], Antton Alberdi^[University of Copenhagen, antton.alberdi@sund.ku.dk] and Ostaizka Aizpurua^[University of Copenhagen, ostaizka.aizpurua@sund.ku.dk]
date: "Last update: `r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
url: https://alberdilab.github.io/Podarcis_adaptation
description: |
  Data analysis code for the study on the recovery of metagenome‑assembled genomes and derived microbial communities from lizard faecal samples.
link-citations: yes
github-repo: alberdilab/Podarcis_adaptation
---

```{r knitr_opts, echo=FALSE}
knitr::opts_chunk$set(
    class.source = "script-source",
    class.output = "script-output",
    comment = NA)
```

# Introduction

This webbook contains all the code used for data analysis in study of the individual-level metagenomic data of Podarcis muralis and Podarcis liolepis lizards from different environments during an experimental setup.

## Prepare the R environment

### Environment

To reproduce all the analyses locally, clone this repository in your computer using:

```
RStudio > New Project > Version Control > Git
```

And indicating the following git repository:

> https://github.com/alberdilab/Podarcis_adaptation.git

Once the R project has been created, follow the instructions and code chunks shown in this webbook.

### Libraries

The following R packages are required for the data analysis.

```{r load_libraries, warning=FALSE, comments="", message=FALSE}
# Base
library(R.utils)
library(knitr)
library(tidyverse)
library(devtools)
library(tinytable)
library(janitor)
library(readxl)

# For tree handling
library(ape)
library(phyloseq)
library(phytools)

# For plotting
library(ggplot2)
library(ggrepel)
library(ggpubr)
library(ggnewscale)
library(gridExtra)
library(ggtreeExtra)
library(ggtree)
library(ggh4x)
library(rstatix)
#library(ggpmisc)

# For statistics
library(spaa)
library(vegan)
library(Rtsne)
library(geiger)
library(hilldiv2)
library(distillR)
library(broom.mixed)
library(corrplot)
library(nlme)
library(pairwiseAdonis)
library(lme4)
library(emmeans)
library(UpSetR)
library(ANCOMBC)
#library(MASS)
```

<!--chapter:end:index.Rmd-->

# Prepare data

## Load data

Load the original data files outputted by the bioinformatic pipeline.

### Sample metadata

```{r load_sample_metadata, warning=FALSE, comments="", message=FALSE, eval=FALSE}
sample_metadata <- read_tsv("data/metadata.tsv") %>%
  mutate(time_point = sub("^\\d+_", "", time_point)) %>% 
  filter(!time_point %in% c("Antibiotics","Transplant1", "Transplant2")) %>% 
  filter(individual!="LI1_2nd_6") %>% 
  mutate(time_point=str_remove_all(time_point, "Post-"))
```

### Read counts

```{r load_read_counts, warning=FALSE, comments="", message=FALSE, eval=FALSE}
read_counts_raw <- read_tsv("data/genome.count.tsv") %>%
    rename(genome=1)

#Transformation of read_counts to combine data from both sequence rounds
merge_and_rename <- function(read_counts_raw) {
  read_counts_raw %>%
    # Gather the columns into long format
    pivot_longer(cols = -genome, names_to = "col") %>%
    # Extract prefix
    mutate(prefix = gsub("^(.*?)_.*", "\\1", col)) %>%
    # Group by prefix and genome, then summarize
    group_by(prefix, genome) %>%
    summarize(value = sum(value)) %>%
    # Spread the data back to wide format
    pivot_wider(names_from = prefix, values_from = value)
}

read_counts <- merge_and_rename(read_counts_raw) %>% 
    select(c("genome",sample_metadata$Tube_code))
```

### Genome base hits

```{r load_genome_hits, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_hits_raw <- read_tsv("data/genome.covered_bases.tsv") %>%
    rename(genome=1)

#Transformation of genome_hits to combine data from both sequence rounds
merge_and_rename <- function(genome_hits_raw) {
  genome_hits_raw %>%
    # Gather the columns into long format
    pivot_longer(cols = -genome, names_to = "col") %>%
    # Extract prefix
    mutate(prefix = gsub("^(.*?)_.*", "\\1", col)) %>%
    # Group by prefix and genome, then summarize
    group_by(prefix, genome) %>%
    summarize(value = sum(value)) %>%
    # Spread the data back to wide format
    pivot_wider(names_from = prefix, values_from = value)
}

genome_hits <- merge_and_rename(genome_hits_raw)%>% 
    select(c("genome",sample_metadata$Tube_code))
```

### Genome taxonomy

```{r load_genome_taxonomy, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_taxonomy <- read_tsv("data/gtdbtk.summary.tsv") %>%
  select(mag_id = user_genome, classification) %>%
  separate(
    classification,
    into = c("domain", "phylum", "class", "order", "family", "genus", "species"),
    sep = ";") %>%
    rename(genome=1)
```

### Genome quality

```{r load_genome_quality, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_quality <- read_tsv("data/quality_report.tsv") %>%
  select(
    genome = Name, 
    completeness = Completeness, 
    contamination = Contamination,
    length = Genome_Size, 
    gc = GC_Content
  )
genome_quality<-genome_quality %>% 
  mutate (genome=str_remove_all(genome,".fa"))

#Filter MAGs with over 70% completeness and less than 10% contamination
genome_quality <- genome_quality %>%
  filter(completeness > 70 & contamination < 10)
```

### Genome tree

```{r load_genome_tree, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_tree <- read_tree("data/gtdbtk.backbone.bac120.classify.tree")
genome_tree$tip.label <- str_replace_all(genome_tree$tip.label,"'", "") #remove single quotes in MAG names

#Filter genome_taxonomy to keep MAGs with over 70% completeness and less than 10% contamination
genome_taxonomy <- genome_taxonomy %>%
  semi_join(genome_quality, by = "genome")
genome_tree <- keep.tip(genome_tree, tip=genome_taxonomy$genome) # keep only MAG tips
```

### Genome annotations

```{r load_genome_annotations, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_annotations <- read_tsv("data/annotations.tsv.xz") %>%
    rename(gene=1, genome=2, contig=3)

#Filter only the MAGs with over 70% completeness and less than 10% contamination
genome_annotations <- genome_annotations %>%
  semi_join(genome_quality, by = "genome")
```

## Create working objects

Transform the original data files into working objects for downstream analyses.

### Merge genome taxonomy and quality

```{r generate_genome_metadata, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_metadata <- genome_taxonomy %>%
  inner_join(genome_quality,by=join_by(genome==genome)) #join quality
```

### Calculate genome coverage

```{r calculate_genome_coverage, warning=FALSE, comments="", message=FALSE, eval=FALSE}
#Filter genome_hits for the MAGs with over 70% completeness and less than 10% contamination
genome_hits <- genome_hits %>%
  semi_join(genome_quality, by = "genome")

genome_coverage <- genome_hits %>%
  mutate(across(where(is.numeric), ~ ./genome_metadata$length))
```

## Filtering

Two samples are removed from the analysis due to their low sequencing depth.

```{r filterging, echo=TRUE, warning=FALSE, eval=FALSE}
#Counts_raw
columns_to_exclude <- c("AD91", "AC85") # Columns to exclude ("AD16","AD23", "AD25", )
read_counts <- read_counts %>%
  select(-columns_to_exclude)

#Coverage_table
genome_coverage <- genome_coverage %>%
  select(-columns_to_exclude)

#Metadata
sample_metadata <- sample_metadata %>%
  filter(Tube_code %in% colnames(read_counts))
```

### Filter reads by coverage

```{r filter_coverage, warning=FALSE, comments="", message=FALSE, eval=FALSE}
#Filter read_counts for the MAGs with over 70% completeness and less than 10% contamination
read_counts <- read_counts %>%
  semi_join(genome_quality, by = "genome")

min_coverage=0.3
read_counts_filt <- genome_coverage %>%
  mutate(across(where(is.numeric), ~ ifelse(. > min_coverage, 1, 0))) %>%
  mutate(across(-1, ~ . * read_counts[[cur_column()]])) 
```

### Transform reads into genome counts

```{r calculate_genome_counts_unfiltered, warning=FALSE, comments="", message=FALSE, eval=FALSE}
readlength=150
genome_counts <- read_counts %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

```{r calculate_genome_counts_filtered, warning=FALSE, comments="", message=FALSE, eval=FALSE}
readlength=150
genome_counts_filt <- read_counts_filt %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

### Distill annotations into GIFTs 

```{r distill_annotations, comment="", message=FALSE, eval=FALSE, warning=FALSE, eval=FALSE}
genome_gifts <- distill(genome_annotations,GIFT_db2,genomecol=2,annotcol=c(9,10,19), verbosity=F)
```

## Load data statistics

### Raw reads

```{r raw_reads, warning=FALSE, comments="", message=FALSE, eval=FALSE}
raw_reads <-
  "data/multiqc_general_stats_2.txt" %>%
  read_tsv() %>%
  select(
    sample = Sample,
    raw_reads = `total_sequences`
  ) %>%
  mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
  group_by(sample_prefix) %>%
  summarize(raw_reads = sum(raw_reads, na.rm = TRUE)) %>%
  rename(sample = sample_prefix)  %>%
  mutate(sample = str_remove(sample, "^fastp \\|\\s*")) %>% 
  filter(sample %in% sample_metadata$Tube_code)
```


### Quality-filtered reads

```{r filtered_reads, warning=FALSE, comments="", message=FALSE, eval=FALSE}
fastp_reads <-
  "data/multiqc_general_stats.txt" %>%
  read_tsv() %>%
  select(
    sample = Sample,
    trimmed_reads = `total_sequences`
  ) %>%
  mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
  group_by(sample_prefix) %>%
  summarize(trimmed_reads = sum(trimmed_reads, na.rm = TRUE)) %>%
  rename(sample = sample_prefix) %>% 
  filter(!str_detect(sample, "nonlizard \\|")) %>%
  filter(!str_detect(sample, "lizard \\|")) %>%
  filter(!str_detect(sample, "refseq500 \\|"))  %>%
  mutate(sample = str_remove(sample, "^fastp \\|\\s*")) %>% 
  filter(sample %in% sample_metadata$Tube_code)
```

### Host-mapped reads

```{r host_mapped, warning=FALSE, comments="", message=FALSE, eval=FALSE}
host_mapped <-
  "data/multiqc_general_stats.txt" %>%
  read_tsv() %>%
  filter(str_detect(Sample, "lizard")) %>%
  select(
    sample = Sample,
    host_mapped = `reads_mapped`,
    mapping_total = `raw_total_sequences`
  ) %>%
  mutate(
    host_unmapped = mapping_total - host_mapped
  ) %>%
  filter(!is.na(host_mapped)) %>%
  separate(
    col = sample,
    into = c("host_name", "sample"),
    sep = " \\| "
  ) %>%
  rename(mapped = host_mapped, unmapped = host_unmapped) %>%
  select(-mapping_total) %>%
  pivot_longer(-host_name:-sample) %>%
  mutate(
    name = str_glue("{name}_{host_name}")
  ) %>%
  select(-host_name) %>%
  pivot_wider() %>%
  mutate(sample = str_extract(sample, "^[^_]+")) %>%
  group_by(sample) %>%
  summarize(
    mapped_lizard = sum(mapped_lizard),
    unmapped_lizard = sum(unmapped_lizard)
  ) %>% 
  filter(sample %in% sample_metadata$Tube_code)
```

### Prokaryotic fraction

```{r singlem, warning=FALSE, comments="", message=FALSE, eval=FALSE}
singlem <-
  "data/singleM.tsv" %>%
  read_tsv() %>%
  distinct() %>%
  mutate(
    sample = sample %>% str_remove_all("_1$"),
   read_fraction = read_fraction %>% str_remove("%") %>% as.numeric(),
   read_fraction = read_fraction / 100
  ) %>%
  select(
   sample,
   singlem_prokaryotic_bases = bacterial_archaeal_bases,
   singlem_metagenome_size = metagenome_size,
   singlem_read_fraction = read_fraction,
  ) %>%
mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
  group_by(sample_prefix) %>%
  summarize(
    singlem_prokaryotic_bases = sum(singlem_prokaryotic_bases),
    singlem_metagenome_size = sum(singlem_metagenome_size),
    singlem_read_fraction = mean(singlem_read_fraction)
  ) %>%
  rename(sample = sample_prefix)%>% 
  filter(sample %in% sample_metadata$Tube_code)
```

### MAG-mapped reads

```{r mag_mapping, warning=FALSE, comments="", message=FALSE, eval=FALSE}
mag_mapping <-
  read_tsv("data/contig.count.tsv.xz") %>%
  pivot_longer(-sequence_id) %>%
  summarise(value = sum(value), .by = "name") %>%
  rename(sample = name, mapped_mags = value) %>%
  mutate(sample_prefix = str_extract(sample, "^[^_]+")) %>%
    group_by(sample_prefix) %>%
    summarize(
      mapped_mags = sum(mapped_mags)
    ) %>%
    rename(sample = sample_prefix)%>% 
  filter(sample %in% sample_metadata$Tube_code)
```

### Wrap data statistics

```{r wrap_statistics, warning=FALSE, comments="", message=FALSE, eval=FALSE}
data_stats <- raw_reads %>%
  left_join(fastp_reads) %>%
  left_join(host_mapped) %>%
  left_join(singlem) %>%
  left_join(mag_mapping)

data_stats <- data_stats %>%
  filter(!str_detect(sample, "nonlizard \\|")) %>%
  filter(!str_detect(sample, "lizard \\|")) %>%
  filter(!str_detect(sample, "refseq500 \\|"))
```

## Prepare color scheme

[AlberdiLab](www.alberdilab.dk) projects use unified color schemes developed for the [Earth Hologenome Initiative](www.earthhologenome.org), to facilitate figure interpretation.

```{r get_ehi_colors, warning=FALSE, comments="", message=FALSE, eval=FALSE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(phylum, colors) %>% 
    unique() %>%
    arrange(phylum) %>%
    pull(colors, name=phylum)
```

## Wrap working objects

All working objects are wrapped into a single Rdata object to facilitate downstream usage.

```{r wrap_working_objects, warning=FALSE, comments="", message=FALSE, eval=FALSE}
save(sample_metadata, 
     genome_metadata, 
     read_counts, 
     genome_counts, 
     genome_counts_filt, 
     genome_tree,
     genome_gifts, 
     phylum_colors,
     data_stats,
     file = "data/data_27022025.Rdata")
```

<!--chapter:end:01_data_preparation.Rmd-->

# Data statistics

```{r load_data_stats}
load("data/data_27022025.Rdata")
```

## Sequencing reads statistics

```{r reads_stats}
data_stats$raw_reads %>% sum()
data_stats$raw_reads %>% mean()
data_stats$raw_reads %>% sd()
```

## DNA fractions
```{r data_fractions_stats}
#Overall
data_stats %>%
    mutate(mapped_perc=mapped_mags/trimmed_reads) %>%
    summarise(mean=mean(mapped_perc),sd=sd(mapped_perc))
```


```{r data_fractions_plot, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
data_stats %>%
  mutate(
    low_quality = raw_reads - trimmed_reads,
    unmapped_reads = trimmed_reads - mapped_lizard - mapped_mags
  ) %>%
  select(sample, low_quality, mapped_lizard, mapped_mags, unmapped_reads) %>%
  pivot_longer(-sample) %>%
  mutate(name=factor(name,levels=c("low_quality","mapped_lizard","unmapped_reads","mapped_mags"))) %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(!is.na(time_point)) %>%
  ggplot(aes(x = sample, y = value, fill = name)) +
      geom_bar(stat = "identity", position = "fill") +
      scale_fill_manual(name="Sequence type",
                    breaks=c("low_quality","mapped_lizard","unmapped_reads","mapped_mags"),
                    labels=c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                    values=c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c"))+
      facet_grid(~time_point, scales = "free", labeller = labeller(time_point=c("0_Wild"="Wild", "1_Acclimation"="Acclimation", "2_Antibiotics"="Antibiotics",
                                                                                "3_Transplant1"="Transplant1", "4_Transplant2"="Transplant2", "5_Post-FMT1"="Post1",
                                                                                "6_Post-FMT2"="Post2"))) +
      theme_minimal() +
      theme(axis.text.x = element_text(angle = 90, hjust = 1, size=0)) +
      labs(y="DNA sequence fraction",x="Samples")
```

## Recovered microbial fraction

```{r data_estimations_plot, message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
data_stats %>%
  mutate(
    unmapped_reads = trimmed_reads - mapped_lizard - mapped_mags,
    mag_proportion = mapped_mags / (mapped_mags + unmapped_reads),
    singlem_read_fraction = singlem_read_fraction
  ) %>%
  select(sample, mag_proportion, singlem_read_fraction) %>%
  mutate(
    mag_proportion = if_else(singlem_read_fraction == 0, 0, mag_proportion),
    singlem_read_fraction = if_else(singlem_read_fraction == 0, NA, singlem_read_fraction),
    singlem_read_fraction = if_else(singlem_read_fraction < mag_proportion, NA, singlem_read_fraction),
    singlem_read_fraction = if_else(singlem_read_fraction > 1, 1, singlem_read_fraction)
  ) %>%
  pivot_longer(-sample, names_to = "proportion", values_to = "value") %>%
  mutate(
    proportion = factor(
      proportion,
      levels = c("mag_proportion", "singlem_read_fraction")
    )
  ) %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(!is.na(time_point)) %>%
  ggplot(aes(x = sample, y = value, color = proportion)) +
      geom_line(aes(group = sample), color = "#f8a538") +
      geom_point() +
      scale_color_manual(name="Proportion",
                    breaks=c("mag_proportion","singlem_read_fraction"),
                    labels=c("Recovered","Estimated"),
                    values=c("#52e1e8", "#876b53"))+
      theme_minimal() +
      facet_grid(~time_point, scales = "free", space="free", labeller = labeller(time_point=c("0_Wild"="Wild", "1_Acclimation"="Acclimation", "2_Antibiotics"="Antibiotics",
                                                                                "3_Transplant1"="Transplant1", "4_Transplant2"="Transplant2", "5_Post-FMT1"="Post1",
                                                                                "6_Post-FMT2"="Post2")))+
      labs(y = "Samples", x = "Prokaryotic fraction") +
      scale_y_continuous(limits = c(0, 1)) +
      theme(
        axis.text.y = element_text(size = 4),
        axis.text.x = element_text( angle = 90, vjust = 0.5, hjust = 1, size = 0),
        legend.position = "right"
      )
```

### Domain-adjusted mapping rate (DAMR)

```{r damr}
data_stats %>%
  mutate(
    unmapped_reads = trimmed_reads - mapped_lizard - mapped_mags,
    mag_proportion = mapped_mags / (mapped_mags + unmapped_reads),
    singlem_read_fraction = singlem_read_fraction
  ) %>%
  mutate(damr=pmin(1, mag_proportion/singlem_read_fraction)) %>%
  left_join(sample_metadata[c(1,7, 10)], by = join_by(sample == Tube_code)) %>%
  filter(!is.na(time_point)) %>%
  select(sample,damr, time_point, type) %>%
  #group_by(type) %>%
  #summarise(mean=mean(damr),sd=sd(damr)) %>%
  tt()
```

<!--chapter:end:02_data_statistics.Rmd-->

# MAG catalogue

```{r load_data_mag}
load("data/data.Rdata")
```

## Genome phylogeny

```{r genome_phylogeny, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
# Generate the phylum color heatmap
phylum_heatmap <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(genome,phylum) %>%
    mutate(phylum = factor(phylum, levels = unique(phylum))) %>%
    column_to_rownames(var = "genome")

# Generate  basal tree
circular_tree <- force.ultrametric(genome_tree, method="extend") %>% # extend to ultrametric for the sake of visualisation
    ggtree(., layout="fan", open.angle=10, size=0.5)

# Add phylum ring
circular_tree <- gheatmap(circular_tree, phylum_heatmap, offset=0.55, width=0.1, colnames=FALSE) +
        scale_fill_manual(values=phylum_colors) +
        geom_tiplab2(size=1, hjust=-0.1) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))

# Flush color scale to enable a new color scheme in the next ring
circular_tree <- circular_tree + new_scale_fill()

# Add completeness ring
circular_tree <- circular_tree +
        new_scale_fill() +
        scale_fill_gradient(low = "#d1f4ba", high = "#f4baba") +
        geom_fruit(
                data=genome_metadata,
                geom=geom_bar,
                mapping = aes(x=completeness, y=genome, fill=contamination),
                offset = 0.55,
                orientation="y",
              stat="identity")

# Add genome-size ring
circular_tree <-  circular_tree +
        new_scale_fill() +
        scale_fill_manual(values = "#cccccc") +
        geom_fruit(
             data=genome_metadata,
             geom=geom_bar,
             mapping = aes(x=length, y=genome),
                 offset = 0.05,
                 orientation="y",
         stat="identity")

# Add text
circular_tree <-  circular_tree +
        annotate('text', x=3.4, y=0, label='              Phylum', family='ArialMT', size=3.5) +
        annotate('text', x=4.2, y=0, label='                         Genome quality', family='ArialMT', size=3.5) +
        annotate('text', x=4.7, y=0, label='                      Genome size', family='ArialMT', size=3.5)

#Plot circular tree
circular_tree %>% open_tree(30) %>% rotate_tree(90)
```

## Genome quality

```{r genome_quality, message=FALSE, warning=FALSE}
genome_metadata$completeness %>% mean()
genome_metadata$completeness %>% sd()
genome_metadata$contamination %>% mean()
genome_metadata$contamination %>% sd()
```

```{r genome_quality_plot, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}

#Generate quality biplot
genome_biplot <- genome_metadata %>%
  select(c(genome,domain,phylum,completeness,contamination,length)) %>%
  arrange(match(genome, rev(genome_tree$tip.label))) %>% #sort MAGs according to phylogenetic tree
  ggplot(aes(x=completeness,y=contamination,size=length,color=phylum)) +
              geom_point(alpha=0.7) +
                    ylim(c(10,0)) +
                    scale_color_manual(values=phylum_colors) +
                    labs(y= "Contamination", x = "Completeness") +
                    theme_classic() +
                    theme(legend.position = "none")

#Generate contamination boxplot
genome_contamination <- genome_metadata %>%
            ggplot(aes(y=contamination)) +
                    ylim(c(10,0)) +
                    geom_boxplot(colour = "#999999", fill="#cccccc") +
                    theme_void() +
                    theme(legend.position = "none",
                        axis.title.x = element_blank(),
                        axis.title.y = element_blank(),
                        axis.text.y=element_blank(),
                        axis.ticks.y=element_blank(),
                        axis.text.x=element_blank(),
                        axis.ticks.x=element_blank(),
                        plot.margin = unit(c(0, 0, 0.40, 0),"inches")) #add bottom-margin (top, right, bottom, left)

#Generate completeness boxplot
genome_completeness <- genome_metadata %>%
        ggplot(aes(x=completeness)) +
                xlim(c(50,100)) +
                geom_boxplot(colour = "#999999", fill="#cccccc") +
                theme_void() +
                theme(legend.position = "none",
                    axis.title.x = element_blank(),
                    axis.title.y = element_blank(),
                    axis.text.y=element_blank(),
                    axis.ticks.y=element_blank(),
                    axis.text.x=element_blank(),
                    axis.ticks.x=element_blank(),
                    plot.margin = unit(c(0, 0, 0, 0.50),"inches")) #add left-margin (top, right, bottom, left)

#Render composite figure
grid.arrange(grobs = list(genome_completeness,genome_biplot,genome_contamination),
        layout_matrix = rbind(c(1,1,1,1,1,1,1,1,1,1,1,4),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3)))

```


## Functional overview

```{r function_heatmap, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}

# Aggregate basal GIFT into elements
function_table <- genome_gifts %>%
    to.elements(., GIFT_db2)

# Generate  basal tree
function_tree <- force.ultrametric(genome_tree, method="extend") %>%
                ggtree(., size = 0.3) 

#Add phylum colors next to the tree tips
function_tree <- gheatmap(function_tree, phylum_heatmap, offset=0, width=0.1, colnames=FALSE) +
            scale_fill_manual(values=phylum_colors) +
            labs(fill="Phylum")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

#Add functions heatmap
function_tree <- gheatmap(function_tree, function_table, offset=0.5, width=3.5, colnames=FALSE) +
            vexpand(.08) +
            coord_cartesian(clip = "off") +
            scale_fill_gradient(low = "#f4f4f4", high = "steelblue", na.value="white") +
            labs(fill="GIFT")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

# Add completeness barplots
function_tree <- function_tree +
            geom_fruit(data=genome_metadata,
            geom=geom_bar,
            grid.params=list(axis="x", text.size=2, nbreak = 1),
            axis.params=list(vline=TRUE),
            mapping = aes(x=length, y=genome, fill=completeness),
                 offset = 3.8,
                 orientation="y",
                 stat="identity") +
            scale_fill_gradient(low = "#cf8888", high = "#a2cc87") +
            labs(fill="Genome\ncompleteness")

function_tree
```

## Functional ordination

```{r function_ordination, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
# Generate the tSNE ordination
tSNE_function <- Rtsne(X=function_table, dims = 2, check_duplicates = FALSE)

# Plot the ordination
function_ordination <- tSNE_function$Y %>%
                as.data.frame() %>%
                mutate(genome=rownames(function_table)) %>%
                inner_join(genome_metadata, by="genome") %>%
                rename(tSNE1="V1", tSNE2="V2") %>%
                select(genome,phylum,tSNE1,tSNE2, length) %>%
                ggplot(aes(x = tSNE1, y = tSNE2, color = phylum, size=length))+
                            geom_point(shape=16, alpha=0.7) +
                            scale_color_manual(values=phylum_colors) +
                            theme_minimal() +
                labs(color="Phylum", size="Genome size") +
                guides(color = guide_legend(override.aes = list(size = 5))) # enlarge Phylum dots in legend

function_ordination
```

<!--chapter:end:03_mag_catalogue.Rmd-->

# Community composition

```{r load_data_community}
load("data/data_27022025.Rdata")
```

## Taxonomy overview 

### Stacked barplot

```{r taxonomy_barplot, warning=FALSE, fig.height=12, fig.width=20, fig.fullwidth=TRUE}
# Merge data frames based on sample
# transplants_metadata <- sample_metadata %>%
#   mutate(Tube_code = str_remove_all(Tube_code, "_a"))
# transplants_metadata$newID <- paste(transplants_metadata$Tube_code,
#                                     "_",
#                                     transplants_metadata$individual)

merged_data <- genome_counts_filt %>%
  mutate_at(vars(-genome),  ~ . / sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>% #append sample metadata
  filter(count > 0) #filter 0 counts

ggplot(merged_data, aes(
  x = sample,
  y = count,
  fill = phylum,
  group = phylum
)) + #grouping enables keeping the same sorting of taxonomic units
  geom_bar(stat = "identity",
           colour = "white",
           linewidth = 0.1) + #plot stacked bars with white borders
  scale_fill_manual(values = phylum_colors) +
  facet_nested(. ~ time_point + type , scales = "free") + #facet per day and treatment
  guides(fill = guide_legend(ncol = 1)) +
  labs(fill = "Phylum", y = "Relative abundance", x = "Sample") +
  theme(
    axis.text.x = element_text(
      angle = 45,
      hjust = 1,
      size = 0
    ),
    strip.text.x = element_text(
      size = 14,
      colour = "black",
      face = "bold"
    ),
    strip.background = element_rect(fill ="lightgrey"),
    axis.title = element_text(size = 18, face = "bold"),
    panel.background = element_blank(),
    legend.title = element_text(size = 20, face = "bold"),
    legend.text = element_text(size = 16)
  )
```

#### Wild samples
```{r taxonomy_barplot_wild, fig.height=8, fig.width=18, fig.fullwidth=TRUE, warning=FALSE}
merged_data  %>%
  filter(time_point=="Wild")  %>%
  ggplot(aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~ Population,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    labs(fill="Phylum",y = "Relative abundance",x="Sample")+
  theme(
    axis.text.x = element_text(
      angle = 45,
      hjust = 1,
      size = 0
    ),
    strip.text.x = element_text(
      size = 14,
      colour = "black",
      face = "bold"
    ),
    axis.title = element_text(size = 18, face = "bold"),
    panel.background = element_blank(),
    strip.background = element_rect(fill ="lightgrey"),
    legend.title = element_text(size = 20, face = "bold"),
    legend.text = element_text(size = 16)
  )
```

### Phylum relative abundances
```{r taxonomy_phylum_summary, warning=FALSE, comments="", message=FALSE}
phylum_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,phylum) %>%
  summarise(relabun=sum(count))
```

#### Cold and wet population
```{r taxonomy_phylum_summary_cold, warning=FALSE, comments="", message=FALSE}
phylum_summary %>%
    left_join(sample_metadata, by = join_by(sample == Tube_code))  %>%
    group_by(phylum, Population) %>%
    filter(Population=="Cold_wet" & time_point=="Wild") %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T))  %>%
  filter(total_mean!=0) %>% 
    mutate(total=str_c(round(total_mean,2),"±",round(total_sd,2))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(phylum,total) %>% 
    tt()
```

#### Hot and dry population
```{r taxonomy_phylum_summary_hot, warning=FALSE, comments="", message=FALSE}
phylum_summary %>%
    left_join(sample_metadata, by = join_by(sample == Tube_code))  %>%
    group_by(phylum, Population) %>%
    filter(Population=="Hot_dry" & time_point=="Wild") %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T))  %>%
  filter(total_mean!=0) %>% 
    mutate(total=str_c(round(total_mean,2),"±",round(total_sd,2))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(phylum,total) %>% 
    tt()
```


## Taxonomy boxplot

### Family
```{r taxonomy_family_summary, warning=FALSE, comments="", message=FALSE}
family_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>% #append sample metadata
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,family) %>%
  summarise(relabun=sum(count))
```


```{r taxonomy_jitterplot_family, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
family_arrange <- family_summary %>%
    group_by(family) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(family) %>%
    pull()

family_summary %>%
    left_join(genome_metadata %>% select(family,phylum) %>% unique(),by=join_by(family==family)) %>%
    left_join(sample_metadata,by=join_by(sample==Tube_code)) %>%
    filter(family %in% family_arrange[1:20]) %>%
    mutate(family=factor(family,levels=rev(family_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=family, group=family, color=phylum)) +
        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~type)+
        theme_minimal() + 
        labs(y="Family", x="Relative abundance", color="Phylum")

```

### Genus
```{r taxonomy_genus_summary, warning=FALSE, comments="", message=FALSE}
genus_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>% #append sample metadata
  left_join(genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,genus) %>%
  summarise(relabun=sum(count)) %>%
  filter(genus != "g__")
```

```{r taxonomy_jitterplot_genus, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genus_arrange <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=sum(relabun)) %>%
    filter(genus != "g__")%>%
    arrange(-mean) %>%
    select(genus) %>%
    mutate(genus= sub("^g__", "", genus)) %>%
    pull()

genus_summary %>%
    left_join(genome_metadata %>% select(genus,phylum) %>% unique(),by=join_by(genus==genus)) %>%
    left_join(sample_metadata,by=join_by(sample==Tube_code)) %>%
    mutate(genus= sub("^g__", "", genus)) %>%
    filter(genus %in% genus_arrange[1:20]) %>%
    mutate(genus=factor(genus,levels=rev(genus_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=genus, group=genus, color=phylum)) +
        scale_color_manual(values=phylum_colors[-c(3,4,6,8)]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~type)+
        theme_minimal() + 
        labs(y="Genus", x="Relative abundance", color="Phylum")

```




<!--chapter:end:04_community_composition.Rmd-->

# Calculations
```{r load_data_calcu, eval=FALSE}
load("data/data_27022025.Rdata")
load("data/beta_27022025.Rdata")
```

## Alpha diversity

```{r alpha_div, comment="", message=FALSE, warning=FALSE, eval=FALSE}
# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample))
```

## Beta diversity
```{r beta_div, comment="", message=FALSE, warning=FALSE, eval=FALSE}
beta_q0n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1)

beta_q1p <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1, tree = genome_tree)
```

## Phyloseq object
```{r phyloseq, comment="", message=FALSE, warning=FALSE, eval=FALSE}
#Phyloseq object
count_phy <- genome_counts_filt %>%
  column_to_rownames(var="genome")%>%
  otu_table(., taxa_are_rows=T)

sample_info_tab_phy <- sample_metadata%>%
  column_to_rownames(var="Tube_code")%>%
  sample_data()

TAX <- genome_metadata%>%
  column_to_rownames(var="genome")%>%
  select(1:7)%>%
  as.matrix()%>%
  tax_table()
tree <- phy_tree(genome_tree)

physeq_all = phyloseq(count_phy, TAX, sample_info_tab_phy, tree)
```

## Functional data

```{r gift_analyses, warning=FALSE, eval=FALSE}
# Aggregate bundle-level GIFTs into the compound level
GIFTs_elements <- to.elements(genome_gifts, GIFT_db2)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts_filt$genome, ]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>%
  select_if(~ !is.numeric(.) || sum(.) != 0)

elements <- GIFTs_elements_filtered %>%
  as.data.frame()

# Aggregate element-level GIFTs into the function level
GIFTs_functions <- to.functions(GIFTs_elements_filtered, GIFT_db2)
functions <- GIFTs_functions %>%
  as.data.frame()

# Aggregate function-level GIFTs into overall Biosynthesis, Degradation and Structural GIFTs
GIFTs_domains <- to.domains(GIFTs_functions, GIFT_db2)
domains <- GIFTs_domains %>%
  as.data.frame()

# Get community-weighed average GIFTs per sample
GIFTs_elements_community <- to.community(GIFTs_elements_filtered, genome_counts_filt %>% column_to_rownames(., "genome") %>% tss(), GIFT_db2)
GIFTs_functions_community <- to.community(GIFTs_functions, genome_counts_filt %>% column_to_rownames(., "genome") %>% tss(), GIFT_db2)
GIFTs_domains_community <- to.community(GIFTs_domains, genome_counts_filt %>% column_to_rownames(., "genome") %>% tss(), GIFT_db2)

uniqueGIFT_db <- unique(GIFT_db2[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)
```

```{r element_gift, comment="", message=FALSE, warning=FALSE, eval=FALSE}
element_gift <- GIFTs_elements_community %>%
  as.data.frame() %>%
  rownames_to_column(., "Tube_code") %>%
  left_join(sample_metadata[c(1, 7, 10)], by = "Tube_code") 
```

```{r save_beta, comment="", message=FALSE,warning=FALSE, eval=FALSE}
save(richness,
     neutral,
     phylogenetic,
     alpha_div,
     physeq_all,
     beta_q0n, 
     beta_q1n, 
     beta_q1p,
     GIFTs_elements_filtered,
     GIFTs_elements_community,
     GIFTs_functions_community,
     GIFTs_domains_community,
     element_gift,
     uniqueGIFT_db,
     file = "data/calculations_28022025.Rdata")
```

<!--chapter:end:05_Calculations.Rmd-->

# Is the GM of cold- vs warm-adapted populations similar in the wild?
```{r knitr_options_wild, include=FALSE}
knitr::opts_chunk$set(fig.width=10, fig.height=7, fig.fullwidth=TRUE,
fig.path='Figures/', warning=FALSE,
message=FALSE)
set.seed(2802)
```
```{r load_data_wild}
load("data/data_27022025.Rdata")
load("data/calculations_28022025.Rdata")
```

## Shared and unique MAGs
```{r chart1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
locationcolors=c('#c4d7d1','#e08683')
subset_meta <- sample_metadata %>%
  filter(time_point=="Wild")

genome_counts_rel_fil<- genome_counts_filt %>%
  select(one_of(c("genome", subset_meta$Tube_code))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames(., "genome")
  
genome_counts_rel_pa=1*(genome_counts_rel_fil>0)
table_upset_analysis_cont=t(aggregate(t(genome_counts_rel_pa),by=list(subset_meta$Population),FUN=sum)[,-1])
colnames(table_upset_analysis_cont)=levels(as.factor(subset_meta$Population))
table_upset_analysis=(table_upset_analysis_cont>0)*1
table_upset_analysis=data.frame(table_upset_analysis)
table_upset_analysis=apply(table_upset_analysis,2,as.integer)
rownames(table_upset_analysis) <- rownames(genome_counts_rel_pa)

upset(as.data.frame(table_upset_analysis),
  keep.order = T,
  sets = rev(c("Cold_wet","Hot_dry")),
  sets.bar.color= rev(locationcolors),
  mb.ratio = c(0.55, 0.45), order.by = "freq")
```

## Alpha diversity
```{r alpha_div_plot_0, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="Wild") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = Population, group=Population, color=Population, fill=Population)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#008080', "#d57d2c")) +
      scale_fill_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#00808050', "#d57d2c50")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=3, label.x=.58) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_blank(),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")
```

## Beta diversity

***Number of samples used***
```{r beta_div_neutral_wild_samples, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep <- sample_metadata %>%
  filter(time_point == "Wild") %>% 
  select(Tube_code) %>% 
  pull()
subset_meta <- sample_metadata %>%
  filter(time_point == "Wild")

length(samples_to_keep)
```

***Richness***
```{r beta_richness_permanova_1, warning=FALSE, comments="", message=FALSE}
richness <- as.matrix(beta_q0n$S)
richness <- as.dist(richness[rownames(richness) %in% samples_to_keep,
               colnames(richness) %in% samples_to_keep])
betadisper(richness, subset_meta$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(richness ~ Population,
        data = subset_meta %>% arrange(match(Tube_code,labels(richness))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

***Neutral***
```{r beta_neutral_permanova_1, warning=FALSE, comments="", message=FALSE}
neutral <- as.matrix(beta_q1n$S)
neutral <- as.dist(neutral[rownames(neutral) %in% samples_to_keep,
               colnames(neutral) %in% samples_to_keep])
betadisper(neutral, subset_meta$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(neutral ~ Population,
        data = subset_meta %>% arrange(match(Tube_code,labels(neutral))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_1, warning=FALSE, comments="", message=FALSE}
phylo <- as.matrix(beta_q1p$S)
phylo <- as.dist(phylo[rownames(phylo) %in% samples_to_keep,
               colnames(phylo) %in% samples_to_keep])
betadisper(phylo, subset_meta$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(phylo ~ Population,
        data = subset_meta %>% arrange(match(Tube_code,labels(phylo))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

***NMDS***
```{r beta_neutral_nmds, comments="", warning=FALSE, message=FALSE, results="hide"}
beta_q0n_nmds_wild <- richness %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
				vegan::scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta, by = join_by(sample == Tube_code)) %>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=type)) +
				scale_color_manual(name="Population",
                    breaks=c("Cold_wet","Hot_dry"),
                    labels=c("Cold","Hot"),
                    values=c('#008080',"#d57d2c")) +
        scale_shape_manual(name="Type",
                           breaks=c("Control", "Hot_control", "Treatment"),
                           labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                         values=c("circle","square","triangle")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_q1n_nmds_wild <- neutral %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
				vegan::scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta, by = join_by(sample == Tube_code))%>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=type)) +
				scale_color_manual(name="Population",
                    breaks=c("Cold_wet","Hot_dry"),
                    labels=c("Cold","Hot"),
                    values=c('#008080',"#d57d2c")) +
        scale_shape_manual(name="Type",
                           breaks=c("Control", "Hot_control", "Treatment"),
                           labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                         values=c("circle","square","triangle")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_q1p_nmds_wild <- phylo %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
				vegan::scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta, by = join_by(sample == Tube_code))%>%
			group_by(Population) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=type)) +
				scale_color_manual(name="Population",
                    breaks=c("Cold_wet","Hot_dry"),
                    labels=c("Cold","Hot"),
                    values=c('#008080',"#d57d2c")) +
        scale_shape_manual(name="Type",
                           breaks=c("Control", "Hot_control", "Treatment"),
                           labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                         values=c("circle","square","triangle")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (),x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none") 
```

```{r beta_neutral_nmds_plot_final, results=TRUE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggarrange(beta_q0n_nmds_wild, beta_q1n_nmds_wild, beta_q1p_nmds_wild, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

## Differential abundance

```{r}
physeq_wild <- subset_samples(physeq_all, time_point == "Wild")
physeq_wild <- prune_taxa(taxa_sums(physeq_wild)>0, physeq_wild)
```

### Phylum
```{r ancom_rand_pond_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancom_rand_output_wild_phylum = ancombc2(data = physeq_wild, 
                  assay_name = "counts", 
                  tax_level = "phylum",
                  fix_formula = "Population", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = "Population", 
                  struc_zero = TRUE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```
```{r}
ancom_rand_output_wild_phylum$res %>%
  dplyr::select(taxon, lfc_PopulationHot_dry, p_PopulationHot_dry) %>%
  filter(p_PopulationHot_dry < 0.05)
```

### MAG level
```{r ancom_rand_pond_mag_wild, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancom_rand_output = ancombc2(data = physeq_wild, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "Population", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = "Population", 
                  struc_zero = TRUE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

***Structural zeros***
```{r}
# Extract structural zeros
structural_zeros <- ancom_rand_output$zero_ind

# Convert to a data frame for better readability
structural_zeros_df <- as.data.frame(structural_zeros) %>% 
  column_to_rownames(., "taxon")

# Find taxa that are structural zeros in at least one group
structural_zero_taxa <- rownames(structural_zeros_df)[apply(structural_zeros_df, 1, any)]
length(structural_zero_taxa)
```

***Differentially abundant MAGs***
```{r ancom_rand, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_wild@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output$res %>%
  dplyr::select(taxon, lfc_PopulationHot_dry, p_PopulationHot_dry) %>%
  filter(p_PopulationHot_dry < 0.05) %>%
  dplyr::arrange(p_PopulationHot_dry) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_PopulationHot_dry)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```
```{r ancombc_rand_plot_phy, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag%>%
      mutate(genome=factor(taxon,levels=ancombc_rand_table_mag$taxon)) %>%
ggplot(., aes(x=lfc_PopulationHot_dry, y=forcats::fct_reorder(genome,lfc_PopulationHot_dry), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

***Phyla of the significant MAGs in warm***
```{r wild_sign_features_warm, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_PopulationHot_dry>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in cold***
```{r wild_sign_features_cold, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_PopulationHot_dry<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

## Functional capacity
### Metabolic capacity index at functional level

```{r gitfs_elements_wild, warning=FALSE, comments="", message=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="Wild") %>%
  group_by(Population) %>%
  summarise(MCI = mean(value), sd = sd(value))
```
```{r gitfs_functional_wild_3, warning=FALSE, comments="", message=FALSE}
MCI_element_wild <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="Wild")

shapiro.test(MCI_element_wild$value) #normality test
var.test(value ~ Population, data = MCI_element_wild)
t.test(value ~ Population, data=MCI_element_wild, var.equal = TRUE)
```

### Differences in functional pathways
```{r comunity_elem, comment="", message=FALSE, warning=FALSE}
element_gift_wild <- element_gift %>% 
  filter(time_point=="Wild") %>% 
  select(-time_point,-type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 4)], by = "Tube_code")
```

```{r commun_wilcox_elem_wild, comment="", message=FALSE, warning=FALSE}
significant_elements_wild <- element_gift_wild %>%
  pivot_longer(-c(Tube_code,Population), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ Population)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)

element_gift_sig_wild <- element_gift_wild %>%
  select(Tube_code, all_of(intersect(
    significant_elements_wild$trait,
    colnames(element_gift_wild)
  ))) %>%
  left_join(., sample_metadata[c(1, 4)], by = join_by(Tube_code == Tube_code))

difference_table_wild <- element_gift_sig_wild %>%
  select(-Tube_code) %>%
  group_by(Population) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Cold_wet-Hot_dry)%>% 
  mutate(group_color = ifelse(Difference <0, "Hot","Cold")) 
```

```{r commun_wilcox_elem_plot_wild, comment="", message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table_wild %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

<!--chapter:end:06_Wild_populations.Rmd-->

# What is the effect of acclimation?
```{r load_data_acclimation}
load("data/data_27022025.Rdata")
load("data/calculations_28022025.Rdata")
```
```{r knitr_options_acclim, include=FALSE}
knitr::opts_chunk$set(fig.width=10, fig.height=7, fig.fullwidth=TRUE,
fig.path='Figures/', warning=FALSE,
message=FALSE)
set.seed(2802)
```

### Do each population retain the wild GM after acclimation?
#### Cold
##### Alpha diversity
```{r alpha_div_plot_cold, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(type == "Treatment" & time_point %in% c("Acclimation", "Wild")) %>%
  mutate(time_point = factor(time_point, levels = c("Wild", "Acclimation"))) %>%
  mutate(metric = factor(metric, levels = c("richness", "neutral", "phylogenetic", "functional"))) %>%
  ggplot(aes(
    y = value,
    x = time_point,
    group = time_point,
    color = time_point,
    fill = time_point
  )) +
  geom_boxplot(outlier.shape = NA, alpha = 0.5) +
  geom_jitter(alpha = 0.5) +
  scale_color_manual(
    name = "Population",
    breaks = c("Wild", "Acclimation"),
    labels = c("Wild", "Acclimation"),
    values = c('#008080', "#d57d2c")
  ) +
  scale_fill_manual(
    name = "Population",
    breaks = c("Wild", "Acclimation"),
    labels = c("Wild", "Acclimation"),
    values = c('#00808050', "#d57d2c50")
  ) +
  facet_wrap(. ~ metric, scales = "free") +
  coord_cartesian(xlim = c(1, NA)) +
  stat_compare_means(size = 3, label.x = .58) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size = 10),
    axis.text.x = element_blank(),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
  ) +
  ylab("Alpha diversity")
```

##### Beta diversity

```{r beta_div_neutral_accli_wild_cold, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_accli <- sample_metadata %>%
  filter(Population=="Cold_wet" & time_point %in% c("Acclimation", "Wild")) %>% 
  select(Tube_code) %>% 
  pull()
subset_meta_accli <- sample_metadata %>%
  filter(Population=="Cold_wet" & time_point %in% c("Acclimation", "Wild"))
```

***Neutral***
```{r beta_neutral_permanova_1_accli_cold, warning=FALSE, comments="", message=FALSE}
neutral_accli <- as.matrix(beta_q1n$S)
neutral_accli <- as.dist(neutral_accli[rownames(neutral_accli) %in% samples_to_keep_accli,
               colnames(neutral_accli) %in% samples_to_keep_accli])
betadisper(neutral_accli, subset_meta_accli$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_accli, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(neutral_accli ~ time_point,
        data = subset_meta_accli %>% arrange(match(Tube_code,labels(neutral_accli))),
        permutations = 999,
        strata = subset_meta_accli %>% arrange(match(Tube_code,labels(neutral_accli))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

##### Functional differences
***MCI***
```{r gitfs_elements_accli_cold, warning=FALSE, comments="", message=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(Population == "Cold_wet" & time_point %in% c("Acclimation", "Wild")) %>%
  group_by(time_point) %>%
  summarise(MCI = mean(value), sd = sd(value))
```
```{r gitfs_functional_accli_treat, warning=FALSE, comments="", message=FALSE}
MCI_element_accli <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(type == "Treatment" & time_point %in% c("Acclimation", "Wild"))

shapiro.test(MCI_element_accli$value)
var.test(value ~ time_point, data = MCI_element_accli)
t.test(value ~ time_point, data=MCI_element_accli, var.equal = TRUE)
```

***Differences in functional pathways***
```{r comunity_elem_fun, comment="", message=FALSE, warning=FALSE}
element_gift_wild <- element_gift %>% 
  filter(type %in% c("Control", "Treatment") & time_point %in% c("Wild", "Acclimation")) %>% 
  select(-time_point,-type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1,10)], by = "Tube_code")
```
```{r commun_wilcox_elem_wild_acc, comment="", message=FALSE, warning=FALSE}
significant_elements_wild <- element_gift_wild %>%
  pivot_longer(-c(Tube_code,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)

element_gift_sig_wild <- element_gift_wild %>%
  select(Tube_code, all_of(intersect(
    significant_elements_wild$trait,
    colnames(element_gift_wild)
  ))) %>%
  left_join(., sample_metadata[c(1, 10)], by = join_by(Tube_code == Tube_code))

difference_table_wild <- element_gift_sig_wild %>%
  select(-Tube_code) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Wild-Acclimation)%>% 
  mutate(group_color = ifelse(Difference <0, "Acclimation","Wild")) 
difference_table_wild
```

#### Warm
##### Alpha diversity
```{r alpha_div_plot_warm, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(type=="Hot_control" & time_point %in% c("Acclimation", "Wild")) %>%
  mutate(time_point = factor(time_point, levels = c("Wild", "Acclimation"))) %>%
  mutate(metric = factor(metric, levels = c("richness", "neutral", "phylogenetic", "functional"))) %>%
  ggplot(aes(
    y = value,
    x = time_point,
    group = time_point,
    color = time_point,
    fill = time_point
  )) +
  geom_boxplot(outlier.shape = NA, alpha = 0.5) +
  geom_jitter(alpha = 0.5) +
  scale_color_manual(
    name = "Population",
    breaks = c("Wild", "Acclimation"),
    labels = c("Wild", "Acclimation"),
    values = c('#008080', "#d57d2c")
  ) +
  scale_fill_manual(
    name = "Population",
    breaks = c("Wild", "Acclimation"),
    labels = c("Wild", "Acclimation"),
    values = c('#00808050', "#d57d2c50")
  ) +
  facet_wrap(. ~ metric, scales = "free") +
  coord_cartesian(xlim = c(1, NA)) +
  stat_compare_means(size = 3, label.x = .58) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size = 10),
    axis.text.x = element_blank(),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
  ) +
  ylab("Alpha diversity")
```

##### Beta diversity
```{r beta_div_neutral_accli_wild_warm, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_accli <- sample_metadata %>%
  filter(Population=="Hot_dry" & time_point %in% c("Acclimation", "Wild")) %>% 
  select(Tube_code) %>% 
  pull()
subset_meta_accli <- sample_metadata %>%
  filter(Population=="Hot_dry" & time_point %in% c("Acclimation", "Wild"))
```

***Neutral***
```{r beta_neutral_permanova_1_accli, warning=FALSE, comments="", message=FALSE}
neutral_accli <- as.matrix(beta_q1n$S)
neutral_accli <- as.dist(neutral_accli[rownames(neutral_accli) %in% samples_to_keep_accli,
               colnames(neutral_accli) %in% samples_to_keep_accli])
betadisper(neutral_accli, subset_meta_accli$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_accli_hot, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(neutral_accli ~ time_point,
        data = subset_meta_accli %>% arrange(match(Tube_code,labels(neutral_accli))),
        permutations = 999,
        strata = subset_meta_accli %>% arrange(match(Tube_code,labels(neutral_accli))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

##### Functional differences
***MCI***
```{r gitfs_elements_accli_hot, warning=FALSE, comments="", message=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(type == "Hot_control" & time_point %in% c("Acclimation", "Wild")) %>%
  group_by(time_point) %>%
  summarise(MCI = mean(value), sd = sd(value))
```
```{r gitfs_functional_accli_hot, warning=FALSE, comments="", message=FALSE}
MCI_element_accli <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(type == "Hot_control" & time_point %in% c("Acclimation", "Wild"))


shapiro.test(MCI_element_accli$value)
var.test(value ~ time_point, data = MCI_element_accli)
t.test(value ~ time_point, data=MCI_element_accli, var.equal = TRUE)
```

***Differences in functional pathways***
```{r comunity_elem_warm, comment="", message=FALSE, warning=FALSE}
element_gift_warm <- element_gift %>% 
  filter(type=="Hot_control" & time_point %in% c("Wild", "Acclimation")) %>% 
  select(-time_point,-type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1,10)], by = "Tube_code")
```

```{r commun_wilcox_elem_warm_acc, comment="", message=FALSE, warning=FALSE}
significant_elements_warm <- element_gift_warm %>%
  pivot_longer(-c(Tube_code,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>% 
  select(-Elements)

element_gift_sig_warm <- element_gift_warm %>%
  select(Tube_code, all_of(intersect(
    significant_elements_warm$trait,
    colnames(element_gift_warm)
  ))) %>%
  left_join(., sample_metadata[c(1, 10)], by = join_by(Tube_code == Tube_code))

difference_table_warm <- element_gift_sig_warm %>%
  select(-Tube_code) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Wild)%>% 
  mutate(group_color = ifelse(Difference <0, "Wild","Acclimation")) 
```

```{r commun_wilcox_elem_plot_hot, comment="", message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table_warm %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

### Do populations mantain a different GM after acclimation?
#### Alpha diversity
```{r alpha_div_plot_accli, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=8, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="Acclimation") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = Population, group=Population, color=Population, fill=Population)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#008080', "#d57d2c")) +
      scale_fill_manual(name="Population",
          breaks=c("Cold_wet","Hot_dry"),
          labels=c("Cold","Hot"),
          values=c('#00808050', "#d57d2c50")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=3, label.x=.58) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_blank(),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")
```

#### Beta diversity
Number of samples used
```{r beta_div_neutral_accli_samples, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_accli <- sample_metadata %>%
  filter(time_point == "Acclimation") %>% 
  select(Tube_code) %>% 
  pull()
subset_meta_accli <- sample_metadata %>%
  filter(time_point == "Acclimation")

length(samples_to_keep_accli)
```

***Richness***
```{r beta_richness_permanova_1_accli, warning=FALSE, comments="", message=FALSE}
richness_accli <- as.matrix(beta_q0n$S)
richness_accli <- as.dist(richness_accli[rownames(richness_accli) %in% samples_to_keep_accli,
               colnames(richness_accli) %in% samples_to_keep_accli])
betadisper(richness_accli, subset_meta_accli$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_accli, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(richness_accli ~ Population,
        data = subset_meta_accli %>% arrange(match(Tube_code,labels(richness_accli))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

***Neutral***
```{r beta_neutral_permanova_1_acclim, warning=FALSE, comments="", message=FALSE}
neutral_accli <- as.matrix(beta_q1n$S)
neutral_accli <- as.dist(neutral_accli[rownames(neutral_accli) %in% samples_to_keep_accli,
               colnames(neutral_accli) %in% samples_to_keep_accli])
betadisper(neutral_accli, subset_meta_accli$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_acclimation, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(neutral_accli ~ Population,
        data = subset_meta_accli %>% arrange(match(Tube_code,labels(neutral_accli))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_1_accli, warning=FALSE, comments="", message=FALSE}
phylo_accli <- as.matrix(beta_q1p$S)
phylo_accli <- as.dist(phylo_accli[rownames(phylo_accli) %in% samples_to_keep_accli,
               colnames(phylo_accli) %in% samples_to_keep_accli])
betadisper(phylo_accli, subset_meta_accli$Population) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_accli, warning=FALSE, comments="", message=FALSE, results=TRUE}
adonis2(phylo_accli ~ Population,
        data = subset_meta_accli %>% arrange(match(Tube_code,labels(phylo_accli))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

***NMDS***
```{r beta_neutral_nmds_accli, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_q0n_nmds_accli <- richness_accli %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(subset_meta_accli, by = join_by(sample == Tube_code))%>%
  group_by(Population) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=Population)) +
  scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
  scale_shape_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("circle","square","triangle")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
  theme_classic() +
  theme(legend.position="none")

beta_q1n_nmds_accli <- neutral_accli %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(subset_meta_accli, by = join_by(sample == Tube_code))%>%
  group_by(Population) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=Population, shape=type)) +
  scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
  scale_shape_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("circle","square","triangle")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
  theme_classic() +
  theme(legend.position="none")

beta_q1p_nmds_accli <- phylo_accli %>%
  metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace=FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(subset_meta_accli, by = join_by(sample == Tube_code))%>%
  group_by(Population) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=NMDS1,y=NMDS2, color=Population)) +
  scale_color_manual(name="Population",
                     breaks=c("Cold_wet","Hot_dry"),
                     labels=c("Cold","Hot"),
                     values=c('#008080',"#d57d2c")) +
  scale_shape_manual(name="Type",
                     breaks=c("Control", "Hot_control", "Treatment"),
                     labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                     values=c("circle","square","triangle")) +
  geom_point(size=2) +
  geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
  labs(y= element_blank (),x="NMDS1 \n Phylogenetic beta diversity") +
  theme_classic() +
  theme(legend.position="none") 
```

```{r beta_neutral_nmds_plot_final_accli, results=TRUE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
ggarrange(
  beta_q0n_nmds_accli,
  beta_q1n_nmds_accli,
  beta_q1p_nmds_accli,
  ncol = 2,
  nrow = 2,
  common.legend = TRUE,
  legend = "right"
)
```

<!--chapter:end:07_Acclimation.Rmd-->

# FMT
```{r knitr_options_FMT, include=FALSE}
knitr::opts_chunk$set(fig.width=10, fig.height=7, fig.fullwidth=TRUE,
fig.path='Figures/', warning=FALSE,
message=FALSE)
set.seed(2802)
```
```{r load_data_FMT}
load("data/data_27022025.Rdata")
load("data/beta_27022025.Rdata")
```

## FMT1
### Is the GM of the intervention group more similar to warm population after one week from FMT?
#### Alpha
```{r alpha_div_plot_5, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
bxp1<-alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT1") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
        scale_x_discrete(labels = c("Control" = "Cold-Cold", "Hot_control" = "Hot-Hot", "Treatment" = "Cold-Hot")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=3, label.x=.7) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_blank(),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")


# Add significance bars with p-values
bxp1 + 
  geom_signif(comparisons = list(c("Control", "Hot_control"), c("Control", "Treatment"), c("Hot_control", "Treatment")),
              map_signif_level = TRUE)
```

```{r}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT1") 
```

***Richness***
```{r alpha_div_data, comment="", message=FALSE, warning=FALSE, eval=FALSE}
Modelq0GLMMNB <- glm.nb(richness ~ type, data = alpha_div_meta,trace=TRUE)
summary(Modelq0GLMMNB)
emmeans(Modelq0GLMMNB, pairwise ~ type)
```

***Neutral***
```{r alpha_div_data_1, comment="", message=FALSE, warning=FALSE}
Modelq1n <- lm(formula = neutral ~ type, data = alpha_div_meta) 
summary(Modelq1n)
```

***Phylogenetic***
```{r alpha_div_data_2, comment="", message=FALSE, warning=FALSE}
Model_phylo <- lm(formula = phylogenetic ~ type, data = alpha_div_meta) 
summary(Model_phylo)
```

#### Beta diversity

***Number of samples used***
```{r beta_div_neutral_treat1_samples_2, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_post1 <- sample_metadata %>%
  filter(time_point == "FMT1") %>% 
  select(Tube_code) %>% 
  pull()
subset_meta_post1 <- sample_metadata %>%
  filter(time_point == "FMT1")
length(samples_to_keep_post1)
```

***Richness***
```{r beta_richness_permanova_post1_2, warning=FALSE, comments="", message=FALSE}
richness_post1 <- as.matrix(beta_q0n$S)
richness_post1 <- as.dist(richness_post1[rownames(richness_post1) %in% samples_to_keep_post1,
               colnames(richness_post1) %in% samples_to_keep_post1])
betadisper(richness_post1, subset_meta_post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post1_1, warning=FALSE, comments="", message=FALSE}
adonis2(richness_post1 ~ type,
        data = subset_meta_post1 %>% arrange(match(Tube_code,labels(richness_post1))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_richness_permanova_post1_3, warning=FALSE, comments="", message=FALSE}
subset_meta_post1_arrange <- column_to_rownames(subset_meta_post1, "Tube_code")
subset_meta_post1_arrange<-subset_meta_post1_arrange[labels(richness_post1),]

pairwise <- pairwise.adonis(richness_post1, subset_meta_post1_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_neutral_permanova_post1_2, warning=FALSE, comments="", message=FALSE}
neutral_post1 <- as.matrix(beta_q1n$S)
neutral_post1 <- as.dist(neutral_post1[rownames(neutral_post1) %in% samples_to_keep_post1,
               colnames(neutral_post1) %in% samples_to_keep_post1])
betadisper(neutral_post1, subset_meta_post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post1_1, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_post1 ~ type,
        data = subset_meta_post1 %>% arrange(match(Tube_code,labels(neutral_post1))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_neutral_permanova_post1_3, warning=FALSE, comments="", message=FALSE}
subset_meta_post1_arrange <- column_to_rownames(subset_meta_post1, "Tube_code")
subset_meta_post1_arrange<-subset_meta_post1_arrange[labels(neutral_post1),]
pairwise <- pairwise.adonis(neutral_post1, subset_meta_post1_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_post1_2, warning=FALSE, comments="", message=FALSE}
phylo_post1 <- as.matrix(beta_q1p$S)
phylo_post1 <- as.dist(phylo_post1[rownames(phylo_post1) %in% samples_to_keep_post1,
               colnames(phylo_post1) %in% samples_to_keep_post1])
betadisper(phylo_post1, subset_meta_post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_post1_1, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_post1 ~ type,
        data = subset_meta_post1 %>% arrange(match(Tube_code,labels(phylo_post1))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

***NMDS***
```{r beta_neutral_nmds_post1, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post1 <- richness_post1 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post1, by = join_by(sample == Tube_code)) %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_neutral_nmds_post1 <- neutral_post1 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post1, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_phylogenetic_nmds_post1 <- phylo_post1 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post1, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post1_final, fig.height=7}
ggarrange(beta_richness_nmds_post1, beta_neutral_nmds_post1, beta_phylogenetic_nmds_post1, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

#### Functional differences
***MCI***
```{r gitfs_elements_tm5, warning=FALSE, comments="", message=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT1") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_functional_tm5_stats_2, warning=FALSE, comments="", message=FALSE}
MCI_tm5_ele <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point == "FMT1")

shapiro.test(MCI_tm5_ele$value)#normality test

res.aov <- aov(value ~ type, data = MCI_tm5_ele)#anova test
summary(res.aov)
```

***Functional differences***

CC and CI
```{r comunity_elem_7_1, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Control", "Treatment") & time_point == "FMT1") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames()
```

CI and WC
```{r comunity_elem_CI_WC, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Hot_control", "Treatment") &
           time_point == "FMT1") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```

CC and WC
```{r comunity_elem_CC_WC, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Hot_control", "Control") &
           time_point == "FMT1") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```

## FMT2
### Is the GM of the intervention group still more similar to warm population after one week from FMT?
#### Alpha
```{r alpha_div_plot_FMT2, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
bxp1<-alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT2") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
        scale_x_discrete(labels = c("Control" = "Cold-Cold", "Hot_control" = "Hot-Hot", "Treatment" = "Cold-Hot")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=3, label.x=.7) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_blank(),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")


# Add significance bars with p-values
bxp1 + 
  geom_signif(comparisons = list(c("Control", "Hot_control"), c("Control", "Treatment"), c("Hot_control", "Treatment")),
              map_signif_level = TRUE)
```

```{r}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT2") 
```

***Richness***
```{r alpha_div_data_FMT2, comment="", message=FALSE, warning=FALSE, eval=FALSE}
Modelq0GLMMNB <- glm.nb(richness ~ type, data = alpha_div_meta,trace=TRUE)
summary(Modelq0GLMMNB)
emmeans(Modelq0GLMMNB, pairwise ~ type)
```

***Neutral***
```{r alpha_div_data_1_FMT2, comment="", message=FALSE, warning=FALSE}
Modelq1n <- lm(formula = neutral ~ type, data = alpha_div_meta) 
summary(Modelq1n)
```

***Phylogenetic***
```{r alpha_div_data_2_FMT2, comment="", message=FALSE, warning=FALSE}
Model_phylo <- lm(formula = phylogenetic ~ type, data = alpha_div_meta) 
summary(Model_phylo)
```

#### Beta diversity

***Number of samples used***
```{r beta_div_neutral_FMT2, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_FMT2 <- sample_metadata %>%
  filter(time_point == "FMT2") %>% 
  select(Tube_code) %>% 
  pull()
subset_meta_FMT2 <- sample_metadata %>%
  filter(time_point == "FMT2")
length(samples_to_keep_FMT2)
```

***Richness***
```{r beta_richness_permanova_FMT2_2, warning=FALSE, comments="", message=FALSE}
richness_FMT2 <- as.matrix(beta_q0n$S)
richness_FMT2 <- as.dist(richness_FMT2[rownames(richness_FMT2) %in% samples_to_keep_FMT2,
               colnames(richness_FMT2) %in% samples_to_keep_FMT2])
betadisper(richness_FMT2, subset_meta_FMT2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_FMT2_1, warning=FALSE, comments="", message=FALSE}
adonis2(richness_FMT2 ~ type,
        data = subset_meta_FMT2 %>% arrange(match(Tube_code,labels(richness_FMT2))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_richness_permanova_FMT2_3, warning=FALSE, comments="", message=FALSE}
subset_meta_FMT2_arrange <- column_to_rownames(subset_meta_FMT2, "Tube_code")
subset_meta_FMT2_arrange<-subset_meta_FMT2_arrange[labels(richness_FMT2),]

pairwise <- pairwise.adonis(richness_FMT2, subset_meta_FMT2_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_neutral_permanova_FMT2_2, warning=FALSE, comments="", message=FALSE}
neutral_FMT2 <- as.matrix(beta_q1n$S)
neutral_FMT2 <- as.dist(neutral_FMT2[rownames(neutral_FMT2) %in% samples_to_keep_FMT2,
               colnames(neutral_FMT2) %in% samples_to_keep_FMT2])
betadisper(neutral_FMT2, subset_meta_FMT2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_FMT2_1, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_FMT2 ~ type,
        data = subset_meta_FMT2 %>% arrange(match(Tube_code,labels(neutral_FMT2))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_neutral_permanova_FMT2_3, warning=FALSE, comments="", message=FALSE}
subset_meta_FMT2_arrange <- column_to_rownames(subset_meta_FMT2, "Tube_code")
subset_meta_FMT2_arrange<-subset_meta_FMT2_arrange[labels(neutral_FMT2),]
pairwise <- pairwise.adonis(neutral_FMT2, subset_meta_FMT2_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_FMT2_2, warning=FALSE, comments="", message=FALSE}
phylo_FMT2 <- as.matrix(beta_q1p$S)
phylo_FMT2 <- as.dist(phylo_FMT2[rownames(phylo_FMT2) %in% samples_to_keep_FMT2,
               colnames(phylo_FMT2) %in% samples_to_keep_FMT2])
betadisper(phylo_FMT2, subset_meta_FMT2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_FMT2_1, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_FMT2 ~ type,
        data = subset_meta_FMT2 %>% arrange(match(Tube_code,labels(phylo_FMT2))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```
```{r beta_phylo_permanova_FMT2_pair, warning=FALSE, comments="", message=FALSE}
subset_meta_FMT2_arrange <- column_to_rownames(subset_meta_FMT2, "Tube_code")
subset_meta_FMT2_arrange<-subset_meta_FMT2_arrange[labels(phylo_FMT2),]
pairwise <- pairwise.adonis(phylo_FMT2, subset_meta_FMT2_arrange$type, perm=999)
pairwise%>%
  tt()
```

***NMDS***
```{r beta_neutral_nmds_FMT2, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_FMT2 <- richness_FMT2 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_FMT2, by = join_by(sample == Tube_code)) %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_neutral_nmds_FMT2 <- neutral_FMT2 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_FMT2, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_phylogenetic_nmds_FMT2 <- phylo_FMT2 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_FMT2, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_FMT2_final, fig.height=7}
ggarrange(beta_richness_nmds_FMT2, beta_neutral_nmds_FMT2, beta_phylogenetic_nmds_FMT2, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

#### Differentially abundant MAGs
##### Warm vs Intervention
```{r}
physeq_FMT2 <- subset_samples(physeq_all, type %in% c("Hot_control", "Treatment") & time_point == "FMT2")
physeq_FMT2 <- prune_taxa(taxa_sums(physeq_FMT2)>0, physeq_FMT2)
```
```{r ancom_rand_pond_FMT2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancom_rand_output = ancombc2(data = physeq_FMT2, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "Population", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = "Population", 
                  struc_zero = TRUE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```
```{r}
ancom_rand_output$res %>%
  dplyr::select(taxon, lfc_PopulationHot_dry, p_PopulationHot_dry) %>%
  filter(p_PopulationHot_dry < 0.05)
```
```{r ancom_rand_FMT2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_FMT2@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output$res %>%
  dplyr::select(taxon, lfc_PopulationHot_dry, p_PopulationHot_dry) %>%
  filter(p_PopulationHot_dry < 0.05) %>%
  dplyr::arrange(p_PopulationHot_dry) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_PopulationHot_dry)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```
```{r ancombc_rand_plot_phy_FMT2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag%>%
      mutate(genome=factor(taxon,levels=ancombc_rand_table_mag$taxon)) %>%
ggplot(., aes(x=lfc_PopulationHot_dry, y=forcats::fct_reorder(genome,lfc_PopulationHot_dry), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

***Phyla of the significant MAGs in warm control***
```{r sign_features_FMT2_warm, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_PopulationHot_dry>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in intervention group***
```{r sign_features_FMT2_cold, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_PopulationHot_dry<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

##### Intervention vs cold control
```{r}
physeq_FMT2 <- subset_samples(physeq_all, type %in% c("Control", "Treatment") & time_point == "FMT2")
physeq_FMT2 <- prune_taxa(taxa_sums(physeq_FMT2)>0, physeq_FMT2)
```
```{r ancom_rand_pond_FMT2_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancom_rand_output_cold = ancombc2(data = physeq_FMT2, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "type", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = "type", 
                  struc_zero = TRUE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```
```{r}
ancom_rand_output_cold$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05)
```

```{r ancom_rand_FMT2_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_FMT2@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_cold$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05) %>%
  dplyr::arrange(p_typeTreatment) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_typeTreatment)

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```
```{r ancombc_rand_plot_phy_FMT2_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag%>%
      mutate(genome=factor(taxon,levels=ancombc_rand_table_mag$taxon)) %>%
ggplot(., aes(x=lfc_typeTreatment, y=forcats::fct_reorder(genome,lfc_typeTreatment), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```

***Phyla of the significant MAGs in intervention group***
```{r sign_features_FMT2_int, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_typeTreatment>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in cold control***
```{r sign_features_FMT2_cold_con, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_typeTreatment<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

#### Functional differences
***MCI***
```{r gitfs_elements_tm5_FMT2_cold, warning=FALSE, comments="", message=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT2") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_functional_FMT2_cold, warning=FALSE, comments="", message=FALSE}
MCI_FMT2 <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point == "FMT2")

shapiro.test(MCI_FMT2$value)

res.aov <- aov(value ~ type, data = MCI_FMT2)
summary(res.aov)
```

***Functional differences***

CC and CI
```{r comunity_elem_FMT2_cold, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Control", "Treatment") & time_point == "FMT2") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

CI and WC
```{r comunity_elem_FMT2_warm, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Hot_control", "Treatment") &
           time_point == "FMT2") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```

CC and WC
```{r comunity_elem_FMT2_control, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Hot_control", "Control") &
           time_point == "FMT2") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```

<!--chapter:end:08_FMT.Rmd-->

# FMT-2
```{r knitr_options_FMT2, include=FALSE}
knitr::opts_chunk$set(fig.width=10, fig.height=7, fig.fullwidth=TRUE,
fig.path='Figures/', warning=FALSE,
message=FALSE)
set.seed(2802)
```
```{r load_data_FMT2}
load("data/data_27022025.Rdata")
load("data/beta_27022025.Rdata")
load("data/calculations_28022025.Rdata")
```

## From acclimation to FMT1
### What is the effect of FMT and temperature on the GM after 1 week?

CI vs CC

#### Alpha

```{r alpha_div_plot_6, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
bxp1<-alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT1"|time_point=="Acclimation") %>%
  filter(type!="Hot_control") %>%
  mutate(metric = factor(metric, levels = c("richness", "neutral", "phylogenetic", "functional"))) %>%
  ggplot(aes(y = value, x = interaction(time_point, type), group = interaction(time_point, type), color = type, fill = type)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8), alpha = 0.5) +
  scale_color_manual(name="Type",
          breaks=c("Control", "Treatment"),
          labels=c("Cold-control", "Cold-intervention"),
          values=c("#4477AA", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control", "Treatment"),
          labels=c("Cold-control", "Cold-intervention"),
          values=c("#4477AA50","#76b18350")) +
        scale_x_discrete(labels = c("Control" = "Cold-Cold", "Treatment" = "Cold-Hot")) +
  theme_minimal() +
  facet_wrap(. ~ metric,scales = "free") +
  stat_compare_means(size=3) +
  scale_x_discrete(labels = c("Acclimation.Control" = "Acclimation-CC",
                              "FMT.Treatment" = "FMT1-CI",
                              "Acclimation.Treatment" = "Acclimation-CI",
                              "FMT1.Control" = "FMT1-CC")) +
  theme(axis.text.x=element_text(size=8, angle=45))+
  labs(x = "Time Point", y = "value", color = "Type", fill = "Type")

# Add significance bars with p-values
bxp1 + 
  geom_signif(comparisons = list(c("Acclimation-CC", "FMT1.CC"), c("Acclimation-CI", "FMT1-CI")),
              map_signif_level = TRUE)

```

```{r, comparison}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT1"|time_point=="Acclimation") %>%
  filter(type!="Hot_control") 
```

***Richness***
```{r alpha_div_data_post3, comparison, comment="", message=FALSE, warning=FALSE}
Modelq0GLMMNB <- glm.nb(richness ~ type*time_point, data = alpha_div_meta,trace=TRUE)
summary(Modelq0GLMMNB)
emmeans(Modelq0GLMMNB, pairwise ~ type)
```

***Neutral***
```{r alpha_div_data_1_post3, comparison, comment="", message=FALSE, warning=FALSE}
Modelq1n <- lm(formula = neutral ~ type*time_point, data = alpha_div_meta) 
summary(Modelq1n)
```

***Phylogenetic***
```{r alpha_div_data_2_post3, comparison, comment="", message=FALSE, warning=FALSE}
Model_phylo <- lm(formula = phylogenetic ~ type*time_point, data = alpha_div_meta) 
summary(Model_phylo)
```

#### Beta diversity

***Number of samples used***
```{r beta_div_neutral_treat1_samples_2_comparison, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_post3 <- sample_metadata %>%
  filter(time_point=="FMT1"|time_point=="Acclimation") %>% 
  filter(type!="Hot_control") %>%
  select(Tube_code) %>% 
  pull()
subset_meta_post3 <- sample_metadata %>%
  filter(time_point=="FMT1"|time_point=="Acclimation") %>% 
  filter(type!="Hot_control")
length(samples_to_keep_post3)
```

***Richness***
```{r beta_richness_permanova_post3_2_comparison, warning=FALSE, comments="", message=FALSE}
richness_post3 <- as.matrix(beta_q0n$S)
richness_post3 <- as.dist(richness_post3[rownames(richness_post3) %in% samples_to_keep_post3,
               colnames(richness_post3) %in% samples_to_keep_post3])
betadisper(richness_post3, subset_meta_post3$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post3_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(richness_post3 ~ type*time_point,
        data = subset_meta_post3 %>% arrange(match(Tube_code,labels(richness_post3))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_richness_permanova_post3_3_comparison, warning=FALSE, comments="", message=FALSE}
subset_meta_post3_arrange <- column_to_rownames(subset_meta_post3, "Tube_code")
subset_meta_post3_arrange<-subset_meta_post3_arrange[labels(richness_post3),]

pairwise <- pairwise.adonis(richness_post3, subset_meta_post3_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_neutral_permanova_post3_2_comparison, warning=FALSE, comments="", message=FALSE}
neutral_post3 <- as.matrix(beta_q1n$S)
neutral_post3 <- as.dist(neutral_post3[rownames(neutral_post3) %in% samples_to_keep_post3,
               colnames(neutral_post3) %in% samples_to_keep_post3])
betadisper(neutral_post3, subset_meta_post3$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post3_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_post3 ~ type,
        data = subset_meta_post3 %>% arrange(match(Tube_code,labels(neutral_post3))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_neutral_permanova_post3_3_comparison, warning=FALSE, comments="", message=FALSE}
subset_meta_post3_arrange <- column_to_rownames(subset_meta_post3, "Tube_code")
subset_meta_post3_arrange<-subset_meta_post3_arrange[labels(neutral_post3),]
pairwise <- pairwise.adonis(neutral_post3, subset_meta_post3_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_post3_2_comparison, warning=FALSE, comments="", message=FALSE}
phylo_post3 <- as.matrix(beta_q1p$S)
phylo_post3 <- as.dist(phylo_post3[rownames(phylo_post3) %in% samples_to_keep_post3,
               colnames(phylo_post3) %in% samples_to_keep_post3])
betadisper(phylo_post3, subset_meta_post3$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_post3_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_post3 ~ type,
        data = subset_meta_post3 %>% arrange(match(Tube_code,labels(phylo_post3))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

***NMDS***
```{r beta_neutral_nmds_post3_comparison, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post3 <- richness_post3 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post3, by = join_by(sample == Tube_code)) %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Treatment"),
                       labels=c("Cold-Cold", "Cold-Hot"),
                       values=c("#4477AA","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_neutral_nmds_post3 <- neutral_post3 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post3, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Treatment"),
                       labels=c("Cold-Cold", "Cold-Hot"),
                       values=c("#4477AA","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_phylogenetic_nmds_post3 <- phylo_post3 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post3, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Treatment"),
                       labels=c("Cold-Cold", "Cold-Hot"),
                       values=c("#4477AA","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post3_final, fig.height=7}
ggarrange(beta_richness_nmds_post3, beta_neutral_nmds_post3, beta_phylogenetic_nmds_post3, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

***Functional differences***

CC from acclimation to FMT1

```{r comunity_elem_FMT3_cold, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Control") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

CI from acclimation to FMT1

```{r comunity_elem_FMT3_cold_1, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Treatment") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

CI vs WC

#### Alpha

```{r alpha_div_plot_9, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
bxp1<-alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT1"|time_point=="Acclimation") %>%
  filter(type!="Hot_control") %>%
  mutate(metric = factor(metric, levels = c("richness", "neutral", "phylogenetic", "functional"))) %>%
  ggplot(aes(y = value, x = interaction(time_point, type), group = interaction(time_point, type), color = type, fill = type)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8), alpha = 0.5) +
  scale_color_manual(name="Type",
          breaks=c("Hot_control", "Treatment"),
          labels=c("Warm_control", "Cold-intervention"),
          values=c("#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Hot_control", "Treatment"),
          labels=c("Warm-control", "Cold-intervention"),
          values=c("#d57d2c","#76b18350")) +
        scale_x_discrete(labels = c("Hot_control" = "Warm-control", "Treatment" = "Cold-intervention")) +
  theme_minimal() +
  facet_wrap(. ~ metric,scales = "free") +
  stat_compare_means(size=3) +
  scale_x_discrete(labels = c("Acclimation.Hot_control" = "Acclimation-WC",
                              "FMT.Treatment" = "FMT1-CI",
                              "Acclimation.Treatment" = "Acclimation-CI",
                              "FMT1.Hot_control" = "FMT1-WC")) +
  theme(axis.text.x=element_text(size=8, angle=45))+
  labs(x = "Time Point", y = "value", color = "Type", fill = "Type")

# Add significance bars with p-values
bxp1 + 
  geom_signif(comparisons = list(c("Acclimation-WC", "FMT1.WC"), c("Acclimation-CI", "FMT1-CI")),
              map_signif_level = TRUE)

```

```{r, comparison_1}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT1"|time_point=="Acclimation") %>%
  filter(type!="Control") 
```

***Richness***
```{r alpha_div_data_post4, comparison, comment="", message=FALSE, warning=FALSE}
Modelq0GLMMNB <- glm.nb(richness ~ type*time_point, data = alpha_div_meta,trace=TRUE)
summary(Modelq0GLMMNB)
emmeans(Modelq0GLMMNB, pairwise ~ type)
```

***Neutral***
```{r alpha_div_data_1_post4, comparison, comment="", message=FALSE, warning=FALSE}
Modelq1n <- lm(formula = neutral ~ type*time_point, data = alpha_div_meta) 
summary(Modelq1n)
```

***Phylogenetic***
```{r alpha_div_data_2_post4, comparison, comment="", message=FALSE, warning=FALSE}
Model_phylo <- lm(formula = phylogenetic ~ type*time_point, data = alpha_div_meta) 
summary(Model_phylo)
```

#### Beta diversity

***Number of samples used***
```{r beta_div_neutral_treat1_samples_2_post4, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_post4 <- sample_metadata %>%
  filter(time_point=="FMT1"|time_point=="Acclimation") %>% 
  filter(type!="Control") %>%
  select(Tube_code) %>% 
  pull()
subset_meta_post4 <- sample_metadata %>%
  filter(time_point=="FMT1"|time_point=="Acclimation") %>% 
  filter(type!="Control")
length(samples_to_keep_post4)
```

***Richness***
```{r beta_richness_permanova_post4_2_comparison, warning=FALSE, comments="", message=FALSE}
richness_post4 <- as.matrix(beta_q0n$S)
richness_post4 <- as.dist(richness_post4[rownames(richness_post4) %in% samples_to_keep_post4,
               colnames(richness_post4) %in% samples_to_keep_post4])
betadisper(richness_post4, subset_meta_post4$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post4_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(richness_post4 ~ type*time_point,
        data = subset_meta_post4 %>% arrange(match(Tube_code,labels(richness_post4))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_richness_permanova_post4_3_comparison, warning=FALSE, comments="", message=FALSE}
subset_meta_post4_arrange <- column_to_rownames(subset_meta_post4, "Tube_code")
subset_meta_post4_arrange<-subset_meta_post4_arrange[labels(richness_post4),]

pairwise <- pairwise.adonis(richness_post4, subset_meta_post4_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_neutral_permanova_post4_2_comparison, warning=FALSE, comments="", message=FALSE}
neutral_post4 <- as.matrix(beta_q1n$S)
neutral_post4 <- as.dist(neutral_post4[rownames(neutral_post4) %in% samples_to_keep_post4,
               colnames(neutral_post4) %in% samples_to_keep_post4])
betadisper(neutral_post4, subset_meta_post4$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post4_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_post4 ~ type,
        data = subset_meta_post4 %>% arrange(match(Tube_code,labels(neutral_post4))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_neutral_permanova_post4_3_comparison, warning=FALSE, comments="", message=FALSE}
subset_meta_post4_arrange <- column_to_rownames(subset_meta_post4, "Tube_code")
subset_meta_post4_arrange<-subset_meta_post4_arrange[labels(neutral_post4),]
pairwise <- pairwise.adonis(neutral_post4, subset_meta_post4_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_post4_2_comparison, warning=FALSE, comments="", message=FALSE}
phylo_post4 <- as.matrix(beta_q1p$S)
phylo_post4 <- as.dist(phylo_post4[rownames(phylo_post4) %in% samples_to_keep_post4,
               colnames(phylo_post4) %in% samples_to_keep_post4])
betadisper(phylo_post4, subset_meta_post4$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_post4_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_post4 ~ type,
        data = subset_meta_post4 %>% arrange(match(Tube_code,labels(phylo_post4))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

***NMDS***
```{r beta_neutral_nmds_post4_comparison, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post4 <- richness_post4 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post4, by = join_by(sample == Tube_code)) %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Hot_control", "Treatment"),
                       labels=c("Warm-control", "Cold-intervention"),
                       values=c("#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_neutral_nmds_post4 <- neutral_post4 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post4, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Hot_control", "Treatment"),
                       labels=c("Warm-control", "Cold-intervention"),
                       values=c("#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_phylogenetic_nmds_post4 <- phylo_post4 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post4, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Hot_control", "Treatment"),
                       labels=c("Warm-control", "Cold-intervention"),
                       values=c("#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post4_final, fig.height=7}
ggarrange(beta_richness_nmds_post4, beta_neutral_nmds_post4, beta_phylogenetic_nmds_post4, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

***Functional differences***

WC from acclimation to FMT1

```{r comunity_elem_FMT3_cold_2, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Hot_control") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

### What is the effect of FMT and temperature on the GM after 2 weeks?

CI vs CC

#### Alpha

```{r alpha_div_plot_10, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
bxp1<-alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT2"|time_point=="Acclimation") %>%
  filter(type!="Hot_control") %>%
  mutate(metric = factor(metric, levels = c("richness", "neutral", "phylogenetic", "functional"))) %>%
  ggplot(aes(y = value, x = interaction(time_point, type), group = interaction(time_point, type), color = type, fill = type)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8), alpha = 0.5) +
  scale_color_manual(name="Type",
          breaks=c("Control", "Treatment"),
          labels=c("Cold-control", "Cold-intervention"),
          values=c("#4477AA", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control", "Treatment"),
          labels=c("Cold-control", "Cold-intervention"),
          values=c("#4477AA50","#76b18350")) +
        scale_x_discrete(labels = c("Control" = "Cold-Cold", "Treatment" = "Cold-Hot")) +
  theme_minimal() +
  facet_wrap(. ~ metric,scales = "free") +
  stat_compare_means(size=3) +
  scale_x_discrete(labels = c("Acclimation.Control" = "Acclimation-CC",
                              "FMT.Treatment" = "FMT2-CI",
                              "Acclimation.Treatment" = "Acclimation-CI",
                              "FMT2.Control" = "FMT2-CC")) +
  theme(axis.text.x=element_text(size=8, angle=45))+
  labs(x = "Time Point", y = "value", color = "Type", fill = "Type")

# Add significance bars with p-values
bxp1 + 
  geom_signif(comparisons = list(c("Acclimation-CC", "FMT2.CC"), c("Acclimation-CI", "FMT2-CI")),
              map_signif_level = TRUE)

```

```{r, comparison_post3}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT2"|time_point=="Acclimation") %>%
  filter(type!="Hot_control") 
```

***Richness***
```{r alpha_div_data, comparison, comment="", message=FALSE, warning=FALSE}
Modelq0GLMMNB <- glm.nb(richness ~ type*time_point, data = alpha_div_meta,trace=TRUE)
summary(Modelq0GLMMNB)
emmeans(Modelq0GLMMNB, pairwise ~ type)
```

***Neutral***
```{r alpha_div_data_1, comparison, comment="", message=FALSE, warning=FALSE}
Modelq1n <- lm(formula = neutral ~ type*time_point, data = alpha_div_meta) 
summary(Modelq1n)
```

***Phylogenetic***
```{r alpha_div_data_2, comparison, comment="", message=FALSE, warning=FALSE}
Model_phylo <- lm(formula = phylogenetic ~ type*time_point, data = alpha_div_meta) 
summary(Model_phylo)
```

#### Beta diversity

***Number of samples used***
```{r beta_div_neutral_treat1_samples_2_comparison_post5, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_post5 <- sample_metadata %>%
  filter(time_point=="FMT2"|time_point=="Acclimation") %>% 
  filter(type!="Hot_control") %>%
  select(Tube_code) %>% 
  pull()
subset_meta_post5 <- sample_metadata %>%
  filter(time_point=="FMT2"|time_point=="Acclimation") %>% 
  filter(type!="Hot_control")
length(samples_to_keep_post5)
```

***Richness***
```{r beta_richness_permanova_post5_2_comparison, warning=FALSE, comments="", message=FALSE}
richness_post5 <- as.matrix(beta_q0n$S)
richness_post5 <- as.dist(richness_post5[rownames(richness_post5) %in% samples_to_keep_post5,
               colnames(richness_post5) %in% samples_to_keep_post5])
betadisper(richness_post5, subset_meta_post5$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post5_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(richness_post5 ~ type*time_point,
        data = subset_meta_post5 %>% arrange(match(Tube_code,labels(richness_post5))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_richness_permanova_post5_3_comparison, warning=FALSE, comments="", message=FALSE}
subset_meta_post5_arrange <- column_to_rownames(subset_meta_post5, "Tube_code")
subset_meta_post5_arrange<-subset_meta_post5_arrange[labels(richness_post5),]

pairwise <- pairwise.adonis(richness_post5, subset_meta_post5_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_neutral_permanova_post5_2_comparison, warning=FALSE, comments="", message=FALSE}
neutral_post5 <- as.matrix(beta_q1n$S)
neutral_post5 <- as.dist(neutral_post5[rownames(neutral_post5) %in% samples_to_keep_post5,
               colnames(neutral_post5) %in% samples_to_keep_post5])
betadisper(neutral_post5, subset_meta_post5$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post5_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_post5 ~ type,
        data = subset_meta_post5 %>% arrange(match(Tube_code,labels(neutral_post5))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_neutral_permanova_post5_3_comparison, warning=FALSE, comments="", message=FALSE}
subset_meta_post5_arrange <- column_to_rownames(subset_meta_post5, "Tube_code")
subset_meta_post5_arrange<-subset_meta_post5_arrange[labels(neutral_post5),]
pairwise <- pairwise.adonis(neutral_post5, subset_meta_post5_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_post5_2_comparison, warning=FALSE, comments="", message=FALSE}
phylo_post5 <- as.matrix(beta_q1p$S)
phylo_post5 <- as.dist(phylo_post5[rownames(phylo_post5) %in% samples_to_keep_post5,
               colnames(phylo_post5) %in% samples_to_keep_post5])
betadisper(phylo_post5, subset_meta_post5$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_post5_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_post5 ~ type,
        data = subset_meta_post5 %>% arrange(match(Tube_code,labels(phylo_post5))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

***NMDS***
```{r beta_neutral_nmds_post5_comparison, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post5 <- richness_post5 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post5, by = join_by(sample == Tube_code)) %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Treatment"),
                       labels=c("Cold-Cold", "Cold-Hot"),
                       values=c("#4477AA","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_neutral_nmds_post5 <- neutral_post5 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post5, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Treatment"),
                       labels=c("Cold-Cold", "Cold-Hot"),
                       values=c("#4477AA","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_phylogenetic_nmds_post5 <- phylo_post5 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post5, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Control", "Treatment"),
                       labels=c("Cold-Cold", "Cold-Hot"),
                       values=c("#4477AA","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post5_final, fig.height=7}
ggarrange(beta_richness_nmds_post5, beta_neutral_nmds_post5, beta_phylogenetic_nmds_post5, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

***Functional differences***

CC from acclimation to FMT2

```{r comunity_elem_FMT3_cold_5, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(time_point %in% c("FMT2","Acclimation") & type == "Control") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

CI from acclimation to FMT2

```{r comunity_elem_FMT3_cold_6, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(time_point %in% c("FMT2","Acclimation") & type == "Treatment") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

CI vs WC

#### Alpha

```{r alpha_div_plot_8, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
bxp1<-alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT2"|time_point=="Acclimation") %>%
  filter(type!="Hot_control") %>%
  mutate(metric = factor(metric, levels = c("richness", "neutral", "phylogenetic", "functional"))) %>%
  ggplot(aes(y = value, x = interaction(time_point, type), group = interaction(time_point, type), color = type, fill = type)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8), alpha = 0.5) +
  scale_color_manual(name="Type",
          breaks=c("Hot_control", "Treatment"),
          labels=c("Warm_control", "Cold-intervention"),
          values=c("#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Hot_control", "Treatment"),
          labels=c("Warm-control", "Cold-intervention"),
          values=c("#d57d2c","#76b18350")) +
        scale_x_discrete(labels = c("Hot_control" = "Warm-control", "Treatment" = "Cold-intervention")) +
  theme_minimal() +
  facet_wrap(. ~ metric,scales = "free") +
  stat_compare_means(size=3) +
  scale_x_discrete(labels = c("Acclimation.Hot_control" = "Acclimation-WC",
                              "FMT.Treatment" = "FMT2-CI",
                              "Acclimation.Treatment" = "Acclimation-CI",
                              "FMT2.Hot_control" = "FMT2-WC")) +
  theme(axis.text.x=element_text(size=8, angle=45))+
  labs(x = "Time Point", y = "value", color = "Type", fill = "Type")

# Add significance bars with p-values
bxp1 + 
  geom_signif(comparisons = list(c("Acclimation-WC", "FMT2.WC"), c("Acclimation-CI", "FMT2-CI")),
              map_signif_level = TRUE)

```

```{r, comparison_1_post5}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT2"|time_point=="Acclimation") %>%
  filter(type!="Control") 
```

***Richness***
```{r alpha_div_data_post6, comparison, comment="", message=FALSE, warning=FALSE}
Modelq0GLMMNB <- glm.nb(richness ~ type*time_point, data = alpha_div_meta,trace=TRUE)
summary(Modelq0GLMMNB)
emmeans(Modelq0GLMMNB, pairwise ~ type)
```

***Neutral***
```{r alpha_div_data_1_post6, comparison, comment="", message=FALSE, warning=FALSE}
Modelq1n <- lm(formula = neutral ~ type*time_point, data = alpha_div_meta) 
summary(Modelq1n)
```

***Phylogenetic***
```{r alpha_div_data_2_post6, comparison, comment="", message=FALSE, warning=FALSE}
Model_phylo <- lm(formula = phylogenetic ~ type*time_point, data = alpha_div_meta) 
summary(Model_phylo)
```

#### Beta diversity

***Number of samples used***
```{r beta_div_neutral_treat1_samples_2_post6, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_post6 <- sample_metadata %>%
  filter(time_point=="FMT2"|time_point=="Acclimation") %>% 
  filter(type!="Control") %>%
  select(Tube_code) %>% 
  pull()
subset_meta_post6 <- sample_metadata %>%
  filter(time_point=="FMT2"|time_point=="Acclimation") %>% 
  filter(type!="Control")
length(samples_to_keep_post6)
```

***Richness***
```{r beta_richness_permanova_post6_2_comparison, warning=FALSE, comments="", message=FALSE}
richness_post6 <- as.matrix(beta_q0n$S)
richness_post6 <- as.dist(richness_post6[rownames(richness_post6) %in% samples_to_keep_post6,
               colnames(richness_post6) %in% samples_to_keep_post6])
betadisper(richness_post6, subset_meta_post6$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post6_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(richness_post6 ~ type*time_point,
        data = subset_meta_post6 %>% arrange(match(Tube_code,labels(richness_post6))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_richness_permanova_post6_3_comparison, warning=FALSE, comments="", message=FALSE}
subset_meta_post6_arrange <- column_to_rownames(subset_meta_post6, "Tube_code")
subset_meta_post6_arrange<-subset_meta_post6_arrange[labels(richness_post6),]

pairwise <- pairwise.adonis(richness_post6, subset_meta_post6_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_neutral_permanova_post6_2_comparison, warning=FALSE, comments="", message=FALSE}
neutral_post6 <- as.matrix(beta_q1n$S)
neutral_post6 <- as.dist(neutral_post6[rownames(neutral_post6) %in% samples_to_keep_post6,
               colnames(neutral_post6) %in% samples_to_keep_post6])
betadisper(neutral_post6, subset_meta_post6$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post6_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_post6 ~ type,
        data = subset_meta_post6 %>% arrange(match(Tube_code,labels(neutral_post6))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_neutral_permanova_post6_3_comparison, warning=FALSE, comments="", message=FALSE}
subset_meta_post6_arrange <- column_to_rownames(subset_meta_post6, "Tube_code")
subset_meta_post6_arrange<-subset_meta_post6_arrange[labels(neutral_post6),]
pairwise <- pairwise.adonis(neutral_post6, subset_meta_post6_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_post6_2_comparison, warning=FALSE, comments="", message=FALSE}
phylo_post6 <- as.matrix(beta_q1p$S)
phylo_post6 <- as.dist(phylo_post6[rownames(phylo_post6) %in% samples_to_keep_post6,
               colnames(phylo_post6) %in% samples_to_keep_post6])
betadisper(phylo_post6, subset_meta_post6$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_post6_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_post6 ~ type,
        data = subset_meta_post6 %>% arrange(match(Tube_code,labels(phylo_post6))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

***NMDS***
```{r beta_neutral_nmds_post6_comparison, warning=FALSE, comments="", message=FALSE, results="hide"}
beta_richness_nmds_post6 <- richness_post6 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post6, by = join_by(sample == Tube_code)) %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Hot_control", "Treatment"),
                       labels=c("Warm-control", "Cold-intervention"),
                       values=c("#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_neutral_nmds_post6 <- neutral_post6 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post6, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Hot_control", "Treatment"),
                       labels=c("Warm-control", "Cold-intervention"),
                       values=c("#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

beta_phylogenetic_nmds_post6 <- phylo_post6 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post6, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Hot_control", "Treatment"),
                       labels=c("Warm-control", "Cold-intervention"),
                       values=c("#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")
```

```{r beta_neutral_nmds_plot_post6_final, fig.height=7}
ggarrange(beta_richness_nmds_post6, beta_neutral_nmds_post6, beta_phylogenetic_nmds_post6, ncol=2, nrow=2, common.legend = TRUE, legend="right")
```

***Functional differences***

WC from acclimation to FMT2

```{r comunity_elem_FMT3_cold_post6, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(time_point %in% c("FMT2","Acclimation") & type == "Hot_control") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 7)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```



<!--chapter:end:09_FMT-2.Rmd-->

